--- util/BUILD.gn
+++ util/BUILD.gn
@@ -661,8 +661,8 @@ crashpad_static_library("net") {
         deps += [ "//third_party/boringssl" ]
       } else {
         libs = [
-          "crypto",
           "ssl",
+          "crypto",
         ]
       }
     }
@@ -707,8 +707,8 @@ if (!crashpad_is_android && !crashpad_is_ios) {
         deps += [ "//third_party/boringssl" ]
       } else {
         libs = [
-          "crypto",
           "ssl",
+          "crypto",
         ]
       }
     }
