--- third_party/lss/lss.h
+++ third_party/lss/lss.h
@@ -16,9 +16,9 @@
 #define CRASHPAD_THIRD_PARTY_LSS_LSS_H_
 
 #if defined(CRASHPAD_LSS_SOURCE_EXTERNAL)
-#include "third_party/lss/linux_syscall_support.h"
+#include <linux_syscall_support.h>
 #elif defined(CRASHPAD_LSS_SOURCE_EMBEDDED)
-#include "third_party/lss/lss/linux_syscall_support.h"
+#include <linux_syscall_support.h>
 #elif defined(CRASHPAD_LSS_SOURCE_FUCHSIA)
 #include "../../../../third_party/linux-syscall-support/linux_syscall_support.h"
 #else
