diff --git a/include/commata/field_handling.hpp b/include/commata/field_handling.hpp
index 04ab67e..8b438b6 100644
--- a/include/commata/field_handling.hpp
+++ b/include/commata/field_handling.hpp
@@ -6,7 +6,9 @@
 #ifndef COMMATA_GUARD_CD40E918_ACCD_4879_BB48_7D9B8B823369
 #define COMMATA_GUARD_CD40E918_ACCD_4879_BB48_7D9B8B823369
 
+#include <functional>
 #include <type_traits>
+#include <utility>
 
 namespace commata {
 
