diff --git a/wscript b/wscript
index 616b3329..4d3e7f82 100755
--- a/wscript
+++ b/wscript
@@ -7,35 +7,34 @@ import shutil
 
 from waflib.Node import Node
 
 from wafmodules.filetasks import (
     find_resource_or_fail)
 
 import os.path, sys
 sys.path[0:0] = [os.path.join('dependencies', 'AnyPlatform', 'ohWafHelpers')]
 
 from filetasks import gather_files, build_tree, copy_task, find_dir_or_fail, create_copy_task
-from utilfuncs import invoke_test, guess_dest_platform, configure_toolchain, guess_ohnet_location, guess_location, guess_ssl_location, guess_raat_location, guess_libplatform_location, guess_libosa_location, is_core_platform
+from utilfuncs import invoke_test, guess_dest_platform, configure_toolchain, guess_ohnet_location, guess_location, guess_ssl_location, guess_libplatform_location, guess_libosa_location, is_core_platform
 
 def options(opt):
     opt.load('msvs')
     opt.load('msvc')
     opt.load('compiler_cxx')
     opt.load('compiler_c')
     opt.add_option('--ohnet-include-dir', action='store', default=None)
     opt.add_option('--ohnet-lib-dir', action='store', default=None)
     opt.add_option('--testharness-dir', action='store', default=os.path.join('dependencies', 'AnyPlatform', 'testharness'))
     opt.add_option('--ohnet', action='store', default=None)
     opt.add_option('--ssl', action='store', default=None)
     opt.add_option('--libplatform', action='store', default=None)
     opt.add_option('--libosa', action='store', default=None)
-    opt.add_option('--raat', action='store', default=None)
     opt.add_option('--debug', action='store_const', dest="debugmode", const="Debug", default="Release")
     opt.add_option('--release', action='store_const', dest="debugmode",  const="Release", default="Release")
     opt.add_option('--dest-platform', action='store', default=None)
     opt.add_option('--cross', action='store', default=None)
     opt.add_option('--with-default-fpm', action='store_true', default=False)
 
 def configure(conf):
 
     def set_env(conf, varname, value):
         conf.msg(
@@ -53,21 +52,20 @@ def configure(conf):
         except KeyError:
             conf.fatal('Specify --dest-platform')
 
     if is_core_platform(conf):
         guess_libosa_location(conf)
         guess_libplatform_location(conf)
 
     configure_toolchain(conf)
     guess_ohnet_location(conf)
     guess_ssl_location(conf)
-    guess_raat_location(conf)
 
     conf.env.dest_platform = conf.options.dest_platform
     conf.env.testharness_dir = os.path.abspath(conf.options.testharness_dir)
 
     if conf.options.dest_platform.startswith('Windows'):
         conf.env.LIB_OHNET=['ws2_32', 'iphlpapi', 'dbghelp', 'psapi', 'userenv']
     conf.env.STLIB_OHNET=['TestFramework', 'ohNetCore']
 
     if is_core_platform(conf):
         conf.env.prepend_value('STLIB_OHNET', ['target', 'platform'])
