diff --git a/wscript b/wscript
index ee7adbba..38ef5b01 100755
--- a/wscript
+++ b/wscript
@@ -10,21 +10,20 @@ from waflib.Node import Node
 from wafmodules.filetasks import (
     find_resource_or_fail)
 
 import os.path, sys
 sys.path[0:0] = [os.path.join('dependencies', 'AnyPlatform', 'ohWafHelpers')]
 
 from filetasks import gather_files, build_tree, copy_task, find_dir_or_fail, create_copy_task
 from utilfuncs import invoke_test, guess_dest_platform, configure_toolchain, guess_ohnet_location, guess_location, guess_ssl_location, guess_libplatform_location, guess_libosa_location, is_core_platform
 
 def options(opt):
-    opt.load('msvs')
     opt.load('msvc')
     opt.load('compiler_cxx')
     opt.load('compiler_c')
     opt.add_option('--ohnet-include-dir', action='store', default=None)
     opt.add_option('--ohnet-lib-dir', action='store', default=None)
     opt.add_option('--testharness-dir', action='store', default=os.path.join('dependencies', 'AnyPlatform', 'testharness'))
     opt.add_option('--ohnet', action='store', default=None)
     opt.add_option('--ssl', action='store', default=None)
     opt.add_option('--libplatform', action='store', default=None)
     opt.add_option('--libosa', action='store', default=None)
