diff --git a/wscript b/wscript
index d83690c0..3d9977cb 100755
--- a/wscript
+++ b/wscript
@@ -58,20 +58,22 @@ def configure(conf):
     configure_toolchain(conf)
     guess_ohnet_location(conf)
     guess_ssl_location(conf)
 
     conf.env.dest_platform = conf.options.dest_platform
     conf.env.testharness_dir = os.path.abspath(conf.options.testharness_dir)
 
     if conf.options.dest_platform.startswith('Windows'):
         conf.env.LIB_OHNET=['ws2_32', 'iphlpapi', 'dbghelp', 'psapi', 'userenv']
     conf.env.STLIB_OHNET=['TestFramework', 'ohNetCore']
+    if conf.options.dest_platform.startswith('Linux') and 'clang' in conf.env.CC[0]:
+        conf.env.append_value('STLIB_OHNET', ['atomic'])
 
     if is_core_platform(conf):
         conf.env.prepend_value('STLIB_OHNET', ['target', 'platform'])
         conf.env.append_value('DEFINES', ['DEFINE_TRACE', 'NETWORK_NTOHL_LOCAL', 'NOTERMIOS']) # Tell FLAC to use local ntohl implementation
 
     conf.env.INCLUDES = [
         '.',
         conf.path.find_node('.').abspath()
         ]
 
