diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2607e56..3ca4ac5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -73,9 +73,9 @@ check_include_file( "unistd.h"        HAVE_UNISTD_H   )
 check_include_file( "sgidefs.h"       HAVE_SGIDEFS_H  )
 check_include_file( "stdafx.h"        HAVE_STDAFX_H   )
 check_include_file( "Windows.h"       HAVE_WINDOWS_H  )
-check_include_file( "elf.h"           HAVE_ELF_H      ) 
-check_include_file( "libelf.h"        HAVE_LIBELF_H   ) 
-check_include_file( "libelf/libelf.h" HAVE_LIBELF_LIBELF_H) 
+set(HAVE_ELF_H FALSE)
+set(HAVE_LIBELF_H FALSE)
+set(HAVE_LIBELF_LIBELF_H TRUE)
 check_include_file( "alloca.h"        HAVE_ALLOCA_H   )
 check_include_file( "elfaccess.h"     HAVE_ELFACCESS_H)
 check_include_file( "sys/elf_386.h"   HAVE_SYS_ELF_386_H  )
@@ -128,6 +128,8 @@ endif()
 #  It's not really setting the location of libelfheader, 
 #  it is really #  either elf.h, or if that is missing 
 #  it is assuming  elf.h data is in the supplied libelf.
+find_package(libelf REQUIRED CONFIG)
+find_package(ZLIB REQUIRED CONFIG)
 
 if(HAVE_ELF_H)
     set(HAVE_LOCATION_OF_LIBELFHEADER "<elf.h>")
@@ -146,12 +148,13 @@ elseif(HAVE_LIBELF_LIBELF_H)
 endif()
 
 if (HAVE_LIBELF_H OR HAVE_LIBELF_LIBELF_H)
-  set (CMAKE_REQUIRED_LIBRARIES elf)
+  set (CMAKE_REQUIRED_DEFINITIONS -D__LIBELF64=1)
   message(STATUS "libelf header ${PLAIN_JUST_LIBELF} checking for elf64_getehdr")
   check_symbol_exists( elf64_getehdr ${PLAIN_JUST_LIBELF} HAVE_ELF64_GETEHDR)
   message(STATUS "libelf header ${PLAIN_JUST_LIBELF} checking for elf64_getshdr")
   check_symbol_exists( elf64_getshdr ${PLAIN_JUST_LIBELF} HAVE_ELF64_GETSHDR)
   set (CMAKE_REQUIRED_LIBRARIES)
+  set (CMAKE_REQUIRED_DEFINITIONS)
 endif()
 
 option(DWARF_WITH_LIBELF "Use libelf (default is YES)" TRUE)
@@ -166,6 +169,9 @@ if (DWARF_WITH_LIBELF)
   message(STATUS "checking using HAVE_ELF_H ... ${HAVE_ELF_H}")
   message(STATUS "checking using elf header ... ${HAVE_LOCATION_OF_LIBELFHEADER}")
   message(STATUS "checking using libelf header ... ${JUST_LIBELF}")
+
+  set (CMAKE_REQUIRED_DEFINITIONS -D__LIBELF64=1)
+
   check_c_source_compiles("
   #include ${HAVE_LOCATION_OF_LIBELFHEADER}
   int main()      
@@ -202,7 +208,7 @@ if (DWARF_WITH_LIBELF)
   # to set HAVE_LIBELF_OFF64_OK at present. 
   check_c_source_compiles("
   #define _GNU_SOURCE 1
-  #include ${JUST_LIBELF}
+  #include <${JUST_LIBELF}>
   int main()
   { 
       off64_t  p; p = 0;
@@ -210,7 +216,7 @@ if (DWARF_WITH_LIBELF)
   }"  HAVE_LIBELF_OFF64_OK)
 
   check_c_source_compiles("
-  #include ${JUST_LIBELF}
+  #include <${JUST_LIBELF}>
   /* This must be at global scope */
   struct _Elf;
   typedef struct _Elf Elf;
@@ -220,6 +226,10 @@ if (DWARF_WITH_LIBELF)
    int i = 12;
    return 0;
   }" HAVE_STRUCT_UNDERSCORE_ELF)
+
+  set (CMAKE_REQUIRED_LIBRARIES)
+  set (CMAKE_REQUIRED_DEFINITIONS)
+
 endif()
 message(STATUS "Assuming struct Elf for the default libdwarf.h")
 # Because cmake treats ; in an interesting way attempting
@@ -378,10 +388,9 @@ message(STATUS "Checking producer generates only 32bit... ${HAVE_STRICT_DWARF2_3
 
 
 # This references cmake/FindLibElf.cmake. See cmake documentation.
-find_package(LibElf REQUIRED)
 list(APPEND CMAKE_REQUIRED_INCLUDES ${LIBELF_INCLUDE_DIRS})
 
-configure_file(config.h.in.cmake config.h)
+configure_file(${CMAKE_SOURCE_DIR}/config.h.in.cmake ${CMAKE_BINARY_DIR}/config.h)
 
 if(BUILD_NON_SHARED)
 	set(DWARF_TARGETS dwarf-static)
@@ -395,7 +404,6 @@ if(BUILD_SHARED)
 endif()
 
 add_subdirectory(libdwarf)
-add_subdirectory(dwarfdump)
 if ( BUILD_DWARFGEN )
 	if ( DWARF_WITH_LIBELF )
 	    add_subdirectory(dwarfgen)
diff --git a/dwarfdump/CMakeLists.txt b/dwarfdump/CMakeLists.txt
index b94f6c8..6bdb57f 100644
--- a/dwarfdump/CMakeLists.txt
+++ b/dwarfdump/CMakeLists.txt
@@ -31,7 +31,7 @@ set_source_group(CONFIGURATION_FILES "Configuration Files"
   ${CMAKE_SOURCE_DIR}/config.h.in.cmake
   ${CMAKE_BINARY_DIR}/config.h)
 	
-add_executable(dwarfdump ${SOURCES} ${HEADERS} ${CONFIGURATION_FILES})
+add_executable(dwarfdump ${SOURCES} ${HEADERS} ${CONFIGURATION_FILES} ${libelf_LIBRARIES})
 
 set_folder(dwarfdump dwarfdump)
 
diff --git a/dwarfgen/CMakeLists.txt b/dwarfgen/CMakeLists.txt
index 488b820..5bde9eb 100644
--- a/dwarfgen/CMakeLists.txt
+++ b/dwarfgen/CMakeLists.txt
@@ -20,16 +20,16 @@ set_folder(dwarfgen dwarfgen)
 
 target_compile_options(dwarfgen PRIVATE ${DW_FWALLXX})
 
-target_link_libraries(dwarfgen PRIVATE ${dwarf-target} ${DW_FZLIB})
+target_link_libraries(dwarfgen PRIVATE ${dwarf-target} ${DW_FZLIB} ${libelf_LIBRARIES})
 
 set(SUFFIX $<$<EQUAL:${CMAKE_SIZEOF_VOID_P},8>:64>)
 set(LIBDIR lib${SUFFIX})
 set(BINDIR bin${SUFFIX})
 
 install(TARGETS dwarfgen DESTINATION
-	RUNTIME DESTINATION ${BINDIR}		
-	LIBRARY DESTINATION ${LIBDIR}
-	ARCHIVE DESTINATION ${LIBDIR})
+	RUNTIME DESTINATION bin
+	LIBRARY DESTINATION lib
+	ARCHIVE DESTINATION lib)
 
 #install(FILES dwarfgen.conf DESTINATION lib)
 
diff --git a/libdwarf/CMakeLists.txt b/libdwarf/CMakeLists.txt
index c610522..3a69533 100644
--- a/libdwarf/CMakeLists.txt
+++ b/libdwarf/CMakeLists.txt
@@ -81,12 +81,9 @@ foreach(i RANGE ${targetCount})
             ${GENNAMES_OUTPUT} ${CONFIGURATION_FILES})
 
 	set_folder(${target} libdwarf)
-	target_include_directories(${target} PUBLIC 
-            ${LIBELF_INCLUDE_DIRS})
-        target_compile_options(${target} PRIVATE ${DW_FWALL})
 	msvc_posix(${target})
 
-	target_link_libraries(${target} PUBLIC ${LIBELF_LIBRARIES})
+    target_link_libraries(${target} PUBLIC ${libelf_LIBRARIES} ${ZLIB_LIBRARIES})
 	
 	set_target_properties(${target} PROPERTIES OUTPUT_NAME dwarf)
 
@@ -95,15 +92,14 @@ foreach(i RANGE ${targetCount})
 	set(BINDIR bin${SUFFIX})
 
 	install(TARGETS ${target}
-		RUNTIME DESTINATION ${BINDIR}		
-		LIBRARY DESTINATION ${LIBDIR}
-		ARCHIVE DESTINATION ${LIBDIR})
+		RUNTIME DESTINATION bin
+    LIBRARY DESTINATION lib
+		ARCHIVE DESTINATION lib)
 endforeach()
 
-if(UNIX AND BUILD_SHARED)
-	target_link_libraries(dwarf-shared PUBLIC z)
-endif()
-		
+install(FILES ${CMAKE_BINARY_DIR}/libdwarf/libdwarf.h DESTINATION include)
+install(FILES dwarf.h DESTINATION include)
+
 if (DO_TESTING)
     set_source_group(TESTLEB "Source Files" dwarf_leb_test.c 
         dwarf_leb.c pro_encode_nm.c )
