diff --git a/cmake/XalanDLL.cmake b/cmake/XalanDLL.cmake
index 858c00c0d..2ae023913 100644
--- a/cmake/XalanDLL.cmake
+++ b/cmake/XalanDLL.cmake
@@ -34,3 +34,8 @@ endif()
 if(MSVC)
   add_definitions(-D_CRT_SECURE_NO_WARNINGS=1 -D_SCL_SECURE_NO_WARNINGS=1)
 endif()
+
+set(XALAN_STATIC_LIBRARY)
+if(NOT BUILD_SHARED_LIBS)
+  set(XALAN_STATIC_LIBRARY 1)
+endif()
diff --git a/src/xalanc/Include/PlatformDefinitions.hpp.in b/src/xalanc/Include/PlatformDefinitions.hpp.in
index 57829a706..5a55b43a5 100644
--- a/src/xalanc/Include/PlatformDefinitions.hpp.in
+++ b/src/xalanc/Include/PlatformDefinitions.hpp.in
@@ -59,7 +59,9 @@
 //  These defines provide the platform specific keywords that they need
 //  to do this.
 // ---------------------------------------------------------------------------
-#if defined(_MSC_VER)
+#cmakedefine XALAN_STATIC_LIBRARY 1
+
+#if defined(_MSC_VER) && !defined(XALAN_STATIC_LIBRARY)
 #define XALAN_PLATFORM_EXPORT     __declspec(dllexport)
 #define XALAN_PLATFORM_IMPORT     __declspec(dllimport)
 #define XALAN_PLATFORM_EXPORT_FUNCTION(T) XALAN_PLATFORM_EXPORT T
diff --git a/src/xalanc/XPathCAPI/XPathCAPI.h b/src/xalanc/XPathCAPI/XPathCAPI.h
index b754c0054..8f0958bb5 100644
--- a/src/xalanc/XPathCAPI/XPathCAPI.h
+++ b/src/xalanc/XPathCAPI/XPathCAPI.h
@@ -29,8 +29,8 @@
 
 #else
 
-#define XALAN_XPATHCAPI_EXPORT __declspec(dllimport)
-#define XALAN_XPATHCAPI_EXPORT_FUNCTION(T) T __declspec(dllimport)
+#define XALAN_XPATHCAPI_EXPORT
+#define XALAN_XPATHCAPI_EXPORT_FUNCTION(T) T
 
 #endif
 
