diff --git a/CMakeLists.txt b/CMakeLists.txt
index 72261da..8e02126 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2,6 +2,11 @@ cmake_minimum_required(VERSION 3.5)
 
 project(hdr_histogram VERSION 0.10.0 LANGUAGES C)
 
+include(${PROJECT_BINARY_DIR}/../conanbuildinfo.cmake)
+conan_basic_setup()
+
+set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
+
 include(GNUInstallDirs)
 include(CMakePackageConfigHelpers)
 
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 9dbb311..f5968c7 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -33,22 +33,15 @@ function(hdr_histogram_add_library NAME LIBRARY_TYPE DO_INSTALL)
         ${HDR_HISTOGRAM_SOURCES}
         ${HDR_HISTOGRAM_PRIVATE_HEADERS}
         ${HDR_HISTOGRAM_PUBLIC_HEADERS})
-    target_link_libraries(${NAME}
-        PRIVATE
-            ZLIB::ZLIB
-            Threads::Threads
-            $<$<BOOL:${HAVE_LIBM}>:m>
-            $<$<BOOL:${HAVE_LIBRT}>:rt>
-            $<$<BOOL:${WIN32}>:ws2_32>)
-    target_include_directories(${NAME}
-        PUBLIC
-            $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
-            $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
+    target_link_libraries(${NAME} ${CONAN_LIBS})
+    target_include_directories(${NAME} PRIVATE ${PROJECT_SOURCE_DIR}/src)
     if(DO_INSTALL)
         install(
             TARGETS ${NAME}
             EXPORT ${PROJECT_NAME}-targets
-            DESTINATION ${CMAKE_INSTALL_LIBDIR})
+            ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+            LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+            RUNTIME DESTINATION bin)
     endif()
 endfunction()
 
