diff --git a/CMakeLists.txt b/CMakeLists.txt
index 19e6c2d..feda9a4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -276,10 +276,6 @@ if(TARGET_VK)
     set(MAGNUM_TARGET_VK 1)
 endif()
 
-if(CORRADE_TARGET_EMSCRIPTEN)
-    include(UseEmscripten)
-endif()
-
 if(WITH_OPENGLTESTER)
     if(MAGNUM_TARGET_HEADLESS OR CORRADE_TARGET_EMSCRIPTEN OR CORRADE_TARGET_ANDROID)
         set(WITH_WINDOWLESSEGLAPPLICATION ON)
