--- a/lib/jxl.cmake
+++ b/lib/jxl.cmake
@@ -288,10 +288,8 @@ set(JPEGXL_INTERNAL_SOURCES_ENC
 )
 
 set(JPEGXL_INTERNAL_LIBS
-  brotlicommon-static
-  brotlienc-static
-  brotlidec-static
-  hwy
+  CONAN_PKG::brotli
+  CONAN_PKG::highway
   Threads::Threads
   ${CMAKE_DL_LIBS}
 )
@@ -300,7 +298,7 @@ if (JPEGXL_ENABLE_SKCMS)
   list(APPEND JPEGXL_INTERNAL_FLAGS -DJPEGXL_ENABLE_SKCMS=1)
   list(APPEND JPEGXL_INTERNAL_LIBS skcms)
 else ()
-  list(APPEND JPEGXL_INTERNAL_LIBS lcms2)
+  list(APPEND JPEGXL_INTERNAL_LIBS CONAN_PKG::lcms)
 endif ()
 
 set(OBJ_COMPILE_DEFINITIONS
@@ -321,8 +319,9 @@ set_property(TARGET jxl_dec-obj PROPERTY POSITION_INDEPENDENT_CODE ON)
 target_include_directories(jxl_dec-obj PUBLIC
   ${PROJECT_SOURCE_DIR}
   ${CMAKE_CURRENT_SOURCE_DIR}/include
-  $<TARGET_PROPERTY:hwy,INTERFACE_INCLUDE_DIRECTORIES>
-  $<TARGET_PROPERTY:brotlicommon-static,INTERFACE_INCLUDE_DIRECTORIES>
+  $<TARGET_PROPERTY:CONAN_PKG::highway,INTERFACE_INCLUDE_DIRECTORIES>
+  $<TARGET_PROPERTY:CONAN_PKG::brotli,INTERFACE_INCLUDE_DIRECTORIES>
+  $<TARGET_PROPERTY:CONAN_PKG::lcms,INTERFACE_INCLUDE_DIRECTORIES>
 )
 target_compile_definitions(jxl_dec-obj PUBLIC
   ${OBJ_COMPILE_DEFINITIONS}
@@ -336,8 +335,9 @@ set_property(TARGET jxl_enc-obj PROPERTY POSITION_INDEPENDENT_CODE ON)
 target_include_directories(jxl_enc-obj PUBLIC
   ${PROJECT_SOURCE_DIR}
   ${CMAKE_CURRENT_SOURCE_DIR}/include
-  $<TARGET_PROPERTY:hwy,INTERFACE_INCLUDE_DIRECTORIES>
-  $<TARGET_PROPERTY:brotlicommon-static,INTERFACE_INCLUDE_DIRECTORIES>
+  $<TARGET_PROPERTY:CONAN_PKG::highway,INTERFACE_INCLUDE_DIRECTORIES>
+  $<TARGET_PROPERTY:CONAN_PKG::brotli,INTERFACE_INCLUDE_DIRECTORIES>
+  $<TARGET_PROPERTY:CONAN_PKG::lcms,INTERFACE_INCLUDE_DIRECTORIES>
 )
 target_compile_definitions(jxl_enc-obj PUBLIC
   ${OBJ_COMPILE_DEFINITIONS}
@@ -353,10 +353,10 @@ if (JPEGXL_ENABLE_SKCMS)
   )
 else ()
   target_include_directories(jxl_enc-obj PRIVATE
-    $<TARGET_PROPERTY:lcms2,INCLUDE_DIRECTORIES>
+    $<TARGET_PROPERTY:CONAN_PKG::lcms,INCLUDE_DIRECTORIES>
   )
   target_include_directories(jxl_dec-obj PRIVATE
-    $<TARGET_PROPERTY:lcms2,INCLUDE_DIRECTORIES>
+    $<TARGET_PROPERTY:CONAN_PKG::lcms,INCLUDE_DIRECTORIES>
   )
 endif ()
 
@@ -392,7 +392,7 @@ add_library(jxl_dec-static STATIC
   $<TARGET_OBJECTS:jxl_dec-obj>
 )
 target_link_libraries(jxl_dec-static
-  PUBLIC ${JPEGXL_COVERAGE_FLAGS} ${JPEGXL_INTERNAL_LIBS} hwy)
+  PUBLIC ${JPEGXL_COVERAGE_FLAGS} ${JPEGXL_INTERNAL_LIBS})
 target_include_directories(jxl_dec-static PUBLIC
   "${PROJECT_SOURCE_DIR}"
   "${CMAKE_CURRENT_SOURCE_DIR}/include"
@@ -407,7 +407,7 @@ add_library(jxl-static STATIC
   $<TARGET_OBJECTS:jxl_dec-obj>
 )
 target_link_libraries(jxl-static
-  PUBLIC ${JPEGXL_COVERAGE_FLAGS} ${JPEGXL_INTERNAL_LIBS} hwy)
+  PUBLIC ${JPEGXL_COVERAGE_FLAGS} ${JPEGXL_INTERNAL_LIBS})
 target_include_directories(jxl-static PUBLIC
   "${PROJECT_SOURCE_DIR}"
   "${CMAKE_CURRENT_SOURCE_DIR}/include"
