--- CMakeLists.txt
+++ CMakeLists.txt
@@ -502,7 +502,7 @@
 # executables
 ########################################################################
 add_executable(
+    zmakecert EXCLUDE_FROM_ALL
-    zmakecert
     "${SOURCE_DIR}/src/zmakecert.c"
 )
 if (TARGET czmq)
@@ -522,9 +522,9 @@
     ${OPTIONAL_LIBRARIES_STATIC}
 )
 endif()
-install(TARGETS zmakecert
-    RUNTIME DESTINATION bin
-)
+#install(TARGETS zmakecert
+#    RUNTIME DESTINATION bin
+#)
 add_executable(
     zsp
     "${SOURCE_DIR}/src/zsp.c"
