--- Findlibcurl.cmake
+++ Findlibcurl.cmake
@@ -3,7 +3,7 @@
 #  Read the zproject/README.md for information about making permanent changes. #
 ################################################################################

-if (NOT MSVC)
+if (0)
     include(FindPkgConfig)
     pkg_check_modules(PC_LIBCURL "libcurl")
     if (PC_LIBCURL_FOUND)
@@ -17,17 +17,9 @@
     endif(PC_LIBCURL_FOUND)
 endif (NOT MSVC)

-find_path (
-    LIBCURL_INCLUDE_DIRS
-    NAMES curl/curl.h
-    HINTS ${PC_LIBCURL_INCLUDE_HINTS}
-)
-
-find_library (
-    LIBCURL_LIBRARIES
-    NAMES curl
-    HINTS ${PC_LIBCURL_LIBRARY_HINTS}
-)
+set(LIBCURL_INCLUDE_DIRS ${CONAN_INCLUDE_DIRS_CURL})
+
+set(LIBCURL_LIBRARIES CONAN_PKG::CURL)

 include(FindPackageHandleStandardArgs)

--- Findlibmicrohttpd.cmake	2019-02-10 18:47:44.000000000 +0100
+++ Findlibmicrohttpd.cmake	2019-12-14 14:54:53.667244809 +0100
@@ -3,7 +3,7 @@
 #  Read the zproject/README.md for information about making permanent changes. #
 ################################################################################

-if (NOT MSVC)
+if (0)
     include(FindPkgConfig)
     pkg_check_modules(PC_LIBMICROHTTPD "libmicrohttpd")
     if (PC_LIBMICROHTTPD_FOUND)
@@ -17,17 +17,9 @@
     endif(PC_LIBMICROHTTPD_FOUND)
 endif (NOT MSVC)

-find_path (
-    LIBMICROHTTPD_INCLUDE_DIRS
-    NAMES microhttpd.h
-    HINTS ${PC_LIBMICROHTTPD_INCLUDE_HINTS}
-)
-
-find_library (
-    LIBMICROHTTPD_LIBRARIES
-    NAMES microhttpd
-    HINTS ${PC_LIBMICROHTTPD_LIBRARY_HINTS}
-)
+set(LIBMICROHTTPD_INCLUDE_DIRS ${CONAN_INCLUDE_DIRS_LIBMICROHTTPD})
+link_directories(${CONAN_LIB_DIRS_LIBMICROHTTPD})
+set(LIBLIBMICROHTTPD_LIBRARIES_DIRS ${CONAN_LIBS_LIBLIBMICROHTTPD} ${CONAN_SYSTEM_LIBS_LIBLIBMICROHTTPD})

 include(FindPackageHandleStandardArgs)

--- Findlibsodium.cmake	2019-02-10 18:47:44.000000000 +0100
+++ Findlibsodium.cmake	2019-12-14 14:54:53.667244809 +0100
@@ -3,7 +3,7 @@
 #  Please refer to the README for information about making permanent changes.  #
 ################################################################################

-if (NOT MSVC)
+if (0)
     include(FindPkgConfig)
     pkg_check_modules(PC_LIBSODIUM "libsodium")
     if (NOT PC_LIBSODIUM_FOUND)
@@ -18,17 +18,10 @@
     endif(PC_LIBSODIUM_FOUND)
 endif (NOT MSVC)

-find_path (
-    LIBSODIUM_INCLUDE_DIRS
-    NAMES sodium.h
-    HINTS ${PC_LIBSODIUM_INCLUDE_HINTS}
-)
-
-find_library (
-    LIBSODIUM_LIBRARIES
-    NAMES sodium
-    HINTS ${PC_LIBSODIUM_LIBRARY_HINTS}
-)
+set(LIBSODIUM_INCLUDE_DIRS ${CONAN_INCLUDE_DIRS_LIBSODIUM})
+link_directories(${CONAN_LIB_DIRS_LIBSODIUM})
+set(LIBSODIUM_LIBRARIES ${CONAN_LIBS_LIBSODIUM} ${CONAN_SYSTEM_LIBS_LIBSODIUM})
+add_definitions(${CONAN_DEFINES_LIBSODIUM})

 include(FindPackageHandleStandardArgs)

--- Findlibzmq.cmake	2019-02-10 18:47:44.000000000 +0100
+++ Findlibzmq.cmake	2019-12-14 14:54:53.667244809 +0100
@@ -3,7 +3,7 @@
 #  Read the zproject/README.md for information about making permanent changes. #
 ################################################################################

-if (NOT MSVC)
+if (0)
     include(FindPkgConfig)
     pkg_check_modules(PC_LIBZMQ "libzmq")
     if (PC_LIBZMQ_FOUND)
@@ -17,19 +17,16 @@
     endif(PC_LIBZMQ_FOUND)
 endif (NOT MSVC)

-find_path (
-    LIBZMQ_INCLUDE_DIRS
-    NAMES zmq.h
-    HINTS ${PC_LIBZMQ_INCLUDE_HINTS}
-)
-
-find_library (
-    LIBZMQ_LIBRARIES
-    NAMES zmq
-    HINTS ${PC_LIBZMQ_LIBRARY_HINTS}
-)
-
+set(LIBZMQ_INCLUDE_DIRS ${CONAN_INCLUDE_DIRS_ZEROMQ})
+link_directories(${CONAN_LIB_DIRS_ZEROMQ})
+set(LIBZMQ_LIBRARIES ${CONAN_LIBS_ZEROMQ} ${CONAN_SYSTEM_LIBS_ZEROMQ})
 include(FindPackageHandleStandardArgs)
+add_definitions(${CONAN_DEFINES_ZEROMQ})
+
+if(DEFINED CONAN_INCLUDE_DIRS_LIBSODIUM)
+    find_package(libsodium REQUIRED)
+    list(APPEND LIBZMQ_LIBRARIES ${LIBSODIUM_LIBRARIES})
+endif()

 find_package_handle_standard_args(
     LIBZMQ
--- Findlz4.cmake	2019-02-10 18:47:44.000000000 +0100
+++ Findlz4.cmake	2019-12-14 14:54:53.667244809 +0100
@@ -3,7 +3,7 @@
 #  Read the zproject/README.md for information about making permanent changes. #
 ################################################################################

-if (NOT MSVC)
+if (0)
     include(FindPkgConfig)
     pkg_check_modules(PC_LZ4 "liblz4")
     if (PC_LZ4_FOUND)
@@ -17,17 +17,10 @@
     endif(PC_LZ4_FOUND)
 endif (NOT MSVC)

-find_path (
-    LZ4_INCLUDE_DIRS
-    NAMES lz4.h
-    HINTS ${PC_LZ4_INCLUDE_HINTS}
-)
-
-find_library (
-    LZ4_LIBRARIES
-    NAMES lz4
-    HINTS ${PC_LZ4_LIBRARY_HINTS}
-)
+set(LZ4_INCLUDE_DIRS ${CONAN_INCLUDE_DIRS_LZ4})
+link_directories(${CONAN_LIB_DIRS_LZ4})
+set(LZ4_LIBRARIES ${CONAN_LIBS_LZ4} ${CONAN_SYSTEM_LIBS_LZ4})
+add_definitions(${CONAN_DEFINES_LZ4})

 include(FindPackageHandleStandardArgs)

--- Finduuid.cmake	2019-02-10 18:47:44.000000000 +0100
+++ Finduuid.cmake	2019-12-14 14:54:53.668244813 +0100
@@ -3,7 +3,7 @@
 #  Read the zproject/README.md for information about making permanent changes. #
 ################################################################################

-if (NOT MSVC)
+if (0)
     include(FindPkgConfig)
     pkg_check_modules(PC_UUID "uuid")
     if (PC_UUID_FOUND)
@@ -17,17 +17,10 @@
     endif(PC_UUID_FOUND)
 endif (NOT MSVC)

-find_path (
-    UUID_INCLUDE_DIRS
-    NAMES uuid/uuid.h
-    HINTS ${PC_UUID_INCLUDE_HINTS}
-)
-
-find_library (
-    UUID_LIBRARIES
-    NAMES uuid
-    HINTS ${PC_UUID_LIBRARY_HINTS}
-)
+set(UUID_INCLUDE_DIRS ${CONAN_INCLUDE_DIRS_LIBUUID})
+link_directories(${CONAN_LIB_DIRS_LIBUUID})
+set(UUID_LIBRARIES ${CONAN_LIBS_LIBUUID} ${CONAN_SYSTEM_LIBS_LIBUUID})
+add_definitions(${CONAN_DEFINES_LIBUUID})

 include(FindPackageHandleStandardArgs)

