diff --git a/src/zsock.c b/src/zsock.c
index 46bbb63..36ad5e0 100644
--- a/src/zsock.c
+++ b/src/zsock.c
@@ -2376,6 +2376,7 @@ zsock_test (bool verbose)
     zstr_free (&addr);
     zstr_free (&message);
 
+#ifndef __APPLE__
     // ZMQ_DGRAM ipv4 multicast test
     zsock_t* mdgramr = zsock_new_dgram ("udp://225.25.25.25:7777");
     assert (mdgramr);
@@ -2389,6 +2390,7 @@ zsock_test (bool verbose)
 
     char *mdmessage, *maddr;
 
+    // this call blocks forever on MacOS
     zmsg_t *mdmsg = zmsg_recv( mdgramr );
     assert (mdmsg);
     maddr = zmsg_popstr (mdmsg);
@@ -2400,6 +2402,7 @@ zsock_test (bool verbose)
     zstr_free (&mdmessage);
     zstr_free (&maddr);
     zstr_free (&mdmessage);
+#endif
 
 //    // ipv6 (not supported yet)
 //    zsys_set_ipv6(1);
diff --git a/src/zsys.c b/src/zsys.c
index 7c7301b..f384051 100644
--- a/src/zsys.c
+++ b/src/zsys.c
@@ -334,9 +334,7 @@ zsys_init (void)
         zsys_set_thread_name_prefix (s_thread_name_prefix);
 
     if (getenv ("ZSYS_THREAD_NAME_PREFIX_STR"))
-        zsys_set_thread_name_prefix_str (getenv ("ZSYS_THREAD_NAME_PREFIX"));
-    else
-        zsys_set_thread_name_prefix_str (s_thread_name_prefix_str);
+       zsys_set_thread_name_prefix_str (getenv ("ZSYS_THREAD_NAME_PREFIX_STR"));
 
     return s_process_ctx;
 }
