diff --git a/a/CMakeLists.txt b/b/CMakeLists.txt
index d51cba0..d12a024 100644
--- a/a/CMakeLists.txt
+++ b/b/CMakeLists.txt
@@ -130,19 +130,19 @@ set(OPTIONAL_LIBRARIES_STATIC)
 ########################################################################
 # LIBZMQ dependency
 ########################################################################
-find_package(libzmq REQUIRED)
-IF (LIBZMQ_FOUND)
-    include_directories(${LIBZMQ_INCLUDE_DIRS})
-    list(APPEND MORE_LIBRARIES ${LIBZMQ_LIBRARIES})
+find_package(ZeroMQ REQUIRED)
+IF (ZeroMQ_FOUND)
+    include_directories(${ZeroMQ_INCLUDE_DIRS})
+    list(APPEND MORE_LIBRARIES ${ZeroMQ_LIBRARIES})
   IF (PC_LIBZMQ_FOUND)
       set(pkg_config_names_private "${pkg_config_names_private} libzmq")
       list(APPEND OPTIONAL_LIBRARIES_STATIC ${PC_LIBZMQ_STATIC_LDFLAGS})
   ELSE (PC_LIBZMQ_FOUND)
       set(pkg_config_libs_private "${pkg_config_libs_private} -lzmq")
   ENDIF (PC_LIBZMQ_FOUND)
-ELSE (LIBZMQ_FOUND)
+ELSE (ZeroMQ_FOUND)
     message( FATAL_ERROR "libzmq not found." )
-ENDIF (LIBZMQ_FOUND)
+ENDIF (ZeroMQ_FOUND)
 
 ########################################################################
 # UUID dependency
@@ -419,6 +419,7 @@ if (CZMQ_BUILD_SHARED)
   ENDIF(APPLE)
 
   set_target_properties (czmq PROPERTIES
+    LINKER_LANGUAGE CXX
     PUBLIC_HEADER "${public_headers}"
     DEFINE_SYMBOL "CZMQ_EXPORTS"
     SOVERSION "4"
@@ -544,6 +545,7 @@ install(FILES ${CMAKE_CURRENT_BINARY_DIR}/czmqConfig.cmake
 ########################################################################
 # executables
 ########################################################################
+if(0) # disable executables
 add_executable(
     zmakecert
     "${SOURCE_DIR}/src/zmakecert.c"
@@ -631,10 +633,12 @@ target_link_libraries(
     ${OPTIONAL_LIBRARIES_STATIC}
 )
 endif()
+endif() # disable executables
 
 ########################################################################
 # tests
 ########################################################################
+if(0) # disable tests
 set(CLASSTEST_TIMEOUT 60 CACHE STRING "Timeout of the selftest of a class")
 set(TOTAL_TIMEOUT 600 CACHE STRING "Timout of the total testsuite")
 
@@ -743,6 +747,7 @@ foreach(TEST_CLASS ${TEST_CLASSES})
 endforeach(TEST_CLASS)
 
 include(CTest)
+endif() # disable tests
 
 ########################################################################
 # cleanup
