diff --git a/CMakeLists.txt b/CMakeLists.txt
index de4e1509..2098507d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -9,7 +9,6 @@
 cmake_minimum_required(VERSION 2.8.8)
 project(czmq)
 enable_language(C)
-enable_testing()
 
 set(SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}")
 set(CMAKE_EXPORT_COMPILE_COMMANDS 1)
@@ -113,6 +112,10 @@ if (CYGWIN)
     set(MORE_LIBRARIES)
 endif()
 
+if (UNIX AND NOT APPLE)
+    set(MORE_LIBRARIES m)
+endif()
+
 list(APPEND CMAKE_MODULE_PATH "${SOURCE_DIR}")
 set(OPTIONAL_LIBRARIES)
 set(OPTIONAL_LIBRARIES_STATIC)
@@ -525,168 +528,6 @@ endif()
 install(TARGETS zmakecert
     RUNTIME DESTINATION bin
 )
-add_executable(
-    zsp
-    "${SOURCE_DIR}/src/zsp.c"
-)
-if (TARGET czmq)
-target_link_libraries(
-    zsp
-    czmq
-    ${LIBZMQ_LIBRARIES}
-    ${OPTIONAL_LIBRARIES}
-)
-endif()
-if (NOT TARGET czmq AND TARGET czmq-static)
-target_link_libraries(
-    zsp
-    czmq-static
-    ${LIBZMQ_LIBRARIES}
-    ${OPTIONAL_LIBRARIES}
-    ${OPTIONAL_LIBRARIES_STATIC}
-)
-endif()
-add_executable(
-    test_randof
-    "${SOURCE_DIR}/src/test_randof.c"
-)
-if (TARGET czmq)
-target_link_libraries(
-    test_randof
-    czmq
-    ${LIBZMQ_LIBRARIES}
-    ${OPTIONAL_LIBRARIES}
-)
-endif()
-if (NOT TARGET czmq AND TARGET czmq-static)
-target_link_libraries(
-    test_randof
-    czmq-static
-    ${LIBZMQ_LIBRARIES}
-    ${OPTIONAL_LIBRARIES}
-    ${OPTIONAL_LIBRARIES_STATIC}
-)
-endif()
-add_executable(
-    czmq_selftest
-    "${SOURCE_DIR}/src/czmq_selftest.c"
-)
-if (TARGET czmq)
-target_link_libraries(
-    czmq_selftest
-    czmq
-    ${LIBZMQ_LIBRARIES}
-    ${OPTIONAL_LIBRARIES}
-)
-endif()
-if (NOT TARGET czmq AND TARGET czmq-static)
-target_link_libraries(
-    czmq_selftest
-    czmq-static
-    ${LIBZMQ_LIBRARIES}
-    ${OPTIONAL_LIBRARIES}
-    ${OPTIONAL_LIBRARIES_STATIC}
-)
-endif()
-
-########################################################################
-# tests
-########################################################################
-set(CLASSTEST_TIMEOUT 60 CACHE STRING "Timeout of the selftest of a class")
-set(TOTAL_TIMEOUT 600 CACHE STRING "Timout of the total testsuite")
-
-if(UNIX)
-    find_program(MEMORYCHECK_COMMAND valgrind)
-    set(MEMORYCHECK_COMMAND_OPTIONS "--leak-check=full --show-reachable=yes
-                                     --error-exitcode=1
-                                     --suppressions=src/.valgrind.supp")
-endif()
-
-set(TEST_CLASSES
-    zactor
-    zarmour
-    zcert
-    zcertstore
-    zchunk
-    zclock
-    zconfig
-    zdigest
-    zdir
-    zdir_patch
-    zfile
-    zframe
-    zhash
-    zhashx
-    ziflist
-    zlist
-    zlistx
-    zloop
-    zmsg
-    zpoller
-    zsock
-    zstr
-    zsys
-    zuuid
-    zauth
-    zbeacon
-    zgossip
-    zmonitor
-    zproxy
-    zrex
-)
-
-IF (ENABLE_DRAFTS)
-    list (APPEND TEST_CLASSES
-    zargs
-    zproc
-    ztimerset
-    ztrie
-    zhttp_client
-    zhttp_server
-    zhttp_server_options
-    zhttp_request
-    zhttp_response
-    )
-ENDIF (ENABLE_DRAFTS)
-
-IF (ENABLE_DRAFTS)
-    list (APPEND TEST_CLASSES
-    private_classes
-    )
-ENDIF (ENABLE_DRAFTS)
-
-
-add_custom_target(
-    copy-selftest-ro ALL
-    COMMAND ${CMAKE_COMMAND} -E copy_directory ${CMAKE_SOURCE_DIR}/src/selftest-ro ${CMAKE_BINARY_DIR}/src/selftest-ro
-)
-
-add_custom_target(
-    make-selftest-rw ALL
-    COMMAND ${CMAKE_COMMAND} -E make_directory ${CMAKE_BINARY_DIR}/src/selftest-rw
-)
-
-set_directory_properties(
-    PROPERTIES
-    ADDITIONAL_MAKE_CLEAN_FILES "${CMAKE_BINARY_DIR}/src/selftest-ro;${CMAKE_BINARY_DIR}/src/selftest-rw"
-)
-
-foreach(TEST_CLASS ${TEST_CLASSES})
-    add_test(
-        NAME ${TEST_CLASS}
-        COMMAND czmq_selftest --continue --verbose --test ${TEST_CLASS}
-    )
-    set_tests_properties(
-        ${TEST_CLASS}
-        PROPERTIES TIMEOUT ${CLASSTEST_TIMEOUT}
-    )
-    set_tests_properties(
-        ${TEST_CLASS}
-        PROPERTIES DEPENDS "copy-selftest-ro;make-selftest-rw"
-    )
-endforeach(TEST_CLASS)
-
-include(CTest)
 
 ########################################################################
 # cleanup
@@ -707,7 +548,6 @@ set(cmake_generated ${CMAKE_BINARY_DIR}/CMakeCache.txt
                     ${CMAKE_BINARY_DIR}/src/czmq_selftest
                     ${CMAKE_BINARY_DIR}/src/zmakecert
                     ${CMAKE_BINARY_DIR}/src/zsp
-                    ${CMAKE_BINARY_DIR}/src/test_randof
                     ${CMAKE_BINARY_DIR}/src/czmq_selftest
 )
 
