diff --git a/gst/matroska/meson.build b/gst/matroska/meson.build
--- a/gst/matroska/meson.build
+++ b/gst/matroska/meson.build
@@ -12,7 +12,7 @@
   'lzo.c',
 ]
 
-bz2_dep = cc.find_library('bz2', required : get_option('bz2'))
+bz2_dep = dependency('bzip2', required : get_option('bz2'))
 cdata.set('HAVE_BZ2', bz2_dep.found() and cc.has_header('bzlib.h'))
 
 gstmatroska = library('gstmatroska',
diff --git a/sys/v4l2/meson.build b/sys/v4l2/meson.build
--- a/sys/v4l2/meson.build
+++ b/sys/v4l2/meson.build
@@ -41,10 +41,8 @@
   have_v4l2 = false
   message('V4L2 plugin is disabled')
 else
-  have_v4l2 = cc.has_header('linux/videodev2.h') or cc.has_header('sys/videodev2.h') or cc.has_header('sys/videoio.h')
-  if v4l2.enabled() and not have_v4l2
-    error('V4L2 is requested but headers were not found')
-  endif
+  libv4l2_dep = dependency('libv4l2', required: v4l2.enabled())
+  have_v4l2 = true
 endif
 
 cdata.set('GST_V4L2_ENABLE_PROBE', get_option('v4l2-probe'))
