--- CMakeLists.txt
+++ CMakeLists.txt
@@ -5,12 +5,14 @@
 include(CTest)
 include(GNUInstallDirs)
 
-add_library(kcp STATIC ikcp.c)
+add_library(kcp ikcp.c)
 
 install(FILES ikcp.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
 
 install(TARGETS kcp
     EXPORT kcp-targets
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
     ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
     INCLUDES DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
 )
