--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -19,7 +19,6 @@ set(DEFAULT_BUILD_TYPE Release)
 # Require CMake 3.13+ with VS generator for complete support of VS versions
 # and support by AppVeyor.
 if(${CMAKE_GENERATOR} MATCHES "Visual Studio")
-  cmake_minimum_required(VERSION 3.13 FATAL_ERROR)
 endif()

 list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake")
@@ -224,7 +223,7 @@ add_subdirectory(src)

 if(BUILD_SHARED_LIBS)
   target_compile_definitions(geos
-    PRIVATE $<$<CXX_COMPILER_ID:MSVC>:GEOS_DLL_EXPORT>)
+    PRIVATE $<IF:$<CXX_COMPILER_ID:MSVC>,GEOS_DLL_EXPORT,DLL_EXPORT>) # fix to not build inline obj if shared and inline with MinGW

   set_target_properties(geos PROPERTIES VERSION ${GEOS_VERSION})
   set_target_properties(geos PROPERTIES SOVERSION ${GEOS_VERSION})
@@ -238,7 +237,7 @@ target_link_libraries(geos_c PRIVATE geos)

 if(BUILD_SHARED_LIBS)
   target_compile_definitions(geos_c
-    PRIVATE $<$<CXX_COMPILER_ID:MSVC>:GEOS_DLL_EXPORT>)
+    PRIVATE $<IF:$<CXX_COMPILER_ID:MSVC>,GEOS_DLL_EXPORT,DLL_EXPORT>) # fix to not build inline obj if shared and inline with MinGW

   set_target_properties(geos_c PROPERTIES VERSION ${CAPI_VERSION})
   if(NOT WIN32)
@@ -259,12 +258,10 @@ endif()
 #-----------------------------------------------------------------------------
 # Benchmarks
 #-----------------------------------------------------------------------------
-add_subdirectory(benchmarks)

 #-----------------------------------------------------------------------------
 # Documentation/Examples
 #-----------------------------------------------------------------------------
-add_subdirectory(doc)

 #-----------------------------------------------------------------------------
 # Install and export targets - support 'make install' or equivalent
@@ -281,7 +278,7 @@ configure_file(cmake/geos-config.cmake

 install(TARGETS geos geos_cxx_flags
   EXPORT geos-targets
-  LIBRARY DESTINATION lib NAMELINK_SKIP
+  LIBRARY DESTINATION lib
   ARCHIVE DESTINATION lib
   RUNTIME DESTINATION bin
   INCLUDES DESTINATION include
@@ -319,7 +316,6 @@ endif()
 install(FILES "${CMAKE_CURRENT_BINARY_DIR}/capi/geos_c.h"
   DESTINATION include)

-add_subdirectory(tools)

 #-----------------------------------------------------------------------------
 # Uninstall
@@ -379,9 +375,6 @@ add_custom_target(check COMMAND ${CMAKE_BUILD_TOOL} test)
 # "make distcheck" workalike
 #-----------------------------------------------------------------------------
 if(NOT _is_multi_config_generator)
-  find_package(MakeDistCheck)
-  AddMakeDistCheck()
-  message(STATUS "GEOS: Configured 'distcheck' target")
 endif()

 unset(_is_multi_config_generator)
