--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -19,7 +19,6 @@ set(DEFAULT_BUILD_TYPE Release)
 # Require CMake 3.13+ with VS generator for complete support of VS versions
 # and support by AppVeyor.
 if(${CMAKE_GENERATOR} MATCHES "Visual Studio")
-  cmake_minimum_required(VERSION 3.13 FATAL_ERROR)
 endif()
 
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake")
@@ -290,7 +289,6 @@ endif()
 #-----------------------------------------------------------------------------
 # Documentation/Examples
 #-----------------------------------------------------------------------------
-add_subdirectory(doc)
 
 #-----------------------------------------------------------------------------
 # Install and export targets - support 'make install' or equivalent
@@ -346,7 +344,6 @@ endif()
 install(FILES "${CMAKE_CURRENT_BINARY_DIR}/capi/geos_c.h"
   DESTINATION include)
 
-add_subdirectory(tools)
 
 #-----------------------------------------------------------------------------
 # Uninstall
@@ -405,7 +402,7 @@ add_custom_target(check COMMAND ${CMAKE_BUILD_TOOL} test)
 #-----------------------------------------------------------------------------
 # "make distcheck" workalike
 #-----------------------------------------------------------------------------
-if(NOT _is_multi_config_generator)
+if(0)
   find_package(MakeDistCheck)
   AddMakeDistCheck()
   message(STATUS "GEOS: Configured 'distcheck' target")
