--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -441,7 +441,7 @@ add_custom_target(check COMMAND ${CMAKE_BUILD_TOOL} test)
 #-----------------------------------------------------------------------------
 # "make distcheck" workalike
 #-----------------------------------------------------------------------------
-if(NOT _is_multi_config_generator)
+if(0)
   find_package(MakeDistCheck)
   AddMakeDistCheck()
   message(STATUS "GEOS: Configured 'distcheck' target")
