diff --git source/CMakeLists.txt source/CMakeLists.txt
index e81afd730..6f53cd86c 100755
--- source/CMakeLists.txt
+++ source/CMakeLists.txt
@@ -48,7 +48,7 @@ else()
 endif()
 set(X86_ALIASES x86 i386 i686 x86_64 amd64)
 set(ARM_ALIASES armv6l armv7l)
-set(ARM64_ALIASES arm64 arm64e aarch64)
+set(ARM64_ALIASES arm64 ARM64 arm64e aarch64)
 set(RISCV64_ALIASES riscv64)
 list(FIND X86_ALIASES "${SYSPROC}" X86MATCH)
 list(FIND ARM_ALIASES "${SYSPROC}" ARMMATCH)
