diff --git a/CHOLMOD/CMakeLists.txt b/CHOLMOD/CMakeLists.txt
index 89640b443..b3cf64a21 100644
--- a/CHOLMOD/CMakeLists.txt
+++ b/CHOLMOD/CMakeLists.txt
@@ -523,10 +523,14 @@ if ( CHOLMOD_HAS_OPENMP )
     message ( STATUS "OpenMP C include:        ${OpenMP_C_INCLUDE_DIRS}" )
     message ( STATUS "OpenMP C flags:          ${OpenMP_C_FLAGS}" )
     if ( BUILD_SHARED_LIBS )
-        target_link_libraries ( CHOLMOD PRIVATE OpenMP::OpenMP_C )
+        target_link_libraries ( CHOLMOD PRIVATE    
+				$<$<LINK_LANGUAGE:CXX>:OpenMP::OpenMP_CXX>
+                $<$<LINK_LANGUAGE:C>:OpenMP::OpenMP_C>)
     endif ( )
     if ( BUILD_STATIC_LIBS )
-        target_link_libraries ( CHOLMOD_static PRIVATE OpenMP::OpenMP_C )
+        target_link_libraries ( CHOLMOD_static PRIVATE
+				$<$<LINK_LANGUAGE:CXX>:OpenMP::OpenMP_CXX>
+                $<$<LINK_LANGUAGE:C>:OpenMP::OpenMP_C>)
         list ( APPEND CHOLMOD_STATIC_LIBS ${OpenMP_C_LIBRARIES} )
     endif ( )
 else ( )
