--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -138,13 +138,10 @@
 # OPENMP is experimental. We experienced some slowdown with it
 option(G2O_USE_OPENMP "Build g2o with OpenMP support (EXPERIMENTAL)" OFF)
 if(G2O_USE_OPENMP)
-  find_package(OpenMP)
-  if(OPENMP_FOUND)
-    set (G2O_OPENMP 1)
-    set(g2o_C_FLAGS "${g2o_C_FLAGS} ${OpenMP_C_FLAGS}")
-    set(g2o_CXX_FLAGS "${g2o_CXX_FLAGS} -DEIGEN_DONT_PARALLELIZE ${OpenMP_CXX_FLAGS}")
-    message(STATUS "Compiling with OpenMP support")
-  endif(OPENMP_FOUND)
+  find_package(OpenMP REQUIRED)
+  set(G2O_OPENMP 1)
+  set(g2o_CXX_FLAGS "${g2o_CXX_FLAGS} -DEIGEN_DONT_PARALLELIZE")
+  link_libraries(OpenMP::OpenMP_CXX)
 endif(G2O_USE_OPENMP)
 
 # OpenGL is used in the draw actions for the different types, as well
