Enforces that either static or dynamic libs are built

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -598,7 +598,9 @@ if(CGNS_BUILD_SHARED)
 endif(CGNS_BUILD_SHARED)
 
 # Set the install path of the static library
+if(NOT CGNS_BUILD_SHARED)
 install(TARGETS cgns_static ARCHIVE DESTINATION lib)
+endif()
 # Set the install path of the shared library
 if(CGNS_BUILD_SHARED)
   # for windows, need to install both cgnsdll.dll and cgnsdll.lib
@@ -650,7 +652,6 @@ install(FILES ${headers}
 # Tools #
 #########
 
-add_subdirectory(tools)
 
 #########
 # Tests #
