diff --git a/src/polylineencoder.h b/src/polylineencoder.h
index 7a05c95..55adc6d 100644
--- a/src/polylineencoder.h
+++ b/src/polylineencoder.h
@@ -32,7 +32,7 @@
 #include <string>
 #include <vector>
 
-namespace gepaf
+namespace gepaf2
 {
 
 //! Implements Google's Encoded Polyline Algorithm Format
@@ -306,5 +306,10 @@ void PolylineEncoder<Digits>::clear()
 
 } // namespace
 
+namespace gepaf
+{
+    using PolylineEncoder = gepaf2::PolylineEncoder<5>;
+}
+
 #endif // POLYLINEENCODER_H
 
