diff --git OpenEXR/IlmImf/ImfDwaCompressor.cpp OpenEXR/IlmImf/ImfDwaCompressor.cpp
index 59d1d5d..585a3e6 100644
--- OpenEXR/IlmImf/ImfDwaCompressor.cpp
+++ OpenEXR/IlmImf/ImfDwaCompressor.cpp
@@ -159,6 +159,7 @@
 #include <limits>
 
 #include <cstddef>
+#include <cstdint>
 
 
 // Windows specific addition to prevent the indirect import of the redefined min/max macros
diff --git OpenEXR/IlmImf/ImfHuf.cpp OpenEXR/IlmImf/ImfHuf.cpp
index 271849b..165fac5 100644
--- OpenEXR/IlmImf/ImfHuf.cpp
+++ OpenEXR/IlmImf/ImfHuf.cpp
@@ -53,6 +53,7 @@
 #include <cstring>
 #include <cassert>
 #include <algorithm>
+#include <cstdint>
 
 
 using namespace std;
diff --git OpenEXR/IlmImf/ImfMisc.cpp OpenEXR/IlmImf/ImfMisc.cpp
index d2c8478..0451a33 100644
--- OpenEXR/IlmImf/ImfMisc.cpp
+++ OpenEXR/IlmImf/ImfMisc.cpp
@@ -54,6 +54,8 @@
 #include <ImfTileDescription.h>
 #include "ImfNamespace.h"
 
+#include <cstdint>
+
 OPENEXR_IMF_INTERNAL_NAMESPACE_SOURCE_ENTER
 
 using IMATH_NAMESPACE::Box2i;
