diff --git a/bin/nvcc.profile b/bin/nvcc.profile
index da6d811..ba0a952 100644
--- a/bin/nvcc.profile
+++ b/bin/nvcc.profile
@@ -7,7 +7,7 @@ PATH            += $(TOP)/$(_NVVM_BRANCH_)/bin;$(_HERE_);$(TOP)/lib;
 
 INCLUDES        +=  "-I$(TOP)/include" $(_SPACE_)
 
-LIBRARIES        =+ $(_SPACE_) "/LIBPATH:$(TOP)/lib/$(_WIN_PLATFORM_)"
+LIBRARIES        =+ $(_SPACE_) "/LIBPATH:$(TOP)/lib"
 
 CUDAFE_FLAGS    +=
 PTXAS_FLAGS     +=

