--- a/H/bool.h
+++ b/H/bool.h
@@ -30,7 +30,7 @@
 ****************************************************************************/
 
 
-#if !defined( BOOL_DEFINED )  &&  !defined( bool ) && !(__WATCOMC__ >= 1070 && defined(__cplusplus))
+#if ( __STDC_VERSION__ < 202311L ) && !defined( BOOL_DEFINED )  &&  !defined( bool ) && !( __WATCOMC__ >= 1070 && defined(__cplusplus) )
     #define BOOL_DEFINED
     typedef unsigned char bool;
 #endif
