From 3370aeda76edb6fc395e5f72790fa2af2ca90db1 Mon Sep 17 00:00:00 2001
From: Martin Valgur <martin.valgur@gmail.com>
Date: Wed, 10 Apr 2024 10:29:53 +0300
Subject: [PATCH] Fix an invalid `install(cmake/FindTBB.cmake)`

---
 CMakeLists.txt | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index a2fd69f..094bb1a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -274,10 +274,6 @@ if(BUILD_HELPER)
     "${CMAKE_CURRENT_BINARY_DIR}/small_gicp-config-version.cmake"
     DESTINATION ${CMAKE_CONFIG_INSTALL_DIR}
   )
-  install(FILES
-    "${CMAKE_CURRENT_SOURCE_DIR}/cmake/FindTBB.cmake"
-    DESTINATION ${CMAKE_CONFIG_INSTALL_DIR}
-  )
 endif()
 
 if(BUILD_PYTHON_BINDINGS)
