--- a/modules/GEL/CMakeLists.txt	2021-06-08 16:55:54.939874984 +0000
+++ b/modules/GEL/CMakeLists.txt	2021-06-21 11:26:33.069936488 +0000
@@ -77,13 +77,6 @@
   set (CMAKE_BUILD_TYPE Release CACHE STRING "Setting build mode to Release" FORCE)
 endif()
 
-# output location
-set (CMAKE_RUNTIME_OUTPUT_DIRECTORY                ${PROJECT_SOURCE_DIR}/bin/${OS}-${ARCH})
-set (CMAKE_RUNTIME_OUTPUT_DIRECTORY_DEBUG          ${PROJECT_SOURCE_DIR}/bin/${OS}-${ARCH})
-set (CMAKE_RUNTIME_OUTPUT_DIRECTORY_MINSIZEREL     ${PROJECT_SOURCE_DIR}/bin/${OS}-${ARCH})
-set (CMAKE_RUNTIME_OUTPUT_DIRECTORY_RELEASE        ${PROJECT_SOURCE_DIR}/bin/${OS}-${ARCH})
-set (CMAKE_RUNTIME_OUTPUT_DIRECTORY_RELWITHDEBINFO ${PROJECT_SOURCE_DIR}/bin/${OS}-${ARCH})
-
 # find dependencies
 find_package (CHAI3D 3.2 REQUIRED)
 include_directories (${CHAI3D_INCLUDE_DIRS})
@@ -120,8 +113,8 @@
 # Linux global build options
 elseif (${CMAKE_SYSTEM_NAME} MATCHES Linux)
   add_definitions (-DLINUX)
-  set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC -march=native -Wno-deprecated -std=c++0x")
-  set (CMAKE_C_FLAGS   "${CMAKE_C_FLAGS}   -fPIC -march=native -Wno-deprecated")
+  set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -march=native -Wno-deprecated -std=c++0x")
+  set (CMAKE_C_FLAGS   "${CMAKE_C_FLAGS}   -march=native -Wno-deprecated")
 
 # Mac OS X global build options
 elseif (${CMAKE_SYSTEM_NAME} MATCHES Darwin)
@@ -153,8 +146,13 @@
 # build flags
 add_definitions (${PROJECT_DEFINITIONS})
 
-# static library
-add_library (chai3d-GEL STATIC ${source})
+# library
+add_library (chai3d-GEL SHARED ${source})
+set_target_properties (chai3d-GEL PROPERTIES VERSION ${PROJECT_VERSION})
+if (MSVC)
+  set_target_properties (chai3d-GEL PROPERTIES WINDOWS_EXPORT_ALL_SYMBOLS ON)
+endif (MSVC)
+target_link_libraries (chai3d-GEL PUBLIC chai3d)
 
 # library exports
 set (CHAI3D-GEL_INCLUDE_DIRS "${PROJECT_SOURCE_DIR}/src")
@@ -162,16 +160,6 @@
 
 
 #
-# executables
-#
-
-# examples
-if (EXISTS ${PROJECT_SOURCE_DIR}/examples)
-  add_subdirectory (${PROJECT_SOURCE_DIR}/examples)
-endif ()
-
-
-#
 # export package
 #
 
