diff --git a/CMakeLists.txt b/CMakeLists.txt
index d6368ca8..dcdeb223 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -354,7 +354,19 @@ endif()
 
 if(BUILD_WITH_HPP_FCL_SUPPORT)
   set(COAL_DISABLE_HPP_FCL_WARNINGS ON)
-  add_project_dependency(hpp-fcl REQUIRED)
+  find_package(coal REQUIRED)
+  if(NOT TARGET hpp-fcl::hpp-fcl)
+    add_library(hpp-fcl::hpp-fcl INTERFACE IMPORTED)
+
+    # Compute the installation prefix relative to this file.
+    # This code is taken from generated cmake xxxTargets.cmake.
+    set_target_properties(
+      hpp-fcl::hpp-fcl
+      PROPERTIES INTERFACE_COMPILE_DEFINITIONS
+      "COAL_BACKWARD_COMPATIBILITY_WITH_HPP_FCL"
+      INTERFACE_INCLUDE_DIRECTORIES "${coal_INCLUDE_DIR}"
+      INTERFACE_LINK_LIBRARIES "coal::coal")
+  endif()
 endif()
 
 if(BUILD_WITH_ACCELERATE_SUPPORT)
