diff -urN ./a/cmake/helpers/CheckDependentLibraries.cmake ./b/cmake/helpers/CheckDependentLibraries.cmake
--- ./a/cmake/helpers/CheckDependentLibraries.cmake
+++ ./b/cmake/helpers/CheckDependentLibraries.cmake
@@ -11,7 +11,7 @@
 include(CheckFunctionExists)
 include(CMakeDependentOption)
 include(FeatureSummary)
-include(DefineFindPackage2)
+include(ConanFindPackage)
 include(CheckSymbolExists)
 
 option(
@@ -109,9 +109,8 @@
   string(TOUPPER ${name} key)
   set(_find_dependency "")
   set(_find_dependency_args "")
-  if(FIND_PACKAGE2_${name}_ENABLED)
-    find_package2(${name} QUIET OUT_DEPENDENCY _find_dependency)
-  else()
+  find_package2(${name} QUIET)
+  if (FALSE)
     set(_find_package_args)
     if (_GCP_VERSION)
       list(APPEND _find_package_args ${_GCP_VERSION})
@@ -368,7 +367,7 @@
 endif ()
 
 # First check with CMake config files (starting at version 8, due to issues with earlier ones), and then fallback to the FindPROJ module.
-find_package(PROJ 9 CONFIG QUIET)
+find_package2(PROJ 9 CONFIG REQUIRED)
 if (NOT PROJ_FOUND)
   find_package(PROJ 8 CONFIG QUIET)
 endif()
@@ -457,8 +456,8 @@
   NAMES json-c
   TARGETS json-c::json-c JSONC::JSONC
 )
-gdal_internal_library(JSONC REQUIRED)
-if(TARGET json-c::json-c)
+find_package2(JSONC REQUIRED)
+if(FALSE)
   get_target_property(include_dirs json-c::json-c INTERFACE_INCLUDE_DIRECTORIES)
   find_path(GDAL_JSON_INCLUDE_DIR NAMES json.h PATHS ${include_dirs} PATH_SUFFIXES json-c NO_DEFAULT_PATH)
   list(APPEND include_dirs "${GDAL_JSON_INCLUDE_DIR}")
@@ -527,6 +526,16 @@
 gdal_check_package(SQLite3 "Enable SQLite3 support (used by SQLite/Spatialite, GPKG, Rasterlite, MBTiles, etc.)"
                    CAN_DISABLE RECOMMENDED)
 if (SQLite3_FOUND)
+  set(CMAKE_REQUIRED_INCLUDES ${SQLite3_INCLUDE_DIRS})
+  check_symbol_exists(sqlite3_mutex_alloc sqlite3ext.h SQLite3_HAS_MUTEX_ALLOC)
+  check_symbol_exists(sqlite3_column_table_name sqlite3ext.h SQLite3_HAS_COLUMN_METADATA)
+  check_symbol_exists(sqlite3_rtree_query_callback sqlite3.h SQLite3_HAS_RTREE)
+  check_symbol_exists(sqlite3_load_extension sqlite3ext.h SQLite3_HAS_LOAD_EXTENSION)
+  # https://www.sqlite.org/compile.html recommends to build with -DSQLITE_OMIT_PROGRESS_CALLBACK
+  # "for applications that are able to use them"... This is sometimes wrongly
+  # understood as recommended in all situations.
+  check_symbol_exists(sqlite3_progress_handler sqlite3.h SQLite3_HAS_PROGRESS_HANDLER)
+
   if (NOT DEFINED SQLite3_HAS_COLUMN_METADATA)
     message(FATAL_ERROR "missing SQLite3_HAS_COLUMN_METADATA")
   endif ()
@@ -566,9 +575,9 @@
 gdal_check_package(SPATIALITE "Enable spatialite support for sqlite3" VERSION 4.1.2 CAN_DISABLE)
 gdal_check_package(RASTERLITE2 "Enable RasterLite2 support for sqlite3" VERSION 1.1.0 CAN_DISABLE)
 
-find_package(LibKML COMPONENTS DOM ENGINE)
+find_package2(LibKML COMPONENTS kmlengine kmldom kmlbase)
 if (GDAL_USE_LIBKML)
-  if (NOT LibKML_FOUND)
+  if (FALSE)
     message(FATAL_ERROR "Configured to use GDAL_USE_LIBKML, but not found")
   endif ()
 endif ()
@@ -733,7 +742,7 @@
 gdal_check_package(HEIF "HEIF >= 1.1" CAN_DISABLE)
 
 # OpenJPEG's cmake-CONFIG is broken, so call module explicitly
-find_package(OpenJPEG MODULE)
+find_package2(OpenJPEG MODULE)
 if (GDAL_USE_OPENJPEG)
   if (NOT OPENJPEG_FOUND)
     message(FATAL_ERROR "Configured to use GDAL_USE_OPENJPEG, but not found")
