diff -Naur a/src/gn/escape.h b/src/gn/escape.h
--- a/src/gn/escape.h	2025-06-03 17:10:03.000000000 +0800
+++ b/src/gn/escape.h	2025-06-03 17:26:04.873946600 +0800
@@ -6,6 +6,7 @@
 #define TOOLS_GN_ESCAPE_H_
 
 #include <iosfwd>
+#include <string>
 #include <string_view>
 
 enum EscapingMode {
@@ -33,7 +34,7 @@
   ESCAPE_NINJA_PREFORMATTED_COMMAND,
 
   // Shell escaping as described by JSON Compilation Database spec:
-  // Parameters use shell quoting and shell escaping of quotes, with ‘"’ and ‘\’
+  // Parameters use shell quoting and shell escaping of quotes, with '"' and '\'
   // being the only special characters.
   ESCAPE_COMPILATION_DATABASE,
 };
