diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1984ac9..75e904d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1051,8 +1051,8 @@ if(PULSEAUDIO_FOUND)
         set(HAVE_PULSEAUDIO 1)
         set(BACKENDS  "${BACKENDS} PulseAudio${IS_LINKED},")
         set(ALC_OBJS  ${ALC_OBJS} alc/backends/pulseaudio.cpp alc/backends/pulseaudio.h)
-        add_backend_libs(${PULSEAUDIO_LIBRARIES})
-        set(INC_PATHS ${INC_PATHS} ${PULSEAUDIO_INCLUDE_DIRS})
+        add_backend_libs(${PULSEAUDIO_LIBRARY})
+        set(INC_PATHS ${INC_PATHS} ${PULSEAUDIO_INCLUDE_DIR})
     endif()
 endif()
 if(ALSOFT_REQUIRE_PULSEAUDIO AND NOT HAVE_PULSEAUDIO)
