diff --git a/CMakeLists.txt b/CMakeLists.txt
index 62b56b1..4c97a06 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -82,13 +82,13 @@ ign_create_packages()
 #============================================================================
 # Create documentation
 #============================================================================
-configure_file(${CMAKE_SOURCE_DIR}/api.md.in ${CMAKE_BINARY_DIR}/api.md)
-configure_file(${CMAKE_SOURCE_DIR}/tutorials.md.in ${CMAKE_BINARY_DIR}/tutorials.md)
+configure_file(${PROJECT_SOURCE_DIR}/api.md.in ${PROJECT_BINARY_DIR}/api.md)
+configure_file(${PROJECT_SOURCE_DIR}/tutorials.md.in ${PROJECT_BINARY_DIR}/tutorials.md)
 
 
 ign_create_docs(
-  API_MAINPAGE_MD "${CMAKE_BINARY_DIR}/api.md"
-  TUTORIALS_MAINPAGE_MD "${CMAKE_BINARY_DIR}/tutorials.md"
+  API_MAINPAGE_MD "${PROJECT_BINARY_DIR}/api.md"
+  TUTORIALS_MAINPAGE_MD "${PROJECT_BINARY_DIR}/tutorials.md"
   TAGFILES
     "${IGNITION-MATH_DOXYGEN_TAGFILE} = ${IGNITION-MATH_API_URL}"
 )
 
