diff --git a/CMakeLists.txt b/CMakeLists.txt
index 62b56b1..9a5054a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -10,7 +10,7 @@ project(ignition-msgs5 VERSION 5.3.0)
 #============================================================================
 # If you get an error at this line, you need to install ignition-cmake
 find_package(ignition-cmake2 REQUIRED)
-
+include(IgnCMake)
 #============================================================================
 # Configure the project
 #============================================================================
@@ -48,7 +48,7 @@ ign_find_package(ignition-tools
 
 #--------------------------------------
 # Find Tinyxml2
-ign_find_package(TINYXML2 REQUIRED PRIVATE PRETTY tinyxml2)
+ign_find_package(tinyxml2 REQUIRED PRIVATE PRETTY tinyxml2)
 
 #============================================================================
 # Configure the build
@@ -82,13 +82,13 @@ ign_create_packages()
 #============================================================================
 # Create documentation
 #============================================================================
-configure_file(${CMAKE_SOURCE_DIR}/api.md.in ${CMAKE_BINARY_DIR}/api.md)
-configure_file(${CMAKE_SOURCE_DIR}/tutorials.md.in ${CMAKE_BINARY_DIR}/tutorials.md)
+configure_file(${PROJECT_SOURCE_DIR}/api.md.in ${PROJECT_BINARY_DIR}/api.md)
+configure_file(${PROJECT_SOURCE_DIR}/tutorials.md.in ${PROJECT_BINARY_DIR}/tutorials.md)
 
 
 ign_create_docs(
-  API_MAINPAGE_MD "${CMAKE_BINARY_DIR}/api.md"
-  TUTORIALS_MAINPAGE_MD "${CMAKE_BINARY_DIR}/tutorials.md"
+  API_MAINPAGE_MD "${PROJECT_BINARY_DIR}/api.md"
+  TUTORIALS_MAINPAGE_MD "${PROJECT_BINARY_DIR}/tutorials.md"
   TAGFILES
     "${IGNITION-MATH_DOXYGEN_TAGFILE} = ${IGNITION-MATH_API_URL}"
 )
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index f68bf3d..88f3575 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -181,7 +181,7 @@ target_link_libraries(${PROJECT_LIBRARY_TARGET_NAME}
     protobuf::libprotobuf
     ${ignition-math${IGN_MATH_VER}_LIBRARIES}
   PRIVATE
-    TINYXML2::TINYXML2
+    tinyxml2::tinyxml2
 )
 
 target_include_directories(${PROJECT_LIBRARY_TARGET_NAME}
@@ -206,7 +206,7 @@ ign_build_tests(TYPE UNIT
   SOURCES
     ${gtest_sources}
   LIB_DEPS
-      TINYXML2::TINYXML2
+      tinyxml2::tinyxml2
 )
 
 ##################################################
