--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -72,54 +72,6 @@
 install(FILES ${PROJECT_BINARY_DIR}/${PROJECT_NAME}_export.h
   DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/${PROJECT_NAME})
 
-if(WIN32 AND NOT CYGWIN)
-  set(CMAKE_CONFIG_INSTALL_DIR CMake)
-else()
-  set(CMAKE_CONFIG_INSTALL_DIR ${CMAKE_INSTALL_LIBDIR}/${PROJECT_NAME}/cmake/)
-endif()
-
-include(CMakePackageConfigHelpers)
-set(PKG_NAME ${PROJECT_NAME})
-set(PKG_LIBRARIES ${PROJECT_NAME})
-set(cmake_conf_file "${PROJECT_NAME}-config.cmake")
-configure_package_config_file("${cmake_conf_file}.in" "${CMAKE_BINARY_DIR}/${cmake_conf_file}"
-                              INSTALL_DESTINATION ${CMAKE_CONFIG_INSTALL_DIR}
-                              PATH_VARS CMAKE_INSTALL_FULL_INCLUDEDIR
-                              NO_SET_AND_CHECK_MACRO
-                              NO_CHECK_REQUIRED_COMPONENTS_MACRO)
-set(cmake_conf_version_file "${PROJECT_NAME}-config-version.cmake")
-# Use write_basic_package_version_file to generate a ConfigVersion file that
-# allow users of gazebo to specify the API or version to depend on
-# TODO: keep this instruction until deprecate Ubuntu/Precise and update with
-# https://github.com/Kitware/CMake/blob/v2.8.8/Modules/CMakePackageConfigHelpers.cmake
-include(WriteBasicConfigVersionFile)
-write_basic_config_version_file(
-  ${CMAKE_CURRENT_BINARY_DIR}/${cmake_conf_version_file}
-  VERSION "${CONSOLE_BRIDGE_VERSION}"
-  COMPATIBILITY SameMajorVersion)
-
-install(EXPORT ${PROJECT_NAME}-targets
-  FILE ${PROJECT_NAME}-targets.cmake
-  NAMESPACE ${PROJECT_NAME}::
-  DESTINATION ${CMAKE_CONFIG_INSTALL_DIR})
-install(FILES
-        "${CMAKE_BINARY_DIR}/${cmake_conf_file}"
-        "${CMAKE_BINARY_DIR}/${cmake_conf_version_file}"
-  DESTINATION ${CMAKE_CONFIG_INSTALL_DIR} COMPONENT cmake)
-
-if (NOT MSVC)
-  set(PKG_DESC "Console Bridge")
-  set(PKG_CB_LIBS "-l${PROJECT_NAME}")
-  set(pkg_conf_file "console_bridge.pc")
-  configure_file("${pkg_conf_file}.in" "${CMAKE_BINARY_DIR}/${pkg_conf_file}" @ONLY)
-  install(FILES "${CMAKE_BINARY_DIR}/${pkg_conf_file}"
-    DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig/ COMPONENT pkgconfig)
-endif()
-
-SET_DIRECTORY_PROPERTIES(PROPERTIES
-  ADDITIONAL_MAKE_CLEAN_FILES ${CMAKE_BINARY_DIR}/console_bridge-config.cmake
-  ADDITIONAL_MAKE_CLEAN_FILES ${CMAKE_BINARY_DIR}/console_bridge.pc)
-
 if(BUILD_TESTING)
   enable_testing()
   add_subdirectory(test)
