commit 74eeb8a432fa08f3827ebfaba730c1b7b0e2c615
Author: dvirtz <dvirtz@gmail.com>
Date:   Mon Jul 5 09:51:24 2021 +0300

    build standalone cmake

diff --git a/aws-cpp-sdk-cdi/CMakeLists.txt b/aws-cpp-sdk-cdi/CMakeLists.txt
index 9ca1679..b4f1393 100644
--- a/aws-cpp-sdk-cdi/CMakeLists.txt
+++ b/aws-cpp-sdk-cdi/CMakeLists.txt
@@ -1,4 +1,5 @@
-add_project(aws-cpp-sdk-cdi "C++ SDK for the AWS cdi service" aws-cpp-sdk-core)
+find_package(AWSSDK REQUIRED)
+add_project(aws-cpp-sdk-cdi "C++ SDK for the AWS cdi service" AWS::aws-sdk-cpp-core)
 
 file(GLOB AWS_CDI_HEADERS
     "include/aws/cdi/*.h"
