diff --git a/src/est/est_locl.h b/src/est/est_locl.h
index 62dcbea..b16f62d 100644
--- a/src/est/est_locl.h
+++ b/src/est/est_locl.h
@@ -590,7 +590,7 @@ typedef struct est_oid_list {
 /*
  * Index used to link the EST Ctx into the SSL structures
  */
-int e_ctx_ssl_exdata_index;
+extern int e_ctx_ssl_exdata_index;
 
 LIBEST_TEST_API void est_log (EST_LOG_LEVEL lvl, char *format, ...);
 LIBEST_TEST_API void est_log_backtrace (void);
