diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5087361..64b44f6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -7,6 +7,7 @@ if(NOT DEFINED PROJECT_NAME)
 else()
   set(NOT_SUBPROJECT OFF)
 endif()
+set(NOT_SUBPROJECT ON)
 
 option(CATCH_INSTALL_DOCS "Install documentation alongside library" ON)
 option(CATCH_INSTALL_EXTRAS "Install extras (CMake scripts, debugger helpers) alongside library" ON)
