diff --git a/cmake/Macros.cmake b/cmake/Macros.cmake
index 257b52d..a44e562 100644
--- a/cmake/Macros.cmake
+++ b/cmake/Macros.cmake
@@ -12,4 +12,12 @@ function(core_add_library name)
   add_library(${name} OBJECT ${SOURCES} ${HEADERS})
   target_include_directories(${name} PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>)
   set(CORE_LIBRARIES "${name};${CORE_LIBRARIES}" CACHE INTERNAL "")
+
+  # no need to install core libs if we build shared library
+  if(NOT BUILD_SHARED_LIBS)
+    install(TARGETS ${name} EXPORT libnfs
+            RUNTIME DESTINATION bin
+            ARCHIVE DESTINATION lib
+            LIBRARY DESTINATION lib)
+  endif()
 endfunction()
