--- CMakeLists.txt
+++ CMakeLists.txt
@@ -39,7 +39,7 @@ ign_find_package(
 ########################################
 # Include swig
 find_package(SWIG QUIET)
-if (NOT SWIG_FOUND)
+if (1)
   IGN_BUILD_WARNING("Swig is missing: Language interfaces are disabled.")
   message (STATUS "Searching for swig - not found.")
 else()
@@ -47,7 +47,7 @@ else()
 endif()
 
 # Include other languages if swig was found
-if (SWIG_FOUND)
+if (0)
   ########################################
   # Include ruby
   find_package(Ruby 1.9 QUIET)
@@ -74,9 +74,9 @@ ign_create_packages()
 #============================================================================
 # Configure documentation
 #============================================================================
-configure_file(${CMAKE_SOURCE_DIR}/api.md.in ${CMAKE_BINARY_DIR}/api.md)
-configure_file(${CMAKE_SOURCE_DIR}/tutorials.md.in ${CMAKE_BINARY_DIR}/tutorials.md)
+configure_file(${PROJECT_SOURCE_DIR}/api.md.in ${PROJECT_BINARY_DIR}/api.md)
+configure_file(${PROJECT_SOURCE_DIR}/tutorials.md.in ${PROJECT_BINARY_DIR}/tutorials.md)
 
 ign_create_docs(
-  API_MAINPAGE_MD "${CMAKE_BINARY_DIR}/api.md"
-  TUTORIALS_MAINPAGE_MD "${CMAKE_BINARY_DIR}/tutorials.md")
+  API_MAINPAGE_MD "${PROJECT_BINARY_DIR}/api.md"
+  TUTORIALS_MAINPAGE_MD "${PROJECT_BINARY_DIR}/tutorials.md")
