diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9f5dcce..4d6d981 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -527,7 +527,7 @@ endif()
 
 add_subdirectory(src)
 
-add_subdirectory(tools)
+# add_subdirectory(tools)
 
 add_subdirectory(extension)
 
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 752e0d6..cad0320 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -15,16 +15,18 @@ if(AMALGAMATION_BUILD)
     add_definitions(/bigobj)
   endif()
 
-  add_library(duckdb SHARED "${PROJECT_SOURCE_DIR}/src/amalgamation/duckdb.cpp")
-  target_link_libraries(duckdb ${CMAKE_DL_LIBS})
-  link_threads(duckdb)
-  link_extension_libraries(duckdb)
-
-  add_library(duckdb_static STATIC
-              "${PROJECT_SOURCE_DIR}/src/amalgamation/duckdb.cpp")
-  target_link_libraries(duckdb_static ${CMAKE_DL_LIBS})
-  link_threads(duckdb_static)
-  link_extension_libraries(duckdb_static)
+  if (NOT BUILD_STATIC_LIBS)
+    add_library(duckdb SHARED "${PROJECT_SOURCE_DIR}/src/amalgamation/duckdb.cpp")
+    target_link_libraries(duckdb ${CMAKE_DL_LIBS})
+    link_threads(duckdb)
+    link_extension_libraries(duckdb)
+  else()
+    add_library(duckdb_static STATIC
+                "${PROJECT_SOURCE_DIR}/src/amalgamation/duckdb.cpp")
+    target_link_libraries(duckdb_static ${CMAKE_DL_LIBS})
+    link_threads(duckdb_static)
+    link_extension_libraries(duckdb_static)
+  endif()
 
   install(FILES "${PROJECT_SOURCE_DIR}/src/amalgamation/duckdb.hpp"
                 "${PROJECT_SOURCE_DIR}/src/include/duckdb.h"
@@ -152,8 +154,17 @@ if(BUILD_PYTHON
   endif()
 endif()
 
-install(
-  TARGETS duckdb duckdb_static
+
+if (NOT BUILD_STATIC_LIBS)
+  install(
+    TARGETS duckdb
+    EXPORT "${DUCKDB_EXPORT_SET}"
+    LIBRARY DESTINATION "${INSTALL_LIB_DIR}"
+    ARCHIVE DESTINATION "${INSTALL_LIB_DIR}")
+else()
+  install(
+  TARGETS duckdb_static
   EXPORT "${DUCKDB_EXPORT_SET}"
   LIBRARY DESTINATION "${INSTALL_LIB_DIR}"
   ARCHIVE DESTINATION "${INSTALL_LIB_DIR}")
+endif()
\ No newline at end of file
