--- cpp/cmake_modules/FindThrift.cmake
+++ cpp/cmake_modules/FindThrift.cmake
@@ -39,12 +39,14 @@
   endif()
 endfunction(EXTRACT_THRIFT_VERSION)

-if(MSVC AND NOT THRIFT_MSVC_STATIC_LIB_SUFFIX)
-  set(THRIFT_MSVC_STATIC_LIB_SUFFIX md)
+if(MSVC AND NOT THRIFT_STATIC_LIB_SUFFIX)
+  set(THRIFT_STATIC_LIB_SUFFIX md)
+elseif(${UPPERCASE_BUILD_TYPE} STREQUAL "DEBUG")
+  set(THRIFT_STATIC_LIB_SUFFIX d)
 endif()

 if(Thrift_ROOT)
-  find_library(THRIFT_STATIC_LIB thrift${THRIFT_MSVC_STATIC_LIB_SUFFIX}
+  find_library(THRIFT_STATIC_LIB thrift${THRIFT_STATIC_LIB_SUFFIX}
                PATHS ${Thrift_ROOT}
                PATH_SUFFIXES "lib/${CMAKE_LIBRARY_ARCHITECTURE}" "lib")
   find_path(THRIFT_INCLUDE_DIR thrift/Thrift.h
@@ -61,7 +63,7 @@

     list(APPEND THRIFT_PC_LIBRARY_DIRS "${THRIFT_PC_LIBDIR}")

-    find_library(THRIFT_STATIC_LIB thrift${THRIFT_MSVC_STATIC_LIB_SUFFIX}
+    find_library(THRIFT_STATIC_LIB thrift${THRIFT_STATIC_LIB_SUFFIX}
                  PATHS ${THRIFT_PC_LIBRARY_DIRS}
                  NO_DEFAULT_PATH)
     find_program(THRIFT_COMPILER thrift
@@ -70,7 +72,7 @@
                  PATH_SUFFIXES "bin")
     set(THRIFT_VERSION ${THRIFT_PC_VERSION})
   else()
-    find_library(THRIFT_STATIC_LIB thrift${THRIFT_MSVC_STATIC_LIB_SUFFIX}
+    find_library(THRIFT_STATIC_LIB thrift${THRIFT_STATIC_LIB_SUFFIX}
                  PATH_SUFFIXES "lib/${CMAKE_LIBRARY_ARCHITECTURE}" "lib")
     find_path(THRIFT_INCLUDE_DIR thrift/Thrift.h PATH_SUFFIXES "include")
     find_program(THRIFT_COMPILER thrift PATH_SUFFIXES "bin")
