diff --git a/cpp/cmake_modules/ThirdpartyToolchain.cmake b/cpp/cmake_modules/ThirdpartyToolchain.cmake
index 722b3b56de..284a6b96c4 100644
--- a/cpp/cmake_modules/ThirdpartyToolchain.cmake
+++ b/cpp/cmake_modules/ThirdpartyToolchain.cmake
@@ -2231,6 +2231,10 @@ endif()
 # mimalloc - Cross-platform high-performance allocator, from Microsoft

 if(ARROW_MIMALLOC)
+  find_package(mimalloc REQUIRED CONFIG)
+endif()
+
+if(0)
   if(NOT ARROW_ENABLE_THREADING)
     message(FATAL_ERROR "Can't use mimalloc with ARROW_ENABLE_THREADING=OFF")
   endif()
