diff --git a/src/sail-codecs/jpegxl/CMakeLists.txt b/src/sail-codecs/jpegxl/CMakeLists.txt
index 16971a48..477d48f6 100644
--- a/src/sail-codecs/jpegxl/CMakeLists.txt
+++ b/src/sail-codecs/jpegxl/CMakeLists.txt
@@ -1,3 +1,4 @@
+if(0)
 # Don't use SAIL_CODEC_JPEGXL_REQUIRED_OPTION as it requires CMake 3.18
 #
 find_library(JPEGXL_LIBRARY jxl                 NAMES jxl jxl-static)
@@ -56,3 +57,13 @@ sail_codec(NAME jpegxl
             DEPENDENCY_INCLUDE_DIRS ${JPEGXL_INCLUDE_DIRS}
             DEPENDENCY_LIBS ${BROTLI_COMMON_LIBRARY} ${BROTLI_DEC_LIBRARY} ${HWY_LIBRARY}
                             ${JPEGXL_LIBRARY} ${JPEGXL_THREADS_LIBRARY})
+endif()
+# Find libjxl package
+find_package(libjxl REQUIRED CONFIG)
+
+# Common codec configuration
+sail_codec(NAME jpegxl
+           SOURCES helpers.h helpers.c jpegxl.c memory.h memory.c
+           ICON jpegxl.png)
+
+target_link_libraries(${SAIL_CODEC_TARGET} PRIVATE libjxl::libjxl)
