diff --git a/cmake/sail_check_init_once_execute_once.cmake b/cmake/sail_check_init_once_execute_once.cmake
index 6fed498e..290e1708 100644
--- a/cmake/sail_check_init_once_execute_once.cmake
+++ b/cmake/sail_check_init_once_execute_once.cmake
@@ -4,7 +4,8 @@ function(sail_check_init_once_execute_once)
     cmake_push_check_state(RESET)
         check_c_source_compiles(
         "
-            #include \"Windows.h\"
+            #define _WIN32_WINNT 0x0600
+            #include <Windows.h>
             static BOOL CALLBACK OnceHandler(PINIT_ONCE InitOnce, PVOID Parameter, PVOID *lpContext)
             {
                 (void)InitOnce;
