diff --git a/src/sail-codecs/jpegxl/CMakeLists.txt b/src/sail-codecs/jpegxl/CMakeLists.txt
index dfd9d04..df32469 100644
--- a/src/sail-codecs/jpegxl/CMakeLists.txt
+++ b/src/sail-codecs/jpegxl/CMakeLists.txt
@@ -1,3 +1,5 @@
+
+if(0)
 include(FindPackageHandleStandardArgs)
 include(SelectLibraryConfigurations)
 
@@ -91,14 +93,14 @@ if (NOT BUILD_SHARED_LIBS)
     endif()
 endif()
 
+endif()
 
+find_package(libjxl REQUIRED CONFIG)
 # Common codec configuration
 #
 sail_codec(NAME jpegxl
             SOURCES helpers.h helpers.c jpegxl.c memory.h memory.c
             ICON jpegxl.png
-            DEPENDENCY_COMPILE_DEFINITIONS ${JXL_STATIC_DEFINE}
-            DEPENDENCY_INCLUDE_DIRS ${JPEGXL_INCLUDE_DIR}
-            DEPENDENCY_LIBS ${JPEGXL_LIBRARY} ${JPEGXL_THREADS_LIBRARY} ${JPEGXL_CMS_LIBRARY_LINK}
+            DEPENDENCY_LIBS libjxl::libjxl
                             ${BROTLI_DEC_LIBRARY} ${BROTLI_ENC_LIBRARY} ${BROTLI_COMMON_LIBRARY} ${HWY_LIBRARY} ${LCMS2_LIBRARY}
                             ${EXTRA_LINK})
