--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -24,9 +24,9 @@ if(WIN32 AND BUILD_SHARED_LIBS)
     set(CMAKE_VS_INCLUDE_INSTALL_TO_DEFAULT_BUILD TRUE)
 endif()
 
-set(SRC source_subfolder/cpp/qrcodegen.cpp)
-set(HEADERS source_subfolder/cpp/qrcodegen.hpp)
-set(LIBRARY_NAME qrcodegencpp)
+set(SRC source_subfolder/cpp/QrCode.cpp source_subfolder/cpp/QrSegment.cpp source_subfolder/cpp/BitBuffer.cpp)
+set(HEADERS source_subfolder/cpp/QrCode.hpp source_subfolder/cpp/QrSegment.hpp source_subfolder/cpp/BitBuffer.hpp)
+set(LIBRARY_NAME qrcodegen)
 
 add_library(${LIBRARY_NAME} ${SRC})
 
