--- CMakeLists.txt
+++ CMakeLists.txt
@@ -26,8 +26,6 @@
 endif()
 
 set(SRC source_subfolder/cpp/QrCode.cpp
-        source_subfolder/cpp/QrSegment.cpp
-        source_subfolder/cpp/BitBuffer.cpp
         source_subfolder/cpp/QrCodeGeneratorWorker.cpp)
 
 add_library(qrcodegen ${SRC})
@@ -37,8 +35,6 @@
     LIBRARY DESTINATION lib
     ARCHIVE DESTINATION lib)
 
-set(HEADERS source_subfolder/cpp/QrCode.hpp
-            source_subfolder/cpp/QrSegment.hpp
-            source_subfolder/cpp/BitBuffer.hpp)
+set(HEADERS source_subfolder/cpp/QrCode.hpp)
 
 INSTALL(FILES ${HEADERS} DESTINATION include/qrcodegen)
