--- CMakeLists.txt
+++ CMakeLists.txt
@@ -25,7 +25,7 @@
 endif()
 
 set(SRC
-    source_subfolder/cpp/QrCode.cpp
+    source_subfolder/cpp/QrCode.cpp source_subfolder/cpp/QrSegment.cpp source_subfolder/cpp/BitBuffer.cpp
 )
 
 add_library(qrcodegen ${SRC})
@@ -36,7 +36,7 @@
     ARCHIVE DESTINATION lib)
 
 set(HEADERS
-    source_subfolder/cpp/QrCode.hpp
+    source_subfolder/cpp/QrCode.hpp source_subfolder/cpp/QrSegment.hpp source_subfolder/cpp/BitBuffer.hpp
 )
 
 INSTALL(FILES ${HEADERS} DESTINATION include/qrcodegen)
