diff --git a/CMakeLists.txt b/CMakeLists.txt
index c0fbd70..44ca23d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -454,12 +454,12 @@ IF(ZLIB_FOUND)
     #
     # Test if ZLIB_WINAPI macro is needed to use.
     #
-    TRY_MACRO_FOR_LIBRARY(
-      "${ZLIB_INCLUDE_DIR}" "${ZLIB_LIBRARIES}"
-      RUNS
-      "#include <zlib.h>\nint main() {uLong f = zlibCompileFlags(); return (f&(1U<<10))?0:-1; }"
-      ZLIB_WINAPI)
-    IF(ZLIB_WINAPI)
+    # TRY_MACRO_FOR_LIBRARY(
+    #   "${ZLIB_INCLUDE_DIR}" "${ZLIB_LIBRARIES}"
+    #   RUNS
+    #   "#include <zlib.h>\nint main() {uLong f = zlibCompileFlags(); return (f&(1U<<10))?0:-1; }"
+    #   ZLIB_WINAPI)
+    IF(TRUE)
       ADD_DEFINITIONS(-DZLIB_WINAPI)
     ELSE(ZLIB_WINAPI)
       # Test if a macro is needed for the library.
