diff --git a/CMakeLists.txt b/CMakeLists.txt
index e23338e..582258c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -475,14 +475,14 @@ IF(ZLIB_FOUND)
     #
     # Test if ZLIB_WINAPI macro is needed to use.
     #
-    TRY_MACRO_FOR_LIBRARY(
-      "${ZLIB_INCLUDE_DIR}" "${ZLIB_LIBRARIES}"
-      RUNS
-      "#include <zlib.h>\nint main() {uLong f = zlibCompileFlags(); return (f&(1U<<10))?0:-1; }"
-      ZLIB_WINAPI)
-    IF(ZLIB_WINAPI)
+    # TRY_MACRO_FOR_LIBRARY(
+    #   "${ZLIB_INCLUDE_DIR}" "${ZLIB_LIBRARIES}"
+    #   RUNS
+    #   "#include <zlib.h>\nint main() {uLong f = zlibCompileFlags(); return (f&(1U<<10))?0:-1; }"
+    #   ZLIB_WINAPI)
+    IF(TRUE)
       ADD_DEFINITIONS(-DZLIB_WINAPI)
-    ELSE(ZLIB_WINAPI)
+    ELSE()
       # Test if a macro is needed for the library.
       TRY_MACRO_FOR_LIBRARY(
         "${ZLIB_INCLUDE_DIR}" "${ZLIB_LIBRARIES}"
@@ -492,7 +492,7 @@ IF(ZLIB_FOUND)
       IF(ZLIB_DLL)
         ADD_DEFINITIONS(-DZLIB_DLL)
       ENDIF(ZLIB_DLL)
-    ENDIF(ZLIB_WINAPI)
+    ENDIF()
   ENDIF(WIN32 AND NOT CYGWIN)
 ENDIF(ZLIB_FOUND)
 MARK_AS_ADVANCED(CLEAR ZLIB_INCLUDE_DIR)
