diff --git a/libvips/arithmetic/complex.c b/libvips/arithmetic/complex.c
index 60010d994..fe62ee04f 100644
--- a/libvips/arithmetic/complex.c
+++ b/libvips/arithmetic/complex.c
@@ -449,12 +449,14 @@ G_DEFINE_TYPE(VipsComplex2, vips_complex2, VIPS_TYPE_BINARY);
 #define CROSS(Q, X1, Y1, X2, Y2) \
 	{ \
 		if (((X1) == 0.0 && (Y1) == 0.0) || \
-			((X2) == 0.0 && (Y2) == 0.0)) { \
+			((X2) == 0.0 && (Y2) == 0.0) || \
+			((Y1) == 0.0 && (Y2) == 0.0)) { \
 			Q[0] = 0.0; \
 			Q[1] = 0.0; \
 		} \
 		else if (ABS(Y1) > ABS(Y2)) { \
-			double a = Y2 / Y1; \
+			double y1 = Y1; /* this suppress C2142 (division by zero) error on MSVC */ \
+			double a = Y2 / y1; \
 			double b = Y1 + Y2 * a; \
 			double re = (X1 + X2 * a) / b; \
 			double im = (X2 - X1 * a) / b; \
@@ -464,7 +466,8 @@ G_DEFINE_TYPE(VipsComplex2, vips_complex2, VIPS_TYPE_BINARY);
 			Q[1] = im / mod; \
 		} \
 		else { \
-			double a = Y1 / Y2; \
+			double y2 = Y2; \
+			double a = Y1 / y2; \
 			double b = Y2 + Y1 * a; \
 			double re = (X1 * a + X2) / b; \
 			double im = (X2 * a - X1) / b; \
