diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5712cf8..e92aeab 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -389,7 +389,6 @@ set(KTX_MAIN_SRC
     external/basisu/transcoder/basisu_transcoder.cpp
     external/basisu/transcoder/basisu_transcoder.h
     external/basisu/transcoder/basisu.h
-    external/basisu/zstd/zstd.c
     lib/checkheader.c
     external/dfdutils/createdfd.c
     external/dfdutils/colourspaces.c
@@ -838,6 +837,10 @@ endmacro(common_libktx_settings)
 common_libktx_settings(ktx 1 ${LIB_TYPE})
 common_libktx_settings(ktx_read 0 ${LIB_TYPE})
 
+find_package(zstd CONFIG REQUIRED)
+target_link_libraries(ktx PUBLIC zstd::libzstd)
+target_link_libraries(ktx_read PUBLIC zstd::libzstd)
+
 if(KTX_FEATURE_JNI)
     add_subdirectory(interface/java_binding)
 endif()
@@ -1316,7 +1319,7 @@ if((KTX_FEATURE_TOOLS OR KTX_FEATURE_TESTS) AND NOT TARGET fmt::fmt)
     set(BUILD_SHARED_LIBS OFF)
     set(FMT_INSTALL OFF)
     set(FMT_SYSTEM_HEADERS ON)
-    add_subdirectory(external/fmt)
+    find_package(fmt CONFIG REQUIRED)
     set(BUILD_SHARED_LIBS ${BUILD_SHARED_LIBS_RESET})
 endif()
 if(KTX_FEATURE_TOOLS AND NOT TARGET cxxopts::cxxopts)
