--- a/src/entt/meta/meta.hpp
+++ b/src/entt/meta/meta.hpp
@@ -1688,7 +1688,7 @@ public:
     using reference = value_type;
     using iterator_category = std::input_iterator_tag;
 
-    constexpr meta_iterator() noexcept
+    meta_iterator() noexcept
         : ctx{},
           vtable{},
           handle{} {}
@@ -1782,7 +1782,7 @@ public:
     using reference = value_type;
     using iterator_category = std::input_iterator_tag;
 
-    constexpr meta_iterator() noexcept
+    meta_iterator() noexcept
         : ctx{},
           vtable{},
           handle{} {}
