diff --git a/libtextstyle/lib/obstack.c b/libtextstyle/lib/obstack.c
index a6757b8..7d4439f 100644
--- a/libtextstyle/lib/obstack.c
+++ b/libtextstyle/lib/obstack.c
@@ -326,7 +326,7 @@ int obstack_exit_failure = EXIT_FAILURE;
 #   include <libio/iolibio.h>
 #  endif
 
-static _Noreturn void
+static __attribute_noreturn__ void
 print_and_abort (void)
 {
   /* Don't change any of these strings.  Yes, it would be possible to add
