diff --git a/CMakeLists.txt b/CMakeLists.txt
index 37616eb..161dd8c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -291,14 +291,6 @@ install(FILES include/mimalloc-new-delete.h DESTINATION ${mi_install_dir}/includ
 install(FILES cmake/mimalloc-config.cmake DESTINATION ${mi_install_dir}/cmake)
 install(FILES cmake/mimalloc-config-version.cmake DESTINATION ${mi_install_dir}/cmake)
 
-if(NOT WIN32 AND MI_BUILD_SHARED)
-  # install a symlink in the /usr/local/lib to the versioned library
-  set(mi_symlink "${CMAKE_SHARED_MODULE_PREFIX}${mi_basename}${CMAKE_SHARED_LIBRARY_SUFFIX}")
-  set(mi_soname "mimalloc-${mi_version}/${mi_symlink}.${mi_version}")
-  install(CODE "execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink ${mi_soname} ${mi_symlink} WORKING_DIRECTORY ${mi_install_dir}/..)")
-  install(CODE "MESSAGE(\"-- Symbolic link: ${CMAKE_INSTALL_PREFIX}/lib/${mi_symlink} -> ${mi_soname}\")")
-endif()
-
 # single object file for more predictable static overriding
 if (MI_BUILD_OBJECT)
   add_library(mimalloc-obj OBJECT src/static.c)
