diff --git a/include/mimalloc.h b/include/mimalloc.h
index 381f98c..43ba74f 100644
--- a/include/mimalloc.h
+++ b/include/mimalloc.h
@@ -392,6 +392,7 @@ mi_decl_nodiscard mi_decl_export void* mi_new_reallocn(void* p, size_t newcount,
 
 #include <cstddef>     // std::size_t
 #include <cstdint>     // PTRDIFF_MAX
+#include <cstddef>     // std::size_t
 #if (__cplusplus >= 201103L) || (_MSC_VER > 1900)  // C++11
 #include <type_traits> // std::true_type
 #include <utility>     // std::forward
