From 3216ce53fa7f276ae4407a450cafafa9e809fea1 Mon Sep 17 00:00:00 2001
From: Sam Bisciglia <sbiscigl@asu.edu>
Date: Tue, 12 Apr 2022 14:42:26 -0400
Subject: [PATCH] Fixes cpp 20 builds for windows. (#364)

---
diff --git a/include/aws/crt/StringView.h b/include/aws/crt/StringView.h
index a0fec83..dc2f05b 100644
--- a/include/aws/crt/StringView.h
+++ b/include/aws/crt/StringView.h
@@ -15,7 +15,7 @@
 #include <stddef.h>
 #include <type_traits>
 
-#if __cplusplus >= 201703L || (defined(_MSC_LANG) && _MSC_LANG >= 201703L)
+#if __cplusplus >= 201703L || (defined(_MSVC_LANG) && _MSVC_LANG >= 201703L)
 #    include <string_view>
 #endif
 
@@ -53,7 +53,7 @@ namespace Aws
 
             basic_string_view &operator=(const basic_string_view &) noexcept = default;
 
-#if __cplusplus >= 201703L || (defined(_MSC_LANG) && _MSC_LANG >= 201703L)
+#if __cplusplus >= 201703L || (defined(_MSVC_LANG) && _MSVC_LANG >= 201703L)
             constexpr basic_string_view(const std::basic_string_view<CharT, Traits> &other) noexcept
                 : m_size(other.size()), m_data(other.data())
             {
