--- a/include/qarchive_enums.hpp
+++ b/include/qarchive_enums.hpp
@@ -1,11 +1,12 @@
 #ifndef QARCHIVE_ENUMS_HPP_INCLUDED
 #define QARCHIVE_ENUMS_HPP_INCLUDED
+#include "qarchive_global.hpp"
 class QString;
 namespace QArchive {
 /* A function that converts the enum value to its variable
  * name to be used by users.
 */
-QString errorCodeToString(short);
+QARCHIVE_EXPORT QString errorCodeToString(short);
 
 /*
  * Common error codes , these are most likely will be
