diff --git a/cmake/developer_package/compile_flags/functions.cmake b/cmake/developer_package/compile_flags/functions.cmake
index 2ec16e6424..072ddcd796 100644
--- a/cmake/developer_package/compile_flags/functions.cmake
+++ b/cmake/developer_package/compile_flags/functions.cmake
@@ -458,12 +458,18 @@ function(ov_target_link_libraries_as_system TARGET_NAME LINK_TYPE)
         if(TARGET ${library})
             get_target_property(include_directories ${library} INTERFACE_INCLUDE_DIRECTORIES)
             if(include_directories)
+                if ("${include_directories}" MATCHES "^\\$<BUILD_INTERFACE:.*>$")
+                    string(REGEX REPLACE "^\\$<BUILD_INTERFACE:" "" include_directories "${include_directories}")
+                    string(REGEX REPLACE ">$" "" include_directories "${include_directories}")
+                endif()
                 foreach(include_directory IN LISTS include_directories)
                     # cannot include /usr/include headers as SYSTEM
                     if(NOT "${include_directory}" MATCHES ".*/usr/include.*$")
                         # Note, some include dirs can be wrapper with $<BUILD_INTERFACE:dir1 dir2 ...> and we need to clean it
-                        string(REGEX REPLACE "^\\$<BUILD_INTERFACE:" "" include_directory "${include_directory}")
-                        string(REGEX REPLACE ">$" "" include_directory "${include_directory}")
+                        if ("${include_directory}" MATCHES "^\\$<BUILD_INTERFACE:.*>$")
+                            string(REGEX REPLACE "^\\$<BUILD_INTERFACE:" "" include_directory "${include_directory}")
+                            string(REGEX REPLACE ">$" "" include_directory "${include_directory}")
+                        endif()
                         target_include_directories(${TARGET_NAME} SYSTEM ${LINK_TYPE} $<BUILD_INTERFACE:${include_directory}>)
                     else()
                         set(_system_library ON)
