diff --git a/CMakeLists.txt b/CMakeLists.txt
index e4ed99e..5e155ec 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -204,12 +204,7 @@ endif()
 # ===============================================

 if(STX_ENABLE_BACKTRACE)
-  if(NOT EXISTS third_party/abseil)
-    execute_process(
-      COMMAND git submodule update --init --recursive third_party/abseil
-      WORKING_DIRECTORY ${CMAKE_CURRENT_LIST_DIR})
-  endif()
-  add_subdirectory(third_party/abseil)
+  find_package(absl REQUIRED)
 endif()

 # the atomics library doesn't automatically link on my Android phone
