diff --git a/CMakeLists.txt b/CMakeLists.txt
index 52d295a..d809ffa 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -398,7 +398,6 @@ if (NOT BLEND2D_NO_JIT)
     set(ASMJIT_EMBED TRUE CACHE BOOL "")
   endif()
 
-  include("${ASMJIT_DIR}/CMakeLists.txt")
   list(APPEND BLEND2D_DEPS ${ASMJIT_LIBS})
   list(APPEND BLEND2D_PRIVATE_CFLAGS ${ASMJIT_CFLAGS})
   list(APPEND BLEND2D_PRIVATE_CFLAGS -DASMJIT_NO_STDCXX)
