diff --git a/CMakeLists.txt b/CMakeLists.txt
index 52d295a..b1a78a5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -398,10 +398,10 @@ if (NOT BLEND2D_NO_JIT)
     set(ASMJIT_EMBED TRUE CACHE BOOL "")
   endif()
 
-  include("${ASMJIT_DIR}/CMakeLists.txt")
-  list(APPEND BLEND2D_DEPS ${ASMJIT_LIBS})
-  list(APPEND BLEND2D_PRIVATE_CFLAGS ${ASMJIT_CFLAGS})
-  list(APPEND BLEND2D_PRIVATE_CFLAGS -DASMJIT_NO_STDCXX)
+  find_package(asmjit CONFIG REQUIRED)
+  list(APPEND BLEND2D_DEPS asmjit::asmjit)
+  list(APPEND BLEND2D_PRIVATE_CFLAGS ${asmjit_DEFINITIONS_DEBUG}${asmjit_DEFINITIONS_RELEASE})
+
 
   # A possibility to reduce the resulting binary size by disabling asmjit logging.
   if (BLEND2D_NO_JIT_LOGGING)
