--- CMakeLists.txt
+++ CMakeLists.txt
@@ -45,11 +45,8 @@
 
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
 
-find_package(tinyxml2_vendor QUIET)
 find_package(TinyXML2 REQUIRED)
 
-find_package(urdfdom_headers 1.0 REQUIRED)
-find_package(console_bridge_vendor QUIET) # Provides console_bridge 0.4.0 on platforms without it.
 find_package(console_bridge REQUIRED)
 
 # Control where libraries and executables are placed during the build
