--- w32/common/common-build-settings.vcxproj
+++ w32/common/common-build-settings.vcxproj
@@ -5,7 +5,7 @@
          Only 0 and 1 are used currently -->
     <TargetOSLevel Condition="$(PlatformToolset.EndsWith('_xp'))">0</TargetOSLevel>
     <TargetOSLevel Condition="! $(PlatformToolset.EndsWith('_xp'))">1</TargetOSLevel>
-  </PropertyGroup>
+  </PropertyGroup><Import Project="../../conan/conantoolchain.props" />
   <PropertyGroup>
     <IncludePath>$(SolutionDir);$(MhdW32Common);$(MhdSrc)include;$(IncludePath)</IncludePath>
   </PropertyGroup>
--- w32/common/libmicrohttpd-build-settings.vcxproj
+++ w32/common/libmicrohttpd-build-settings.vcxproj
@@ -31,8 +31,8 @@
   <ItemDefinitionGroup Condition="'$(ConfigurationType)'=='StaticLibrary'">
     <ClCompile>
       <PreprocessorDefinitions>_LIB;MHD_W32LIB;%(PreprocessorDefinitions)</PreprocessorDefinitions>
-      <RuntimeLibrary Condition="'$(UseDebugLibraries)'=='true'">MultiThreadedDebug</RuntimeLibrary>
-      <RuntimeLibrary Condition="'$(UseDebugLibraries)'!='true'">MultiThreaded</RuntimeLibrary>
+      <!--<RuntimeLibrary Condition="'$(UseDebugLibraries)'=='true'">MultiThreadedDebug</RuntimeLibrary>
+      <RuntimeLibrary Condition="'$(UseDebugLibraries)'!='true'">MultiThreaded</RuntimeLibrary>-->
     </ClCompile>
     <Lib>
       <AdditionalDependencies>Ws2_32.lib</AdditionalDependencies>
@@ -45,8 +45,8 @@
   <ItemDefinitionGroup Condition="'$(ConfigurationType)'=='DynamicLibrary'">
     <ClCompile>
       <PreprocessorDefinitions>_USRDLL;MHD_W32DLL;%(PreprocessorDefinitions)</PreprocessorDefinitions>
-      <RuntimeLibrary Condition="'$(UseDebugLibraries)'=='true'">MultiThreadedDebugDLL</RuntimeLibrary>
-      <RuntimeLibrary Condition="'$(UseDebugLibraries)'!='true'">MultiThreadedDLL</RuntimeLibrary>
+      <!--<RuntimeLibrary Condition="'$(UseDebugLibraries)'=='true'">MultiThreadedDebugDLL</RuntimeLibrary>
+      <RuntimeLibrary Condition="'$(UseDebugLibraries)'!='true'">MultiThreadedDLL</RuntimeLibrary>-->
     </ClCompile>
     <Link>
       <AdditionalDependencies>Ws2_32.lib;%(AdditionalDependencies)</AdditionalDependencies>
