--- CMakeLists.txt
+++ CMakeLists.txt
@@ -56,7 +56,7 @@
 endif()
 
 
-find_package(OpenCV REQUIRED COMPONENTS core)
+find_package(OpenCV REQUIRED COMPONENTS core imgproc features2d highgui)
 find_package(Eigen3 REQUIRED)
 
 option(BUILD_SHARED_LIBS "build shared libraries" ON)
@@ -102,7 +102,7 @@
   )
 
 target_link_libraries(vilib PUBLIC
-  opencv_core Eigen3::Eigen ${CUDA_LIBRARIES})
+  opencv_core opencv_imgproc opencv_features2d opencv_highgui Eigen3::Eigen ${CUDA_LIBRARIES})
 
 # Set different compiler options for cxx and nvcc
 
