--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,3 +1,4 @@
+cmake_minimum_required (VERSION 3.13.0)
 project (GeographicLib)
 
 # Version information
@@ -47,7 +48,6 @@ set (LIBVERSION_BUILD 19.2.0)
 string (TOLOWER ${PROJECT_NAME} PROJECT_NAME_LOWER)
 string (TOUPPER ${PROJECT_NAME} PROJECT_NAME_UPPER)
 
-cmake_minimum_required (VERSION 3.7.0) # This version was released 2016-11-11
 
 # User-settable variables
 
