diff --git a/CMakeLists.txt b/CMakeLists.txt
index 64f3b131..2be1f8d1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -755,7 +755,7 @@ if(OpenMP_CXX_FOUND)
   target_link_libraries(libtesseract PUBLIC OpenMP::OpenMP_CXX)
 endif()
 if(LibArchive_FOUND)
-  target_link_libraries(libtesseract PUBLIC ${LibArchive_LIBRARIES})
+  target_link_libraries(libtesseract PUBLIC LibArchive::LibArchive)
 endif(LibArchive_FOUND)
 if(CURL_FOUND)
   if(NOT CURL_LIBRARIES)
@@ -790,7 +790,7 @@ if(SW_BUILD)
     FILE ${CMAKE_CURRENT_BINARY_DIR}/TesseractTargets.cmake
     NAMESPACE Tesseract::)
 else()
-  target_link_libraries(libtesseract PUBLIC ${Leptonica_LIBRARIES})
+  target_link_libraries(libtesseract PUBLIC leptonica)
   export(
     TARGETS libtesseract
     FILE ${CMAKE_CURRENT_BINARY_DIR}/TesseractTargets.cmake
