--- cmake/AwsCFlags.cmake
+++ cmake/AwsCFlags.cmake
@@ -62,8 +62,8 @@
         # Warning disables always go last to avoid future flags re-enabling them
         list(APPEND AWS_C_FLAGS -Wno-long-long)
 
-        # Always enable position independent code, since this code will always end up in a shared lib
-        list(APPEND AWS_C_FLAGS -fPIC)
+        # Don't always set fPIC as conan will handle position independent code.
+        # list(APPEND AWS_C_FLAGS -fPIC)
 
         if (LEGACY_COMPILER_SUPPORT)
             list(APPEND AWS_C_FLAGS -Wno-strict-aliasing)
