diff --git a/CMakeLists.txt b/CMakeLists.txt
index d00a4f9..bc45784 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -197,12 +197,12 @@ if(ENABLE_LZMA)
 endif(ENABLE_LZMA)
 
 if(ENABLE_ZSTD)
-  find_package(Zstd 1.3.6)
-  if(Zstd_FOUND)
+  find_package(zstd)
+  if(zstd_FOUND)
     set(HAVE_LIBZSTD 1)
   else()
     message(WARNING "-- zstd library not found; zstandard support disabled")
-  endif(Zstd_FOUND)
+  endif(zstd_FOUND)
 endif(ENABLE_ZSTD)
 
 if (COMMONCRYPTO_FOUND)
diff --git a/lib/CMakeLists.txt b/lib/CMakeLists.txt
index 1213fa0..b6446cb 100644
--- a/lib/CMakeLists.txt
+++ b/lib/CMakeLists.txt
@@ -147,7 +147,7 @@ endif()
 
 if(HAVE_LIBZSTD)
   target_sources(zip PRIVATE zip_algorithm_zstd.c)
-  target_link_libraries(zip PRIVATE Zstd::Zstd)
+  target_link_libraries(zip PRIVATE $<IF:$<TARGET_EXISTS:zstd::libzstd_shared>,zstd::libzstd_shared,zstd::libzstd_static>)
 endif()
 
 if(HAVE_COMMONCRYPTO)
