--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -107,13 +107,6 @@
     endif()
 endif()
 
-if(NOT CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
-    if (CMAKE_BUILD_TYPE STREQUAL "Debug")
-        message("-- Use flag -fsanitize=address")
-        list(APPEND CXX_FLAGS "-fsanitize=address")
-    endif()
-endif()
-
 if("${CMAKE_SYSTEM_NAME}" STREQUAL "Linux") # uname -s
   if("${CMAKE_SYSTEM_PROCESSOR}" MATCHES "x86_64|aarch64") # uname -p
     set(UTHREAD ON)
