diff --git a/CMakeLists.txt b/CMakeLists.txt
index a19209d..581a308 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -19,26 +19,9 @@ set(libuvc_VERSION ${libuvc_VERSION_MAJOR}.${libuvc_VERSION_MINOR}.${libuvc_VERS
 set(libuvc_DESCRIPTION "A cross-platform library for USB video devices")
 set(libuvc_URL "https://github.com/ktossell/libuvc")
 
-find_package(PkgConfig)
-pkg_check_modules(LIBUSB libusb-1.0)
-
 # Try to find JPEG using a module or pkg-config. If that doesn't work, search for the header.
-find_package(jpeg QUIET)
-if(JPEG_FOUND)
-  set(JPEG_LINK_FLAGS ${JPEG_LIBRARIES})
-else()
-  pkg_check_modules(JPEG QUIET libjpeg)
-  if(JPEG_FOUND)
-      set(JPEG_INCLUDE_DIR ${JPEG_INCLUDE_DIRS})
-      set(JPEG_LINK_FLAGS ${JPEG_LDFLAGS})
-  else()
-    find_path(JPEG_INCLUDE_DIR jpeglib.h)
-    if(JPEG_INCLUDE_DIR)
-      set(JPEG_FOUND ON)
-      set(JPEG_LINK_FLAGS -ljpeg)
-    endif()
-  endif()
-endif()
+find_package(JPEG QUIET)
+set(JPEG_LINK_FLAGS ${JPEG_LIBRARIES})
 
 include(GNUInstallDirs)
 
@@ -54,7 +37,7 @@ SET(SOURCES src/ctrl.c src/ctrl-gen.c src/device.c src/diag.c
 include_directories(
   ${libuvc_SOURCE_DIR}/include
   ${libuvc_BINARY_DIR}/include
-  ${LIBUSB_INCLUDE_DIRS}
+  ${CONAN_INCLUDE_DIRS_LIBUSB}
 )
 
 if(JPEG_FOUND)
@@ -101,7 +84,7 @@ if(BUILD_UVC_SHARED)
     target_link_libraries (uvc ${JPEG_LINK_FLAGS})
   endif(JPEG_FOUND)
 
-  target_link_libraries(uvc ${LIBUSB_LIBRARIES})
+  target_link_libraries(uvc ${CONAN_LIBS_LIBUSB})
 
   #add_executable(test src/test.c)
   #target_link_libraries(test uvc ${LIBUSB_LIBRARIES} opencv_highgui
