diff --git a/include/valijson/adapters/nlohmann_json_adapter.hpp b/include/valijson/adapters/nlohmann_json_adapter.hpp
index 49d27dd..5541adc 100644
--- a/include/valijson/adapters/nlohmann_json_adapter.hpp
+++ b/include/valijson/adapters/nlohmann_json_adapter.hpp
@@ -26,7 +26,7 @@
 #pragma once
 
 #include <string>
-#include <json.hpp>
+#include <nlohmann/json.hpp>
 
 #include <valijson/adapters/adapter.hpp>
 #include <valijson/adapters/basic_adapter.hpp>
diff --git a/include/valijson/utils/nlohmann_json_utils.hpp b/include/valijson/utils/nlohmann_json_utils.hpp
index 8d03899..15edaba 100644
--- a/include/valijson/utils/nlohmann_json_utils.hpp
+++ b/include/valijson/utils/nlohmann_json_utils.hpp
@@ -2,7 +2,7 @@
 
 #include <iostream>
 
-#include <json.hpp>
+#include <nlohmann/json.hpp>
 #include <valijson/utils/file_utils.hpp>
 
 namespace valijson {
