diff --git a/lang/c++/api/buffer/detail/BufferDetail.hh b/lang/c++/api/buffer/detail/BufferDetail.hh
index c481c07..e4002ac 100644
--- a/lang/c++/api/buffer/detail/BufferDetail.hh
+++ b/lang/c++/api/buffer/detail/BufferDetail.hh
@@ -30,6 +30,7 @@
 #include <exception>
 #include <cassert>
 #include <deque>
+#include <iterator>
 
 /** 
  * \file BufferDetail.hh
diff --git a/lang/c++/impl/DataFile.cc b/lang/c++/impl/DataFile.cc
index 035dd27..864731f 100644
--- a/lang/c++/impl/DataFile.cc
+++ b/lang/c++/impl/DataFile.cc
@@ -20,6 +20,7 @@
 #include "Compiler.hh"
 #include "Exception.hh"
 
+#include <iterator>
 #include <sstream>
 
 #include <boost/random/mersenne_twister.hpp>
diff --git a/lang/c++/impl/Stream.cc b/lang/c++/impl/Stream.cc
index 5da5edb..d702841 100644
--- a/lang/c++/impl/Stream.cc
+++ b/lang/c++/impl/Stream.cc
@@ -18,6 +18,7 @@
 
 #include "Stream.hh"
 #include <vector>
+#include <iterator>
 
 namespace avro {
 
diff --git a/lang/c++/impl/parsing/JsonCodec.cc b/lang/c++/impl/parsing/JsonCodec.cc
index ead44ca..65ba10c 100644
--- a/lang/c++/impl/parsing/JsonCodec.cc
+++ b/lang/c++/impl/parsing/JsonCodec.cc
@@ -18,6 +18,7 @@
 
 #define __STDC_LIMIT_MACROS
 
+#include <iterator>
 #include <string>
 #include <map>
 #include <algorithm>
diff --git a/lang/c++/impl/parsing/ResolvingDecoder.cc b/lang/c++/impl/parsing/ResolvingDecoder.cc
index e0d25ed..73b011d 100644
--- a/lang/c++/impl/parsing/ResolvingDecoder.cc
+++ b/lang/c++/impl/parsing/ResolvingDecoder.cc
@@ -18,6 +18,7 @@
 
 #define __STDC_LIMIT_MACROS
 
+#include <iterator>
 #include <string>
 #include <stack>
 #include <map>
diff --git a/lang/c++/impl/parsing/ValidatingCodec.cc b/lang/c++/impl/parsing/ValidatingCodec.cc
index a401afc..391f810 100644
--- a/lang/c++/impl/parsing/ValidatingCodec.cc
+++ b/lang/c++/impl/parsing/ValidatingCodec.cc
@@ -18,6 +18,7 @@
 
 #include "ValidatingCodec.hh"
 
+#include <iterator>
 #include <string>
 #include <map>
 #include <algorithm>
diff --git a/lang/c++/test/CodecTests.cc b/lang/c++/test/CodecTests.cc
index c0ca1e0..10c2ae1 100644
--- a/lang/c++/test/CodecTests.cc
+++ b/lang/c++/test/CodecTests.cc
@@ -16,6 +16,7 @@
  * limitations under the License.
  */
 
+#include <iterator>
 #include <iostream>
 
 #include "Encoder.hh"
