diff --git a/ThirdParty/doubleconversion/vtk_doubleconversion.h.in b/ThirdParty/doubleconversion/vtk_doubleconversion.h.in
index afe28ba6c8..cb7512f7d1 100644
--- a/ThirdParty/doubleconversion/vtk_doubleconversion.h.in
+++ b/ThirdParty/doubleconversion/vtk_doubleconversion.h.in
@@ -19,7 +19,7 @@
 #cmakedefine01 VTK_MODULE_USE_EXTERNAL_vtkdoubleconversion
 
 #if VTK_MODULE_USE_EXTERNAL_vtkdoubleconversion
-# define VTK_DOUBLECONVERSION_HEADER(header) <header>
+# define VTK_DOUBLECONVERSION_HEADER(header) <double-conversion/header>
 #else
 # define VTK_DOUBLECONVERSION_HEADER(header) <vtkdoubleconversion/double-conversion/header>
 #endif
diff --git a/ThirdParty/eigen/CMakeLists.txt b/ThirdParty/eigen/CMakeLists.txt
index 1edabc47a2..aa922dc634 100644
--- a/ThirdParty/eigen/CMakeLists.txt
+++ b/ThirdParty/eigen/CMakeLists.txt
@@ -6,7 +6,7 @@ vtk_module_third_party(
     STANDARD_INCLUDE_DIRS
   EXTERNAL
     PACKAGE Eigen3
-    TARGETS Eigen3::Eigen3
+    TARGETS Eigen3::Eigen
     STANDARD_INCLUDE_DIRS)
 
 configure_file(
diff --git a/ThirdParty/exprtk/CMakeLists.txt b/ThirdParty/exprtk/CMakeLists.txt
index 3a8919aea4..06ecd6f860 100644
--- a/ThirdParty/exprtk/CMakeLists.txt
+++ b/ThirdParty/exprtk/CMakeLists.txt
@@ -2,11 +2,11 @@ vtk_module_third_party(
   INTERNAL
     HEADER_ONLY
     LICENSE_FILES "vtkexprtk/License.txt"
-    VERSION       "2.71"
+    VERSION       "0.0.1"
   EXTERNAL
-    PACKAGE ExprTk
-    VERSION 2.7
-    TARGETS ExprTk::ExprTk
+    PACKAGE exprtk
+    VERSION 0.0.1
+    TARGETS exprtk::exprtk
     STANDARD_INCLUDE_DIRS)
 
 configure_file(
diff --git a/ThirdParty/libproj/CMakeLists.txt b/ThirdParty/libproj/CMakeLists.txt
index 39e1859bb8..a8847b47ca 100644
--- a/ThirdParty/libproj/CMakeLists.txt
+++ b/ThirdParty/libproj/CMakeLists.txt
@@ -4,13 +4,15 @@ vtk_module_third_party(
     VERSION       "4.9.3"
     STANDARD_INCLUDE_DIRS
   EXTERNAL
-    PACKAGE LibPROJ
-    TARGETS LibPROJ::LibPROJ
-    USE_VARIABLES LibPROJ_MAJOR_VERSION
+    PACKAGE proj
+    TARGETS PROJ::proj
+    USE_VARIABLES proj_VERSION_MAJOR
     STANDARD_INCLUDE_DIRS)
 
 if (NOT VTK_MODULE_USE_EXTERNAL_VTK_libproj)
   set(LibPROJ_MAJOR_VERSION "4")
+else ()
+  set(LibPROJ_MAJOR_VERSION "${proj_VERSION_MAJOR}")
 endif ()
 
 configure_file(
diff --git a/ThirdParty/lz4/CMakeLists.txt b/ThirdParty/lz4/CMakeLists.txt
index 90ab696000..f768ea9a2e 100644
--- a/ThirdParty/lz4/CMakeLists.txt
+++ b/ThirdParty/lz4/CMakeLists.txt
@@ -4,8 +4,8 @@ vtk_module_third_party(
     VERSION       "1.9.3"
     STANDARD_INCLUDE_DIRS
   EXTERNAL
-    PACKAGE LZ4
-    TARGETS LZ4::LZ4
+    PACKAGE lz4
+    TARGETS lz4::lz4
     STANDARD_INCLUDE_DIRS)
 
 configure_file(
diff --git a/ThirdParty/lzma/CMakeLists.txt b/ThirdParty/lzma/CMakeLists.txt
index 882d538a0a..4eb8984bf0 100644
--- a/ThirdParty/lzma/CMakeLists.txt
+++ b/ThirdParty/lzma/CMakeLists.txt
@@ -4,12 +4,12 @@ vtk_module_third_party(
     VERSION       "5.2.5"
     STANDARD_INCLUDE_DIRS
   EXTERNAL
-    PACKAGE LZMA
-    TARGETS LZMA::LZMA
+    PACKAGE LibLZMA
+    TARGETS LibLZMA::LibLZMA
     STANDARD_INCLUDE_DIRS)
 
 include(vtkDetectLibraryType)
-vtk_detect_library_shared(vtklzma LZMA::LZMA)
+vtk_detect_library_shared(vtklzma LibLZMA::LibLZMA)
 
 set(LZMA_BUILT_AS_DYNAMIC_LIB "${vtklzma_is_shared}")
 
diff --git a/ThirdParty/utf8/CMakeLists.txt b/ThirdParty/utf8/CMakeLists.txt
index 508d17f6fb..290dc881af 100644
--- a/ThirdParty/utf8/CMakeLists.txt
+++ b/ThirdParty/utf8/CMakeLists.txt
@@ -6,7 +6,7 @@ vtk_module_third_party(
     STANDARD_INCLUDE_DIRS
   EXTERNAL
     PACKAGE utf8cpp
-    TARGETS utf8cpp::utf8cpp
+    TARGETS utf8cpp
     STANDARD_INCLUDE_DIRS)
 
 configure_file(
diff --git a/ThirdParty/freetype/CMakeLists.txt b/ThirdParty/freetype/CMakeLists.txt
index af875da128..14bd5b4a65 100644
--- a/ThirdParty/freetype/CMakeLists.txt
+++ b/ThirdParty/freetype/CMakeLists.txt
@@ -5,12 +5,12 @@ vtk_module_third_party(
     VERSION       "2.11.0"
     STANDARD_INCLUDE_DIRS
   EXTERNAL
-    PACKAGE Freetype
-    TARGETS Freetype::Freetype
+    PACKAGE freetype
+    TARGETS freetype
     STANDARD_INCLUDE_DIRS)
 
 include(vtkDetectLibraryType)
-vtk_detect_library_shared(vtkfreetype Freetype::Freetype)
+vtk_detect_library_shared(vtkfreetype freetype)
 
 set(VTK_FREETYPE_SHARED "${vtkfreetype_is_shared}")

diff --git a/ThirdParty/expat/CMakeLists.txt b/ThirdParty/expat/CMakeLists.txt
index 9de83911d1..ab7e0b2f91 100644
--- a/ThirdParty/expat/CMakeLists.txt
+++ b/ThirdParty/expat/CMakeLists.txt
@@ -4,12 +4,12 @@ vtk_module_third_party(
     VERSION       "2.4.1"
     STANDARD_INCLUDE_DIRS
   EXTERNAL
-    PACKAGE EXPAT
-    TARGETS EXPAT::EXPAT
+    PACKAGE expat
+    TARGETS expat::expat
     STANDARD_INCLUDE_DIRS)

 include(vtkDetectLibraryType)
-vtk_detect_library_shared(vtkexpat EXPAT::EXPAT)
+vtk_detect_library_shared(vtkexpat expat::expat)
 
 if (vtkexpat_is_shared)
   set(XML_STATIC 0)
diff --git a/ThirdParty/theora/CMakeLists.txt b/ThirdParty/theora/CMakeLists.txt
index 4820607e54..43f86fbbe5 100644
--- a/ThirdParty/theora/CMakeLists.txt
+++ b/ThirdParty/theora/CMakeLists.txt
@@ -4,10 +4,10 @@ vtk_module_third_party(
     VERSION       "1.1.1"
     STANDARD_INCLUDE_DIRS
   EXTERNAL
-    PACKAGE THEORA
-    TARGETS THEORA::THEORA
-            THEORA::DEC
-            THEORA::ENC
+    PACKAGE theora
+    TARGETS theora::theora
+            theora::theoradec
+            theora::theoraenc
     STANDARD_INCLUDE_DIRS)
 
 configure_file(
diff --git a/ThirdParty/cgns/CMakeLists.txt b/ThirdParty/cgns/CMakeLists.txt
index 610e0d3c25..ef3f224957 100644
--- a/ThirdParty/cgns/CMakeLists.txt
+++ b/ThirdParty/cgns/CMakeLists.txt
@@ -4,9 +4,9 @@ vtk_module_third_party(
     VERSION       "4.1.2"
     STANDARD_INCLUDE_DIRS
   EXTERNAL
-    PACKAGE CGNS
-    VERSION       "4.1.1"
-    TARGETS CGNS::CGNS
+    PACKAGE cgns
+    VERSION       "4.3.0"
+    TARGETS cgns::cgns
     STANDARD_INCLUDE_DIRS)
 
 configure_file(
diff --git a/Rendering/OpenVR/CMakeLists.txt b/Rendering/OpenVR/CMakeLists.txt
index de0f21e0fa..7f8d28c975 100644
--- a/Rendering/OpenVR/CMakeLists.txt
+++ b/Rendering/OpenVR/CMakeLists.txt
@@ -1,4 +1,4 @@
-vtk_module_find_package(PACKAGE OpenVR)
+vtk_module_find_package(PACKAGE openvr)
 
 set(classes
   vtkOpenVRCamera
@@ -23,7 +23,7 @@ vtk_module_add_module(VTK::RenderingOpenVR
   PRIVATE_HEADERS "${dashboard_header}" ${vtk_object_factory_header} ${geometry_headers})
 vtk_module_link(VTK::RenderingOpenVR
   PUBLIC
-    OpenVR::OpenVR)
+    openvr::openvr)
 
 set(openvr_input_files
   vtk_openvr_actions.json
