diff --git a/GUISupport/Qt/CMakeLists.txt b/GUISupport/Qt/CMakeLists.txt
index 18f35fd6f0..36c4f7a480 100644
--- a/GUISupport/Qt/CMakeLists.txt
+++ b/GUISupport/Qt/CMakeLists.txt
@@ -42,7 +42,6 @@ endif()
 
 vtk_module_find_package(
   PACKAGE     "Qt${vtk_qt_major_version}"
-  VERSION     5.9
   COMPONENTS  ${qt_components}
   FORWARD_VERSION_REQ MINOR
   VERSION_VAR         "Qt${vtk_qt_major_version}_VERSION")
diff --git a/GUISupport/QtQuick/CMakeLists.txt b/GUISupport/QtQuick/CMakeLists.txt
index 8d049be6c2..cbee30b950 100644
--- a/GUISupport/QtQuick/CMakeLists.txt
+++ b/GUISupport/QtQuick/CMakeLists.txt
@@ -15,7 +15,6 @@ set(qt_components
 
 vtk_module_find_package(
   PACKAGE     "Qt${vtk_qt_major_version}"
-  VERSION     5.9
   COMPONENTS  ${qt_components}
   FORWARD_VERSION_REQ MINOR
   VERSION_VAR         "Qt${vtk_qt_major_version}_VERSION")
