diff --git a/CMake/vtkInstallCMakePackage.cmake b/CMake/vtkInstallCMakePackage.cmake
index dd8f7300ab..846a4b0cd9 100644
--- a/CMake/vtkInstallCMakePackage.cmake
+++ b/CMake/vtkInstallCMakePackage.cmake
@@ -70,36 +70,7 @@ configure_file(
   COPYONLY)
 
 set(vtk_cmake_module_files
-  Finddouble-conversion.cmake
-  FindEigen3.cmake
-  FindEXPAT.cmake
-  FindExprTk.cmake
-  FindFFMPEG.cmake
-  FindFontConfig.cmake
-  FindFreetype.cmake
-  FindGL2PS.cmake
-  FindGLEW.cmake
-  FindJOGL.cmake
-  FindJsonCpp.cmake
-  FindLibHaru.cmake
-  FindLibPROJ.cmake
-  FindLibXml2.cmake
-  FindLZ4.cmake
-  FindLZMA.cmake
-  Findmpi4py.cmake
-  FindMySQL.cmake
-  FindNetCDF.cmake
-  FindODBC.cmake
-  FindOGG.cmake
-  FindOpenSlide.cmake
-  FindOpenVR.cmake
-  FindOSMesa.cmake
-  FindPEGTL.cmake
-  FindSDL2.cmake
-  FindTBB.cmake
-  FindTHEORA.cmake
-  Findutf8cpp.cmake
-  FindCGNS.cmake
+   # ALL VTK bundled ThirdParty Finders should be removed for the conan situation
 
   vtkCMakeBackports.cmake
   vtkDetectLibraryType.cmake
@@ -121,25 +92,9 @@ set(vtk_cmake_module_files
   vtk-use-file-deprecated.cmake
   vtk-use-file-error.cmake)
 set(vtk_cmake_patch_files
-  patches/3.13/FindZLIB.cmake
-  patches/3.16/FindPostgreSQL.cmake
-  patches/3.18/FindPython/Support.cmake
-  patches/3.18/FindPython2.cmake
-  patches/3.18/FindPython3.cmake
-  patches/3.19/FindJPEG.cmake
-  patches/3.19/FindLibArchive.cmake
-  patches/3.19/FindSQLite3.cmake
-  patches/3.19/FindX11.cmake
-  patches/3.20/FindGDAL.cmake
-  patches/3.22/FindMPI/fortranparam_mpi.f90.in
-  patches/3.22/FindMPI/libver_mpi.c
-  patches/3.22/FindMPI/libver_mpi.f90.in
-  patches/3.22/FindMPI/mpiver.f90.in
-  patches/3.22/FindMPI/test_mpi.c
-  patches/3.22/FindMPI/test_mpi.f90.in
-  patches/3.22/FindMPI.cmake
-  patches/99/FindHDF5.cmake
-  patches/99/FindOpenGL.cmake)
+   # ALL patches to support old CMake versions should be removed,
+   # we will require a CMake that is recent.
+  )
 
 set(vtk_cmake_files_to_install)
 foreach (vtk_cmake_module_file IN LISTS vtk_cmake_module_files vtk_cmake_patch_files)
