diff --git a/ThirdParty/cgns/vtkcgns/CMakeLists.txt b/ThirdParty/cgns/vtkcgns/CMakeLists.txt
index ef06267fc6..dc4cbb083d 100644
--- a/ThirdParty/cgns/vtkcgns/CMakeLists.txt
+++ b/ThirdParty/cgns/vtkcgns/CMakeLists.txt
@@ -399,7 +399,7 @@ if (CGNS_ENABLE_HDF5 AND HDF5_NEED_MPI)
   set(CGNS_ENABLE_PARALLEL "OFF" CACHE BOOL "Enable or disable parallel interface ?")
   mark_as_advanced(CLEAR CGNS_ENABLE_PARALLEL)
   # Check that HDF5 has parallel support
-  if (NOT (HDF5_IS_PARALLEL OR HDF5_ENABLE_PARALLEL))
+  if (NOT HDF5_ENABLE_PARALLEL)
     message(FATAL_ERROR "HDF5 has been found, but is missing parallel support.")
   endif()
 else (CGNS_ENABLE_HDF5 AND HDF5_NEED_MPI)
diff --git a/ThirdParty/hdf5/CMakeLists.txt b/ThirdParty/hdf5/CMakeLists.txt
index 1d9158af79..d71bfbf6b8 100644
--- a/ThirdParty/hdf5/CMakeLists.txt
+++ b/ThirdParty/hdf5/CMakeLists.txt
@@ -12,18 +12,18 @@ vtk_module_third_party(
     COMPONENTS    C HL
     TARGETS       hdf5::hdf5
                   hdf5::hdf5_hl
-    USE_VARIABLES HDF5_IS_PARALLEL
+                  USE_VARIABLES HDF5_ENABLE_PARALLEL
     STANDARD_INCLUDE_DIRS)
 
-set(vtkhdf5_is_parallel 0)
-if (VTK_MODULE_USE_EXTERNAL_vtkhdf5 AND HDF5_IS_PARALLEL)
+set(vtkhdf5_enable_parallel 0)
+if (VTK_MODULE_USE_EXTERNAL_vtkhdf5 AND HDF5_ENABLE_PARALLEL)
   if (NOT TARGET VTK::mpi)
     message(FATAL_ERROR
       "An external MPI-aware HDF5 requires that VTK be built with MPI support "
       "as well.")
   endif ()
 
-  set(vtkhdf5_is_parallel 1)
+  set(vtkhdf5_enable_parallel 1)
   vtk_module_link(VTK::hdf5
     INTERFACE
       VTK::mpi)
diff --git a/ThirdParty/hdf5/vtk_hdf5.h.in b/ThirdParty/hdf5/vtk_hdf5.h.in
index c4738b2cbf..6be213264a 100644
--- a/ThirdParty/hdf5/vtk_hdf5.h.in
+++ b/ThirdParty/hdf5/vtk_hdf5.h.in
@@ -21,11 +21,11 @@
 #define H5_USE_110_API
 
 #if VTK_MODULE_USE_EXTERNAL_vtkhdf5
-#cmakedefine01 vtkhdf5_is_parallel
-#if vtkhdf5_is_parallel
+#cmakedefine01 vtkhdf5_enable_parallel
+#if vtkhdf5_enable_parallel
 #include <vtk_mpi.h>
 #endif
-#undef vtkhdf5_is_parallel
+#undef vtkhdf5_enable_parallel
 
 # include <hdf5.h>
 # include <hdf5_hl.h>
diff --git a/ThirdParty/xdmf3/vtkxdmf3/core/CMakeLists.txt b/ThirdParty/xdmf3/vtkxdmf3/core/CMakeLists.txt
index 09187666ac..786be33010 100644
--- a/ThirdParty/xdmf3/vtkxdmf3/core/CMakeLists.txt
+++ b/ThirdParty/xdmf3/vtkxdmf3/core/CMakeLists.txt
@@ -34,7 +34,7 @@ if(HDF5_FOUND)
   include_directories(${HDF5_C_INCLUDE_DIR})
   # FIXME: Would like to get this info from HDF5 so we don't have conflicting
   # MPI versions
-  if(HDF5_IS_PARALLEL)
+  if(HDF5_ENABLE_PARALLEL)
     if (NOT MPI_FOUND)
       find_package(MPI REQUIRED)
       if(MPI_FOUND)
