diff -ru Rendering/OpenGL2/vtkTextureObject.cxx Rendering/OpenGL2/vtkTextureObject.cxx
--- Rendering/OpenGL2/vtkTextureObject.cxx	2023-06-20 10:23:31.345709784 +0800
+++ Rendering/OpenGL2/vtkTextureObject.cxx	2023-05-18 13:06:11.900887277 +0800
@@ -435,6 +435,11 @@
         target = GL_TEXTURE_BINDING_CUBE_MAP;
         break;
 #endif
+#if defined(GL_TEXTURE_2D_ARRAY) && defined(GL_TEXTURE_BINDING_2D_ARRAY)
+      case GL_TEXTURE_2D_ARRAY:
+        target = GL_TEXTURE_BINDING_2D_ARRAY;
+        break;
+#endif
       default:
         assert("check: impossible case" && 0);
         break;
