diff --git a/ThirdParty/netcdf/vtknetcdf/CMakeLists.txt b/ThirdParty/netcdf/vtknetcdf/CMakeLists.txt
index daf4a218b84549340006df4e2775fff27a464780..af8c272e7b82f4eee5da69d296ac9eb0bd3547f3 100644
--- a/ThirdParty/netcdf/vtknetcdf/CMakeLists.txt
+++ b/ThirdParty/netcdf/vtknetcdf/CMakeLists.txt
@@ -143,6 +143,7 @@ check_symbol_exists("strdup" "string.h" HAVE_STRDUP)
 check_symbol_exists("strlcat" "string.h" HAVE_STRLCAT)
 check_symbol_exists("st_blocksize" "sys/stat.h" HAVE_STRUCT_STAT_ST_BLKSIZE)
 check_symbol_exists("sysconf" "unistd.h" HAVE_SYSCONF)
+check_symbol_exists("snprintf" "stdio.h" HAVE_SNPRINTF)
 
 set(HAVE_HDF5_H 1)
 check_symbol_exists("H5free_memory" "vtk_hdf5.h" HDF5_HAS_H5FREE)
