diff --git a/CMakeLists.txt b/CMakeLists.txt
index c298a2e..bc0b6c1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -5,6 +5,7 @@
 # Author: Mateusz Loskot <mateusz@loskot.net>
 #
 ###############################################################################
+CMAKE_MINIMUM_REQUIRED(VERSION 3.1)
 PROJECT(GeoTIFF)
 
 SET(GEOTIFF_LIB_NAME geotiff)
@@ -12,7 +13,6 @@ SET(GEOTIFF_LIBRARY_TARGET geotiff_library)
 
 ##############################################################################
 # CMake settings
-CMAKE_MINIMUM_REQUIRED(VERSION 2.6.0)
 
 SET(CMAKE_COLOR_MAKEFILE ON)
 
@@ -54,7 +54,7 @@ IF(CMAKE_BUILD_TYPE MATCHES Debug)
     SET(GEOTIFF_BUILD_PEDANTIC TRUE)
 ENDIF()
 
-if (CMAKE_MAJOR_VERSION GREATER 2)
+if(0) # No ! we want CMP0042 NEW for relocatable shared lib on macOS
 cmake_policy(SET CMP0022 OLD) # interface link libraries
 cmake_policy(SET CMP0042 OLD) # osx rpath
 endif()
@@ -80,8 +80,9 @@ SET(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/lib)
 
 IF(WIN32)
     IF(MSVC)
-        ADD_DEFINITIONS(-DBUILD_AS_DLL=1)
-        ADD_DEFINITIONS(/DW4)
+        IF(BUILD_SHARED_LIBS)
+            ADD_DEFINITIONS(-DBUILD_AS_DLL=1)
+        ENDIF()
         if (NOT (MSVC_VERSION VERSION_LESS 1400))
             ADD_DEFINITIONS(-D_CRT_SECURE_NO_DEPRECATE)
             ADD_DEFINITIONS(-D_CRT_SECURE_NO_WARNINGS)
@@ -92,7 +93,7 @@ IF(WIN32)
 ENDIF()
 
 IF(CMAKE_COMPILER_IS_GNUCC OR CMAKE_COMPILER_IS_GNUCXX)
-    SET(COMPILE_FLAGS "-fPIC -Wall -Wno-long-long")
+    SET(COMPILE_FLAGS "-Wall -Wno-long-long")
     SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${COMPILE_FLAGS} -std=c99")
     SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${COMPILE_FLAGS} -std=c++98")
     IF(GEOTIFF_BUILD_PEDANTIC)
@@ -171,7 +172,7 @@ IF(WITH_TIFF)
       FIND_PACKAGE(TIFF REQUIRED)
     endif ()
 
-    IF(TIFF_FOUND)
+    IF(0)
         # Confirm required API is available
         INCLUDE(CheckFunctionExists)
         SET(CMAKE_REQUIRED_LIBRARIES ${TIFF_LIBRARIES})
@@ -300,7 +301,7 @@ INSTALL(FILES ${GEOTIFF_LIB_HEADERS} DESTINATION include)
 ###############################################################################
 # Build libxtiff library
 
-ADD_SUBDIRECTORY(libxtiff)
+# ADD_SUBDIRECTORY(libxtiff)
 
 ###############################################################################
 # Build libgeotiff library
@@ -345,7 +346,7 @@ IF(UNIX)
       VERSION ${LINK_VERSION}
       SOVERSION ${LINK_SOVERSION}
       CLEAN_DIRECT_OUTPUT 1 )
-   if (APPLE)
+   if (0) # no custom install_name, we want @rpath
       set_target_properties(
         ${GEOTIFF_LIBRARY_TARGET}
         PROPERTIES
