--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -5,6 +5,7 @@
 # Author: Mateusz Loskot <mateusz@loskot.net>
 #
 ###############################################################################
+CMAKE_MINIMUM_REQUIRED(VERSION 3.1)
 PROJECT(GeoTIFF)
 
 SET(GEOTIFF_LIB_NAME geotiff)
@@ -12,7 +13,6 @@ SET(GEOTIFF_LIBRARY_TARGET geotiff_library)
 
 ##############################################################################
 # CMake settings
-CMAKE_MINIMUM_REQUIRED(VERSION 2.6.0)
 
 SET(CMAKE_COLOR_MAKEFILE ON)
 
@@ -54,7 +54,7 @@ IF(CMAKE_BUILD_TYPE MATCHES Debug)
     SET(GEOTIFF_BUILD_PEDANTIC TRUE)
 ENDIF()
 
-if (CMAKE_MAJOR_VERSION GREATER 2)
+if(0) # No ! we want CMP0042 NEW for relocatable shared lib on macOS
 cmake_policy(SET CMP0022 OLD) # interface link libraries
 cmake_policy(SET CMP0042 OLD) # osx rpath
 endif()
@@ -80,8 +80,9 @@ SET(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/lib)
 
 IF(WIN32)
     IF(MSVC)
-        ADD_DEFINITIONS(-DBUILD_AS_DLL=1)
-        ADD_DEFINITIONS(/DW4)
+        IF(BUILD_SHARED_LIBS)
+            ADD_DEFINITIONS(-DBUILD_AS_DLL=1)
+        ENDIF()
         if (NOT (MSVC_VERSION VERSION_LESS 1400))
             ADD_DEFINITIONS(-D_CRT_SECURE_NO_DEPRECATE)
             ADD_DEFINITIONS(-D_CRT_SECURE_NO_WARNINGS)
@@ -92,7 +93,7 @@ IF(WIN32)
 ENDIF()
 
 IF(CMAKE_COMPILER_IS_GNUCC OR CMAKE_COMPILER_IS_GNUCXX)
-    SET(COMPILE_FLAGS "-fPIC -Wall -Wno-long-long")
+    SET(COMPILE_FLAGS "-Wall -Wno-long-long")
     SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${COMPILE_FLAGS} -std=c99")
     SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${COMPILE_FLAGS} -std=c++98")
     IF(GEOTIFF_BUILD_PEDANTIC)
@@ -120,6 +121,7 @@ INCLUDE_DIRECTORIES(${PROJECT_SOURCE_DIR})
 INCLUDE_DIRECTORIES(${PROJECT_SOURCE_DIR}/libxtiff)
 
 # TIFF support - required, default=ON
+if(0)
 SET(WITH_TIFF TRUE CACHE BOOL "Choose if TIFF support should be built")
 
 FIND_PACKAGE(PROJ NO_MODULE QUIET)
@@ -192,6 +194,7 @@ IF(WITH_TIFF)
         ADD_DEFINITIONS(-DHAVE_TIFF=1)
     ENDIF(TIFF_FOUND)
 ENDIF(WITH_TIFF)
+endif()
 
 # Turn off TOWGS84 support
 SET(WITH_TOWGS84 TRUE CACHE BOOL "Build with TOWGS84 support")
@@ -300,7 +303,6 @@ INSTALL(FILES ${GEOTIFF_LIB_HEADERS} DESTINATION include)
 ###############################################################################
 # Build libxtiff library
 
-ADD_SUBDIRECTORY(libxtiff)
 
 ###############################################################################
 # Build libgeotiff library
@@ -345,7 +347,7 @@ IF(UNIX)
       VERSION ${LINK_VERSION}
       SOVERSION ${LINK_SOVERSION}
       CLEAN_DIRECT_OUTPUT 1 )
-   if (APPLE)
+   if (0) # no custom install_name, we want @rpath
       set_target_properties(
         ${GEOTIFF_LIBRARY_TARGET}
         PROPERTIES
@@ -361,11 +363,15 @@ ENDIF(UNIX)
 SET_TARGET_PROPERTIES(${GEOTIFF_LIBRARY_TARGET} PROPERTIES
    OUTPUT_NAME ${GEOTIFF_LIB_NAME})
 
-TARGET_LINK_LIBRARIES(${GEOTIFF_LIBRARY_TARGET}
-    ${TIFF_LIBRARIES}
-    ${PROJ_LIBRARIES}
-    ${ZLIB_LIBRARIES}
-    ${JPEG_LIBRARIES})
+find_package(TIFF REQUIRED)
+target_link_libraries(${GEOTIFF_LIBRARY_TARGET} TIFF::TIFF)
+find_package(proj4 QUIET CONFIG)
+if(TARGET PROJ4::proj)
+    target_link_libraries(${GEOTIFF_LIBRARY_TARGET} PROJ4::proj)
+else()
+    find_package(proj REQUIRED CONFIG)
+    target_link_libraries(${GEOTIFF_LIBRARY_TARGET} PROJ::proj)
+endif()
 
 # INSTALL(TARGETS ${GEOTIFF_ARCHIVE_TARGET} ${GEOTIFF_LIBRARY_TARGET}
 #	RUNTIME DESTINATION ${GEOTIFF_BIN_DIR}
