diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4862f86..0fbc867 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -92,7 +92,7 @@ IF(WIN32)
 ENDIF()
 
 IF(CMAKE_COMPILER_IS_GNUCC OR CMAKE_COMPILER_IS_GNUCXX)
-    SET(COMPILE_FLAGS "-fPIC -Wall -Wno-long-long")
+    SET(COMPILE_FLAGS "-Wall -Wno-long-long")
     SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${COMPILE_FLAGS} -std=c99")
     SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${COMPILE_FLAGS} -std=c++98")
     IF(GEOTIFF_BUILD_PEDANTIC)
@@ -168,7 +168,7 @@ IF(WITH_TIFF)
       FIND_PACKAGE(TIFF REQUIRED)
     endif ()
 
-    IF(TIFF_FOUND)
+    IF(0)
         # Confirm required API is available
         INCLUDE(CheckFunctionExists)
         SET(CMAKE_REQUIRED_LIBRARIES ${TIFF_LIBRARIES})
@@ -258,7 +258,7 @@ INSTALL(FILES ${GEOTIFF_LIB_HEADERS} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
 ###############################################################################
 # Build libxtiff library
 
-ADD_SUBDIRECTORY(libxtiff)
+# ADD_SUBDIRECTORY(libxtiff)
 
 ###############################################################################
 # Build libgeotiff library
@@ -302,7 +302,7 @@ if(UNIX)
       VERSION ${LINK_VERSION}
       SOVERSION ${LINK_SOVERSION}
       CLEAN_DIRECT_OUTPUT 1 )
-   if (APPLE)
+   if (0) # no custom install_name, we want @rpath
       set_target_properties(
         ${GEOTIFF_LIBRARY_TARGET}
         PROPERTIES
