--- CMakeLists.txt
+++ CMakeLists.txt
@@ -9,7 +9,7 @@

 project(squirrel C CXX)

-include_directories(${CMAKE_SOURCE_DIR}/include)
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}/include)

 if(CMAKE_COMPILER_IS_GNUCXX)
   set(SQ_FLAGS -fno-exceptions -fno-strict-aliasing -Wall -Wextra -pedantic -Wcast-qual)
@@ -55,7 +55,7 @@
   set_target_properties(squirrel sqstdlib PROPERTIES SOVERSION 0 VERSION 0.0.0)
 endif()

-if(DEFINED INSTALL_INC_DIR)
+
   set(SQ_PUB_HEADERS include/sqconfig.h
                      include/sqstdaux.h
                      include/sqstdblob.h
@@ -64,5 +64,5 @@
                      include/sqstdstring.h
                      include/sqstdsystem.h
                      include/squirrel.h)
-  install(FILES ${SQ_PUB_HEADERS} DESTINATION ${INSTALL_INC_DIR})
-endif()
+  install(FILES ${SQ_PUB_HEADERS} DESTINATION include)
+
--- sq/CMakeLists.txt
+++ sq/CMakeLists.txt
@@ -3,21 +3,7 @@
 target_link_libraries(sq squirrel sqstdlib)
 install(TARGETS sq RUNTIME DESTINATION ${INSTALL_BIN_DIR})

-if(NOT DEFINED DISABLE_STATIC)
-  add_executable(sq_static sq.c)
-  set_target_properties(sq_static PROPERTIES LINKER_LANGUAGE C)
-  target_link_libraries(sq_static squirrel_static sqstdlib_static)
-  install(TARGETS sq_static RUNTIME DESTINATION ${INSTALL_BIN_DIR})
-endif()
-
 if(DEFINED LONG_OUTPUT_NAMES)
   set_target_properties(sq PROPERTIES OUTPUT_NAME squirrel3)

-  if(NOT DEFINED DISABLE_STATIC)
-    set_target_properties(sq_static PROPERTIES OUTPUT_NAME squirrel3_static)
-  endif()
-endif()
-
-if(CMAKE_COMPILER_IS_GNUCXX AND (NOT DEFINED DISABLE_STATIC))
-  set_target_properties(sq_static PROPERTIES COMPILE_FLAGS "-static -Wl,-static")
 endif()
--- sqstdlib/CMakeLists.txt
+++ sqstdlib/CMakeLists.txt
@@ -7,21 +7,14 @@
                  sqstdstring.cpp
                  sqstdsystem.cpp)

-add_library(sqstdlib SHARED ${SQSTDLIB_SRC})
+add_library(sqstdlib ${SQSTDLIB_SRC})
 target_link_libraries(sqstdlib squirrel)
 install(TARGETS sqstdlib RUNTIME DESTINATION ${INSTALL_BIN_DIR}
                          LIBRARY DESTINATION ${INSTALL_LIB_DIR}
                          ARCHIVE DESTINATION ${INSTALL_LIB_DIR})

-if(NOT DEFINED DISABLE_STATIC)
-  add_library(sqstdlib_static STATIC ${SQSTDLIB_SRC})
-  install(TARGETS sqstdlib_static ARCHIVE DESTINATION ${INSTALL_LIB_DIR})
-endif()

 if(DEFINED LONG_OUTPUT_NAMES)
   set_target_properties(sqstdlib PROPERTIES OUTPUT_NAME sqstdlib3)

-  if(NOT DEFINED DISABLE_STATIC)
-    set_target_properties(sqstdlib_static PROPERTIES OUTPUT_NAME sqstdlib3_static)
-  endif()
 endif()
--- squirrel/CMakeLists.txt
+++ squirrel/CMakeLists.txt
@@ -11,20 +11,13 @@
                  sqtable.cpp
                  sqvm.cpp)

-add_library(squirrel SHARED ${SQUIRREL_SRC})
+add_library(squirrel ${SQUIRREL_SRC})
 install(TARGETS squirrel RUNTIME DESTINATION ${INSTALL_BIN_DIR}
                          LIBRARY DESTINATION ${INSTALL_LIB_DIR}
                          ARCHIVE DESTINATION ${INSTALL_LIB_DIR})

-if(NOT DEFINED DISABLE_STATIC)
-  add_library(squirrel_static STATIC ${SQUIRREL_SRC})
-  install(TARGETS squirrel_static ARCHIVE DESTINATION ${INSTALL_LIB_DIR})
-endif()

 if(DEFINED LONG_OUTPUT_NAMES)
   set_target_properties(squirrel PROPERTIES OUTPUT_NAME squirrel3)

-  if(NOT DEFINED DISABLE_STATIC)
-    set_target_properties(squirrel_static PROPERTIES OUTPUT_NAME squirrel3_static)
-  endif()
 endif()
