--- runtime/Cpp/runtime/src/FlatHashSet.h
+++ runtime/Cpp/runtime/src/FlatHashSet.h
@@ -48,9 +48,9 @@
   using FlatHashSet = absl::flat_hash_set<Key, Hash, Equal, Allocator>;
 #else
   template <typename Key,
-            typename Hash = typename std::unordered_set<Key>::hasher,
-            typename Equal = typename std::unordered_set<Key>::key_equal,
-            typename Allocator = typename std::unordered_set<Key>::allocator_type>
+            typename Hash = std::hash<Key>,
+            typename Equal = std::equal_to<Key>,
+            typename Allocator = std::allocator<Key>>
   using FlatHashSet = std::unordered_set<Key, Hash, Equal, Allocator>;
 #endif
