--- runtime/Cpp/runtime/CMakeLists.txt
+++ runtime/Cpp/runtime/CMakeLists.txt
@@ -75,10 +75,7 @@
   endif()
 endif()
 
-if(CMAKE_SYSTEM_NAME MATCHES "Linux")
-  target_link_libraries(antlr4_shared ${UUID_LIBRARIES})
-  target_link_libraries(antlr4_static ${UUID_LIBRARIES})
-elseif(APPLE)
+if(APPLE)
   target_link_libraries(antlr4_shared ${COREFOUNDATION_LIBRARY})
   target_link_libraries(antlr4_static ${COREFOUNDATION_LIBRARY})
 endif()
