diff --git a/CMakeLists.txt b/CMakeLists.txt
index 53c3ed1..f48be06 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -42,8 +42,9 @@ set_target_properties(djinni_support_lib PROPERTIES
 
 install(
   TARGETS djinni_support_lib
-  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
-  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+  ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+  RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
 )
 
 
