--- cmake/config-ix.cmake
+++ cmake/config-ix.cmake
@@ -166,6 +166,7 @@
         else()
           include_directories(${LIBXML2_INCLUDE_DIR})
         endif()
+        set(LIBXML2_LIBRARIES "xml2;iconv;z")
       endif()
     endif()
   endif()
@@ -319,7 +319,7 @@
     message(FATAL_ERROR "libffi includes are not found.")
   endif()
 
-  find_library(FFI_LIBRARY_PATH ffi PATHS ${FFI_LIBRARY_DIR})
+  find_library(FFI_LIBRARY_PATH NAMES ffi libffi PATHS ${FFI_LIBRARY_DIR})
   if( NOT FFI_LIBRARY_PATH )
     message(FATAL_ERROR "libffi is not found.")
   endif()
--- lib/WindowsManifest/CMakeLists.txt
+++ lib/WindowsManifest/CMakeLists.txt
@@ -8,16 +8,6 @@
 if(LIBXML2_LIBRARIES)
   target_link_libraries(LLVMWindowsManifest PUBLIC ${LIBXML2_LIBRARIES})
 
-  get_filename_component(xml2_library ${LIBXML2_LIBRARIES} NAME)
-  if (CMAKE_STATIC_LIBRARY_PREFIX AND
-      xml2_library MATCHES "^${CMAKE_STATIC_LIBRARY_PREFIX}.*${CMAKE_STATIC_LIBRARY_SUFFIX}$")
-    string(REGEX REPLACE "^${CMAKE_STATIC_LIBRARY_PREFIX}" "" xml2_library ${xml2_library})
-    string(REGEX REPLACE "${CMAKE_STATIC_LIBRARY_SUFFIX}$" "" xml2_library ${xml2_library})
-  elseif (CMAKE_SHARED_LIBRARY_PREFIX AND
-          xml2_library MATCHES "^${CMAKE_SHARED_LIBRARY_PREFIX}.*${CMAKE_SHARED_LIBRARY_SUFFIX}$")
-    string(REGEX REPLACE "^${CMAKE_SHARED_LIBRARY_PREFIX}" "" xml2_library ${xml2_library})
-    string(REGEX REPLACE "${CMAKE_SHARED_LIBRARY_SUFFIX}$" "" xml2_library ${xml2_library})
-  endif()
   set_property(TARGET LLVMWindowsManifest PROPERTY
-    LLVM_SYSTEM_LIBS ${xml2_library})
+      LLVM_SYSTEM_LIBS ${LIBXML2_LIBRARIES})
 endif()
