From afea23abf322fb7eba0f1bb8675e09b1039fb47d Mon Sep 17 00:00:00 2001
From: MHU-valantic <>
Date: Mon, 23 May 2022 18:08:35 +0200
Subject: [PATCH] Apply patch for #1024

https://github.com/civetweb/civetweb/issues/1024#issuecomment-1131456269

diff --git a/cmake/AddCCompilerFlag.cmake b/cmake/AddCCompilerFlag.cmake
index 5eaf890..a1a28a6 100644
--- a/cmake/AddCCompilerFlag.cmake
+++ b/cmake/AddCCompilerFlag.cmake
@@ -24,14 +24,15 @@ function(add_c_compiler_flag FLAG)
   string(REPLACE "+" "X" SANITIZED_FLAG ${SANITIZED_FLAG})
   string(REGEX REPLACE "[^A-Za-z_0-9]" "_" SANITIZED_FLAG ${SANITIZED_FLAG})
   string(REGEX REPLACE "_+" "_" SANITIZED_FLAG ${SANITIZED_FLAG})
-  check_c_compiler_flag(${SANITIZED_FLAG} NO_DIAGNOSTICS_PRODUCED)
-  if(${NO_DIAGNOSTICS_PRODUCED})
+  check_c_compiler_flag("${FLAG}" ${SANITIZED_FLAG})
+  if(${${SANITIZED_FLAG}})
+    set(${SANITIZED_FLAG} ON PARENT_SCOPE )
     set(VARIANT ${ARGV1})
     if(ARGV1)
       string(REGEX REPLACE "[^A-Za-z_0-9]" "_" VARIANT "${VARIANT}")
       string(TOUPPER "_${VARIANT}" VARIANT)
     endif()
-    set(CMAKE_C_FLAGS${VARIANT} "${CMAKE_C_FLAGS${VARIANT}} ${FLAG}" PARENT_SCOPE)
-  endif()
+      set(CMAKE_C_FLAGS${VARIANT} "${CMAKE_C_FLAGS${VARIANT}} ${FLAG}" PARENT_SCOPE)
+	endif()
 endfunction()
 
diff --git a/cmake/AddCXXCompilerFlag.cmake b/cmake/AddCXXCompilerFlag.cmake
index 6607c62..dbb4bf2 100644
--- a/cmake/AddCXXCompilerFlag.cmake
+++ b/cmake/AddCXXCompilerFlag.cmake
@@ -24,14 +24,14 @@ function(add_cxx_compiler_flag FLAG)
   string(REPLACE "+" "X" SANITIZED_FLAG ${SANITIZED_FLAG})
   string(REGEX REPLACE "[^A-Za-z_0-9]" "_" SANITIZED_FLAG ${SANITIZED_FLAG})
   string(REGEX REPLACE "_+" "_" SANITIZED_FLAG ${SANITIZED_FLAG})
-  check_cxx_compiler_flag(${SANITIZED_FLAG} NO_DIAGNOSTICS_PRODUCED)
-  if(${NO_DIAGNOSTICS_PRODUCED})
+  check_cxx_compiler_flag( "${FLAG}" ${SANITIZED_FLAG} )
+  if(${${SANITIZED_FLAG}})
     set(VARIANT ${ARGV1})
     if(ARGV1)
       string(REGEX REPLACE "[^A-Za-z_0-9]" "_" VARIANT "${VARIANT}")
       string(TOUPPER "_${VARIANT}" VARIANT)
     endif()
-    set(CMAKE_CXX_FLAGS${VARIANT} "${CMAKE_CXX_FLAGS${VARIANT}} ${FLAG}" PARENT_SCOPE)
-  endif()
+      set(CMAKE_CXX_FLAGS${VARIANT} "${CMAKE_CXX_FLAGS${VARIANT}} ${FLAG}" PARENT_SCOPE)
+	endif()
 endfunction()
 
