--- a/lib/internalexport_write.def
+++ b/lib/internalexport_write.def
@@ -12,13 +12,6 @@ EXPORTS
     ??0Resampler@basisu@@QEAA@HHHHW4Boundary_Op@01@MMPEBDPEAUContrib_List@01@2MMMM@Z
     ?put_line@Resampler@basisu@@QEAA_NPEBM@Z
     ?get_line@Resampler@basisu@@QEAAPEBMXZ
-    ?lodepng_error_text@@YAPEBDI@Z
-    ?lodepng_get_raw_size@@YA_KIIPEBULodePNGColorMode@@@Z
-    ?lodepng_decode@@YAIPEAPEAEPEAI1PEAULodePNGState@@PEBE_K@Z
-    ?lodepng_inspect@@YAIPEAI0PEAULodePNGState@@PEBE_K@Z
-    ?lodepng_chunk_find_const@@YAPEBEPEBE0QEBD@Z
-    ??0State@lodepng@@QEAA@XZ
-    ??1State@lodepng@@UEAA@XZ
     ??0jpeg_decoder_file_stream@jpgd@@QEAA@XZ
     ??1jpeg_decoder_file_stream@jpgd@@UEAA@XZ
     ?decompress_jpeg_image_from_stream@jpgd@@YAPEAEPEAVjpeg_decoder_stream@1@PEAH11HI@Z
