from https://github.com/KhronosGroup/KTX-Software/pull/422

--- a/utils/ktxapp.h
+++ b/utils/ktxapp.h
@@ -5,6 +5,9 @@
 // SPDX-License-Identifier: Apache-2.0
 
 #include <stdarg.h>
+#if (_MSVC_LANG >= 201703L || __cplusplus >= 201703L)
+#include <algorithm>
+#endif
 #include <vector>
 #include <ktx.h>
 
@@ -24,7 +27,7 @@
 using namespace std;
 
 // clamp is in std:: from c++17.
-#if ((defined(_MSVC_LANG) && _MSVC_LANG < 201703L) || __cplusplus < 201703L)
+#if !(_MSVC_LANG >= 201703L || __cplusplus >= 201703L)
 template <typename T> inline T clamp(T value, T low, T high) {
     return (value < low) ? low : ((value > high) ? high : value);
 }
