--- include/open62541/config.h.in	2024-11-25 13:09:34.000000000 +0100
+++ include/open62541/config.h.in	2025-01-23 12:09:26.931400699 +0100
@@ -122,6 +122,15 @@
 #endif
 
 /**
+ * QNX Flags
+ * -------------------
+ * Defining _QNX_SOURCE will ensure proper qnx types and other features 
+ * are available. */
+#if defined(__QNX__)
+# define _QNX_SOURCE 1
+#endif
+
+/**
  * POSIX Feature Flags
  * -------------------
  * These feature flags have to be set before including the first POSIX
