diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -987,8 +987,8 @@
     set(MDNSD_GENERATED_DIR "${PROJECT_BINARY_DIR}/src_generated/mdnsd")
     file(MAKE_DIRECTORY "${MDNSD_GENERATED_DIR}")
 
-    set(MDNSD_LIBMDNSD_HEADERS 1035.h xht.h sdtxt.h mdnsd.h)
-    set(MDNSD_LIBMDNSD_SOURCES 1035.c xht.c sdtxt.c mdnsd.c log.c)
+    set(MDNSD_LIBMDNSD_HEADERS 1035.h xht.h sdtxt.h mdnsd.h inet.h)
+    set(MDNSD_LIBMDNSD_SOURCES 1035.c xht.c sdtxt.c mdnsd.c log.c inet.c)
 
     foreach(_f IN LISTS MDNSD_LIBMDNSD_HEADERS MDNSD_LIBMDNSD_SOURCES)
         configure_file(
@@ -1015,6 +1015,7 @@
         ${MDNSD_GENERATED_DIR}/sdtxt.c
         ${MDNSD_GENERATED_DIR}/mdnsd.c
         ${MDNSD_GENERATED_DIR}/log.c
+        ${MDNSD_GENERATED_DIR}/inet.c
         PROPERTIES COMPILE_FLAGS
             "-Wno-c++-compat -Wno-variadic-macros -Wno-pedantic -Wno-implicit-function-declaration -Wno-nested-externs -Wno-int-to-pointer-cast")
     # ua_discovery_mdns.c includes mdnsd.h which has GCC-style named variadic macros
@@ -1027,12 +1028,14 @@
         ${MDNSD_GENERATED_DIR}/1035.h
         ${MDNSD_GENERATED_DIR}/xht.h
         ${MDNSD_GENERATED_DIR}/sdtxt.h
-        ${MDNSD_GENERATED_DIR}/mdnsd.h)
+        ${MDNSD_GENERATED_DIR}/mdnsd.h
+        ${MDNSD_GENERATED_DIR}/inet.h)
     list(APPEND lib_sources
         ${PROJECT_SOURCE_DIR}/src/server/ua_discovery_mdns.c
         ${MDNSD_GENERATED_DIR}/1035.c
         ${MDNSD_GENERATED_DIR}/xht.c
         ${MDNSD_GENERATED_DIR}/sdtxt.c
         ${MDNSD_GENERATED_DIR}/mdnsd.c
-        ${MDNSD_GENERATED_DIR}/log.c)
+        ${MDNSD_GENERATED_DIR}/log.c
+        ${MDNSD_GENERATED_DIR}/inet.c)
 elseif(UA_ENABLE_DISCOVERY_MULTICAST_AVAHI)
