diff --git a/src/server/ua_discovery_mdns.c b/src/server/ua_discovery_mdns.c
index 1111111..2222222 100644
--- a/src/server/ua_discovery_mdns.c
+++ b/src/server/ua_discovery_mdns.c
@@ -900,7 +900,9 @@
     int rr = message_parse(&mm, (unsigned char*)msg.data);
     if(rr == 0) { /* 0 = success in new mdnsd API */
-        struct sockaddr_in *sa = (struct sockaddr_in*)infoptr->ai_addr;
-        mdnsd_in(mdnsPrivateData.mdnsDaemon, &mm, sa->sin_addr, sa->sin_port);
+        inet_addr_t from;
+        memset(&from, 0, sizeof(inet_addr_t));
+        memcpy(&from, infoptr->ai_addr, infoptr->ai_addrlen);
+        mdnsd_in(mdnsPrivateData.mdnsDaemon, &mm, &from);
     }
     freeaddrinfo(infoptr);
 }
@@ -914,9 +916,8 @@
-    struct in_addr ip;
-    memset(&ip, 0, sizeof(struct in_addr));
+    inet_addr_t to;
+    memset(&to, 0, sizeof(inet_addr_t));
 
     struct message mm;
     memset(&mm, 0, sizeof(struct message));
 
-    unsigned short sport = 0;
-    while(mdnsd_out(mdnsPrivateData.mdnsDaemon, &mm, &ip, &sport) > 0) {
+    while(mdnsd_out(mdnsPrivateData.mdnsDaemon, &mm, &to) > 0) {
         int len = message_packet_len(&mm);
