diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2fd6ef29..269a9086 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -9,7 +9,6 @@ string(TOLOWER "${CMAKE_BUILD_TYPE}" BUILD_TYPE_LOWER_CASE)
 
 set(CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/tools/cmake")
 find_package(PythonInterp REQUIRED)
-find_package(Git)
 include(AssignSourceGroup)
 include(GNUInstallDirs)
 
@@ -33,12 +32,11 @@ set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
 # Version #
 ###########
 
-include(SetGitBasedVersion)
-
-# The version string will be automatically set based on the git describe output.
-# This will automate the version strings. Just create a new tag, and the version will be set correctly.
-set_open62541_version()
-MESSAGE(STATUS "open62541 Version: ${OPEN62541_VER_MAJOR}.${OPEN62541_VER_MINOR}.${OPEN62541_VER_PATCH}${OPEN62541_VER_LABEL}")
+set(OPEN62541_VER_MINOR "0" CACHE STRING "Set the major version. Used by conan")
+set(OPEN62541_VER_MAJOR "0" CACHE STRING "Set the minor version. Used by conan")
+set(OPEN62541_VER_PATCH "0" CACHE STRING "Set the patch version. Used by conan")
+set(OPEN62541_VER_LABEL "")
+set(OPEN62541_VER_COMMIT "${OPEN62541_VER_MAJOR}.${OPEN62541_VER_MINOR}.${OPEN62541_VER_PATCH}")
 
 #################
 # Build Options #
