--- a/librabbitmq/CMakeLists.txt
+++ b/librabbitmq/CMakeLists.txt
@@ -88,8 +88,8 @@ if (ENABLE_SSL_SUPPORT)
       amqp_openssl_bio.c
       amqp_openssl_bio.h
   )
-  include_directories(${OPENSSL_INCLUDE_DIR})
-  set(AMQP_SSL_LIBS ${OPENSSL_LIBRARIES})
+  include_directories(${OpenSSL_INCLUDE_DIR})
+  set(AMQP_SSL_LIBS ${OpenSSL_LIBRARIES})
   if (APPLE)
     # Apple has deprecated OpenSSL in 10.7+. This disables that warning.
     set_source_files_properties(${AMQP_SSL_SRCS}
