From 6952e852f3a6c13d1b3df8948aeca5be6e200dcd Mon Sep 17 00:00:00 2001
From: Martin Valgur <martin.valgur@gmail.com>
Date: Mon, 5 Aug 2024 18:22:38 +0300
Subject: [PATCH 2/2] cmake fixes

---
 glomap/CMakeLists.txt | 19 +++++++++++++++----
 1 file changed, 15 insertions(+), 4 deletions(-)

diff --git a/glomap/CMakeLists.txt b/glomap/CMakeLists.txt
index c71b966..721d410 100644
--- a/glomap/CMakeLists.txt
+++ b/glomap/CMakeLists.txt
@@ -74,19 +74,21 @@ else()
     target_link_libraries(glomap PUBLIC PoseLib)
 endif()
 
+find_package(CHOLMOD REQUIRED CONFIG)
+
 target_link_libraries(
     glomap
     PUBLIC
         Eigen3::Eigen
         Ceres::ceres
         ${BOOST_LIBRARIES}
-        ${SuiteSparse_CHOLMOD_LIBRARY}
+        SuiteSparse::CHOLMOD
 )
 target_include_directories(
     glomap
     PUBLIC
         ..
-        ${SuiteSparse_CHOLMOD_INCLUDE_DIR}
+        ${CHOLMOD_INCLUDE_DIR}
 )
 
 if(OPENMP_FOUND)
@@ -98,7 +100,6 @@ if(MSVC)
 else()
     target_compile_options(glomap PRIVATE
         -Wall
-        -Werror
         -Wno-sign-compare
         -Wno-unused-variable
     )
@@ -111,9 +112,19 @@ add_executable(glomap_main
     exe/global_mapper.cc)
 target_link_libraries(glomap_main glomap)
 
+include(GNUInstallDirs)
 set_target_properties(glomap_main PROPERTIES OUTPUT_NAME glomap)
-install(TARGETS glomap_main DESTINATION bin)
+install(TARGETS glomap_main DESTINATION ${CMAKE_INSTALL_BINDIR})
 
+install(TARGETS glomap
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+)
+install(DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/"
+    DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/glomap"
+    FILES_MATCHING PATTERN "*.h"
+)
 
 if(TESTS_ENABLED)
     add_executable(glomap_test
-- 
2.25.1

