diff --color -ur a/Makefile.in b/Makefile.in
--- Makefile.in	2021-05-21 15:56:41.000000000 -0300
+++ Makefile.in	2021-05-27 19:01:06.857704652 -0300
@@ -130,8 +130,7 @@
 @ANDROID_HOST_TRUE@	-I$(top_srcdir)/src/common/android/testing/include
 
 # Build as PIC on Linux, for linux_client_unittest_shlib
-@LINUX_HOST_TRUE@am__append_2 = -fPIC
-@LINUX_HOST_TRUE@am__append_3 = -fPIC
+
 libexec_PROGRAMS = $(am__EXEEXT_10)
 bin_PROGRAMS = $(am__EXEEXT_2) $(am__EXEEXT_3) $(am__EXEEXT_4)
 check_PROGRAMS = $(am__EXEEXT_5) $(am__EXEEXT_6) $(am__EXEEXT_7) \
