--- a/src/client/linux/handler/minidump_descriptor.h
+++ b/src/client/linux/handler/minidump_descriptor.h
@@ -30,6 +30,7 @@
 #ifndef CLIENT_LINUX_HANDLER_MINIDUMP_DESCRIPTOR_H_
 #define CLIENT_LINUX_HANDLER_MINIDUMP_DESCRIPTOR_H_
 
+#include <cstdint>
 #include <assert.h>
 #include <sys/types.h>
 
--- a/configure
+++ b/configure
@@ -7476,7 +7476,6 @@
 
 done
 
-as_fn_append WARN_CXXFLAGS " -Werror"
 ac_ext=c
 ac_cpp='$CPP $CPPFLAGS'
 ac_compile='$CC -c $CFLAGS $CPPFLAGS conftest.$ac_ext >&5'
