diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 5078d14..7ec17bc 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -180,4 +180,11 @@ endif()
 source_group(TREE "${CMAKE_CURRENT_SOURCE_DIR}" PREFIX "src" FILES ${BOX2D_SOURCE_FILES})
 source_group(TREE "${CMAKE_CURRENT_SOURCE_DIR}/../include" PREFIX "include" FILES ${BOX2D_API_FILES})
 
-install(TARGETS box2d)
+include(GNUInstallDirs)
+
+install(TARGETS box2d
+	RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+	LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+	ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+)
+install(FILES ${BOX2D_API_FILES} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/box2d)
