diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8c1390e..d6590d7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -53,7 +53,7 @@ set_property(GLOBAL PROPERTY USE_FOLDERS ON)
 set(CMAKE_VERBOSE_MAKEFILE ON)
 
 # The Box2D library uses simde https://github.com/simd-everywhere/simde
-add_subdirectory(extern/simde)
+# add_subdirectory(extern/simde)
 add_subdirectory(src)
 
 # This hides samples, test, and doxygen from apps that use box2d via FetchContent
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 5078d14..7ec17bc 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -78,7 +78,7 @@ add_library(box2d ${BOX2D_SOURCE_FILES} ${BOX2D_API_FILES})
 # turned this off to make Box2D easier to use without cmake
 # include(GenerateExportHeader)
 # generate_export_header(box2d)
-
+find_package(simde REQUIRED CONFIG)
 target_include_directories(box2d
   PUBLIC
     $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../include>
@@ -89,7 +89,7 @@ target_include_directories(box2d
 )
 
 # SIMDE is used to support SIMD math on multiple platforms
-target_link_libraries(box2d PRIVATE simde)
+target_link_libraries(box2d PRIVATE simde::simde)
 
 # Box2D uses C17
 set_target_properties(box2d PROPERTIES
diff --git a/src/contact_solver.c b/src/contact_solver.c
index b190501..c61e250 100644
--- a/src/contact_solver.c
+++ b/src/contact_solver.c
@@ -9,8 +9,8 @@
 #include "core.h"
 #include "solver_set.h"
 #include "world.h"
-#include "x86/avx2.h"
-#include "x86/fma.h"
+#include "simde/x86/avx2.h"
+#include "simde/x86/fma.h"
 
 // Soft contact constraints with sub-stepping support
 // http://mmacklin.com/smallsteps.pdf
diff --git a/src/contact_solver.h b/src/contact_solver.h
index e265e93..dcdc1fb 100644
--- a/src/contact_solver.h
+++ b/src/contact_solver.h
@@ -4,7 +4,7 @@
 #pragma once
 
 #include "solver.h"
-#include "x86/avx.h"
+#include "simde/x86/avx.h"
 
 typedef struct b2ContactSim b2ContactSim;
 
diff --git a/src/solver.c b/src/solver.c
index d83a471..a115417 100644
--- a/src/solver.c
+++ b/src/solver.c
@@ -17,7 +17,7 @@
 #include "world.h"
 
 // for mm_pause
-#include "x86/sse2.h"
+#include "simde/x86/sse2.h"
 
 #include <limits.h>
 #include <stdatomic.h>
