diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5887924..79151a0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -239,7 +239,7 @@ install(TARGETS ${executables}
 install(FILES shapefil.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
 
 # Install pkg-config file
-install(FILES "${SHAPELIB_PC}" DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
+#install(FILES "${SHAPELIB_PC}" DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
 
 if(BUILD_TESTING)
   # Set up tests:
