--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -485,7 +485,6 @@ if(BUILD_ONNX_PYTHON)
     target_link_libraries(onnx_cpp2py_export PRIVATE ${PYTHON_LIBRARIES})
     target_compile_options(onnx_cpp2py_export
                            PRIVATE /MP
-                                   /WX
                                    /wd4800 # disable warning type' : forcing
                                            # value to bool 'true' or 'false'
                                            # (performance warning)
@@ -545,7 +544,6 @@ endif()
 if(MSVC)
   target_compile_options(onnx_proto
                          PRIVATE /MP
-                                 /WX
                                  /wd4800 # disable warning type' : forcing value
                                          # to bool 'true' or 'false'
                                          # (performance warning)
@@ -563,7 +561,6 @@ if(MSVC)
                                  ${EXTRA_FLAGS})
   target_compile_options(onnx
                          PRIVATE /MP
-                                 /WX
                                  /wd4800 # disable warning type' : forcing value
                                          # to bool 'true' or 'false'
                                          # (performance warning)
