From https://github.com/onnx/onnx/pull/7524 and https://github.com/onnx/onnx/pull/7525
Backports visibility fixes

---
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3191bba..aeac87a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -444,8 +444,7 @@ add_library(onnx_object OBJECT ${ONNX_SRCS})
 add_dependencies(onnx_object onnx_proto_object)
 target_include_directories(onnx_object PUBLIC
   $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>)
-set_target_properties(onnx_object PROPERTIES CXX_VISIBILITY_PRESET hidden)
-set_target_properties(onnx_object PROPERTIES VISIBILITY_INLINES_HIDDEN ON)
+set_source_files_properties(${ONNX_SRCS} PROPERTIES CXX_VISIBILITY_PRESET hidden)
 target_include_directories(onnx_object PUBLIC $<BUILD_INTERFACE:${ONNX_ROOT}>)
 add_onnx_global_defines(onnx_object)
 target_link_libraries(onnx_proto_object PUBLIC ${LINKED_PROTOBUF_TARGET})
diff --git a/onnx/checker.h b/onnx/checker.h
index 9035ea3..87ea67d 100644
--- a/onnx/checker.h
+++ b/onnx/checker.h
@@ -158,29 +158,29 @@ void check_function(const FunctionProto& function, const CheckerContext&, const
 // Check schema compatibility for 2 opset versions for a given node.
 // Checks whether the schema for 2 versions is same, this is true when the opschema
 // does not change between versions.
-void check_opset_compatibility(
+ONNX_API void check_opset_compatibility(
     const NodeProto& node,
     const CheckerContext& ctx,
     const std::unordered_map<std::string, int>& func_opset_imports,
     const std::unordered_map<std::string, int>& model_opset_imports);
 
 // Checks all model local functions present in ModelProto
-void check_model_local_functions(
+ONNX_API void check_model_local_functions(
     const ModelProto& model,
     const CheckerContext& ctx,
     const LexicalScopeContext& parent_lex);
 
-void check_model(
+ONNX_API void check_model(
     const ModelProto& model,
     bool full_check = false,
     bool skip_opset_compatibility_check = false,
     bool check_custom_domain = false);
-void check_model(
+ONNX_API void check_model(
     const std::string& model_path,
     bool full_check = false,
     bool skip_opset_compatibility_check = false,
     bool check_custom_domain = false);
-std::string resolve_external_data_location(
+ONNX_API std::string resolve_external_data_location(
     const std::string& base_dir,
     const std::string& location,
     const std::string& tensor_name);
diff --git a/onnx/common/ir_pb_converter.h b/onnx/common/ir_pb_converter.h
index 7ae6843..aad3965 100644
--- a/onnx/common/ir_pb_converter.h
+++ b/onnx/common/ir_pb_converter.h
@@ -43,7 +43,7 @@ void ExportModelProto(ModelProto* p_m, const std::shared_ptr<Graph>& g);
 
 std::unique_ptr<Graph> ImportModelProto(const ModelProto& mp);
 
-ModelProto PrepareOutput(const ModelProto& mp_in);
+ONNX_API ModelProto PrepareOutput(const ModelProto& mp_in);
 
 void assertNonNull(const std::shared_ptr<Graph>& g);
 } // namespace ONNX_NAMESPACE
