--- CMakeLists.txt
+++ CMakeLists.txt
@@ -91,6 +91,14 @@ add_library(${PROJECT_NAME}
     ${HEADERS}
 )

+if(BUILD_SHARED_LIBS AND MSVC)
+  target_sources(${PROJECT_NAME} PRIVATE "tommath.def")
+endif()
+
+if(WIN32 AND NOT MSVC)
+  target_link_libraries(${PROJECT_NAME} PRIVATE crypt32)
+endif()
+
 target_include_directories(${PROJECT_NAME} PUBLIC
     $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
     $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/${PROJECT_NAME}>
