# They forgot to export the mocktracer dll

--- mocktracer/CMakeLists.txt
+++ mocktracer/CMakeLists.txt
@@ -21,6 +21,7 @@ if (BUILD_SHARED_LIBS)
   install(TARGETS opentracing_mocktracer
           COMPONENT DIST
           EXPORT OpenTracingTargets
+          RUNTIME DESTINATION ${LIB_INSTALL_DIR}
           LIBRARY DESTINATION ${LIB_INSTALL_DIR}
           ARCHIVE DESTINATION ${LIB_INSTALL_DIR})

