From b1c65354a8321e0288882b03b3313f26ed484d93 Mon Sep 17 00:00:00 2001
From: Sergey Bobrenok
Date: Sun, 22 Dec 2024 20:52:43 +0700
Subject: [PATCH] Make.Rules: Make compile tools configurable

Signed-off-by: Sergey Bobrenok
---
 Make.Rules | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/Make.Rules b/Make.Rules
index a6b7742..b345d24 100644
--- a/Make.Rules
+++ b/Make.Rules
@@ -65,11 +65,11 @@ LIBCAP_INCLUDES = -I$(KERNEL_HEADERS) -I$(topdir)/libcap/include
 DEFINES := -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64
 
 SUDO := sudo
-CC := $(CROSS_COMPILE)gcc
+CC ?= $(CROSS_COMPILE)gcc
 LD := $(CC) -Wl,-x -shared -Wl,-shared
-AR := $(CROSS_COMPILE)ar
-RANLIB := $(CROSS_COMPILE)ranlib
-OBJCOPY := $(CROSS_COMPILE)objcopy
+AR ?= $(CROSS_COMPILE)ar
+RANLIB ?= $(CROSS_COMPILE)ranlib
+OBJCOPY ?= $(CROSS_COMPILE)objcopy
 
 # Reference:
 #   CPPFLAGS used for building .o files from .c & .h files
-- 
2.49.0

