diff --git a/iceoryx_meta/CMakeLists.txt b/iceoryx_meta/CMakeLists.txt
index 726b7db0..bf3464e2 100644
--- a/iceoryx_meta/CMakeLists.txt
+++ b/iceoryx_meta/CMakeLists.txt
@@ -16,6 +16,8 @@
 # SPDX-License-Identifier: Apache-2.0
 cmake_minimum_required(VERSION 3.5)
 project(iceoryx_meta)
+include(${CONAN_BUILD_INFO_FILE})
+conan_basic_setup()
 
 if(NOT CMAKE_BUILD_TYPE)
   set(CMAKE_BUILD_TYPE Release)
