diff --git a/iceoryx_platform/cmake/IceoryxPackageHelper.cmake b/iceoryx_platform/cmake/IceoryxPackageHelper.cmake
index c597836..7c1e3f3 100644
--- a/iceoryx_platform/cmake/IceoryxPackageHelper.cmake
+++ b/iceoryx_platform/cmake/IceoryxPackageHelper.cmake
@@ -252,7 +252,6 @@ Macro(iox_add_executable)
         endif()
     endif()
 
-    set_target_properties( ${IOX_TARGET} PROPERTIES POSITION_INDEPENDENT_CODE ON )
 
     foreach(INTERFACE ${IOX_BUILD_INTERFACE})
         target_include_directories(${IOX_TARGET}
@@ -328,7 +327,6 @@ Macro(iox_add_library)
         SOVERSION ${PROJECT_VERSION_MAJOR}
     )
 
-    set_target_properties( ${IOX_TARGET} PROPERTIES POSITION_INDEPENDENT_CODE ON )
 
     if ( IOX_USE_C_LANGUAGE )
         target_compile_options(${IOX_TARGET} PRIVATE ${ICEORYX_C_FLAGS} ${ICEORYX_C_WARNINGS} ${ICEORYX_SANITIZER_FLAGS} ${ICEORYX_GRCOV_FLAGS})
