diff --git a/OpenHome/Media/Codec/AacFdkBase.cpp b/OpenHome/Media/Codec/AacFdkBase.cpp
index 65f45365..e870cfe9 100644
--- a/OpenHome/Media/Codec/AacFdkBase.cpp
+++ b/OpenHome/Media/Codec/AacFdkBase.cpp
@@ -7,7 +7,7 @@
 #include <string.h>
 
 extern "C" {
-#include <aacdecoder_lib.h>
+#include <fdk-aac/aacdecoder_lib.h>
 }
 
 
diff --git a/OpenHome/Media/Codec/AacFdkBase.h b/OpenHome/Media/Codec/AacFdkBase.h
index ade972ac..f9969605 100644
--- a/OpenHome/Media/Codec/AacFdkBase.h
+++ b/OpenHome/Media/Codec/AacFdkBase.h
@@ -4,7 +4,7 @@
 #include <OpenHome/Media/Codec/Container.h>
 
 extern "C" {
-#include <aacdecoder_lib.h>
+#include <fdk-aac/aacdecoder_lib.h>
 }
 
 
