diff --git a/OpenHome/Media/Codec/Mpeg4.cpp b/OpenHome/Media/Codec/Mpeg4.cpp
index ac0cc212..c45387e7 100755
--- a/OpenHome/Media/Codec/Mpeg4.cpp
+++ b/OpenHome/Media/Codec/Mpeg4.cpp
@@ -2452,6 +2452,7 @@ TUint SeekTable::AudioSampleFromCodecSample(TUint aCodecSample) const
     // Use entries from stts box to find audio sample that start at given codec sample;
     TUint totalCodecSamples = 0;
     TUint totalAudioSamples = 0;
+    (void) totalAudioSamples;
     for (TUint entry = 0; entry < iAudioSamplesPerSample.size(); entry++) {
         const TUint sampleCount = iAudioSamplesPerSample[entry].iSampleCount;
         const TUint audioSamples = iAudioSamplesPerSample[entry].iAudioSamples;
