diff --git a/CMakeLists.txt b/CMakeLists.txt
index ed51adc..b9427b8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -5,6 +5,9 @@
 cmake_minimum_required(VERSION 2.8.11)
 project (mFAST CXX)
 
+include(${PROJECT_BINARY_DIR}/../conanbuildinfo.cmake)
+conan_basic_setup()
+
 # options to build tests, examples and packages
 option(BUILD_TESTS "Build tests" ON)
 option(BUILD_EXAMPLES "Build examples" ON)
@@ -60,8 +63,6 @@ if(BUILD_PACKAGES)
 
 endif(BUILD_PACKAGES)
 
-find_package(Boost 1.56.0 REQUIRED)
-include_directories(${Boost_INCLUDE_DIR})
 include_directories(${CMAKE_CURRENT_SOURCE_DIR}/tinyxml2)
 include_directories ("${PROJECT_SOURCE_DIR}/src")
 
