diff --git a/src/mfast/json/CMakeLists.txt b/src/mfast/json/CMakeLists.txt
index 2018f37..e160b2a 100644
--- a/src/mfast/json/CMakeLists.txt
+++ b/src/mfast/json/CMakeLists.txt
@@ -9,7 +9,7 @@ SET(LIBRARY_OUTPUT_PATH ${PROJECT_BINARY_DIR}/lib)
 set(mfast_json_SRCS  ${sources} ${headers})
 
 add_library(mfast_json_static STATIC ${mfast_json_SRCS})
-target_link_libraries(mfast_json_static mfast_static)
+target_link_libraries(mfast_json_static mfast_static ${CONAN_LIBS})
 
 if (UNIX)
   set_target_properties(mfast_json_static PROPERTIES OUTPUT_NAME mfast_json)
@@ -35,7 +35,7 @@ if (BUILD_SHARED_LIBS)
   if (CMAKE_COMPILER_IS_GNUCXX OR ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang"))
     set_target_properties(mfast_json PROPERTIES COMPILE_FLAGS -fvisibility=hidden)
   endif()
-  target_link_libraries(mfast_json mfast)
+  target_link_libraries(mfast_json mfast ${CONAN_LIBS})
   set_target_properties(mfast_json PROPERTIES
                         VERSION "${MFAST_VERSION}"
                         SOVERSION "${MFAST_SOVERSION}")
