--- CMakeLists.txt
+++ CMakeLists.txt
@@ -521,14 +521,14 @@
     set(PIXARLOG_SUPPORT TRUE)
   endif()
 endif()
-
 # JPEG
 option(jpeg "use libjpeg (required for JPEG compression)" ON)
 if (jpeg)
   find_package(JPEG)
+  find_package(libjpeg-turbo)
 endif()
 set(JPEG_SUPPORT FALSE)
-if(JPEG_FOUND)
+if(JPEG_FOUND OR libjpeg-turbo_FOUND)
   set(JPEG_SUPPORT TRUE)
 endif()
 
@@ -680,8 +680,8 @@
 if(ZLIB_INCLUDE_DIRS)
   list(APPEND TIFF_INCLUDES ${ZLIB_INCLUDE_DIRS})
 endif()
-if(JPEG_INCLUDE_DIR)
-  list(APPEND TIFF_INCLUDES ${JPEG_INCLUDE_DIR})
+if(JPEG_INCLUDE_DIR OR libjpeg-turbo_INCLUDE_DIRS)
+  list(APPEND TIFF_INCLUDES ${JPEG_INCLUDE_DIR} ${libjpeg-turbo_INCLUDE_DIRS})
 endif()
 if(JPEG12_INCLUDE_DIR)
   list(APPEND TIFF_INCLUDES ${JPEG12_INCLUDE_DIR})
@@ -701,8 +701,8 @@
 if(ZLIB_LIBRARIES)
   list(APPEND TIFF_LIBRARY_DEPS ${ZLIB_LIBRARIES})
 endif()
-if(JPEG_LIBRARIES)
-  list(APPEND TIFF_LIBRARY_DEPS ${JPEG_LIBRARIES})
+if(JPEG_LIBRARIES OR libjpeg-turbo_LIBRARIES)
+  list(APPEND TIFF_LIBRARY_DEPS ${JPEG_LIBRARIES} ${libjpeg-turbo_LIBRARIES})
 endif()
 if(JPEG12_LIBRARIES)
   list(APPEND TIFF_LIBRARY_DEPS ${JPEG12_LIBRARIES})
