diff --git a/cmake/JPEGCodec.cmake b/cmake/JPEGCodec.cmake
index 8455a3ec..09fe975a 100644
--- a/cmake/JPEGCodec.cmake
+++ b/cmake/JPEGCodec.cmake
@@ -42,25 +42,7 @@ endif()
 if (JPEG_SUPPORT)
     # Check for jpeg12_read_scanlines() which has been added in libjpeg-turbo 2.2
     # for dual 8/12 bit mode.
-    include(CheckCSourceCompiles)
-    include(CMakePushCheckState)
-    cmake_push_check_state(RESET)
-    set(CMAKE_REQUIRED_INCLUDES "${JPEG_INCLUDE_DIRS}")
-    set(CMAKE_REQUIRED_LIBRARIES "${JPEG_LIBRARIES}")
-    check_c_source_compiles(
-        "
-        #include <stddef.h>
-        #include <stdio.h>
-        #include \"jpeglib.h\"
-        int main()
-        {
-            jpeg_read_scanlines(0,0,0);
-            jpeg12_read_scanlines(0,0,0);
-            return 0;
-        }
-        "
-        HAVE_JPEGTURBO_DUAL_MODE_8_12)
-    cmake_pop_check_state()
+    set(HAVE_JPEGTURBO_DUAL_MODE_8_12 FALSE)
 endif()
 
 if (NOT HAVE_JPEGTURBO_DUAL_MODE_8_12)
