--- a/src/lib_proj.cmake
+++ b/src/lib_proj.cmake
@@ -311,7 +311,7 @@ source_group("Source Files\\Transformations"
 source_group("Source Files\\ISO19111"
   FILES ${SRC_LIBPROJ_ISO19111})
 
-include_directories(${CMAKE_SOURCE_DIR}/include)
+include_directories(${PROJ4_SOURCE_DIR}/include)
 
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
 source_group("CMake Files" FILES CMakeLists.txt)
@@ -440,8 +440,7 @@ if(USE_THREAD AND Threads_FOUND AND CMAKE_USE_PTHREADS_INIT)
   target_link_libraries(${PROJ_CORE_TARGET} ${CMAKE_THREAD_LIBS_INIT})
 endif()
 
-include_directories(${SQLITE3_INCLUDE_DIR})
-target_link_libraries(${PROJ_CORE_TARGET} ${SQLITE3_LIBRARY})
+target_link_libraries(${PROJ_CORE_TARGET} CONAN_PKG::nlohmann_json CONAN_PKG::sqlite3)
 
 if(MSVC AND BUILD_LIBPROJ_SHARED)
   target_compile_definitions(${PROJ_CORE_TARGET}
