diff -ru a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2021-02-26 02:37:25.000000000 +0800
+++ b/CMakeLists.txt	2022-12-14 21:26:48.404915901 +0800
@@ -121,6 +121,8 @@
 include(ProjMac)
 include(policies)
 
+find_package(nlohmann_json REQUIRED)
+
 ################################################################################
 # Check for sqlite3
 ################################################################################
diff -ru a/src/lib_proj.cmake b/src/lib_proj.cmake
--- a/src/lib_proj.cmake	2021-02-20 19:15:52.000000000 +0800
+++ b/src/lib_proj.cmake	2022-12-14 21:25:41.713734988 +0800
@@ -394,21 +394,18 @@
   target_link_libraries(proj PRIVATE ${CMAKE_THREAD_LIBS_INIT})
 endif()
 
-target_include_directories(proj PRIVATE ${SQLITE3_INCLUDE_DIR})
-target_link_libraries(proj PRIVATE ${SQLITE3_LIBRARY})
+target_link_libraries(proj PUBLIC nlohmann_json::nlohmann_json PRIVATE SQLite::SQLite3)
 
 if(TIFF_ENABLED)
   target_compile_definitions(proj PRIVATE -DTIFF_ENABLED)
-  target_include_directories(proj PRIVATE ${TIFF_INCLUDE_DIR})
-  target_link_libraries(proj PRIVATE ${TIFF_LIBRARY})
+  target_link_libraries(proj PRIVATE TIFF::TIFF)
 endif()
 
 if(CURL_ENABLED)
   target_compile_definitions(proj PRIVATE -DCURL_ENABLED)
-  target_include_directories(proj PRIVATE ${CURL_INCLUDE_DIR})
   target_link_libraries(proj
     PRIVATE
-      ${CURL_LIBRARY}
+      CURL::libcurl
       $<$<CXX_COMPILER_ID:MSVC>:ws2_32>
       $<$<CXX_COMPILER_ID:MSVC>:wldap32>
       $<$<CXX_COMPILER_ID:MSVC>:advapi32>
