--- a/src/grids.cpp
+++ b/src/grids.cpp
@@ -309,8 +309,11 @@ constexpr uint16 TIFFTAG_GEOTRANSMATRIX = 34264;
 constexpr uint16 TIFFTAG_GEOKEYDIRECTORY = 34735;
 constexpr uint16 TIFFTAG_GEODOUBLEPARAMS = 34736;
 constexpr uint16 TIFFTAG_GEOASCIIPARAMS = 34737;
+#ifndef TIFFTAG_GDAL_METADATA
+// Starting with libtiff > 4.1.0, those symbolic names are #define in tiff.h
 constexpr uint16 TIFFTAG_GDAL_METADATA = 42112;
 constexpr uint16 TIFFTAG_GDAL_NODATA = 42113;
+#endif
 
 // ---------------------------------------------------------------------------
 
