--- a/dtoa_milo.h
+++ b/dtoa_milo.h
@@ -1,19 +1,18 @@
 #pragma once
 #include <assert.h>
 #include <math.h>
+#include <stdint.h>
+#include <string.h>
 
 #if defined(_MSC_VER)
-#include "msinttypes/stdint.h"
 #include <intrin.h>
-#else
-#include <stdint.h>
-#endif
-
+#elif (__GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 6)) && defined(__x86_64__)
 namespace gcc_ints
 {
     __extension__ typedef __int128 int128;
     __extension__ typedef unsigned __int128 uint128;
 }
+#endif
 
 #define UINT64_C2(h, l) ((static_cast<uint64_t>(h) << 32) | static_cast<uint64_t>(l))
 
