diff --git a/CMakeLists.txt b/CMakeLists.txt
index 66e3e7c3..32e64b5d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -53,7 +53,6 @@ elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC")
 endif ()
 
 list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake)
-include(SelectMSVCRuntime)
 
 option(GOOGLE_CLOUD_CPP_ENABLE_MACOS_OPENSSL_CHECK
        "If enabled, check that the user has defined OPENSSL_ROOT_DIR on macOS"
diff --git a/cmake/GoogleCloudCppCommon.cmake b/cmake/GoogleCloudCppCommon.cmake
index b487a1bc..880c98fe 100644
--- a/cmake/GoogleCloudCppCommon.cmake
+++ b/cmake/GoogleCloudCppCommon.cmake
@@ -17,9 +17,6 @@
 # Get the destination directories based on the GNU recommendations.
 include(GNUInstallDirs)
 
-# Pick the right MSVC runtime libraries.
-include(SelectMSVCRuntime)
-
 # Enable Werror
 include(EnableWerror)
 
diff --git a/examples/CMakeLists.txt b/examples/CMakeLists.txt
index ec8b7395..91a7b32d 100644
--- a/examples/CMakeLists.txt
+++ b/examples/CMakeLists.txt
@@ -20,9 +20,6 @@ if (NOT GOOGLE_CLOUD_CPP_ENABLE_EXAMPLES)
     return()
 endif ()
 
-# Pick the right MSVC runtime libraries.
-include(SelectMSVCRuntime)
-
 if (bigtable IN_LIST GOOGLE_CLOUD_CPP_ENABLE AND storage IN_LIST
                                                  GOOGLE_CLOUD_CPP_ENABLE)
     add_executable(gcs2cbt gcs2cbt.cc)
diff --git a/external/googleapis/CMakeLists.txt b/external/googleapis/CMakeLists.txt
index 15dcc92f..b6146bfe 100644
--- a/external/googleapis/CMakeLists.txt
+++ b/external/googleapis/CMakeLists.txt
@@ -163,8 +163,6 @@ externalproject_add(
 
 google_cloud_cpp_find_proto_include_dir(PROTO_INCLUDE_DIR)
 
-include(SelectMSVCRuntime)
-
 google_cloud_cpp_add_protos_property()
 
 function (external_googleapis_short_name var proto)
