diff --git a/CMakeLists.txt b/CMakeLists.txt
index c37734d1..54cf7478 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -150,7 +150,6 @@ endmacro()
 macro(jerry_add_compile_warnings)
   foreach(_warning ${ARGV})
     jerry_add_compile_flags(-W${_warning})
-    jerry_add_compile_flags(-Werror=${_warning})
   endforeach()
 endmacro()
 
@@ -194,7 +193,7 @@ if(USING_GCC OR USING_CLANG)
   # Turn off stack protector
   jerry_add_compile_flags(-fno-builtin -fno-stack-protector)
   jerry_add_compile_warnings(all extra format-nonliteral init-self conversion sign-conversion format-security missing-declarations shadow strict-prototypes undef old-style-definition)
-  jerry_add_compile_flags(-Wno-stack-protector -Wno-attributes -Werror)
+  jerry_add_compile_flags(-Wno-stack-protector -Wno-attributes)
 endif()
 
 if(USING_GCC)
