diff --git a/libmdnsd/CMakeLists.txt b/libmdnsd/CMakeLists.txt
index 3c29b08..73b3b1d 100644
--- a/libmdnsd/CMakeLists.txt
+++ b/libmdnsd/CMakeLists.txt
@@ -46,8 +46,12 @@ endif()
 configure_file("mdnsd_config.h.in" "${PROJECT_BINARY_DIR}/mdnsd_config.h")
 include_directories(${PROJECT_BINARY_DIR})
 
-set(LIBRARY_FILES mdnsd.c mdnsd.h 1035.c 1035.h sdtxt.c sdtxt.h xht.c xht.h)
+set(PUBLIC_INCLUDES mdnsd.h 1035.h sdtxt.h xht.h)
+  if(WIN32 AND CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
+  list(APPEND PUBLIC_INCLUDES ms_stdint.h)
+endif()
 
+set(LIBRARY_FILES mdnsd.c 1035.c xht.c sdtxt.c)
 
 if (MDNSD_BUILD_FUZZING_CORPUS)
     set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
@@ -63,9 +67,9 @@ if(MDNSD_COMPILE_AS_CXX)
 endif()
 
 if(MDNSD_BUILD_OSS_FUZZ)
-    add_library(libmdnsd ${LIBRARY_FILES})
+    add_library(libmdnsd ${LIBRARY_FILES} ${PUBLIC_INCLUDES})
 else()
-    add_library(libmdnsd SHARED ${LIBRARY_FILES})
+    add_library(libmdnsd SHARED ${LIBRARY_FILES} ${PUBLIC_INCLUDES})
 endif()
 
 if(WIN32 OR MINGW)
@@ -73,3 +77,23 @@ if(WIN32 OR MINGW)
 endif()
 
 target_compile_definitions(libmdnsd PRIVATE -DMDNSD_DYNAMIC_LINKING_EXPORT)
+
+include(GNUInstallDirs)
+
+install(
+    TARGETS ${PROJECT_NAME}
+    EXPORT ${PROJECT_NAME}Targets
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+)
+
+install(
+    FILES ${PUBLIC_INCLUDES}
+    DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/${PROJECT_NAME}
+)
+
+install(
+  FILES ${PROJECT_BINARY_DIR}/mdnsd_config.h
+  DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/${PROJECT_NAME}
+)
