diff --git a/CMakeLists.txt b/CMakeLists.txt
index 51879ed..e1226ff 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -145,8 +145,7 @@ endif()
 
 if(CMAKE_COMPILER_IS_MSVC)
   add_definitions("-DBOOST_ALL_NO_LIB -D_SCL_SECURE_NO_WARNINGS -D_CRT_SECURE_NO_WARNINGS -DNOMINMAX -DPCL_ONLY_CORE_POINT_TYPES ${SSE_DEFINITIONS}")
-  
-  string(APPEND CMAKE_CXX_FLAGS " /bigobj")
+
   if("${CMAKE_CXX_FLAGS}" STREQUAL "${CMAKE_CXX_FLAGS_DEFAULT}")
     string(APPEND CMAKE_CXX_FLAGS " /fp:precise ${SSE_FLAGS} ${AVX_FLAGS}")
 
@@ -203,6 +202,7 @@ if(CMAKE_COMPILER_IS_MSVC)
       endif()
     endif()
   endif()
+  string(APPEND CMAKE_CXX_FLAGS " /bigobj")
 
   if(CMAKE_GENERATOR STREQUAL "Ninja")
     string(APPEND CMAKE_C_FLAGS " /FS")
