--- lang/c++/api/buffer/detail/BufferDetail.hh
+++ lang/c++/api/buffer/detail/BufferDetail.hh
@@ -30,6 +30,7 @@
 #include <exception>
 #include <cassert>
 #include <deque>
+#include <iterator>
 
 /**
  * \file BufferDetail.hh
--- lang/c++/impl/DataFile.cc
+++ lang/c++/impl/DataFile.cc
@@ -20,6 +20,7 @@
 #include "Compiler.hh"
 #include "Exception.hh"
 
+#include <iterator>
 #include <sstream>
 
 #include <boost/random/mersenne_twister.hpp>
--- lang/c++/impl/Stream.cc
+++ lang/c++/impl/Stream.cc
@@ -17,6 +17,7 @@
  */
 
 #include "Stream.hh"
+#include <iterator>
 #include <vector>
 
 namespace avro {
--- lang/c++/impl/parsing/JsonCodec.cc
+++ lang/c++/impl/parsing/JsonCodec.cc
@@ -23,6 +23,7 @@
 #include <algorithm>
 #include <ctype.h>
 #include <memory>
+#include <iterator>
 #include <boost/math/special_functions/fpclassify.hpp>
 
 #include "ValidatingCodec.hh"
--- lang/c++/impl/parsing/ResolvingDecoder.cc
+++ lang/c++/impl/parsing/ResolvingDecoder.cc
@@ -24,6 +24,7 @@
 #include <algorithm>
 #include <memory>
 #include <ctype.h>
+#include <iterator>
 
 #include "ValidatingCodec.hh"
 #include "Symbol.hh"
--- lang/c++/impl/parsing/ValidatingCodec.cc
+++ lang/c++/impl/parsing/ValidatingCodec.cc
@@ -22,6 +22,7 @@
 #include <map>
 #include <algorithm>
 #include <memory>
+#include <iterator>
 #include <boost/any.hpp>
 
 #include "ValidSchema.hh"
--- lang/c++/test/CodecTests.cc
+++ lang/c++/test/CodecTests.cc
@@ -30,6 +30,7 @@
 #include <stack>
 #include <string>
 #include <functional>
+#include <iterator>
 #include <boost/bind.hpp>
 
 #include <boost/test/included/unit_test_framework.hpp>
--- lang/c++/test/DataFileTests.cc
+++ lang/c++/test/DataFileTests.cc
@@ -25,6 +25,7 @@
 
 #include <thread>
 #include <chrono>
+#include <iterator>
 
 #include <sstream>
 
