diff --git a/lang/c++/CMakeLists.txt b/lang/c++/CMakeLists.txt
index 19059a4..e1f23c0 100644
--- a/lang/c++/CMakeLists.txt
+++ b/lang/c++/CMakeLists.txt
@@ -197,10 +197,10 @@ target_include_directories(avrocpp PUBLIC
 enable_testing()
 
 macro (unittest name)
-    add_executable (${name} test/${name}.cc)
-    target_link_libraries (${name} avrocpp ${Boost_LIBRARIES} ${SNAPPY_LIBRARIES})
-    add_test (NAME ${name} WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-        COMMAND ${CMAKE_CURRENT_BINARY_DIR}/${name})
+    # add_executable (${name} test/${name}.cc)
+    # target_link_libraries (${name} avrocpp ${Boost_LIBRARIES} ${SNAPPY_LIBRARIES})
+    # add_test (NAME ${name} WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+    #     COMMAND ${CMAKE_CURRENT_BINARY_DIR}/${name})
 endmacro (unittest)
 
 unittest (buffertest)
@@ -217,14 +217,14 @@ unittest (CompilerTests)
 unittest (AvrogencppTestReservedWords)
 unittest (CommonsSchemasTests)
 
-add_dependencies (AvrogencppTestReservedWords cpp_reserved_words_hh)
+# add_dependencies (AvrogencppTestReservedWords cpp_reserved_words_hh)
 
-add_dependencies (AvrogencppTests bigrecord_hh bigrecord_r_hh bigrecord2_hh
-    tweet_hh
-    union_array_union_hh union_map_union_hh union_conflict_hh
-    recursive_hh reuse_hh circulardep_hh tree1_hh tree2_hh crossref_hh
-    primitivetypes_hh empty_record_hh cpp_reserved_words_union_typedef_hh
-    union_empty_record_hh)
+# add_dependencies (AvrogencppTests bigrecord_hh bigrecord_r_hh bigrecord2_hh
+#     tweet_hh
+#     union_array_union_hh union_map_union_hh union_conflict_hh
+#     recursive_hh reuse_hh circulardep_hh tree1_hh tree2_hh crossref_hh
+#     primitivetypes_hh empty_record_hh cpp_reserved_words_union_typedef_hh
+#     union_empty_record_hh)
 
 include (InstallRequiredSystemLibraries)
 
