diff --git a/CMakeLists.txt b/CMakeLists.txt
index c8fa3c2a2..15dd0f60e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -296,7 +296,7 @@ if(MSVC)
     # - w4100 - unreferenced formal parameter
     # - w4201 - nonstandard extension used: nameless struct/union
     # - w4505 - unreferenced local function has been removed
-    target_compile_options(Diligent-BuildSettings INTERFACE /W4 /WX /wd4100 /wd4201 /wd4505 /MP)
+    target_compile_options(Diligent-BuildSettings INTERFACE /W1 /wd4100 /wd4201 /wd4505 /MP)
     # In all release modes also:
     # - disable w4189 - local variable is initialized but not referenced
     # - Disable RTTI (/GR-)
@@ -343,7 +343,7 @@ endif()
 if (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
     target_compile_options(Diligent-BuildSettings INTERFACE
     	# All warnings are errors
-        -Werror
+        
         # Some extra warnings
         -Wall -Wextra -Wuninitialized -Wconditional-uninitialized -Wextra-tokens -Wpointer-arith -Wloop-analysis -Wunused
         # Disable few warnings
@@ -427,7 +427,6 @@ add_subdirectory(Primitives)
 add_subdirectory(Platforms)
 add_subdirectory(Common)
 add_subdirectory(Graphics)
-add_subdirectory(Tests)
 
 
 # Installation instructions
