diff --git a/Graphics/ShaderTools/CMakeLists.txt b/Graphics/ShaderTools/CMakeLists.txt
index dc47ac4b..83630de3 100644
--- a/Graphics/ShaderTools/CMakeLists.txt
+++ b/Graphics/ShaderTools/CMakeLists.txt
@@ -147,13 +147,13 @@ if(ENABLE_SPIRV)
     if (${USE_GLSLANG})
         target_link_libraries(Diligent-ShaderTools
         PRIVATE
-            SPIRV
-            glslang
+            spirv-tools::SPIRV-Tools-opt
+            glslang::SPIRV glslang::glslang glslang::MachineIndependent glslang::GenericCodeGen glslang::OSDependent glslang::OGLCompiler glslang::HLSL
         )
 
         target_include_directories(Diligent-ShaderTools
         PRIVATE
-            ../../ThirdParty/glslang
+            ${CONAN_INCLUDE_DIRS_GLSLANG}/glslang/
         )
     endif()
 endif()
