diff --git a/ThirdParty/CMakeLists.txt b/ThirdParty/CMakeLists.txt
index a893fe6a5..75f581d16 100644
--- a/ThirdParty/CMakeLists.txt
+++ b/ThirdParty/CMakeLists.txt
@@ -65,7 +65,7 @@ if(VULKAN_SUPPORTED OR METAL_SUPPORTED)
     set(SPIRV_CROSS_SKIP_INSTALL ON CACHE BOOL "Skip SPIRV-Cross installation.")
     set(SPIRV_CROSS_NAMESPACE_OVERRIDE diligent_spirv_cross CACHE STRING "Override the namespace used in the C++ API.")
 
-    list(APPEND THIRD_PARTY_LIBS CONAN_PKG::spirv-tools CONAN_PKG::spirv-cross CONAN_PKG::glslang)
+    list(APPEND THIRD_PARTY_LIBS CONAN_PKG::spirv-tools CONAN_PKG::spirv-cross CONAN_PKG::glslang xxhash::xxhash)
 
     # Make sure that symbols do not leak out when third-party
     # libs are linked into shared libraries
@@ -85,8 +85,3 @@ if(${DILIGENT_BUILD_GOOGLE_TEST})
 endif()
 
 option(BUILD_SHARED_LIBS "Build shared library" OFF)
-set(XXHASH_BUILD_XXHSUM OFF)
-set(XXHASH_BUILD_ENABLE_INLINE_API OFF)
-add_subdirectory(xxHash/cmake_unofficial)
-set_directory_root_folder("xxHash/cmake_unofficial" "DiligentCore/ThirdParty/xxHash")
-
