diff --git a/CMakeLists.txt b/CMakeLists.txt
index bc7de0420..5eeafaa36 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -375,7 +375,8 @@ elseif (CMAKE_CXX_COMPILER_ID MATCHES "GNU")
     if (CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL "9.0")
         message(WARNING "gcc 9.0 and above seemingly produce invalid binary code with O2 and O3 optimization levels. Optimization in release configurations will be downgraded to O1. It is strongly recommended to use clang.")
         set(GCC_OPTIMIZATION_LEVEL "-O1")
-        target_compile_options(Diligent-BuildSettings INTERFACE $<$<CONFIG:Release,RelWithDebInfo>:${GCC_OPTIMIZATION_LEVEL}>)
+        target_compile_options(Diligent-BuildSettings INTERFACE $<$<CONFIG:Release>:${GCC_OPTIMIZATION_LEVEL}>)
+        target_compile_options(Diligent-BuildSettings INTERFACE $<$<CONFIG:RelWithDebInfo>:${GCC_OPTIMIZATION_LEVEL}>)
     endif()
 endif()
 
