diff --git a/Graphics/ShaderTools/CMakeLists.txt b/Graphics/ShaderTools/CMakeLists.txt
index 015425a9..53507f61 100644
--- a/Graphics/ShaderTools/CMakeLists.txt
+++ b/Graphics/ShaderTools/CMakeLists.txt
@@ -152,8 +152,8 @@ if(ENABLE_SPIRV)
     if (${USE_GLSLANG})
         target_link_libraries(Diligent-ShaderTools
         PRIVATE
-            SPIRV
-            glslang
+            SPIRV-Tools-opt
+            glslang::SPIRV glslang::glslang glslang::MachineIndependent glslang::GenericCodeGen glslang::OSDependent glslang::OGLCompiler glslang::HLSL
         )
     endif()
 endif()
diff --git a/ThirdParty/CMakeLists.txt b/ThirdParty/CMakeLists.txt
index 278a72df..f0183b05 100644
--- a/ThirdParty/CMakeLists.txt
+++ b/ThirdParty/CMakeLists.txt
@@ -64,7 +64,7 @@ if (VULKAN_SUPPORTED OR METAL_SUPPORTED)
         install(FILES "${spirv-tools_SOURCE_DIR}/LICENSE" DESTINATION "Licenses/ThirdParty/${DILIGENT_CORE_DIR}" RENAME SPIRV-Tools-License.txt)
     endif()
 
-    if (NOT ${DILIGENT_NO_GLSLANG} AND (NOT TARGET glslang))
+    if (NOT ${DILIGENT_NO_GLSLANG} AND (NOT TARGET glslang::glslang))
         set(ENABLE_GLSLANG_BINARIES OFF CACHE BOOL "Do not build glslang binaries")
         set(ENABLE_SPVREMAPPER OFF CACHE BOOL "Do not build SPIRV remapper")
         set(SKIP_GLSLANG_INSTALL ON CACHE BOOL "Skip glslang installation")
