diff --git a/ThirdParty/CMakeLists.txt b/ThirdParty/CMakeLists.txt
index f0183b05..dd7d2e3d 100644
--- a/ThirdParty/CMakeLists.txt
+++ b/ThirdParty/CMakeLists.txt
@@ -26,7 +26,6 @@ if (VULKAN_SUPPORTED OR METAL_SUPPORTED)
         # Note that SPIRV-Headers_SOURCE_DIR variable is normally created by project(SPIRV-Headers)
         set(SPIRV-Headers_SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/SPIRV-Headers")
         target_include_directories(SPIRV-Headers INTERFACE "${SPIRV-Headers_SOURCE_DIR}/include")
-        install(FILES "${SPIRV-Headers_SOURCE_DIR}/LICENSE" DESTINATION "Licenses/ThirdParty/${DILIGENT_CORE_DIR}" RENAME SPIRV-Headers-License.txt)
     endif()
 
     if (NOT TARGET Vulkan::Headers)
@@ -37,7 +36,6 @@ if (VULKAN_SUPPORTED OR METAL_SUPPORTED)
         set(Vulkan-Headers_SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/Vulkan-Headers")
         add_library(Vulkan::Headers ALIAS Vulkan-Headers)
         target_include_directories(Vulkan-Headers INTERFACE "${Vulkan-Headers_SOURCE_DIR}/include")
-        install(FILES "${Vulkan-Headers_SOURCE_DIR}/LICENSE.txt" DESTINATION "Licenses/ThirdParty/${DILIGENT_CORE_DIR}" RENAME Vulkan-Headers-License.txt)
     endif()
 
     option(DILIGENT_NO_GLSLANG "Do NOT build glslang compiler" OFF)
@@ -61,7 +59,6 @@ if (VULKAN_SUPPORTED OR METAL_SUPPORTED)
         set_target_properties(${SPIRV_TOOLS_EXCLUDE_TARGETS} PROPERTIES EXCLUDE_FROM_ALL 1 EXCLUDE_FROM_DEFAULT_BUILD 1)
 
         set(THIRD_PARTY_LIBS SPIRV-Tools-static SPIRV-Tools-opt)
-        install(FILES "${spirv-tools_SOURCE_DIR}/LICENSE" DESTINATION "Licenses/ThirdParty/${DILIGENT_CORE_DIR}" RENAME SPIRV-Tools-License.txt)
     endif()
 
     if (NOT ${DILIGENT_NO_GLSLANG} AND (NOT TARGET glslang::glslang))
@@ -101,7 +98,6 @@ if (VULKAN_SUPPORTED OR METAL_SUPPORTED)
             list(APPEND THIRD_PARTY_LIBS spirv-cross-msl spirv-cross-glsl)
         endif()
 
-        install(FILES "${SPIRV-Cross_SOURCE_DIR}/LICENSE" DESTINATION "Licenses/ThirdParty/${DILIGENT_CORE_DIR}" RENAME SPIRV-Cross-License.txt)
     endif()
 
     if (DILIGENT_INSTALL_CORE)
