diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0fc8aafc..aa8b8cf8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -311,7 +311,7 @@ if(MSVC)
     # - w4100 - unreferenced formal parameter
     # - w4201 - nonstandard extension used: nameless struct/union
     # - w4505 - unreferenced local function has been removed
-    target_compile_options(Diligent-BuildSettings INTERFACE /W4 /wd4100 /wd4201 /wd4505 /MP ${DILIGENT_MSVC_COMPILE_OPTIONS})
+    target_compile_options(Diligent-BuildSettings INTERFACE /W0 /wd4100 /wd4201 /wd4505 /MP ${DILIGENT_MSVC_COMPILE_OPTIONS})
 
     # In all release modes also:
     # - disable w4189 - local variable is initialized but not referenced
