--- Source/Modules/main.cxx
+++ Source/Modules/main.cxx
@@ -879,6 +879,32 @@ static void getoptions(int argc, char *a
   }
 }
 
+#if defined(HAVE_UNISTD_H) && !defined(_WIN32)
+#include <libgen.h>
+#include <unistd.h>
+#include <dlfcn.h>
+
+static String *get_exe_path(void) {
+    Dl_info info;
+    if (dladdr("main", &info)) {
+        char realp_buffer[PATH_MAX];
+        char* res = NULL;
+
+        res = realpath(info.dli_fname, realp_buffer);
+        if (!res) {
+         return NewString(SWIG_LIB);
+        }
+
+        const char* dir = dirname(realp_buffer);
+        char dest_buf[PATH_MAX];
+        strcpy(dest_buf, dir);
+        strcat(dest_buf, "/swiglib");
+        return NewStringWithSize(dest_buf, strlen(dest_buf));
+    }
+    return NewString(SWIG_LIB);
+}
+#endif
+
 int SWIG_main(int argc, char *argv[], const TargetLanguageModule *tlm) {
   char *c;
 
@@ -939,12 +965,14 @@ int SWIG_main(int argc, char *argv[], co
     char *p;
     if (!(GetModuleFileName(0, buf, MAX_PATH) == 0 || (p = strrchr(buf, '\\')) == 0)) {
       *(p + 1) = '\0';
-      SwigLib = NewStringf("%sLib", buf); // Native windows installation path
+      SwigLib = NewStringf("%sswiglib", buf); // Native windows installation path
     } else {
       SwigLib = NewStringf("");	// Unexpected error
     }
     if (Len(SWIG_LIB_WIN_UNIX) > 0)
       SwigLibWinUnix = NewString(SWIG_LIB_WIN_UNIX); // Unix installation path using a drive letter (for msys/mingw)
+#elif defined(HAVE_UNISTD_H) && !defined(_WIN32)
+    SwigLib = get_exe_path();
 #else
     SwigLib = NewString(SWIG_LIB);
 #endif
