diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index fa08094a..516806dd 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -55,8 +55,8 @@ if(ENABLE_SHM)
   else()
     set(iceoryx_required QUIET)
   endif()
-  find_package(iceoryx_binding_c ${iceoryx_required})
-  set(ENABLE_SHM ${iceoryx_binding_c_FOUND} CACHE STRING "" FORCE)
+  find_package(iceoryx ${iceoryx_required})
+  set(ENABLE_SHM ${iceoryx_FOUND} CACHE STRING "" FORCE)
   if(ENABLE_SHM AND APPLE)
     get_filename_component(iceoryx_libdir "${ICEORYX_LIB}" DIRECTORY)
     set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_RPATH};${iceoryx_libdir}")
