diff --git a/src/core/CMakeLists.txt b/src/core/CMakeLists.txt
index 0c786dc2..4773dccf 100644
--- a/src/core/CMakeLists.txt
+++ b/src/core/CMakeLists.txt
@@ -74,13 +74,7 @@ install(
   LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}" COMPONENT lib
   ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}" COMPONENT lib
 )
-if (MSVC)
-  install(FILES $<TARGET_PDB_FILE:ddsc>
-    DESTINATION "${CMAKE_INSTALL_BINDIR}"
-    COMPONENT dev
-    OPTIONAL
-  )
-endif()
+
 
 if(BUILD_TESTING AND BUILD_IDLC)
   add_subdirectory(xtests)
diff --git a/src/idl/CMakeLists.txt b/src/idl/CMakeLists.txt
index c4e9bb8f..c025b232 100644
--- a/src/idl/CMakeLists.txt
+++ b/src/idl/CMakeLists.txt
@@ -120,13 +120,7 @@ install(
   RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}" COMPONENT lib
   LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}" COMPONENT lib
   ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}" COMPONENT lib)
-if (MSVC)
-  install(FILES $<TARGET_PDB_FILE:idl>
-    DESTINATION "${CMAKE_INSTALL_BINDIR}"
-    COMPONENT dev
-    OPTIONAL
-  )
-endif()
+
 
 if(BUILD_TESTING)
   add_subdirectory(tests)
diff --git a/src/security/builtin_plugins/access_control/CMakeLists.txt b/src/security/builtin_plugins/access_control/CMakeLists.txt
index c088aea8..edea1028 100644
--- a/src/security/builtin_plugins/access_control/CMakeLists.txt
+++ b/src/security/builtin_plugins/access_control/CMakeLists.txt
@@ -55,10 +55,3 @@ install(
   LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}" COMPONENT lib
   ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}" COMPONENT lib
 )
-if (MSVC)
-  install(FILES $<TARGET_PDB_FILE:dds_security_ac>
-    DESTINATION "${CMAKE_INSTALL_BINDIR}"
-    COMPONENT dev
-    OPTIONAL
-  )
-endif()
diff --git a/src/security/builtin_plugins/authentication/CMakeLists.txt b/src/security/builtin_plugins/authentication/CMakeLists.txt
index 9c4d3b90..929e13a8 100644
--- a/src/security/builtin_plugins/authentication/CMakeLists.txt
+++ b/src/security/builtin_plugins/authentication/CMakeLists.txt
@@ -54,10 +54,3 @@ install(
   RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}" COMPONENT lib
   LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}" COMPONENT lib
   ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}" COMPONENT lib)
-if (MSVC)
-  install(FILES $<TARGET_PDB_FILE:dds_security_auth>
-    DESTINATION "${CMAKE_INSTALL_BINDIR}"
-    COMPONENT dev
-    OPTIONAL
-  )
-endif()
diff --git a/src/security/builtin_plugins/cryptographic/CMakeLists.txt b/src/security/builtin_plugins/cryptographic/CMakeLists.txt
index 0c2f20fd..55dc3454 100644
--- a/src/security/builtin_plugins/cryptographic/CMakeLists.txt
+++ b/src/security/builtin_plugins/cryptographic/CMakeLists.txt
@@ -62,12 +62,6 @@ install(
   LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}" COMPONENT lib
   ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}" COMPONENT lib
 )
-if (MSVC)
-  install(FILES $<TARGET_PDB_FILE:dds_security_crypto>
-    DESTINATION "${CMAKE_INSTALL_BINDIR}"
-    COMPONENT dev
-    OPTIONAL
-  )
-endif()
+
 
 add_coverage(dds_security_crypto)
diff --git a/src/tools/ddsperf/CMakeLists.txt b/src/tools/ddsperf/CMakeLists.txt
index ddf87b23..b7377788 100644
--- a/src/tools/ddsperf/CMakeLists.txt
+++ b/src/tools/ddsperf/CMakeLists.txt
@@ -23,10 +23,3 @@ install(
   DESTINATION "${CMAKE_INSTALL_BINDIR}"
   COMPONENT dev
 )
-if (MSVC)
-  install(FILES $<TARGET_PDB_FILE:ddsperf>
-    DESTINATION "${CMAKE_INSTALL_BINDIR}"
-    COMPONENT dev
-    OPTIONAL
-  )
-endif()
diff --git a/src/tools/pubsub/CMakeLists.txt b/src/tools/pubsub/CMakeLists.txt
index 0338efd2..62a94949 100644
--- a/src/tools/pubsub/CMakeLists.txt
+++ b/src/tools/pubsub/CMakeLists.txt
@@ -21,10 +21,3 @@ install(
   DESTINATION "${CMAKE_INSTALL_BINDIR}"
   COMPONENT dev
 )
-if (MSVC)
-  install(FILES $<TARGET_PDB_FILE:pubsub>
-    DESTINATION "${CMAKE_INSTALL_BINDIR}"
-    COMPONENT dev
-    OPTIONAL
-  )
-endif()
