diff --git a/thirdparty/thirdparty-conan-build.cmake b/thirdparty/thirdparty-conan-build.cmake
index 5aecfbd..e1a15ce 100644
--- a/thirdparty/thirdparty-conan-build.cmake
+++ b/thirdparty/thirdparty-conan-build.cmake
@@ -8,7 +8,7 @@ find_package(string-view-lite)
 find_package(Boost)
 set(CONAN_BOOST_PACKAGE_NAME Boost::Boost)
 find_package(fmt)
-find_package(rapidjson)
+find_package(RapidJSON)
 
-set(JINJA2_PRIVATE_LIBS_INT ${CONAN_BOOST_PACKAGE_NAME} fmt::fmt rapidjson::rapidjson)
+set(JINJA2_PRIVATE_LIBS_INT ${CONAN_BOOST_PACKAGE_NAME} fmt::fmt RapidJSON::RapidJSON)
 set(JINJA2_PUBLIC_LIBS_INT expected-lite::expected-lite variant-lite::variant-lite optional-lite::optional-lite string-view-lite::string-view-lite)
