diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1f0cde1..4ad03b6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -70,11 +70,13 @@ if(AVIF_LOCAL_LIBGAV1)
     enable_language(CXX)
 endif()
 
+if(0)
 if(APPLE)
     set(XCRUN xcrun)
 else()
     set(XCRUN)
 endif()
+endif()
 
 if(BUILD_SHARED_LIBS)
     set(AVIF_LIBRARY_PREFIX "${CMAKE_SHARED_LIBRARY_PREFIX}")
@@ -209,6 +211,7 @@ else(libsharpyuv_FOUND)
 endif(libsharpyuv_FOUND)
 # ---------------------------------------------------------------------------------------
 
+if(0)
 # Enable all warnings
 include(CheckCCompilerFlag)
 if(CMAKE_C_COMPILER_ID MATCHES "Clang")
@@ -257,6 +260,7 @@ if(AVIF_ENABLE_COVERAGE)
         message(WARNING "libavif: Ignoring request for coverage (AVIF_ENABLE_COVERAGE); only clang is currently supported.")
     endif()
 endif()
+endif()
 
 if(AVIF_ENABLE_EXPERIMENTAL_YCGCO_R)
     add_compile_definitions(AVIF_ENABLE_EXPERIMENTAL_YCGCO_R)
