--- cmake/IgnUtils.cmake
+++ cmake/IgnUtils.cmake
@@ -1391,7 +1391,7 @@ macro(_ign_add_library_or_component)
     # Generate export macro headers
     # Note: INTERFACE libraries do not need the export header
     set(binary_include_dir
-      "${CMAKE_BINARY_DIR}/include/${include_dir}")
+      "${PROJECT_BINARY_DIR}/include/${include_dir}")
 
     set(implementation_file_name "${binary_include_dir}/detail/Export.hh")

