diff --git a/cmake/opentelemetry-proto.cmake b/cmake/opentelemetry-proto.cmake
index 8d8f868..2a78f98 100644
--- a/cmake/opentelemetry-proto.cmake
+++ b/cmake/opentelemetry-proto.cmake
@@ -214,6 +214,10 @@ else()
     ${METRICS_SERVICE_PB_CPP_FILE})
 endif()
 
+if(TARGET gRPC::grpc++)
+  target_link_libraries(opentelemetry_proto PUBLIC gRPC::grpc++)
+endif()
+
 if(needs_proto_download)
   add_dependencies(opentelemetry_proto opentelemetry-proto)
 endif()
