diff --git a/a/CMakeLists.txt b/b/CMakeLists.txt
index 09c21fd..a8d7d16 100644
--- a/a/CMakeLists.txt
+++ b/b/CMakeLists.txt
@@ -203,7 +203,6 @@ if(WITH_JAEGER)
   find_package(Thrift QUIET)
   if(Thrift_FOUND)
     find_package(Boost REQUIRED)
-    include_directories(${Boost_INCLUDE_DIR})
   else()
     # Install Thrift and propagate via vcpkg toolchain file
     if(WIN32 AND (NOT DEFINED CMAKE_TOOLCHAIN_FILE))
