diff --git a/CMakeLists.txt b/CMakeLists.txt
index a1b69340..d4f52514 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -209,7 +209,6 @@ if(WITH_JAEGER)
   find_package(Thrift QUIET)
   if(Thrift_FOUND)
     find_package(Boost REQUIRED)
-    include_directories(${Boost_INCLUDE_DIR})
   else()
     # Install Thrift and propagate via vcpkg toolchain file
     if(WIN32 AND (NOT DEFINED CMAKE_TOOLCHAIN_FILE))
