diff --git a/libmdnsd/CMakeLists.txt b/libmdnsd/CMakeLists.txt
index b7d2a4d..a302987 100644
--- a/libmdnsd/CMakeLists.txt
+++ b/libmdnsd/CMakeLists.txt
@@ -73,7 +73,7 @@ endif()
 
 add_library(libmdnsd ${LIBRARY_FILES} ${PUBLIC_INCLUDES})
 set_property(TARGET libmdnsd PROPERTY OUTPUT_NAME "mdnsd")
-set_property(TARGET libmdnsd PROPERTY DEFINE_SYMBOL "MDNSD_DYNAMIC_LINKING_EXPORT")
+target_compile_definitions(libmdnsd PUBLIC MDNSD_DYNAMIC_LINKING_EXPORT)
 
 if(WIN32)
   target_link_libraries(libmdnsd PUBLIC wsock32 ws2_32)
