diff --git a/CMakeLists.txt b/CMakeLists.txt
index b7d5036..2032d2b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -15,11 +15,9 @@ if(MDNSD_BUILD_FUZZING)
     add_definitions(-DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION)
 endif()
 
-set(BUILD_SHARED_LIBS_DEFAULT ON)
 if(MDNSD_BUILD_OSS_FUZZ)
-    set(BUILD_SHARED_LIBS_DEFAULT OFF)
+    set(BUILD_SHARED_LIBS OFF)
 endif()
-option(BUILD_SHARED_LIBS "Build libmdnsd as a shared library" ${BUILD_SHARED_LIBS_DEFAULT})
 
 if(MDNSD_BUILD_FUZZING_CORPUS)
     add_definitions(-DMDNSD_DEBUG_DUMP_PKGS_FILE)
@@ -51,8 +49,8 @@ if(MDNSD_BUILD_FUZZING OR MDNSD_BUILD_OSS_FUZZ)
     add_subdirectory(tests/fuzz)
 endif()
 
-add_executable(mdnsd mdnsd.c)
-target_link_libraries(mdnsd PRIVATE libmdnsd)
+add_executable(mdns_daemon mdnsd.c)
+target_link_libraries(mdns_daemon PRIVATE libmdnsd)
 
 add_executable(mquery mquery.c)
 target_link_libraries(mquery PRIVATE libmdnsd)
diff --git a/libmdnsd/CMakeLists.txt b/libmdnsd/CMakeLists.txt
index b7d2a4d..b3f5e16 100644
--- a/libmdnsd/CMakeLists.txt
+++ b/libmdnsd/CMakeLists.txt
@@ -72,8 +72,8 @@ if(MDNSD_COMPILE_AS_CXX)
 endif()
 
 add_library(libmdnsd ${LIBRARY_FILES} ${PUBLIC_INCLUDES})
-set_property(TARGET libmdnsd PROPERTY OUTPUT_NAME "mdnsd")
-set_property(TARGET libmdnsd PROPERTY DEFINE_SYMBOL "MDNSD_DYNAMIC_LINKING_EXPORT")
+set_target_properties(libmdnsd PROPERTIES OUTPUT_NAME "mdnsd")
+target_compile_definitions(libmdnsd PUBLIC MDNSD_DYNAMIC_LINKING_EXPORT)
 
 if(WIN32)
   target_link_libraries(libmdnsd PUBLIC wsock32 ws2_32)
