diff --git a/libmdnsd/CMakeLists.txt b/libmdnsd/CMakeLists.txt
index 3c29b08..105eb52 100644
--- a/libmdnsd/CMakeLists.txt
+++ b/libmdnsd/CMakeLists.txt
@@ -25,7 +25,7 @@ if(NOT MDNSD_COMPILE_AS_CXX AND (CMAKE_COMPILER_IS_GNUCC OR "x${CMAKE_C_COMPILER
     add_definitions(-Wpedantic)
   endif()
   if(NOT WIN32 AND NOT CYGWIN)
-    add_definitions(-Wshadow -Wconversion -fvisibility=hidden -fPIC)
+    add_definitions(-Wshadow -Wconversion -fvisibility=hidden)
   endif()
 endif()
 
