diff --git a/CMakeLists.txt b/CMakeLists.txt
index c48f798..ff622b7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -27,19 +27,6 @@ endif()
 
 option(MDNSD_ENABLE_SANITIZERS "enable sanitizers for mdnsd ON)" ON)
 
-# Debug
-string(TOLOWER "${CMAKE_BUILD_TYPE}" BUILD_TYPE_LOWER_CASE)
-if(MDNSD_ENABLE_SANITIZERS AND BUILD_TYPE_LOWER_CASE STREQUAL "debug" AND UNIX)
-    if("x${CMAKE_C_COMPILER_ID}" STREQUAL "xClang")
-        # Add default sanitizer settings when using clang and Debug build.
-        # This allows e.g. CLion to find memory locations for SegFaults
-        message(STATUS "Sanitizer enabled")
-        set(SANITIZER_FLAGS "-g -fno-omit-frame-pointer -gline-tables-only -fsanitize=address -fsanitize-address-use-after-scope -fsanitize-coverage=trace-pc-guard,trace-cmp -fsanitize=leak -fsanitize=undefined")
-        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${SANITIZER_FLAGS}")
-        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${SANITIZER_FLAGS}")
-    endif()
-endif()
-
 add_subdirectory(libmdnsd)
 
 #option(MDNSD_BUILD_UNIT_TESTS "Build the unit tests" OFF)
