diff --git a/cmake/helpers/GdalDriverHelper.cmake b/cmake/helpers/GdalDriverHelper.cmake
--- a/cmake/helpers/GdalDriverHelper.cmake
+++ b/cmake/helpers/GdalDriverHelper.cmake
@@ -249,6 +249,7 @@
                 target_compile_options(${_TARGET} PRIVATE ${_res})
             endif ()
             get_property(_res TARGET ${_LIB} PROPERTY INTERFACE_LINK_LIBRARIES)
+            list(REMOVE_ITEM _res ${_LIB})
             if (_res)
                 gdal_target_interfaces(${_TARGET} ${_res})
             endif ()
diff --git a/cmake/helpers/GdalGenerateConfig.cmake b/cmake/helpers/GdalGenerateConfig.cmake
--- a/cmake/helpers/GdalGenerateConfig.cmake
+++ b/cmake/helpers/GdalGenerateConfig.cmake
@@ -50,6 +50,7 @@
             if(TARGET "${_lib}")
                 get_target_property(_link_libraries ${_lib} INTERFACE_LINK_LIBRARIES)
                 get_target_property(_type ${_lib} TYPE)
+                list(REMOVE_ITEM _link_libraries ${_lib})
                 if(_link_libraries AND NOT TYPE STREQUAL "SHARED_LIBRARY")
                     list(INSERT ARGN 0 ${_link_libraries})
                 endif()
