diff --git cmake/OpenEXRSetup.cmake cmake/OpenEXRSetup.cmake
index c3da7225..eacbcfe7 100644
--- cmake/OpenEXRSetup.cmake
+++ cmake/OpenEXRSetup.cmake
@@ -267,7 +267,7 @@ if (NOT OPENEXR_FORCE_INTERNAL_OPENJPH)
     endif()
 
     message(STATUS "Using OpenJPH ${openjph_VERSION} from ${openjph_DIR}")
-    set(EXR_OPENJPH_LIB openjph)
+    set(EXR_OPENJPH_LIB openjph::openjph)
   else()
     # If not found, try pkgconfig
     find_package(PkgConfig)
