diff --git a/cmake/deps.cmake b/cmake/deps.cmake
index f19c0378..7b2d7cd7 100644
--- a/cmake/deps.cmake
+++ b/cmake/deps.cmake
@@ -24,7 +24,8 @@ check_c_source_compiles("
 # Check for libraries.
 find_package(Threads)
 if(Threads_FOUND)
-  if(CMAKE_USE_PTHREADS_INIT)
+  # work around cmake bug on QNX (https://chromium-review.googlesource.com/c/webm/libwebp/+/2637274)
+  if(CMAKE_USE_PTHREADS_INIT AND NOT CMAKE_SYSTEM_NAME STREQUAL "QNX")
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -pthread")
   endif()
   foreach(PTHREAD_TEST HAVE_PTHREAD_PRIO_INHERIT PTHREAD_CREATE_UNDETACHED)
