(required for static MSVC)
--- aclocal.m4
+++ aclocal.m4
@@ -8506,7 +8506,7 @@
 	esac
 
 	# either way, check for the library header files
-	AC_CHECK_HEADERS(pcre2-posix.h pcreposix.h)
+	AC_CHECK_HEADERS(pcre2posix.h pcreposix.h)
 fi
 ])dnl
 dnl ---------------------------------------------------------------------------
--- configure
+++ configure
@@ -7516,7 +7516,7 @@
 
 	# either way, check for the library header files
 
-for ac_header in pcre2-posix.h pcreposix.h
+for ac_header in pcre2posix.h pcreposix.h
 do
 as_ac_Header=`echo "ac_cv_header_$ac_header" | $as_tr_sh`
 echo "$as_me:7522: checking for $ac_header" >&5
--- form/fty_regex.c
+++ form/fty_regex.c
@@ -39,8 +39,8 @@
 
 #if HAVE_REGEX_H_FUNCS || HAVE_LIB_PCRE2	/* We prefer POSIX regex */
 
-#if HAVE_PCRE2_POSIX_H
-#include <pcre2-posix.h>
+#if HAVE_PCRE2POSIX_H
+#include <pcre2posix.h>
 #elif HAVE_PCREPOSIX_H
 #include <pcreposix.h>
 #else
--- include/ncurses_defs
+++ include/ncurses_defs
@@ -120,7 +120,7 @@
 HAVE_NC_ALLOC_H
 HAVE_NEWPAD	1
 HAVE_PANEL_H
-HAVE_PCRE2_POSIX_H
+HAVE_PCRE2POSIX_H
 HAVE_PCREPOSIX_H
 HAVE_POLL
 HAVE_POLL_H
