From b4ee22aedf3b084df93532d14dada7a7e4ade518 Mon Sep 17 00:00:00 2001
From: Anonymous Maarten <anonymous.maarten@gmail.com>
Date: Sat, 14 Mar 2020 11:40:00 +0100
Subject: [PATCH 09/15] Added --enable-fvisibility configure option

---
 aclocal.m4   | 32 +++++++++++++++++++++++++++++++-
 configure.in |  2 ++
 2 files changed, 33 insertions(+), 1 deletion(-)

diff --git a/aclocal.m4 b/aclocal.m4
index e6ccc42f..55c65d1a 100644
--- a/aclocal.m4
+++ b/aclocal.m4
@@ -9068,4 +9068,34 @@ make an error
 	fi
 fi
 fi # cf_cv_posix_visible
-])
+])dnl ---------------------------------------------------------------------------
+dnl CF_WITH_FVISIBILITY version: 1 updated: 2020/03/14 10:53:46
+dnl ----------------
+dnl Check whether the compiler understands -fvisibility=hidden
+AC_DEFUN([CF_WITH_FVISIBILITY],[
+AC_MSG_CHECKING(if you want to use fvisibility)
+NCURSES_EXPORT=
+AC_ARG_WITH(fvisibility,
+	[  --with-fvisibility      use -fvisibility=hidden],
+	[cf_with_fvisibility=$withval],
+	[cf_with_fvisibility=no])
+
+if test "$cf_with_fvisibility" = yes; then
+    _save_cflags="$CFLAGS"
+    CFLAGS=-fvisibility=hidden
+    AC_TRY_COMPILE([],[
+__attribute__ ((visibility("default"))) int somefunc() {return 42;}
+],
+    [cf_fvisibility_set_ok=yes],
+    [AC_MSG_ERROR([The compiler does not support the -fvisibility argument])])
+    CFLAGS=$_save_cflags
+    if test "$cf_with_fvisibility" = yes; then
+        CF_ADD_CFLAGS([-fvisibility=hidden]) dnl FIXME: also need to test CXX if cxx enabled + CF_ADD_CXXFLAGS
+        NCURSES_EXPORT="__attribute__ ((visibility(\"default\")))"
+    fi
+else
+    cf_fvisibility_set_ok=no
+fi
+AC_SUBST(NCURSES_EXPORT)
+AC_MSG_RESULT($cf_fvisibility_set_ok)
+])dnl
diff --git a/configure.in b/configure.in
index fc00f9dc..0c8bea3a 100644
--- a/configure.in
+++ b/configure.in
@@ -1083,6 +1083,8 @@ AC_ARG_WITH(rcs-ids,
 AC_MSG_RESULT($with_rcs_ids)
 test "x$with_rcs_ids" = xyes && AC_DEFINE(USE_RCS_IDS,1,[Define to 1 to compile-in RCS identifiers])
 
+CF_WITH_FVISIBILITY
+
 ###############################################################################
 CF_MAN_PAGES([ captoinfo clear infocmp infotocap reset tabs tic toe tput tset ])
 
-- 
2.21.1

