--- include/curses.h.in
+++ include/curses.h.in
@@ -298,7 +298,7 @@
 /* line graphics */
 
 #if @BROKEN_LINKER@ || NCURSES_REENTRANT
-NCURSES_WRAPPED_VAR(chtype*, acs_map);
+NCURSES_WRAPPED_VAR(NCURSES_IMPEXP chtype*, acs_map);
 #define acs_map NCURSES_PUBLIC_VAR(acs_map())
 #else
 extern NCURSES_EXPORT_VAR(chtype) acs_map[];
--- include/MKterm.h.awk.in	2020-02-03 00:34:34.000000000 +0100
+++ include/MKterm.h.awk.in	2020-02-20 04:25:49.184821200 +0100
@@ -253,15 +253,15 @@
 	print  "#endif"
 	print  ""
 	print  "#if @BROKEN_LINKER@ || @cf_cv_enable_reentrant@"
-	print  "NCURSES_WRAPPED_VAR(NCURSES_CONST char * const *, boolnames);"
-	print  "NCURSES_WRAPPED_VAR(NCURSES_CONST char * const *, boolcodes);"
-	print  "NCURSES_WRAPPED_VAR(NCURSES_CONST char * const *, boolfnames);"
-	print  "NCURSES_WRAPPED_VAR(NCURSES_CONST char * const *, numnames);"
-	print  "NCURSES_WRAPPED_VAR(NCURSES_CONST char * const *, numcodes);"
-	print  "NCURSES_WRAPPED_VAR(NCURSES_CONST char * const *, numfnames);"
-	print  "NCURSES_WRAPPED_VAR(NCURSES_CONST char * const *, strnames);"
-	print  "NCURSES_WRAPPED_VAR(NCURSES_CONST char * const *, strcodes);"
-	print  "NCURSES_WRAPPED_VAR(NCURSES_CONST char * const *, strfnames);"
+	print  "NCURSES_WRAPPED_VAR(NCURSES_IMPEXP NCURSES_CONST char * const *, boolnames);"
+	print  "NCURSES_WRAPPED_VAR(NCURSES_IMPEXP NCURSES_CONST char * const *, boolcodes);"
+	print  "NCURSES_WRAPPED_VAR(NCURSES_IMPEXP NCURSES_CONST char * const *, boolfnames);"
+	print  "NCURSES_WRAPPED_VAR(NCURSES_IMPEXP NCURSES_CONST char * const *, numnames);"
+	print  "NCURSES_WRAPPED_VAR(NCURSES_IMPEXP NCURSES_CONST char * const *, numcodes);"
+	print  "NCURSES_WRAPPED_VAR(NCURSES_IMPEXP NCURSES_CONST char * const *, numfnames);"
+	print  "NCURSES_WRAPPED_VAR(NCURSES_IMPEXP NCURSES_CONST char * const *, strnames);"
+	print  "NCURSES_WRAPPED_VAR(NCURSES_IMPEXP NCURSES_CONST char * const *, strcodes);"
+	print  "NCURSES_WRAPPED_VAR(NCURSES_IMPEXP NCURSES_CONST char * const *, strfnames);"
 	print  ""
 	print  "#define boolnames  NCURSES_PUBLIC_VAR(boolnames())"
 	print  "#define boolcodes  NCURSES_PUBLIC_VAR(boolcodes())"
--- ncurses/build.priv.h
+++ ncurses/build.priv.h
@@ -88,7 +88,7 @@
 /* declare these, to avoid needing term.h */
 #if BROKEN_LINKER || USE_REENTRANT
 #define NCURSES_ARRAY(name) \
-	NCURSES_WRAPPED_VAR(NCURSES_CONST char * const *, name)
+	NCURSES_WRAPPED_VAR(NCURSES_IMPEXP NCURSES_CONST char * const *, name)
 
     NCURSES_ARRAY(boolnames);
     NCURSES_ARRAY(boolfnames);
