--- CMakeLists.txt
+++ CMakeLists.txt
@@ -291,7 +291,13 @@
         CLEAN_DIRECT_OUTPUT 1
         OUTPUT_NAME ${PROJECT_NAME})
 
-target_link_libraries(fbow ${OpenCV_LIBS} ${OpenMP_CXX_LIBRARIES})
+target_link_libraries(fbow
+    opencv_core
+    opencv_features2d
+    opencv_highgui
+    "$<$<BOOL:${USE_CONTRIB}>:opencv_xfeatures2d>"
+    OpenMP::OpenMP_CXX
+)
 target_include_directories(fbow PUBLIC
         "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
         "$<INSTALL_INTERFACE:include>")
