--- SConstruct
+++ SConstruct
@@ -140,8 +140,8 @@
     EnumVariable('MSVC_VERSION',
                  "Visual C++ to use for building (E.g. 11.0, 9.0)",
                  None,
-                 allowed_values=('14.0', '12.0',
+                 allowed_values=('16.0', '15.0', '14.0', '12.0',
                                  '11.0', '10.0', '9.0', '8.0', '6.0')
                 ),
 
     # We always documented that we handle an install layout, but in fact we
@@ -273,11 +273,11 @@
   # Choose runtime and optimization
   if debug:
     # Disable optimizations for debugging, use debug DLL runtime
-    env.Append(CCFLAGS=['/Od', '/MDd'])
+    env.Append(CCFLAGS=['/Od'])
     env.Append(CPPDEFINES=['DEBUG', '_DEBUG'])
   else:
     # Optimize for speed, use DLL runtime
-    env.Append(CCFLAGS=['/O2', '/MD'])
+    env.Append(CCFLAGS=['/O2'])
     env.Append(CPPDEFINES=['NDEBUG'])
     env.Append(LINKFLAGS=['/RELEASE'])
 
@@ -316,8 +316,8 @@
 
   env.Append(LIBS=[apr_libs, apu_libs])
   if not env.get('SOURCE_LAYOUT', None):
-    env.Append(LIBPATH=['$APR/lib', '$APU/lib'],
-               CPPPATH=['$APR/include/apr-1', '$APU/include/apr-1'])
+    env.Append(LIBPATH=['$APR/lib', '$APU/lib'],
+               CPPPATH=['$APR/include', '$APU/include'])
   elif aprstatic:
     env.Append(LIBPATH=['$APR/LibR','$APU/LibR'],
                CPPPATH=['$APR/include', '$APU/include'])
@@ -337,7 +337,7 @@
   # openssl
   env.Append(LIBS=['libeay32.lib', 'ssleay32.lib'])
   if not env.get('SOURCE_LAYOUT', None):
-    env.Append(CPPPATH=['$OPENSSL/include/openssl'],
+    env.Append(CPPPATH=['$OPENSSL/include', '$OPENSSL/include/openssl'],
                LIBPATH=['$OPENSSL/lib'])
   elif 0: # opensslstatic:
     env.Append(CPPPATH=['$OPENSSL/inc32'],
