--- SConstruct	2015-09-17 14:46:24.000000000 +0200
+++ SConstruct	2023-10-23 12:11:41.087125000 +0200
@@ -117,6 +117,7 @@
                   None),
   RawListVariable('CPPFLAGS', "Extra flags for the C preprocessor "
                   "(space separated)", None), 
+  RawListVariable('ZLIB_LIBNAME', 'Name of the zlib library file', 'zlib.lib'),
   )
 
 if sys.platform == 'win32':
@@ -326,7 +327,7 @@
                CPPPATH=['$APR/include', '$APU/include'])
 
   # zlib
-  env.Append(LIBS=['zlib.lib'])
+  env.Append(LIBS=[env['ZLIB_LIBNAME'][0]])
   if not env.get('SOURCE_LAYOUT', None):
     env.Append(CPPPATH=['$ZLIB/include'],
                LIBPATH=['$ZLIB/lib'])
