--- src/x86/win64.S
+++ src/x86/win64.S
@@ -170,7 +170,7 @@
 	mov	rcx, QWORD PTR RVALUE[rbp]
 	movzx   rax, ax
 	mov	QWORD PTR [rcx], rax
-	jmp	SHORT ret_void$
+	jmp ret_void$
 
 ret_sint16$:
  	cmp	DWORD PTR CIF_FLAGS[rbp], FFI_TYPE_SINT16
@@ -179,7 +179,7 @@
 	mov	rcx, QWORD PTR RVALUE[rbp]
 	movsx   rax, ax
 	mov	QWORD PTR [rcx], rax
-	jmp	SHORT ret_void$
+	jmp ret_void$
 
 ret_uint32$:
  	cmp	DWORD PTR CIF_FLAGS[rbp], FFI_TYPE_UINT32
@@ -188,7 +188,7 @@
 	mov	rcx, QWORD PTR RVALUE[rbp]
 	mov     eax, eax
 	mov	QWORD PTR [rcx], rax
-	jmp	SHORT ret_void$
+	jmp ret_void$
 
 ret_sint32$:
  	cmp	DWORD PTR CIF_FLAGS[rbp], FFI_TYPE_SINT32
@@ -197,7 +197,7 @@
 	mov	rcx, QWORD PTR RVALUE[rbp]
 	cdqe
 	mov	QWORD PTR [rcx], rax
-	jmp	SHORT ret_void$
+	jmp ret_void$
 
 ret_float$:
  	cmp	DWORD PTR CIF_FLAGS[rbp], FFI_TYPE_FLOAT
@@ -205,7 +205,7 @@
 
  	mov	rax, QWORD PTR RVALUE[rbp]
  	movss	DWORD PTR [rax], xmm0
- 	jmp	SHORT ret_void$
+ 	jmp ret_void$
 
 ret_double$:
  	cmp	DWORD PTR CIF_FLAGS[rbp], FFI_TYPE_DOUBLE
@@ -213,7 +213,7 @@
 
  	mov	rax, QWORD PTR RVALUE[rbp]
  	movlpd	QWORD PTR [rax], xmm0
- 	jmp	SHORT ret_void$
+ 	jmp ret_void$
 
 ret_uint64$:
   	cmp	DWORD PTR CIF_FLAGS[rbp], FFI_TYPE_UINT64
@@ -221,7 +221,7 @@
 
  	mov	rcx, QWORD PTR RVALUE[rbp]
  	mov	QWORD PTR [rcx], rax
- 	jmp	SHORT ret_void$
+ 	jmp ret_void$
 
 ret_sint64$:
   	cmp	DWORD PTR CIF_FLAGS[rbp], FFI_TYPE_SINT64
@@ -229,7 +229,7 @@
 
  	mov	rcx, QWORD PTR RVALUE[rbp]
  	mov	QWORD PTR [rcx], rax
- 	jmp	SHORT ret_void$
+ 	jmp ret_void$
 
 ret_pointer$:
   	cmp	DWORD PTR CIF_FLAGS[rbp], FFI_TYPE_POINTER
@@ -237,7 +237,7 @@
 
  	mov	rcx, QWORD PTR RVALUE[rbp]
  	mov	QWORD PTR [rcx], rax
- 	jmp	SHORT ret_void$
+ 	jmp ret_void$
 
 ret_int$:
   	cmp	DWORD PTR CIF_FLAGS[rbp], FFI_TYPE_INT
@@ -246,7 +246,7 @@
 	mov	rcx, QWORD PTR RVALUE[rbp]
 	cdqe
 	mov	QWORD PTR [rcx], rax
- 	jmp	SHORT ret_void$
+ 	jmp ret_void$
 
 ret_void$:
 	xor	rax, rax
