From e061243958d63080956887e121d275533570f7cc Mon Sep 17 00:00:00 2001
From: Pan7 <panter@gmx.net>
Date: Mon, 14 Sep 2015 08:25:09 +0200
Subject: [PATCH] Fix configuring msvc compiler

---
 msvcc.sh | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/msvcc.sh b/msvcc.sh
index 65fbfef7e..1a6e73f1f 100755
--- a/msvcc.sh
+++ b/msvcc.sh
@@ -55,6 +55,10 @@ while [ $# -gt 0 ]
 do
   case $1
   in
+    --version)
+      args="-help"
+      shift 1
+    ;;
     -fexceptions)
       # Don't enable exceptions for now.
       #args="$args -EHac"
