--- a/third-party/rsutils/src/control-c-handler.cpp
+++ b/third-party/rsutils/src/control-c-handler.cpp
@@ -6,6 +6,7 @@
 
 #include <signal.h>
 #include <atomic>
+#include <stdexcept>
 
 
 namespace rsutils {
--- a/third-party/rsutils/src/special-folder.cpp
+++ b/third-party/rsutils/src/special-folder.cpp
@@ -4,6 +4,7 @@
 #include <rsutils/os/special-folder.h>
 #include <rsutils/easylogging/easyloggingpp.h>
 #include <locale.h>
+#include <stdexcept>
 
 #ifdef _WIN32
