diff --git a/CMake/install_config.cmake b/CMake/install_config.cmake
index 9f5666611..2ca5625ce 100644
--- a/CMake/install_config.cmake
+++ b/CMake/install_config.cmake
@@ -48,4 +48,4 @@ install(FILES "${CMAKE_CURRENT_BINARY_DIR}/config/realsense2.pc"
         DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig"
 )
 
-install(CODE "execute_process(COMMAND ldconfig)")
+# install(CODE "execute_process(COMMAND ldconfig)")
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0c15d9510..2c55dbe7b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -3,7 +3,7 @@ cmake_minimum_required(VERSION 3.1.0)
 project(librealsense2 LANGUAGES CXX C)
 
 include(CMake/lrs_options.cmake)
-include(CMake/connectivity_check.cmake)
+set(INTERNET_CONNECTION ON)
 
 #Deprecation message, should be removed in future releases
 if(${FORCE_LIBUVC} OR ${FORCE_WINUSB_UVC} OR ${ANDROID_USB_HOST_UVC})
