From 34d1bf2f1dcc138f7cb3a54daf771931cd799785 Mon Sep 17 00:00:00 2001
From: rezso <rezso@rezso.net>
Date: Sat, 8 Feb 2020 19:08:06 +0100
Subject: [PATCH] Add install target to CMake build

---
 CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 60fd8a1a7..228535770 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -127,3 +127,5 @@ endforeach()
 
 enable_testing()
 add_test(NinjaTest ninja_test)
+
+install(TARGETS ninja DESTINATION bin)
