diff --git a/CMakeLists.txt b/CMakeLists.txt
index c939dec..1fdca97 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -395,6 +395,8 @@ function(b_embed TARGET FILENAME)
     # Add the generated files to the target
     target_include_directories(${TARGET} PUBLIC ${EMBED_BINARY_DIR}/autogen/${TARGET_ID}/include)
     target_sources(${TARGET} PRIVATE ${CPP_FILE} ${EMBED_BINARY_DIR}/embed_impl.cpp ${EMBED_HPP})
+    find_package(Threads REQUIRED)
+    target_link_libraries(${TARGET} PRIVATE Threads::Threads)
 
     if (MSVC)
         target_sources(${TARGET} PRIVATE ${FULL_PATH})
