--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -647,7 +647,7 @@ configure_package_config_file(
 install(
     TARGETS ${INSTALL_TARGETS}
     EXPORT "${TARGETS_EXPORT_NAME}"
-    RUNTIME DESTINATION "bin"
+    DESTINATION "bin"
     COMPONENT h3
 )
 
@@ -656,6 +656,7 @@ install(
     EXPORT "${TARGETS_EXPORT_NAME}"
     LIBRARY DESTINATION "lib"
     ARCHIVE DESTINATION "lib"
+    RUNTIME DESTINATION "bin"
     COMPONENT libh3
 )
 
