diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0b0cc090..04a3b32b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -463,15 +463,10 @@ INCLUDE(character_sets)
 INCLUDE(compile_flags)
 INCLUDE(cpu_info)
 INCLUDE(zlib)
-INCLUDE(lz4)
-INCLUDE(re2)
-INCLUDE(icu)
-INCLUDE(libevent)
 INCLUDE(ssl)
 INCLUDE(sasl)
 INCLUDE(rpc)
 INCLUDE(readline)
-INCLUDE(protobuf)
 INCLUDE(package_name)
 INCLUDE(libutils)
 INCLUDE(plugin)
@@ -481,7 +476,6 @@ INCLUDE(install_layout)
 INCLUDE(mysql_add_executable)
 INCLUDE(mysqlgcs)
 INCLUDE(curl)
-INCLUDE(rapidjson)
 
 OPTION(WITH_JEMALLOC "Use jemalloc rather than builtin malloc/free etc." OFF)
 IF(WITH_JEMALLOC)
@@ -941,7 +935,6 @@ ELSE()
   SET(DEFAULT_TMPDIR "\"${TMPDIR}\"")
 ENDIF()
 
-INCLUDE(cmake/boost.cmake)
 
 IF (LINUX)
   OPTION(WITH_SYSTEMD "Enable installation of systemd support files" OFF)
@@ -1115,25 +1108,9 @@ MYSQL_CHECK_SSL()
 MYSQL_CHECK_SSL_DLLS()
 
-# Add system/bundled editline.
-MYSQL_CHECK_EDITLINE()
-# Add libevent
-MYSQL_CHECK_LIBEVENT()
-# Add lz4 library
-MYSQL_CHECK_LZ4()
-# Add re2 library
-MYSQL_CHECK_RE2()
-# Add icu library
-MYSQL_CHECK_ICU()
 # Add SASL library
 MYSQL_CHECK_SASL()
-# Add protoc and libprotobuf
-IF(NOT WITHOUT_SERVER)
-  MYSQL_CHECK_PROTOBUF()
-ENDIF()
 # Try and set CURL_LIBRARY
 MYSQL_CHECK_CURL()
-# Add RapidJSON library.
-MYSQL_CHECK_RAPIDJSON()
 
 MACRO(MY_INCLUDE_SYSTEM_DIRECTORIES LIBRARY)
   IF(${WITH_${LIBRARY}} STREQUAL "bundled")
@@ -1248,7 +1225,6 @@ IF(WITH_UNIT_TESTS)
   ADD_SUBDIRECTORY(unittest/mytap/t)
 ENDIF()
 
-ADD_SUBDIRECTORY(client)
 ADD_SUBDIRECTORY(utilities)
 ADD_SUBDIRECTORY(share)
 ADD_SUBDIRECTORY(libservices)
@@ -1290,7 +1266,4 @@ IF(NOT WITHOUT_SERVER)
 ENDIF()
 
-IF(UNIX)
-  ADD_SUBDIRECTORY(man)
-ENDIF()
 
 IF(NOT WITHOUT_SERVER)
diff --git a/cmake/zlib.cmake b/cmake/zlib.cmake
--- a/cmake/zlib.cmake
+++ b/cmake/zlib.cmake
@@ -51,28 +51,6 @@ MACRO (MYSQL_CHECK_ZLIB_WITH_COMPRESS)
   IF(WITH_ZLIB STREQUAL "bundled")
     MYSQL_USE_BUNDLED_ZLIB()
   ELSE()
-    SET(ZLIB_FIND_QUIETLY TRUE)
-    INCLUDE(FindZLIB)
-    IF(ZLIB_FOUND)
-      INCLUDE(CheckFunctionExists)
-      SET(SAVE_CMAKE_REQUIRED_LIBRARIES ${CMAKE_REQUIRED_LIBRARIES})
-      SET(CMAKE_REQUIRED_LIBRARIES ${CMAKE_REQUIRED_LIBRARIES} z)
-      CHECK_FUNCTION_EXISTS(crc32 HAVE_CRC32)
-      CHECK_FUNCTION_EXISTS(compressBound HAVE_COMPRESSBOUND)
-      CHECK_FUNCTION_EXISTS(deflateBound HAVE_DEFLATEBOUND)
-      SET(CMAKE_REQUIRED_LIBRARIES ${SAVE_CMAKE_REQUIRED_LIBRARIES})
-      IF(HAVE_CRC32 AND HAVE_COMPRESSBOUND AND HAVE_DEFLATEBOUND)
-        SET(ZLIB_LIBRARY ${ZLIB_LIBRARIES} CACHE INTERNAL "System zlib library")
-        SET(WITH_ZLIB "system" CACHE STRING
-          "Which zlib to use (possible values are 'bundled' or 'system')")
-        SET(ZLIB_SOURCES "")
-      ELSE()
-        SET(ZLIB_FOUND FALSE CACHE INTERNAL "Zlib found but not usable")
-        MESSAGE(STATUS "system zlib found but not usable")
-      ENDIF()
-    ENDIF()
-    IF(NOT ZLIB_FOUND)
-      MYSQL_USE_BUNDLED_ZLIB()
-    ENDIF()
+    SET(ZLIB_LIBRARY "${CONAN_LIBS_ZLIB}" CACHE INTERNAL "System zlib library")
   ENDIF()
 ENDMACRO()
