--- cmake/ssl.cmake
+++ cmake/ssl.cmake
@@ -277,11 +277,11 @@
     ENDIF()
 
     FIND_LIBRARY(OPENSSL_LIBRARY
-                 NAMES ssl libssl ssleay32 ssleay32MD
+                 NAMES ssl libssl ssleay32 ssleay32MD ssld libssld
                  HINTS ${OPENSSL_ROOT_DIR}/lib)
     FIND_LIBRARY(CRYPTO_LIBRARY
-                 NAMES crypto libcrypto libeay32
+                 NAMES crypto libcrypto libeay32 cryptod libcryptod
                  HINTS ${OPENSSL_ROOT_DIR}/lib)
 
     IF(OPENSSL_INCLUDE_DIR)
       # Verify version number. Version information looks like:
