diff --git a/cmake/package_name.cmake b/cmake/package_name.cmake
index 192035dd..dedc59c6 100644
--- a/cmake/package_name.cmake
+++ b/cmake/package_name.cmake
@@ -94,7 +94,7 @@ MACRO(GET_PACKAGE_FILE_NAME Var)
 
       STRING(REGEX MATCH
         "ProductVersion:[\n\t ]*([0-9]+)\\.([0-9]+)" UNUSED ${SW_VERS_PRODUCTVERSION})
-      IF(NOT CMAKE_MATCH_1 OR NOT CMAKE_MATCH_2)
+      IF(CMAKE_MATCH_1 STREQUAL "" OR CMAKE_MATCH_2 STREQUAL "")
         MESSAGE(FATAL_ERROR "Could not run sw_vers")
       ENDIF()
 
