diff --git a/Installation/lib/cmake/CGAL/CGALConfig.cmake b/Installation/lib/cmake/CGAL/CGALConfig.cmake
index bb8bdcc9c93..903887a5661 100644
--- a/Installation/lib/cmake/CGAL/CGALConfig.cmake
+++ b/Installation/lib/cmake/CGAL/CGALConfig.cmake
@@ -98,6 +98,9 @@ foreach(cgal_lib ${CGAL_LIBRARIES})
     set(CGALConfig_all_targets_are_defined FALSE)
   endif()
 endforeach()
+
+list(APPEND CMAKE_MODULE_PATH ${CGAL_MODULES_DIR})
+
 if(CGALConfig_all_targets_are_defined)
   return()
 endif()
@@ -108,8 +111,6 @@ if(NOT CGAL_FOUND)
   return()
 endif()
 
-list(APPEND CMAKE_MODULE_PATH ${CGAL_MODULES_DIR})
-
 include( ${CGAL_MODULES_DIR}/CGAL_SCM.cmake )
 CGAL_detect_git(${CGAL_CONFIG_DIR}/../../../..)
 
