diff --git a/src/json.hpp b/src/json.hpp
index 620536e..6003f25 100644
--- a/src/json.hpp
+++ b/src/json.hpp
@@ -48,13 +48,13 @@ static void delete_(T* ptr) {
 #define RAPIDJSON_NEW(x) ::datastax::internal::json::new_<x>
 #define RAPIDJSON_DELETE(x) ::datastax::internal::json::delete_(x)
 
-#include "third_party/rapidjson/rapidjson/document.h"
-#include "third_party/rapidjson/rapidjson/stringbuffer.h"
+#include <rapidjson/document.h>
+#include <rapidjson/stringbuffer.h>
 #ifndef JSON_DEBUG
-#include "third_party/rapidjson/rapidjson/writer.h"
+#include <rapidjson/writer.h>
 #define JSON_WRITE_TYPE Writer
 #else
-#include "third_party/rapidjson/rapidjson/prettywriter.h"
+#include <rapidjson/prettywriter.h>
 #define JSON_WRITE_TYPE PrettyWriter
 #endif
 
