diff --git a/CMakeLists.txt b/CMakeLists.txt
index bb66da4..9a3ddfa 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,6 +1,7 @@
 cmake_minimum_required(VERSION 3.15)
 project(jtjson)
 
+if(0) # disable CXX_STANDARD and vendored in double-conversion library
 set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
@@ -16,13 +17,19 @@ add_library(double-conversion STATIC
     double-conversion/strtod.cc
 )
 target_include_directories(double-conversion PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
+endif()
+
+find_library(LIBM m)
+find_package(double-conversion REQUIRED CONFIG)
+find_package(jtckdint REQUIRED CONFIG)
 
 # Main JSON library
 add_library(json STATIC
     json.cpp
 )
-target_link_libraries(json PRIVATE double-conversion)
+target_link_libraries(json PRIVATE $<$<BOOL:${LIBM}>:${LIBM}> double-conversion::double-conversion jtckdint::jtckdint)
 
+if(0) # Disable tests
 # Tests
 add_executable(json_test json_test.cpp)
 target_link_libraries(json_test PRIVATE json)
@@ -45,3 +52,13 @@ add_test(
     COMMAND jsontestsuite_test
     WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
 )
+endif()
+
+include(GNUInstallDirs)
+install(FILES json.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+install(
+    TARGETS json
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+)
