diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0d00a90..5984c15 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -131,4 +131,14 @@ if(MINISCRIPT_BUILD_TESTING)
 	endif()
 endif()
 
-install(TARGETS miniscript-cpp minicmd)
+include(GNUInstallDirs)
+install(
+    TARGETS miniscript-cpp minicmd
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+)
+install(
+	FILES ${MINISCRIPT_HEADERS}
+	DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/MiniScript
+)
\ No newline at end of file
