--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -120,6 +120,7 @@ endif ()
 # ==============================================================================
 # Check C++11 support
 # ==============================================================================
+if(0)
 include(CXX11)
 check_for_cxx11_compiler(CXX11_COMPILER)
 # If a C++11 compiler is available, then set the appropriate flags
@@ -130,6 +131,7 @@ if (CXX11_COMPILER)
 else (CXX11_COMPILER)
   message(FATAL_ERROR "A CXX11 compiler is required.")
 endif (CXX11_COMPILER)
+endif()
 
 # ==============================================================================
 # OpenMP detection
