--- a/cli/CMakeLists.txt	2023-01-28 10:42:04.000000000 +0100
+++ b/cli/CMakeLists.txt	2023-01-31 07:59:36.540879100 +0100
@@ -61,10 +61,9 @@
 add_dependencies(cppcheck copy_cfg)
 add_dependencies(cppcheck copy_addons)
 add_dependencies(cppcheck copy_platforms)
-add_dependencies(cppcheck run-dmake)
 
 install(TARGETS cppcheck
-    RUNTIME DESTINATION ${CMAKE_INSTALL_FULL_BINDIR}
+    DESTINATION ${CMAKE_INSTALL_FULL_BINDIR}
     COMPONENT applications)
 
 install(FILES ${addons}
