--- a/cli/CMakeLists.txt	2022-07-12 23:11:29.000000000 +0200
+++ b/cli/CMakeLists.txt	2022-11-23 22:01:29.111581500 +0100
@@ -45,10 +45,9 @@
 
 add_dependencies(cppcheck copy_cfg)
 add_dependencies(cppcheck copy_addons)
-add_dependencies(cppcheck run-dmake)
 
 install(TARGETS cppcheck
-    RUNTIME DESTINATION ${CMAKE_INSTALL_FULL_BINDIR}
+    DESTINATION ${CMAKE_INSTALL_FULL_BINDIR}
     COMPONENT applications)
 
 install(FILES ${addons}
