commit 3b6a5638cb6c0e7b5680adae04ca1a59d86fe508
Author: Martin Delille <martin@delille.org>
Date:   Fri Sep 2 15:11:19 2022 +0200

    cppcheck-htmlreport python3

diff --git a/htmlreport/cppcheck-htmlreport b/htmlreport/cppcheck-htmlreport
index 5c0a78adf..ce5956c1e 100755
--- a/htmlreport/cppcheck-htmlreport
+++ b/htmlreport/cppcheck-htmlreport
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 from __future__ import unicode_literals
 
