--- a/cmake/findDependencies.cmake	2022-11-08 20:22:59.000000000 +0100
+++ b/cmake/findDependencies.cmake	2022-11-23 21:29:50.222152900 +0100
@@ -36,7 +36,7 @@
 
 if (HAVE_RULES)
     find_path(PCRE_INCLUDE pcre.h)
-    find_library(PCRE_LIBRARY pcre)
+    find_library(PCRE_LIBRARY NAMES pcre pcred)
     if (NOT PCRE_LIBRARY OR NOT PCRE_INCLUDE)
         message(FATAL_ERROR "pcre dependency for RULES has not been found")
     endif()
