--- a/cmake/findDependencies.cmake	2023-09-09 15:10:31.000000000 +0200
+++ b/cmake/findDependencies.cmake	2023-09-11 08:44:40.220863900 +0200
@@ -31,7 +31,7 @@
 
 if (HAVE_RULES)
     find_path(PCRE_INCLUDE pcre.h)
-    find_library(PCRE_LIBRARY pcre)
+    find_library(PCRE_LIBRARY NAMES pcre pcred)
     if (NOT PCRE_LIBRARY OR NOT PCRE_INCLUDE)
         message(FATAL_ERROR "pcre dependency for RULES has not been found")
     endif()
