diff --git a/cmake/Config.cmake b/cmake/Config.cmake
index 2ce5ac4..3cd3a07 100644
--- a/cmake/Config.cmake
+++ b/cmake/Config.cmake
@@ -39,13 +39,12 @@ elseif(CMAKE_SYSTEM_NAME MATCHES "^NetBSD$")
     set(SFML_OS_NETBSD 1)
     # don't use the OpenGL ES implementation on NetBSD
     set(OPENGL_ES 0)
-elseif(${CMAKE_SYSTEM_NAME} STREQUAL "Darwin")
-    if(IOS)
+elseif(${CMAKE_SYSTEM_NAME} STREQUAL "iOS")
         set(SFML_OS_IOS 1)
 
         # use the OpenGL ES implementation on iOS
         set(OPENGL_ES 1)
-    else()
+elseif(${CMAKE_SYSTEM_NAME} STREQUAL "Darwin")
         set(SFML_OS_MACOSX 1)
 
         # don't use the OpenGL ES implementation on Mac OS X
@@ -58,7 +57,6 @@ elseif(${CMAKE_SYSTEM_NAME} STREQUAL "Darwin")
             message(FATAL_ERROR "Unsupported version of OS X: ${MACOSX_VERSION_RAW}")
             return()
         endif()
-    endif()
 elseif(${CMAKE_SYSTEM_NAME} STREQUAL "Android")
     set(SFML_OS_ANDROID 1)

