diff --git a/src/SFML/Graphics/CMakeLists.txt b/src/SFML/Graphics/CMakeLists.txt
index 1e77eec..42bcbd5 100644
--- a/src/SFML/Graphics/CMakeLists.txt
+++ b/src/SFML/Graphics/CMakeLists.txt
@@ -111,13 +111,16 @@ endif()
 
 # find external libraries
 if(SFML_OS_ANDROID)
-    target_link_libraries(sfml-graphics PRIVATE z)
+    find_package(ZLIB REQUIRED)
+    target_link_libraries(sfml-graphics PRIVATE ZLIB::ZLIB)
 elseif(SFML_OS_IOS)
-    target_link_libraries(sfml-graphics PRIVATE z bz2)
+    find_package(ZLIB REQUIRED)
+    find_package(BZip2 REQUIRED)
+    target_link_libraries(sfml-graphics PRIVATE ZLIB::ZLIB BZip2::BZip2)
 endif()
 
-find_package(Freetype REQUIRED)
-target_link_libraries(sfml-graphics PRIVATE Freetype::Freetype)
+find_package(freetype REQUIRED)
+target_link_libraries(sfml-graphics PRIVATE freetype)
 
 # add preprocessor symbols
 target_compile_definitions(sfml-graphics PRIVATE "STBI_FAILURE_USERMSG")
