--- src/CMakeLists.txt
+++ src/CMakeLists.txt
@@ -120,7 +120,7 @@
 # TINYSPLINE_DLANG_INTERMEDIATE_FILE
 #   Full path of D intermediate file.
 ###############################################################################
-set(TINYSPLINE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib CACHE INTERNAL "")
+set(TINYSPLINE_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/lib CACHE INTERNAL "")
 
 # Generic no-config case (e.g. MinGW)
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY_BACKUP ${CMAKE_RUNTIME_OUTPUT_DIRECTORY})
@@ -857,7 +857,7 @@
 	"-l${TINYSPLINE_PKGCONFIG_C_LINK_LIBRARIES}")
 configure_file("${CMAKE_CURRENT_SOURCE_DIR}/pkg/tinyspline.pc.in"
 	"${TINYSPLINE_C_LIBRARY_OUTPUT_NAME}.pc" @ONLY)
-install(FILES "${CMAKE_BINARY_DIR}/src/${TINYSPLINE_C_LIBRARY_OUTPUT_NAME}.pc"
+install(FILES "${CMAKE_CURRENT_BINARY_DIR}/${TINYSPLINE_C_LIBRARY_OUTPUT_NAME}.pc"
 	DESTINATION "${TINYSPLINE_INSTALL_PKGCONFIG_PATH}")
 
 # C++ library
@@ -920,7 +920,7 @@
 		"-l${TINYSPLINE_PKGCONFIG_CXX_LINK_LIBRARIES}")
 	configure_file("${CMAKE_CURRENT_SOURCE_DIR}/pkg/tinysplinecxx.pc.in"
 		"${TINYSPLINE_CXX_LIBRARY_OUTPUT_NAME}.pc" @ONLY)
-	install(FILES "${CMAKE_BINARY_DIR}/src/${TINYSPLINE_CXX_LIBRARY_OUTPUT_NAME}.pc"
+	install(FILES "${CMAKE_CURRENT_BINARY_DIR}/${TINYSPLINE_CXX_LIBRARY_OUTPUT_NAME}.pc"
 		DESTINATION "${TINYSPLINE_INSTALL_PKGCONFIG_PATH}")
 endif()
 
@@ -1073,7 +1073,7 @@
 			endif()
 		endif()
 		set(TINYSPLINE_MACOSX_RELINK_SCRIPT
-			"${CMAKE_BINARY_DIR}/macosx-relink.sh")
+			"${CMAKE_CURRENT_BINARY_DIR}/macosx-relink.sh")
 		configure_file(
 			"${CMAKE_CURRENT_SOURCE_DIR}/pkg/macosx-relink.sh.in"
 			${TINYSPLINE_MACOSX_RELINK_SCRIPT}
@@ -1438,7 +1438,7 @@
 get_filename_component(TINYSPLINE_LIB_DIR ${TINYSPLINE_OUTPUT_DIRECTORY} NAME)
 
 # Dub
-file(REMOVE "${CMAKE_BINARY_DIR}/dub.json")
+file(REMOVE "${CMAKE_CURRENT_BINARY_DIR}/dub.json")
 if(${TINYSPLINE_WITH_DLANG})
 	set(TINYSPLINE_DUB_COPY_FILES "\"${TINYSPLINE_LIB_DIR}/*.d\"")
 	set(TINYSPLINE_DUB_COPY_FILES "${TINYSPLINE_DUB_COPY_FILES}\
@@ -1449,7 +1449,7 @@
 	endforeach()
 	configure_file(
 		"${CMAKE_CURRENT_SOURCE_DIR}/pkg/dub.json.in"
-		"${CMAKE_BINARY_DIR}/dub.json"
+		"${CMAKE_CURRENT_BINARY_DIR}/dub.json"
 		@ONLY)
 endif()
 
@@ -1479,12 +1479,12 @@
 		"tinyspline${TINYSPLINE_LUAROCKS_SPEC}.rockspec")
 	configure_file(
 		"${CMAKE_CURRENT_SOURCE_DIR}/pkg/tinyspline.rockspec.in"
-		"${CMAKE_BINARY_DIR}/${TINYSPLINE_LUAROCKS_SPEC}"
+		"${CMAKE_CURRENT_BINARY_DIR}/${TINYSPLINE_LUAROCKS_SPEC}"
 		@ONLY)
 endif()
 
 # NuGet
-file(REMOVE "${CMAKE_BINARY_DIR}/.nuspec")
+file(REMOVE "${CMAKE_CURRENT_BINARY_DIR}/.nuspec")
 if(${TINYSPLINE_WITH_CSHARP})
 	get_filename_component(TINYSPLINE_NUGET_INTERFACE_FILE
 		${TINYSPLINE_CSHARP_INTERFACE_FILE}
@@ -1531,21 +1531,21 @@
 	endforeach()
 	configure_file(
 		"${CMAKE_CURRENT_SOURCE_DIR}/pkg/.nuspec.in"
-		"${CMAKE_BINARY_DIR}/.nuspec"
+		"${CMAKE_CURRENT_BINARY_DIR}/.nuspec"
 		@ONLY)
 endif()
 
 # Maven
-file(REMOVE "${CMAKE_BINARY_DIR}/pom.xml")
+file(REMOVE "${CMAKE_CURRENT_BINARY_DIR}/pom.xml")
 if(${TINYSPLINE_WITH_JAVA})
 	configure_file(
 		"${CMAKE_CURRENT_SOURCE_DIR}/pkg/pom.xml.in"
-		"${CMAKE_BINARY_DIR}/pom.xml"
+		"${CMAKE_CURRENT_BINARY_DIR}/pom.xml"
 		@ONLY)
 endif()
 
 # Distutils
-file(REMOVE "${CMAKE_BINARY_DIR}/setup.py")
+file(REMOVE "${CMAKE_CURRENT_BINARY_DIR}/setup.py")
 if(${TINYSPLINE_WITH_PYTHON})
 	get_filename_component(TINYSPLINE_SETUP_INTERFACE_FILE
 		${TINYSPLINE_PYTHON_INTERFACE_FILE}
@@ -1561,12 +1561,12 @@
 	set(TINYSPLINE_SETUP_FILE_ARRAY "[${TINYSPLINE_SETUP_FILE_ARRAY}]")
 	configure_file(
 		"${CMAKE_CURRENT_SOURCE_DIR}/pkg/setup.py.in"
-		"${CMAKE_BINARY_DIR}/setup.py"
+		"${CMAKE_CURRENT_BINARY_DIR}/setup.py"
 		@ONLY)
 endif()
 
 # Gem
-file(REMOVE "${CMAKE_BINARY_DIR}/tinyspline.gemspec")
+file(REMOVE "${CMAKE_CURRENT_BINARY_DIR}/tinyspline.gemspec")
 if(${TINYSPLINE_WITH_RUBY})
 	set(TINYSPLINE_RUBY_REQ_VERSION
 			"${RUBY_VERSION_MAJOR}.${RUBY_VERSION_MINOR}")
@@ -1578,7 +1578,7 @@
 	endforeach()
 	configure_file(
 		"${CMAKE_CURRENT_SOURCE_DIR}/pkg/tinyspline.gemspec.in"
-		"${CMAKE_BINARY_DIR}/tinyspline.gemspec"
+		"${CMAKE_CURRENT_BINARY_DIR}/tinyspline.gemspec"
 		@ONLY)
 endif()
 
