find gklib from conan
--- a/libmetis/CMakeLists.txt
+++ b/libmetis/CMakeLists.txt
@@ -3,9 +3,11 @@ include_directories(.)
 # Find sources.
 file(GLOB metis_sources *.c)
 # Build libmetis.
-add_library(metis ${METIS_LIBRARY_TYPE} ${GKlib_sources} ${metis_sources})
+add_library(metis ${METIS_LIBRARY_TYPE} ${metis_sources})
+find_package(gklib CONFIG REQUIRED)
+target_link_libraries(metis PRIVATE gklib::gklib)
 if(UNIX)
-  target_link_libraries(metis m)
+  target_link_libraries(metis PRIVATE m)
 endif()
 
 if(METIS_INSTALL)
