--- a/libshaderc_spvc/src/spvc.cc
+++ b/libshaderc_spvc/src/spvc.cc
@@ -15,9 +15,9 @@
 #include "shaderc/spvc.h"
 
 #include "libshaderc_util/exceptions.h"
-#include "spirv-cross/spirv_glsl.hpp"
-#include "spirv-cross/spirv_hlsl.hpp"
-#include "spirv-cross/spirv_msl.hpp"
+#include "spirv_glsl.hpp"
+#include "spirv_hlsl.hpp"
+#include "spirv_msl.hpp"
 #include "spirv-tools/libspirv.hpp"
 #include "spirv-tools/optimizer.hpp"
 
@@ -169,7 +169,7 @@ size_t shaderc_spvc_compile_options_set_for_fuzzing(
     shaderc_spvc_compile_options_t options, const uint8_t* data, size_t size) {
   if (!data || size < sizeof(*options)) return 0;
 
-  memcpy(options, data, sizeof(*options));
+  memcpy(static_cast<void*>(options), data, sizeof(*options));
   return sizeof(*options);
 }
 
