diff --git a/include.h b/include.h
index e5b8d1e..a55f70d 100644
--- a/include.h
+++ b/include.h
@@ -1,4 +1,4 @@
-/// \defgroup rbd_common RBD common library 
+/// \defgroup rbd_common RBD common library
 ///@{
 
 /// \file include.h
@@ -7,9 +7,13 @@
 #ifndef INCLUDE_LIB
 #define INCLUDE_LIB
 
+#if !defined(_CRT_SECURE_NO_WARNINGS)
+#define _CRT_SECURE_NO_WARNINGS
+#endif
+
 //#define use_namespace                   // define name spaces
 
-//#define SETUP_C_SUBSCRIPTS              // allow element access via A[i][j]
+#define SETUP_C_SUBSCRIPTS              // allow element access via A[i][j]
 
 //#define OPT_COMPATIBLE                  // for use with opt++
 
 
