diff --git a/include/SpiceZpl.h b/include/SpiceZpl.h
index b2e0565..a4251b4 100644
--- a/include/SpiceZpl.h
+++ b/include/SpiceZpl.h
@@ -111,7 +111,7 @@
 #define HAVE_PLATFORM_MACROS_H
  
  
-   #define   CSPICE_PC_CYGWIN
+   #define   CSPICE_PC_CYGWIN_64BIT_GCC
  
 #endif
  
diff --git a/src/cspice/SpiceZpl.h b/src/cspice/SpiceZpl.h
index b2e0565..a4251b4 100644
--- a/src/cspice/SpiceZpl.h
+++ b/src/cspice/SpiceZpl.h
@@ -111,7 +111,7 @@
 #define HAVE_PLATFORM_MACROS_H
  
  
-   #define   CSPICE_PC_CYGWIN
+   #define   CSPICE_PC_CYGWIN_64BIT_GCC
  
 #endif
  
diff --git a/src/cspice/zzplatfm.c b/src/cspice/zzplatfm.c
index 26a0c61..251471e 100644
--- a/src/cspice/zzplatfm.c
+++ b/src/cspice/zzplatfm.c
@@ -349,7 +349,7 @@ static integer c__6 = 6;
 	s_copy(attcpy + 32, "PC", (ftnlen)32, (ftnlen)2);
 	s_copy(attcpy + 64, "MICROSOFT WINDOWS/CYGWIN", (ftnlen)32, (ftnlen)
 		24);
-	s_copy(attcpy + 96, "GCC", (ftnlen)32, (ftnlen)3);
+	s_copy(attcpy + 96, "GCC/64BIT", (ftnlen)32, (ftnlen)9);
 	s_copy(attcpy + 128, "LTL-IEEE", (ftnlen)32, (ftnlen)8);
 	s_copy(attcpy + 160, "LF", (ftnlen)32, (ftnlen)2);
 	s_copy(attcpy + 192, "BIG-IEEE LTL-IEEE", (ftnlen)32, (ftnlen)17);
