see https://github.com/cddlib/cddlib/issues/63

--- a/lib-src/cdd.h
+++ b/lib-src/cdd.h
@@ -27,6 +27,8 @@
 #endif
 #endif
 
+#include <stdio.h>
+
 #if defined(__cplusplus)
 extern "C" {
 #endif
--- a/lib-src/cddtypes.h
+++ b/lib-src/cddtypes.h
@@ -13,6 +13,9 @@
 #define  __CDDTYPES_H
 #endif  /* __CDDTYPES_H */
 
+#include "cddmp.h"
+#include "setoper.h"
+
 #define dd_COPYRIGHT   "Copyright (C) Komei Fukuda, fukuda@math.ethz.ch"
 #define dd_DDVERSION   "Version 0.94m"
 #include <time.h>
--- a/lib-src/cddtypes_f.h
+++ b/lib-src/cddtypes_f.h
@@ -14,6 +14,9 @@
 #define  _CDDTYPES_HF
 #endif  /* _CDDTYPES_HF */
 
+#include "cddmp_f.h"
+#include "setoper.h"
+
 #define ddf_COPYRIGHT   "Copyright (C) Komei Fukuda, fukuda@math.ethz.ch"
 #define ddf_DDVERSION   "Version 0.94m"
 #include <time.h>
