diff --git a/influxdb.hpp b/influxdb.hpp
index 025f01a..3c4fa84 100644
--- a/influxdb.hpp
+++ b/influxdb.hpp
@@ -6,15 +6,17 @@
 
   Please see LICENSE file or visit https://github.com/orca-zhang/influxdb-cpp for details.
  */
+
+#ifndef INFLUXDB_CPP_HPP
+#define INFLUXDB_CPP_HPP
+
 #include <sstream>
 #include <cstring>
 #include <cstdio>
 
 #ifdef _WIN32
-    #define NOMINMAX
     #include <windows.h>
     #include <algorithm>
-    #pragma comment(lib, "ws2_32")
     typedef struct iovec { void* iov_base; size_t iov_len; } iovec;
     inline __int64 writev(int sock, struct iovec* iov, int cnt) {
         __int64 r = send(sock, (const char*)iov->iov_base, iov->iov_len, 0);
@@ -300,3 +302,5 @@ namespace influxdb_cpp {
         }
     }
 }
+
+#endif // INFLUXDB_CPP_HPP
