--- CMakeLists.txt	2022-11-18 03:26:35.000000000 -0300
+++ CMakeLists.txt	2022-12-09 15:00:53.805137270 -0300
@@ -397,7 +397,7 @@
       file(CREATE_LINK \${OLD_REL} \$ENV{DESTDIR}/\${NEW_ABS} SYMBOLIC)")
   endfunction()
   mold_install_relative_symlink(${CMAKE_INSTALL_BINDIR}/mold
-    ${CMAKE_INSTALL_LIBEXECDIR}/mold/ld)
+    ${CMAKE_INSTALL_BINDIR}/ld)
   mold_install_relative_symlink(${CMAKE_INSTALL_BINDIR}/mold
     ${CMAKE_INSTALL_BINDIR}/ld.mold)
   mold_install_relative_symlink(${CMAKE_INSTALL_BINDIR}/mold
