Use external re2
Fix installation of dll

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -33,7 +33,7 @@ include(cmake/setup_build.cmake)
 include(cmake/utils.cmake)
 include(GNUInstallDirs)
 
-add_subdirectory(third_party)
+find_package(re2 REQUIRED CONFIG)
 add_subdirectory(effcee)
 add_subdirectory(fuzzer)
 
--- a/effcee/CMakeLists.txt
+++ b/effcee/CMakeLists.txt
@@ -4,8 +4,8 @@ add_library(effcee
 effcee_default_compile_options(effcee)
 # We need to expose RE2's StringPiece.
 target_include_directories(effcee
-  PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/.. ${EFFCEE_RE2_DIR})
-target_link_libraries(effcee PUBLIC re2 ${CMAKE_THREADS_LIB_INIT})
+  PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/..)
+target_link_libraries(effcee PUBLIC re2::re2 ${CMAKE_THREADS_LIB_INIT})
 
 # TODO(dneto): Avoid installing gtest and gtest_main. ?!
 install(
@@ -14,6 +14,7 @@ install(
   DESTINATION
     include/effcee)
 install(TARGETS effcee
+  RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
   LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
   ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 
