diff --git a/effcee/CMakeLists.txt b/effcee/CMakeLists.txt
index 840c87f..af94ca9 100644
--- a/effcee/CMakeLists.txt
+++ b/effcee/CMakeLists.txt
@@ -13,9 +13,7 @@ install(
     effcee.h
   DESTINATION
     include/effcee)
-install(TARGETS effcee
-  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
-  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(TARGETS effcee DESTINATION ${CMAKE_INSTALL_LIBDIR})
 
 if(EFFCEE_BUILD_TESTING)
   add_executable(effcee-test
