--- tools/build/src/build/configure.jam.orig	2020-12-03 00:02:49.000000000 -0500
+++ tools/build/src/build/configure.jam	2022-01-28 20:32:41.260634800 -0500
@@ -454,7 +454,7 @@
     return [ find-builds-raw $(p) : $(ps) : $(what) :
         $(3) : $(4) : $(5) : $(6) : $(7) : $(8) : $(9) :
         $(10) : $(11) : $(12) : $(13) : $(14) : $(15) :
-        $(16) : $(17) : $(18) : $(19)  ] ;
+        $(16) : $(17) : $(18) ] ;
 }
 
 
@@ -518,7 +518,7 @@
     {
         local project = [ project.current ] ;
         self.message = $(message) ;
-        for i in 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19
+        for i in 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17
         {
             local name = [ CALC $(i) - 1 ] ;
             self.targets.$(name) = $($(i)[1]) ;
@@ -537,7 +537,7 @@
     }
     rule all-properties ( )
     {
-        local i = 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 ;
+        local i = 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 ;
         return $(self.props.$(i)) ;
     }
     rule check ( properties * )
@@ -559,9 +559,7 @@
             : $(self.targets.14) $(self.what.14)
             : $(self.targets.15) $(self.what.15)
             : $(self.targets.16) $(self.what.16)
-            : $(self.targets.17) $(self.what.17)
-            : $(self.targets.18) $(self.what.18)
-            : $(self.targets.19) $(self.what.19) ] ;
+            : $(self.targets.17) $(self.what.17) ] ;
         if $(self.props.$(i))
         {
             return [ property.evaluate-conditionals-in-context $(self.props.$(i)) : $(properties) ] ;
--- tools/build/src/util/indirect.jam.orig	2020-12-03 00:02:49.000000000 -0500
+++ tools/build/src/util/indirect.jam	2022-01-28 20:32:41.249509200 -0500
@@ -102,7 +102,7 @@
 {
     return [ modules.call-in [ get-module $(r) ] : [ get-rule $(r) ] $(args) :
         $(2) : $(3) : $(4) : $(5) : $(6) : $(7) : $(8) : $(9) : $(10) : $(11) :
-        $(12) : $(13) : $(14) : $(15) : $(16) : $(17) : $(18) : $(19) ] ;
+        $(12) : $(13) : $(14) : $(15) : $(16) : $(17) : $(18) ] ;
 }
 
 
