diff --git a/boost/process/v2/detail/utf8.hpp b/boost/process/v2/detail/utf8.hpp
index 86add628d..83a192cc0 100644
--- a/boost/process/v2/detail/utf8.hpp
+++ b/boost/process/v2/detail/utf8.hpp
@@ -24,7 +24,6 @@ BOOST_PROCESS_V2_DECL std::size_t convert_to_wide(const  char   * in, std::size_
 template<typename CharOut, typename Traits = std::char_traits<CharOut>, 
          typename Allocator = std::allocator<CharOut>, typename CharIn,
          typename = typename std::enable_if<std::is_same<CharOut, CharIn>::value>::type>
-BOOST_PROCESS_V2_DECL 
 std::basic_string<CharOut, Traits, Allocator> conv_string(
     const CharIn * data, std::size_t size, 
     const Allocator allocator = Allocator{})
@@ -36,7 +35,6 @@ std::basic_string<CharOut, Traits, Allocator> conv_string(
 template<typename CharOut, typename Traits = std::char_traits<CharOut>, 
          typename Allocator = std::allocator<CharOut>,
          typename = typename std::enable_if<std::is_same<CharOut, char>::value>::type>
-BOOST_PROCESS_V2_DECL 
 std::basic_string<CharOut, Traits, Allocator> conv_string(
     const wchar_t * data, std::size_t size, 
     const Allocator allocator = Allocator{})
@@ -60,7 +58,6 @@ std::basic_string<CharOut, Traits, Allocator> conv_string(
 template<typename CharOut, typename Traits = std::char_traits<CharOut>, 
          typename Allocator = std::allocator<CharOut>,
          typename = typename std::enable_if<std::is_same<CharOut, wchar_t>::value>::type>
-BOOST_PROCESS_V2_DECL 
 std::basic_string<CharOut, Traits, Allocator> conv_string(
     const char * data, std::size_t size, 
     const Allocator allocator = Allocator{})
