--- bootstrap.bat	2019-12-17 10:36:22.554737800 +0100
+++ bootstrap.patched.bat	2019-12-17 10:12:52.452738800 +0100
@@ -10,15 +10,16 @@
 
 ECHO Building Boost.Build engine
 if exist ".\tools\build\src\engine\b2.exe" del tools\build\src\engine\b2.exe
-pushd tools\build\src\engine
+set working_dir=%cd%
+pushd %~dp0\tools\build\src\engine
 
-call .\build.bat %* > ..\..\..\..\bootstrap.log
+call .\build.bat %* > %working_dir%\bootstrap.log
 @ECHO OFF
 
 popd
 
-if exist ".\tools\build\src\engine\b2.exe" (
-   copy .\tools\build\src\engine\b2.exe . > nul
+if exist "%~dp0\tools\build\src\engine\b2.exe" (
+   copy %~dp0\tools\build\src\engine\b2.exe . > nul
    goto :bjam_built)
 
 goto :bjam_failure
