From c70876cb1159ecc757d460102d42ed6b6daac66f Mon Sep 17 00:00:00 2001
From: Joakim Haugen <joakim.haugen@sintef.no>
Date: Fri, 15 Sep 2023 15:26:38 +0200
Subject: [PATCH] Enable testing

---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b01aba5..eff4b40 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -291,6 +291,7 @@ if(FMILIB_BUILD_SHARED_LIB)
 endif()
 
 if(FMILIB_BUILD_TESTS)
+        enable_testing()
 	include(runtime_test)
 	configure_file (
 		"${FMILibrary_SOURCE_DIR}/Config.cmake/config_test.h.cmake"
-- 
2.30.2

