diff --git a/openmp/CMakeLists.txt b/openmp/CMakeLists.txt
index 6f0d9e728002..cf7bd7590f43 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -21,12 +21,6 @@ if (OPENMP_STANDALONE_BUILD OR "${CMAKE_SOURCE_DIR}" STREQUAL "${CMAKE_CURRENT_S
   # Do not use OPENMP_LIBDIR_SUFFIX directly, use OPENMP_INSTALL_LIBDIR.
   set(OPENMP_INSTALL_LIBDIR "lib${OPENMP_LIBDIR_SUFFIX}")
 
-  # Group test settings.
-  set(OPENMP_TEST_C_COMPILER ${CMAKE_C_COMPILER} CACHE STRING
-    "C compiler to use for testing OpenMP runtime libraries.")
-  set(OPENMP_TEST_CXX_COMPILER ${CMAKE_CXX_COMPILER} CACHE STRING
-    "C++ compiler to use for testing OpenMP runtime libraries.")
-  set(OPENMP_LLVM_TOOLS_DIR "" CACHE PATH "Path to LLVM tools for testing.")
 else()
   set(OPENMP_ENABLE_WERROR ${LLVM_ENABLE_WERROR})
   # If building in tree, we honor the same install suffix LLVM uses.
@@ -45,15 +39,6 @@ endif()
 include(config-ix)
 include(HandleOpenMPOptions)
 
-# Set up testing infrastructure.
-include(OpenMPTesting)
-
-set(OPENMP_TEST_FLAGS "" CACHE STRING
-  "Extra compiler flags to send to the test compiler.")
-set(OPENMP_TEST_OPENMP_FLAGS ${OPENMP_TEST_COMPILER_OPENMP_FLAGS} CACHE STRING
-  "OpenMP compiler flag to use for testing OpenMP runtime libraries.")
-
-
 # Build host runtime library.
 add_subdirectory(runtime)
 
@@ -78,18 +63,3 @@ if (OPENMP_ENABLE_LIBOMPTARGET)
 
   add_subdirectory(libomptarget)
 endif()
-
-set(ENABLE_OMPT_TOOLS ON)
-# Currently tools are not tested well on Windows or MacOS X.
-if (APPLE OR WIN32)
-  set(ENABLE_OMPT_TOOLS OFF)
-endif()
-
-option(OPENMP_ENABLE_OMPT_TOOLS "Enable building ompt based tools for OpenMP."
-       ${ENABLE_OMPT_TOOLS})
-if (OPENMP_ENABLE_OMPT_TOOLS)
-  add_subdirectory(tools)
-endif()
-
-# Now that we have seen all testsuites, create the check-openmp target.
-construct_check_openmp_target()
