diff --git a/CMakeLists.txt b/CMakeLists.txt
index ba03695..ebd83a3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -55,8 +55,8 @@ if (CPPKAFKA_RDKAFKA_STATIC_LIB)
 endif()
 
 # Look for Boost (just need boost.optional headers here)
-find_package(Boost REQUIRED ${FIND_PACKAGE_QUIET})
-find_package(RdKafka REQUIRED ${FIND_PACKAGE_QUIET})
+find_package(Boost REQUIRED CONFIG ${FIND_PACKAGE_QUIET})
+find_package(RdKafka REQUIRED CONFIG ${FIND_PACKAGE_QUIET})
 
 if (Boost_FOUND)
     find_package(Boost COMPONENTS program_options ${FIND_PACKAGE_QUIET})
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 5d31078..5ac2258 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -33,7 +33,12 @@ add_library(cppkafka ${CPPKAFKA_LIBRARY_TYPE} ${SOURCES})
 set_target_properties(cppkafka PROPERTIES VERSION ${CPPKAFKA_VERSION}
                                           SOVERSION ${CPPKAFKA_VERSION})
 
-set(DEPENDENCIES ${RDKAFKA_LIBRARY})
+if ("${CMAKE_BUILD_TYPE}" STREQUAL "Debug")
+    set(DEPENDENCIES ${RdKafka_LIBS_DEBUG})
+else()
+    set(DEPENDENCIES ${RdKafka_LIBS_RELEASE})
+endif()
+
 if (WIN32)
     # On windows ntohs and related are in ws2_32
     set(DEPENDENCIES ${DEPENDENCIES} ws2_32.lib)
