From bb01af1c36cebd08d407c9a06e5a3c906bd9a660 Mon Sep 17 00:00:00 2001
From: Mark Jan van Kampen <mjvk@allseas.com>
Date: Wed, 27 Oct 2021 06:49:11 +0200
Subject: [PATCH] Adds missing includes to utf8_strings.h

---
 include/litehtml/utf8_strings.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/include/litehtml/utf8_strings.h b/include/litehtml/utf8_strings.h
index c5b2421..35d474f 100644
--- a/include/litehtml/utf8_strings.h
+++ b/include/litehtml/utf8_strings.h
@@ -1,6 +1,9 @@
 #ifndef LH_UTF8_STRINGS_H
 #define LH_UTF8_STRINGS_H
 
+#include "os_types.h"
+#include "types.h"
+
 namespace litehtml
 {
 	class utf8_to_wchar
