--- CMakeLists.txt
+++ CMakeLists.txt
@@ -20,7 +20,7 @@ add_library(ls-qpack "")
 target_include_directories(ls-qpack PUBLIC .)
 target_sources(ls-qpack PRIVATE lsqpack.c)
 
-target_include_directories(ls-qpack PRIVATE deps/xxhash/)
 if(LSQPACK_XXH)
+    find_package(xxHash REQUIRED CONFIG)
-    target_sources(ls-qpack PRIVATE deps/xxhash/xxhash.c)
+    target_link_libraries(ls-qpack PUBLIC xxHash::xxhash)
 endif()
