--- a/src/cspice/rdtext.c
+++ b/src/cspice/rdtext.c
@@ -41,7 +41,7 @@ static integer c__1 = 1;
     integer unit, i__;
     extern /* Subroutine */ int chkin_(char *, ftnlen), errch_(char *, char *,
 	     ftnlen, ftnlen);
-    static integer index, units[96];
+    static integer index, units[64];
     extern integer isrchi_(integer *, integer *, integer *);
     integer number;
     extern /* Subroutine */ int sigerr_(char *, ftnlen), chkout_(char *, 
@@ -637,7 +637,7 @@ static integer c__1 = 1;
 /*           need a free logical unit. But only if we don't */
 /*           have too many files open already. */
 
-	    if (n == 96) {
+	    if (n == 64) {
 		setmsg_("Too many files open already.", (ftnlen)28);
 		sigerr_("SPICE(TOOMANYFILESOPEN)", (ftnlen)23);
 		chkout_("RDTEXT", (ftnlen)6);
@@ -676,12 +676,12 @@ static integer c__1 = 1;
 /*               - The index of the file within the UNITS array. */
 
 	    ++n;
-	    units[(i__1 = n - 1) < 96 && 0 <= i__1 ? i__1 : s_rnge("units", 
+	    units[(i__1 = n - 1) < 64 && 0 <= i__1 ? i__1 : s_rnge("units", 
 		    i__1, "rdtext_", (ftnlen)659)] = unit;
 	    index = n;
 	}
 	s_copy(lstfil, file, (ftnlen)255, file_len);
-	lstunt = units[(i__1 = index - 1) < 96 && 0 <= i__1 ? i__1 : s_rnge(
+	lstunt = units[(i__1 = index - 1) < 64 && 0 <= i__1 ? i__1 : s_rnge(
 		"units", i__1, "rdtext_", (ftnlen)665)];
     }
 
@@ -711,15 +711,15 @@ L100001:
     *eof = iostat < 0;
     if (iostat != 0) {
 	cl__1.cerr = 0;
-	cl__1.cunit = units[(i__1 = index - 1) < 96 && 0 <= i__1 ? i__1 : 
+	cl__1.cunit = units[(i__1 = index - 1) < 64 && 0 <= i__1 ? i__1 : 
 		s_rnge("units", i__1, "rdtext_", (ftnlen)689)];
 	cl__1.csta = 0;
 	f_clos(&cl__1);
 	i__1 = n;
 	for (i__ = index + 1; i__ <= i__1; ++i__) {
-	    units[(i__2 = i__ - 2) < 96 && 0 <= i__2 ? i__2 : s_rnge("units", 
+	    units[(i__2 = i__ - 2) < 64 && 0 <= i__2 ? i__2 : s_rnge("units", 
 		    i__2, "rdtext_", (ftnlen)692)] = units[(i__3 = i__ - 1) < 
-		    96 && 0 <= i__3 ? i__3 : s_rnge("units", i__3, "rdtext_", 
+		    64 && 0 <= i__3 ? i__3 : s_rnge("units", i__3, "rdtext_", 
 		    (ftnlen)692)];
 	}
 	--n;
@@ -971,11 +971,11 @@ L_cltext:
     index = isrchi_(&number, &n, units);
     if (index > 0) {
 	cl__1.cerr = 0;
-	cl__1.cunit = units[(i__1 = index - 1) < 96 && 0 <= i__1 ? i__1 : 
+	cl__1.cunit = units[(i__1 = index - 1) < 64 && 0 <= i__1 ? i__1 : 
 		s_rnge("units", i__1, "rdtext_", (ftnlen)952)];
 	cl__1.csta = 0;
 	f_clos(&cl__1);
-	if (units[(i__1 = index - 1) < 96 && 0 <= i__1 ? i__1 : s_rnge("units"
+	if (units[(i__1 = index - 1) < 64 && 0 <= i__1 ? i__1 : s_rnge("units"
 		, i__1, "rdtext_", (ftnlen)954)] == lstunt) {
 	    s_copy(lstfil, " ", (ftnlen)255, (ftnlen)1);
 	}
@@ -985,9 +985,9 @@ L_cltext:
 
 	i__1 = n;
 	for (i__ = index + 1; i__ <= i__1; ++i__) {
-	    units[(i__2 = i__ - 2) < 96 && 0 <= i__2 ? i__2 : s_rnge("units", 
+	    units[(i__2 = i__ - 2) < 64 && 0 <= i__2 ? i__2 : s_rnge("units", 
 		    i__2, "rdtext_", (ftnlen)963)] = units[(i__3 = i__ - 1) < 
-		    96 && 0 <= i__3 ? i__3 : s_rnge("units", i__3, "rdtext_", 
+		    64 && 0 <= i__3 ? i__3 : s_rnge("units", i__3, "rdtext_", 
 		    (ftnlen)963)];
 	}
 	--n;
--- a/src/cspice/stmp03.c
+++ b/src/cspice/stmp03.c
@@ -27,7 +27,7 @@
     extern /* Subroutine */ int chkin_(char *, ftnlen);
     extern doublereal dpmax_(void);
     extern /* Subroutine */ int errdp_(char *, doublereal *, ftnlen);
-    static doublereal pairs[20], lbound;
+    static doublereal pairs[18], lbound;
     extern /* Subroutine */ int sigerr_(char *, ftnlen), chkout_(char *, 
 	    ftnlen), setmsg_(char *, ftnlen);
 
@@ -563,8 +563,8 @@
 
     if (first) {
 	first = FALSE_;
-	for (i__ = 1; i__ <= 20; ++i__) {
-	    pairs[(i__1 = i__ - 1) < 20 && 0 <= i__1 ? i__1 : s_rnge("pairs", 
+	for (i__ = 1; i__ <= 18; ++i__) {
+	    pairs[(i__1 = i__ - 1) < 18 && 0 <= i__1 ? i__1 : s_rnge("pairs", 
 		    i__1, "stmp03_", (ftnlen)589)] = 1. / ((doublereal) i__ * 
 		    (doublereal) (i__ + 1));
 	}
@@ -694,8 +694,8 @@
 /*     LPAIR3 will be 18. */
 
     *c3 = 1.;
-    for (i__ = 20; i__ >= 4; i__ += -2) {
-	*c3 = 1. - *x * pairs[(i__1 = i__ - 1) < 20 && 0 <= i__1 ? i__1 : 
+    for (i__ = 18; i__ >= 4; i__ += -2) {
+	*c3 = 1. - *x * pairs[(i__1 = i__ - 1) < 18 && 0 <= i__1 ? i__1 : 
 		s_rnge("pairs", i__1, "stmp03_", (ftnlen)733)] * *c3;
     }
     *c3 = pairs[1] * *c3;
@@ -722,8 +722,8 @@
 /*     LPAIR2 will be 17. */
 
     *c2 = 1.;
-    for (i__ = 19; i__ >= 3; i__ += -2) {
-	*c2 = 1. - *x * pairs[(i__1 = i__ - 1) < 20 && 0 <= i__1 ? i__1 : 
+    for (i__ = 17; i__ >= 3; i__ += -2) {
+	*c2 = 1. - *x * pairs[(i__1 = i__ - 1) < 18 && 0 <= i__1 ? i__1 : 
 		s_rnge("pairs", i__1, "stmp03_", (ftnlen)764)] * *c2;
     }
     *c2 = pairs[0] * *c2;
