--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -26,12 +26,18 @@
 
 find_package(OpenGL REQUIRED)
 
-find_package(glm)
-find_package(glfw3)
-find_package(Eigen3)
-find_package(PNG)
-find_package(JPEG)
-find_package(assimp QUIET)
+find_package(glm REQUIRED)
+find_package(glfw3 REQUIRED)
+find_package(Eigen3 REQUIRED)
+find_package(PNG REQUIRED)
+find_package(JPEG REQUIRED)
+find_package(assimp REQUIRED)
+
+find_package(imgui REQUIRED CONFIG)
+find_package(imguizmo REQUIRED CONFIG)
+find_package(implot REQUIRED CONFIG)
+find_package(portable-file-dialogs REQUIRED CONFIG)
+find_package(khrplatform REQUIRED CONFIG)
 
 if(NOT glm_FOUND)
   message(STATUS "System GLM not found. Download GLM 1.0.1.")
@@ -94,7 +100,7 @@
 endif()
 
 if(${assimp_FOUND})
-  list(APPEND EXTRA_LIBRARIES assimp)
+  list(APPEND EXTRA_LIBRARIES assimp::assimp)
   list(APPEND EXTRA_SOURCE src/glk/io/mesh_io.cpp)
 endif()
 
@@ -115,20 +121,8 @@
 add_library(iridescence
   # GL and IMGUI
   thirdparty/gl3w/gl3w.cpp
-  thirdparty/imgui/imgui.cpp
-  thirdparty/imgui/imgui_demo.cpp
-  thirdparty/imgui/imgui_draw.cpp
-  thirdparty/imgui/imgui_tables.cpp
-  thirdparty/imgui/imgui_widgets.cpp
-  thirdparty/ImGuizmo/ImCurveEdit.cpp
-  thirdparty/ImGuizmo/ImGradient.cpp
-  thirdparty/ImGuizmo/ImGuizmo.cpp
-  thirdparty/ImGuizmo/ImSequencer.cpp
-  thirdparty/implot/implot.cpp
-  thirdparty/implot/implot_demo.cpp
-  thirdparty/implot/implot_items.cpp
-  thirdparty/imgui/backends/imgui_impl_glfw.cpp
-  thirdparty/imgui/backends/imgui_impl_opengl3.cpp
+  ${IMGUI_BACKENDS}/imgui_impl_glfw.cpp
+  ${IMGUI_BACKENDS}/imgui_impl_opengl3.cpp
   # glk
   src/glk/path_std.cpp
   src/glk/mesh.cpp
@@ -195,19 +189,19 @@
 )
 target_include_directories(iridescence PUBLIC
   $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
-  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/thirdparty/imgui>
-  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/thirdparty/imgui/examples>
-  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/thirdparty/imgui/backends>
-  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/thirdparty/ImGuizmo>
-  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/thirdparty/implot>
-  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/thirdparty/portable-file-dialogs>
   $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/thirdparty/gl3w>
   $<INSTALL_INTERFACE:include/iridescence>
+  ${IMGUI_BACKENDS}
 )
 target_link_libraries(iridescence PUBLIC
   Eigen3::Eigen
   OpenGL::GL
   glfw
+  imgui::imgui
+  imguizmo::imguizmo
+  implot::implot
+  portable-file-dialogs::portable-file-dialogs
+  khrplatform::khrplatform
   ${EXTRA_LIBRARIES}
 )
 target_link_libraries(iridescence PRIVATE
@@ -287,10 +281,6 @@
 
 install(DIRECTORY include/ DESTINATION ${INCLUDE_INSTALL_DIR})
 install(DIRECTORY thirdparty/gl3w/GL/ DESTINATION ${INCLUDE_INSTALL_DIR}/GL)
-install(FILES thirdparty/imgui/imgui.h thirdparty/imgui/imconfig.h thirdparty/imgui/imgui_internal.h thirdparty/imgui/imstb_textedit.h DESTINATION  ${INCLUDE_INSTALL_DIR})
-install(FILES thirdparty/implot/implot.h thirdparty/implot/implot_internal.h DESTINATION  ${INCLUDE_INSTALL_DIR})
-install(FILES thirdparty/portable-file-dialogs/portable-file-dialogs.h DESTINATION  ${INCLUDE_INSTALL_DIR})
-install(FILES thirdparty/ImGuizmo/ImGuizmo.h DESTINATION  ${INCLUDE_INSTALL_DIR})
 install(DIRECTORY data/ DESTINATION ${CMAKE_INSTALL_DATADIR}/iridescence/data)
 
 install(TARGETS iridescence
