--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -16,12 +16,8 @@ if( EXISTS "${LOC_PATH}" )
 endif()
 
 # Set c++ standard options
-set( CMAKE_CXX_STANDARD 17 )
-set( CMAKE_CXX_STANDARD_REQUIRED ON )
-set( CMAKE_CXX_EXTENSIONS OFF )
 
 # Fetching dependencies
-add_subdirectory( deps )
 
 # Include directories
 include_directories( ${CMAKE_CURRENT_SOURCE_DIR}/include )
@@ -32,11 +28,13 @@ add_library( arsenalgear STATIC
     src/system.cpp
     src/utils.cpp
 )
+target_compile_features(arsenalgear PUBLIC cxx_std_17)
 add_library( arsenalgear::arsenalgear ALIAS arsenalgear )
 
 # Compile tests
 option( ARSENALGEAR_TESTS "Enable / disable tests." ON )
 if( ARSENALGEAR_TESTS )
+    add_subdirectory( deps )
     add_subdirectory( test )
 else()
     message( STATUS "Skipping tests." )
