From 54c5a0f4cfe79e6c2118c77132302d45b43e093c Mon Sep 17 00:00:00 2001
From: Chris Mc <prince.chrismc@gmail.com>
Date: Sun, 17 Jan 2021 16:26:31 -0500
Subject: [PATCH] Update CMakeLists.txt

---
 CMakeLists.txt | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 762c561..0f53b1e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -86,6 +86,10 @@ if (REDIS_PLUS_PLUS_BUILD_SHARED)
         target_compile_definitions(${SHARED_LIB} PRIVATE NOMINMAX)
         set_target_properties(${SHARED_LIB} PROPERTIES CXX_STANDARD ${REDIS_PLUS_PLUS_CXX_STANDARD} CXX_EXTENSIONS OFF)
         set_target_properties(${SHARED_LIB} PROPERTIES WINDOWS_EXPORT_ALL_SYMBOLS TRUE)
+        target_link_libraries(${SHARED_LIB} ws2_32)
+        if (REDIS_PLUS_PLUS_USE_TLS)
+            target_link_libraries(${SHARED_LIB} crypt32)
+        endif()
     endif()
 
     target_link_libraries(${SHARED_LIB} ${HIREDIS_LIB})
