--- a/lib/identify.c
+++ b/lib/identify.c
@@ -196,9 +196,9 @@ static void threshold(struct quirc *q)
 		threshold_s = THRESHOLD_S_MIN;
 
 	for (y = 0; y < q->h; y++) {
-		int row_average[q->w];
+		int *row_average = malloc(q->w * sizeof(int));
 
-		memset(row_average, 0, sizeof(row_average));
+		memset(row_average, 0, q->w * sizeof(int));
 
 		for (x = 0; x < q->w; x++) {
 			int w, u;
@@ -229,6 +229,7 @@ static void threshold(struct quirc *q)
 		}
 
 		row += q->w;
+		free(row_average);
 	}
 }
 
