diff --git a/cmake/external/abseil-cpp.cmake b/cmake/external/abseil-cpp.cmake
index 54d2f9c5c1..3195ef15a0 100644
--- a/cmake/external/abseil-cpp.cmake
+++ b/cmake/external/abseil-cpp.cmake
@@ -22,6 +22,7 @@ FetchContent_Declare(
     URL ${DEP_URL_abseil_cpp}
     URL_HASH SHA1=${DEP_SHA1_abseil_cpp}
     PATCH_COMMAND ${ABSL_PATCH_COMMAND}
+    FIND_PACKAGE_ARGS REQUIRED CONFIG NAMES absl
 )
 
 onnxruntime_fetchcontent_makeavailable(abseil_cpp)
diff --git a/cmake/external/eigen.cmake b/cmake/external/eigen.cmake
index 266dd534af..3c7cb77891 100644
--- a/cmake/external/eigen.cmake
+++ b/cmake/external/eigen.cmake
@@ -14,8 +14,11 @@ else ()
         FetchContent_Declare(
         eigen
         URL https://gitlab.com/libeigen/eigen/-/archive/d10b27fe37736d2944630ecd7557cefa95cf87c9/eigen-d10b27fe37736d2944630ecd7557cefa95cf87c9.zip
+        FIND_PACKAGE_ARGS REQUIRED CONFIG NAMES Eigen3
 		)
     endif()
-    FetchContent_Populate(eigen)
-    set(eigen_INCLUDE_DIRS  "${eigen_SOURCE_DIR}")
+    # FetchContent_Populate(eigen)
+    # set(eigen_INCLUDE_DIRS  "${eigen_SOURCE_DIR}")
+    onnxruntime_fetchcontent_makeavailable(eigen)
+    get_target_property(eigen_INCLUDE_DIRS Eigen3::Eigen INTERFACE_INCLUDE_DIRECTORIES)
 endif()
diff --git a/cmake/external/onnxruntime_external_deps.cmake b/cmake/external/onnxruntime_external_deps.cmake
index 9effd1a2db..361141c593 100644
--- a/cmake/external/onnxruntime_external_deps.cmake
+++ b/cmake/external/onnxruntime_external_deps.cmake
@@ -113,7 +113,7 @@ if(CMAKE_CROSSCOMPILING AND NOT ONNX_CUSTOM_PROTOC_EXECUTABLE AND NOT CMAKE_OSX_
     elseif(CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL "x86")
       FetchContent_Declare(protoc_binary URL ${DEP_URL_protoc_win32} URL_HASH SHA1=${DEP_SHA1_protoc_win32})
       FetchContent_Populate(protoc_binary)
-    endif() 
+    endif()
     if(protoc_binary_SOURCE_DIR)
       message("Use prebuilt protoc")
       set(ONNX_CUSTOM_PROTOC_EXECUTABLE ${protoc_binary_SOURCE_DIR}/bin/protoc.exe)
@@ -152,7 +152,7 @@ FetchContent_Declare(
   URL ${DEP_URL_protobuf}
   URL_HASH SHA1=${DEP_SHA1_protobuf}
   PATCH_COMMAND ${ONNXRUNTIME_PROTOBUF_PATCH_COMMAND}
-  FIND_PACKAGE_ARGS 3.21.12 NAMES Protobuf
+  FIND_PACKAGE_ARGS NAMES Protobuf
 )
 set(protobuf_BUILD_TESTS OFF CACHE BOOL "Build protobuf tests" FORCE)
 if (CMAKE_SYSTEM_NAME STREQUAL "Android")
@@ -173,15 +173,17 @@ FetchContent_Declare(
       date
       URL ${DEP_URL_date}
       URL_HASH SHA1=${DEP_SHA1_date}
+      FIND_PACKAGE_ARGS NAMES date
     )
 onnxruntime_fetchcontent_makeavailable(date)
-
+add_library(date_interface ALIAS date::date)
 
 
 FetchContent_Declare(
   mp11
   URL ${DEP_URL_mp11}
   URL_HASH SHA1=${DEP_SHA1_mp11}
+  FIND_PACKAGE_ARGS NAMES Boost
 )
 
 set(JSON_BuildTests OFF CACHE INTERNAL "")
@@ -300,10 +302,12 @@ FetchContent_Declare(
     safeint
     URL ${DEP_URL_safeint}
     URL_HASH SHA1=${DEP_SHA1_safeint}
+    FIND_PACKAGE_ARGS NAMES safeint
 )
 
 # The next line will generate an error message "fatal: not a git repository", but it is ok. It is from flatbuffers
 onnxruntime_fetchcontent_makeavailable(Protobuf nlohmann_json mp11 re2 safeint GSL flatbuffers)
+add_library(Boost::mp11 ALIAS Boost::headers)
 if(NOT flatbuffers_FOUND)
   if(NOT TARGET flatbuffers::flatbuffers)
     add_library(flatbuffers::flatbuffers ALIAS flatbuffers)
@@ -328,6 +332,10 @@ namespace std { using ::getenv; }
       target_compile_options(flatc PRIVATE /FI${CMAKE_BINARY_DIR}/gdk_cstdlib_wrapper.h)
     endif()
   endif()
+else()
+  if(NOT TARGET flatbuffers::flatbuffers)
+    add_library(flatbuffers::flatbuffers ALIAS flatbuffers::flatbuffers_shared)
+  endif()
 endif()
 
 if (onnxruntime_BUILD_UNIT_TESTS)
@@ -396,6 +404,7 @@ FetchContent_Declare(
   URL ${DEP_URL_onnx}
   URL_HASH SHA1=${DEP_SHA1_onnx}
   PATCH_COMMAND ${ONNXRUNTIME_ONNX_PATCH_COMMAND}
+  FIND_PACKAGE_ARGS NAMES onnx ONNX
 )
 
 
@@ -423,8 +432,8 @@ endif()
 set(GSL_TARGET "Microsoft.GSL::GSL")
 set(GSL_INCLUDE_DIR "$<TARGET_PROPERTY:${GSL_TARGET},INTERFACE_INCLUDE_DIRECTORIES>")
 
-add_library(safeint_interface INTERFACE)
-target_include_directories(safeint_interface INTERFACE ${safeint_SOURCE_DIR})
+add_library(safeint_interface ALIAS safeint::safeint)
+#target_include_directories(safeint_interface INTERFACE ${safeint_SOURCE_DIR})
 
 # XNNPACK EP
 if (onnxruntime_USE_XNNPACK)
@@ -453,9 +462,9 @@ set(onnxruntime_EXTERNAL_LIBRARIES ${onnxruntime_EXTERNAL_LIBRARIES_XNNPACK} WIL
 # The other libs do not have the problem. All the sources are already there. We can compile them in any order.
 set(onnxruntime_EXTERNAL_DEPENDENCIES onnx_proto flatbuffers::flatbuffers)
 
-target_compile_definitions(onnx PUBLIC $<TARGET_PROPERTY:onnx_proto,INTERFACE_COMPILE_DEFINITIONS> PRIVATE "__ONNX_DISABLE_STATIC_REGISTRATION")
+# target_compile_definitions(onnx PUBLIC $<TARGET_PROPERTY:onnx_proto,INTERFACE_COMPILE_DEFINITIONS> PRIVATE "__ONNX_DISABLE_STATIC_REGISTRATION")
 if (NOT onnxruntime_USE_FULL_PROTOBUF)
-  target_compile_definitions(onnx PUBLIC "__ONNX_NO_DOC_STRINGS")
+  # target_compile_definitions(onnx PUBLIC "__ONNX_NO_DOC_STRINGS")
 endif()
 
 if (onnxruntime_RUN_ONNX_TESTS)
diff --git a/cmake/external/wil.cmake b/cmake/external/wil.cmake
index d38535c4a1..8b181871f9 100644
--- a/cmake/external/wil.cmake
+++ b/cmake/external/wil.cmake
@@ -11,12 +11,18 @@ FetchContent_Declare(
 )
 #We can not use FetchContent_MakeAvailable(microsoft_wil) at here, since their cmake file
 #always executes install command without conditions.
-FetchContent_Populate(microsoft_wil)
-if(NOT wil_FOUND)
+#FetchContent_Populate(microsoft_wil)
+#if(NOT wil_FOUND)
+#  add_library(WIL INTERFACE)
+#  add_library(WIL::WIL ALIAS WIL)
+#
+#  # The interface's include directory.
+#  target_include_directories(WIL INTERFACE
+#    $<BUILD_INTERFACE:${microsoft_wil_SOURCE_DIR}/include>)
+#endif()
+if(WIN32)
+  FetchContent_MakeAvailable(microsoft_wil)
+else()
   add_library(WIL INTERFACE)
   add_library(WIL::WIL ALIAS WIL)
-
-  # The interface's include directory.
-  target_include_directories(WIL INTERFACE
-    $<BUILD_INTERFACE:${microsoft_wil_SOURCE_DIR}/include>)
-endif()
\ No newline at end of file
+endif()
diff --git a/cmake/external/xnnpack.cmake b/cmake/external/xnnpack.cmake
index 1fc2c6ccdc..fd96487552 100644
--- a/cmake/external/xnnpack.cmake
+++ b/cmake/external/xnnpack.cmake
@@ -25,12 +25,19 @@ set(FXDIV_SOURCE_DIR ${fxdiv_SOURCE_DIR})
 
 FetchContent_Declare(pthreadpool URL ${DEP_URL_pthreadpool} URL_HASH SHA1=${DEP_SHA1_pthreadpool})
 onnxruntime_fetchcontent_makeavailable(pthreadpool)
-FetchContent_Declare(googlexnnpack URL ${DEP_URL_googlexnnpack}  URL_HASH SHA1=${DEP_SHA1_googlexnnpack}
-PATCH_COMMAND ${Patch_EXECUTABLE} --binary --ignore-whitespace -p1 < ${PROJECT_SOURCE_DIR}/patches/xnnpack/AddEmscriptenAndIosSupport.patch)
+FetchContent_Declare(
+    googlexnnpack
+    URL ${DEP_URL_googlexnnpack}
+    URL_HASH SHA1=${DEP_SHA1_googlexnnpack}
+    PATCH_COMMAND ${Patch_EXECUTABLE} --binary --ignore-whitespace -p1 < ${PROJECT_SOURCE_DIR}/patches/xnnpack/AddEmscriptenAndIosSupport.patch
+    FIND_PACKAGE_ARGS NAMES xnnpack
+)
 
 onnxruntime_fetchcontent_makeavailable(googlexnnpack)
-set(XNNPACK_DIR ${googlexnnpack_SOURCE_DIR})
-set(XNNPACK_INCLUDE_DIR ${XNNPACK_DIR}/include)
+# set(XNNPACK_DIR ${googlexnnpack_SOURCE_DIR})
+# set(XNNPACK_INCLUDE_DIR ${XNNPACK_DIR}/include)
+
+add_library(XNNPACK ALIAS xnnpack::xnnpack)
 
 set(onnxruntime_EXTERNAL_LIBRARIES_XNNPACK XNNPACK pthreadpool)
 
diff --git a/cmake/onnxruntime_common.cmake b/cmake/onnxruntime_common.cmake
index 0410d3361c..685df7a487 100644
--- a/cmake/onnxruntime_common.cmake
+++ b/cmake/onnxruntime_common.cmake
@@ -195,7 +195,7 @@ if (ARM64 OR ARM OR X86 OR X64 OR X86_64)
     # Its functionality in detecting x86 cpu features are lacking, so is support for Windows.
     if (CPUINFO_SUPPORTED)
       onnxruntime_add_include_to_target(onnxruntime_common cpuinfo::cpuinfo)
-      list(APPEND onnxruntime_EXTERNAL_LIBRARIES cpuinfo::cpuinfo cpuinfo::clog)
+      list(APPEND onnxruntime_EXTERNAL_LIBRARIES cpuinfo::cpuinfo)
     endif()
   endif()
 endif()
diff --git a/cmake/onnxruntime_providers.cmake b/cmake/onnxruntime_providers.cmake
index 2839713406..13a294e4fe 100644
--- a/cmake/onnxruntime_providers.cmake
+++ b/cmake/onnxruntime_providers.cmake
@@ -1711,10 +1711,6 @@ if (onnxruntime_USE_XNNPACK)
   add_dependencies(onnxruntime_providers_xnnpack onnx ${onnxruntime_EXTERNAL_DEPENDENCIES})
   set_target_properties(onnxruntime_providers_xnnpack PROPERTIES FOLDER "ONNXRuntime")
 
-  install(DIRECTORY ${ONNXRUNTIME_INCLUDE_DIR}/core/providers/xnnpack
-    DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/onnxruntime/core/providers
-  )
-
   set_target_properties(onnxruntime_providers_xnnpack PROPERTIES LINKER_LANGUAGE CXX)
 
   if (NOT onnxruntime_BUILD_SHARED_LIB)
