diff --git a/onnxruntime/core/session/utils.cc b/onnxruntime/core/session/utils.cc
--- a/onnxruntime/core/session/utils.cc
+++ b/onnxruntime/core/session/utils.cc
@@ -549,7 +549,7 @@
                                                             true,
                                                             ProviderLibraryPathType::Absolute);
   bool is_provider_bridge = provider_library->Load() == Status::OK();  // library has GetProvider
-  LOGS_DEFAULT(INFO) << "Loading EP library: " << library_path
+  LOGS_DEFAULT(INFO) << "Loading EP library: " << resolved_library_path.generic_string()
                      << (is_provider_bridge ? " as a provider bridge" : " as a plugin");

   // create EpLibraryPlugin to ensure CreateEpFactories and ReleaseEpFactory are available
