diff --git a/sml-1.1.4/include/boost/ b/sml-1.1.4/include/boost/
index b75b41f..a4e6430 100644

--- a/include/boost/sml.hpp
+++ b/include/boost/sml.hpp
@@ -549,6 +549,8 @@ template <class T>
 const char *get_type_name() {
 #if defined(_MSC_VER) && !defined(__clang__)
   return detail::get_type_name<T, 39>(__FUNCSIG__, make_index_sequence<sizeof(__FUNCSIG__) - 39 - 8>{});
+#elif defined(__clang__) && __clang_major__ >= 12
+  return detail::get_type_name<T, 50>(__PRETTY_FUNCTION__, make_index_sequence<sizeof(__PRETTY_FUNCTION__) - 50 - 2>{});
 #elif defined(__clang__)
   return detail::get_type_name<T, 63>(__PRETTY_FUNCTION__, make_index_sequence<sizeof(__PRETTY_FUNCTION__) - 63 - 2>{});
 #elif defined(__GNUC__)
