diff --git a/extern/filereaderlp/reader.cpp b/extern/filereaderlp/reader.cpp
index 86151120..71f6747a 100644
--- a/extern/filereaderlp/reader.cpp
+++ b/extern/filereaderlp/reader.cpp
@@ -2,6 +2,7 @@
 
 #include "builder.hpp"
 
+#include <cctype>
 #include <cstring>
 #include <cassert>
 #include <iostream>
diff --git a/src/lp_data/HighsOptions.cpp b/src/lp_data/HighsOptions.cpp
index 903a7c71..6e66e78e 100644
--- a/src/lp_data/HighsOptions.cpp
+++ b/src/lp_data/HighsOptions.cpp
@@ -17,6 +17,7 @@
 
 #include <algorithm>
 #include <cassert>
+#include <cctype>
 
 // void setLogOptions();
 
diff --git a/src/presolve/ICrash.cpp b/src/presolve/ICrash.cpp
index 12a6fc3f..e7bf86a4 100644
--- a/src/presolve/ICrash.cpp
+++ b/src/presolve/ICrash.cpp
@@ -14,6 +14,7 @@
 #include "presolve/ICrash.h"
 
 #include <algorithm>
+#include <cctype>
 #include <chrono>
 #include <cmath>
 #include <iomanip>
diff --git a/src/util/HighsTimer.h b/src/util/HighsTimer.h
index 6f1d3047..cabef8c4 100644
--- a/src/util/HighsTimer.h
+++ b/src/util/HighsTimer.h
@@ -16,6 +16,7 @@
 #ifndef UTIL_HIGHSTIMER_H_
 #define UTIL_HIGHSTIMER_H_
 
+#include <algorithm>
 #include <cassert>
 #include <chrono>
 #include <cstdio>
