diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1e0d0c3..b46fa5f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -13,9 +13,9 @@ set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 
 find_package("OpenSSL")
 find_package("Boost")
-find_package("ms_gsl")
+find_package("Microsoft.GSL")
 find_package("LibXml2")
-find_package("libarchive")
+find_package("LibArchive")
 find_package("nlohmann_json")
 find_package("jwt-cpp")
 
