--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -44,12 +44,12 @@
 set(EVENT__LIBRARY_TYPE DEFAULT CACHE STRING
     "Set library type to SHARED/STATIC/BOTH (default SHARED for MSVC, otherwise BOTH)")

 project(libevent C)

-set(CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake/")
+list(APPEND CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake/")
 string(REGEX MATCH "SunOS" SOLARIS "${CMAKE_SYSTEM_NAME}")


 include(CheckTypeSize)
 include(CheckFunctionExistsEx)
 include(CheckFileOffsetBits)
