diff --git a/src/odr/internal/pdf/pdf_object.cpp b/src/odr/internal/pdf/pdf_object.cpp
index 2e21d39..14c1b8c 100644
--- a/src/odr/internal/pdf/pdf_object.cpp
+++ b/src/odr/internal/pdf/pdf_object.cpp
@@ -149,39 +149,39 @@ std::string Dictionary::to_string() const {
 
 namespace odr::internal {
 
-std::ostream &pdf::operator<<(std::ostream &out,
+std::ostream &::odr::internal::pdf::operator<<(std::ostream &out,
                               const StandardString &standard_string) {
   standard_string.to_stream(out);
   return out;
 }
 
-std::ostream &pdf::operator<<(std::ostream &out, const HexString &hex_string) {
+std::ostream &::odr::internal::pdf::operator<<(std::ostream &out, const HexString &hex_string) {
   hex_string.to_stream(out);
   return out;
 }
 
-std::ostream &pdf::operator<<(std::ostream &out, const Name &name) {
+std::ostream &::odr::internal::pdf::operator<<(std::ostream &out, const Name &name) {
   name.to_stream(out);
   return out;
 }
 
-std::ostream &pdf::operator<<(std::ostream &out,
+std::ostream &::odr::internal::pdf::operator<<(std::ostream &out,
                               const ObjectReference &object_reference) {
   object_reference.to_stream(out);
   return out;
 }
 
-std::ostream &pdf::operator<<(std::ostream &out, const Object &object) {
+std::ostream &::odr::internal::pdf::operator<<(std::ostream &out, const Object &object) {
   object.to_stream(out);
   return out;
 }
 
-std::ostream &pdf::operator<<(std::ostream &out, const Array &array) {
+std::ostream &::odr::internal::pdf::operator<<(std::ostream &out, const Array &array) {
   array.to_stream(out);
   return out;
 }
 
-std::ostream &pdf::operator<<(std::ostream &out, const Dictionary &dictionary) {
+std::ostream &::odr::internal::pdf::operator<<(std::ostream &out, const Dictionary &dictionary) {
   dictionary.to_stream(out);
   return out;
 }
