diff -u cmake/FindEigen.cmake cmake_conan/FindEigen.cmake
--- cmake/FindEigen.cmake	2018-03-21 21:00:14.000000000 -0700
+++ cmake_conan/FindEigen.cmake	2020-01-06 09:32:00.889700000 -0800
@@ -180,12 +180,12 @@
 if (NOT EIGEN_FOUND)
   # Search user-installed locations first, so that we prefer user installs
   # to system installs where both exist.
-  list(APPEND EIGEN_CHECK_INCLUDE_DIRS
-    /usr/local/include
-    /usr/local/homebrew/include # Mac OS X
-    /opt/local/var/macports/software # Mac OS X.
-    /opt/local/include
-    /usr/include)
+  list(APPEND EIGEN_CHECK_INCLUDE_DIRS ${CMAKE_INCLUDE_PATH})
+    #/usr/local/include
+    #/usr/local/homebrew/include # Mac OS X
+    #/opt/local/var/macports/software # Mac OS X.
+    #/opt/local/include
+    #/usr/include)
   # Additional suffixes to try appending to each search path.
   list(APPEND EIGEN_CHECK_PATH_SUFFIXES
     eigen3 # Default root directory for Eigen.
Only in cmake: FindEigen_conan.cmake
diff -u cmake/FindGflags.cmake cmake_conan/FindGflags.cmake
--- cmake/FindGflags.cmake	2018-03-21 21:00:14.000000000 -0700
+++ cmake_conan/FindGflags.cmake	2020-01-06 09:26:02.613681000 -0800
@@ -434,21 +434,21 @@
 
   # Search user-installed locations first, so that we prefer user installs
   # to system installs where both exist.
-  list(APPEND GFLAGS_CHECK_INCLUDE_DIRS
-    /usr/local/include
-    /usr/local/homebrew/include # Mac OS X
-    /opt/local/var/macports/software # Mac OS X.
-    /opt/local/include
-    /usr/include)
+  list(APPEND GFLAGS_CHECK_INCLUDE_DIRS ${CMAKE_INCLUDE_PATH})
+    #/usr/local/include
+    #/usr/local/homebrew/include # Mac OS X
+    #/opt/local/var/macports/software # Mac OS X.
+    #/opt/local/include
+    #/usr/include)
   list(APPEND GFLAGS_CHECK_PATH_SUFFIXES
     gflags/include # Windows (for C:/Program Files prefix).
     gflags/Include ) # Windows (for C:/Program Files prefix).
 
-  list(APPEND GFLAGS_CHECK_LIBRARY_DIRS
-    /usr/local/lib
-    /usr/local/homebrew/lib # Mac OS X.
-    /opt/local/lib
-    /usr/lib)
+  list(APPEND GFLAGS_CHECK_LIBRARY_DIRS ${CMAKE_LIB_PATH})
+    #/usr/local/lib
+    #/usr/local/homebrew/lib # Mac OS X.
+    #/opt/local/lib
+    #/usr/lib)
   list(APPEND GFLAGS_CHECK_LIBRARY_SUFFIXES
     gflags/lib # Windows (for C:/Program Files prefix).
     gflags/Lib ) # Windows (for C:/Program Files prefix).
Only in cmake: FindGflags_conan.cmake
diff -u cmake/FindGlog.cmake cmake_conan/FindGlog.cmake
--- cmake/FindGlog.cmake	2018-03-21 21:00:14.000000000 -0700
+++ cmake_conan/FindGlog.cmake	2020-01-06 09:30:26.033695000 -0800
@@ -228,12 +228,12 @@
 
   # Search user-installed locations first, so that we prefer user installs
   # to system installs where both exist.
-  list(APPEND GLOG_CHECK_INCLUDE_DIRS
-    /usr/local/include
-    /usr/local/homebrew/include # Mac OS X
-    /opt/local/var/macports/software # Mac OS X.
-    /opt/local/include
-    /usr/include)
+  list(APPEND GLOG_CHECK_INCLUDE_DIRS ${CMAKE_INCLUDE_PATH})
+    #/usr/local/include
+    #/usr/local/homebrew/include # Mac OS X
+    #/opt/local/var/macports/software # Mac OS X.
+    #/opt/local/include
+    #/usr/include)
   # Windows (for C:/Program Files prefix).
   list(APPEND GLOG_CHECK_PATH_SUFFIXES
     glog/include
@@ -243,11 +243,11 @@
     google-glog/include # CMake installs with project name prefix.
     google-glog/Include)
 
-  list(APPEND GLOG_CHECK_LIBRARY_DIRS
-    /usr/local/lib
-    /usr/local/homebrew/lib # Mac OS X.
-    /opt/local/lib
-    /usr/lib)
+  list(APPEND GLOG_CHECK_LIBRARY_DIRS ${CMAKE_LIB_PATH})
+    #/usr/local/lib
+    #/usr/local/homebrew/lib # Mac OS X.
+    #/opt/local/lib
+    #/usr/lib)
   # Windows (for C:/Program Files prefix).
   list(APPEND GLOG_CHECK_LIBRARY_SUFFIXES
     glog/lib
Only in cmake: FindGlog_conan.cmake
