diff --git a/cmake/ext_libcurl.cmake b/cmake/ext_libcurl.cmake
index 5e1f665..b30da62 100644
--- a/cmake/ext_libcurl.cmake
+++ b/cmake/ext_libcurl.cmake
@@ -3,7 +3,7 @@ if(libcurl_link_lib)
 endif()
 
 
-include(FindCURL)
+find_package(CURL REQUIRED)
 
 if(NOT CURL_FOUND)
     message(FATAL_ERROR "curl dependency not found")
diff --git a/cmake/ext_uriparser.cmake b/cmake/ext_uriparser.cmake
index 89a08c3..09f8bbc 100644
--- a/cmake/ext_uriparser.cmake
+++ b/cmake/ext_uriparser.cmake
@@ -2,7 +2,9 @@ if(uriparser_link_lib)
     return()
 endif()
 
-
+find_package(uriparser REQUIRED)
+set(uriparser_link_lib uriparser::uriparser)
+return()
 if(MSVC OR BMX_BUILD_URIPARSER_SOURCE)
     include(FetchContent)
 
diff --git a/cmake/ext_uuid.cmake b/cmake/ext_uuid.cmake
index ee34277..9bec461 100644
--- a/cmake/ext_uuid.cmake
+++ b/cmake/ext_uuid.cmake
@@ -4,6 +4,9 @@ endif()
 
 
 if(UNIX AND NOT APPLE)
+    find_package(libuuid REQUIRED)
+    set(uuid_link_lib libuuid::libuuid)
+    return()
     find_library(uuid_lib
         NAMES uuid
     )
