diff --git a/CMakeLists.txt b/CMakeLists.txt
index 69deeaf..4623ab2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -91,7 +91,10 @@ else()
     #add_library(${PROJECT_NAME} STATIC ${SRCS})
     add_library(${PROJECT_NAME} STATIC ${src_MAIN} ${src_LINUX_TCP})
 endif()
-
+if(AMQP-CPP_LINUX_TCP)
+find_package(OpenSSL REQUIRED)
+target_link_libraries(${PROJECT_NAME} OpenSSL::SSL OpenSSL::Crypto)
+endif()
 # install rules
 # ------------------------------------------------------------------------------------------------------
 
@@ -100,7 +103,7 @@ if(AMQP-CPP_BUILD_SHARED)
     install(TARGETS ${PROJECT_NAME} EXPORT ${PROJECT_NAME}Config
             ARCHIVE DESTINATION lib
             LIBRARY DESTINATION lib
-            RUNTIME DESTINATION lib
+            RUNTIME DESTINATION bin
     )
 else()
     # copy static lib
