--- a/cmake/target_default_compile_options.cmake
+++ b/cmake/target_default_compile_options.cmake
@@ -3,6 +3,8 @@ cmake_minimum_required(VERSION 3.5)
 # do not rely on cmake presets
 # and use target configuration
 
+# Do not reset CMAKE_*_FLAGS* !
+if(0)
 set(CMAKE_C_FLAGS "")
 set(CMAKE_C_FLAGS_DEBUG "")
 set(CMAKE_C_FLAGS_RELEASE "")
@@ -14,6 +16,7 @@ set(CMAKE_CXX_FLAGS_DEBUG "")
 set(CMAKE_CXX_FLAGS_RELEASE "")
 set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "")
 set(CMAKE_CXX_FLAGS_MINSIZEREL "")
+endif()
 
 # this seems redundant as the target
 # property is set. but for apple clang
@@ -37,21 +40,15 @@ function(target_default_compile_options_c THIS)
             OR CMAKE_C_COMPILER_ID STREQUAL "GNU")
 
         target_compile_options(${THIS} PRIVATE
-                -fPIC
                 -fvisibility=hidden
                 -fno-stack-protector
-                $<$<CONFIG:DEBUG>:-g -O0 -DDEBUG>
-                $<$<CONFIG:RELEASE>:-O3 -DNDEBUG>
-                $<$<CONFIG:RELWITHDEBINFO>:-g -O2 -DNDEBUG>
-                $<$<CONFIG:MINSIZEREL>:-Os -DNDEBUG>)
+                $<$<CONFIG:DEBUG>:-DDEBUG>
+                $<$<CONFIG:RELEASE>:-DNDEBUG>
+                $<$<CONFIG:RELWITHDEBINFO>:-DNDEBUG>
+                $<$<CONFIG:MINSIZEREL>:-DNDEBUG>)
 
     elseif (CMAKE_C_COMPILER_ID STREQUAL "MSVC")
 
-        target_compile_options(${THIS} PRIVATE
-                $<$<CONFIG:DEBUG>:/Zi /Od /Ob0 /RTC1>
-                $<$<CONFIG:RELEASE>:/O2 /Ob2 /DNDEBUG>
-                $<$<CONFIG:RELWITHDEBINFO>:/Zi /O2 /Ob1 /DNDEBUG>
-                $<$<CONFIG:MINSIZEREL>:/O1 /Ob1 /DNDEBUG>)
 
     else ()
 
