--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -8,6 +8,11 @@ project(cppzmq VERSION ${DETECTED_CPPZMQ_VERSION})
 
 if (NOT TARGET libzmq AND NOT TARGET libzmq-static)
   find_package(ZeroMQ QUIET)
+  add_library(libzmq INTERFACE)
+  target_link_libraries(libzmq INTERFACE ZeroMQ::ZeroMQ)
+  add_library(libzmq-static INTERFACE)
+  target_link_libraries(libzmq-static INTERFACE ZeroMQ::ZeroMQ)
+  install(TARGETS libzmq libzmq-static EXPORT ${PROJECT_NAME}-targets)
 
   # libzmq autotools install: fallback to pkg-config
   if(NOT ZeroMQ_FOUND)
