diff --git a/CMakeLists.txt b/CMakeLists.txt
index 114c3ba..31b1583 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -82,6 +82,8 @@ if(CMAKE_VERSION VERSION_LESS 3.0)
 else()
     cmake_policy(SET CMP0048 NEW)
     project(fastcdr VERSION "${LIB_VERSION_STR}" LANGUAGES C CXX)
+    set(CMAKE_CXX_STANDARD 11)
+    set(CMAKE_CXX_STANDARD_REQUIRED ON)
 endif()
 
 set(PROJECT_NAME_STYLED "FastCDR")
