diff --git a/CMakeLists.txt b/CMakeLists.txt
index cd0fcaaca..67e0808ae 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -109,6 +109,7 @@ if (sqlite3)
 endif()
 
 find_package(Iconv REQUIRED)
+get_target_property(ICONV_INCLUDE_DIR Iconv::Iconv INTERFACE_INCLUDE_DIRECTORIES)
 include_directories(${ICONV_INCLUDE_DIR})
 
 
diff --git a/addon/doxyapp/CMakeLists.txt b/addon/doxyapp/CMakeLists.txt
index ae52cabc3..73e1e3d29 100644
--- a/addon/doxyapp/CMakeLists.txt
+++ b/addon/doxyapp/CMakeLists.txt
@@ -1,4 +1,5 @@
 find_package(Iconv)
+get_target_property(ICONV_INCLUDE_DIR Iconv::Iconv INTERFACE_INCLUDE_DIRECTORIES)
 
 include_directories(
 	${CMAKE_SOURCE_DIR}/src
@@ -26,7 +27,7 @@ mscgen
 doxygen_version
 doxycfg
 vhdlparser
-${ICONV_LIBRARIES}
+Iconv::Iconv
 ${CMAKE_THREAD_LIBS_INIT}
 ${SQLITE3_LIBRARIES}
 ${EXTRA_LIBS}
diff --git a/addon/doxyparse/CMakeLists.txt b/addon/doxyparse/CMakeLists.txt
index 2387f1b3c..cbf451037 100644
--- a/addon/doxyparse/CMakeLists.txt
+++ b/addon/doxyparse/CMakeLists.txt
@@ -1,4 +1,5 @@
 find_package(Iconv)
+get_target_property(ICONV_INCLUDE_DIR Iconv::Iconv INTERFACE_INCLUDE_DIRECTORIES)
 
 include_directories(
 	${CMAKE_SOURCE_DIR}/src
@@ -26,7 +27,7 @@ mscgen
 doxygen_version
 doxycfg
 vhdlparser
-${ICONV_LIBRARIES}
+Iconv::Iconv
 ${CMAKE_THREAD_LIBS_INIT}
 ${SQLITE3_LIBRARIES}
 ${EXTRA_LIBS}
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 23460d004..8f2c51012 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -325,7 +325,7 @@ target_link_libraries(doxygen
     doxygen_version
     vhdlparser
     ${SQLITE3_LIBRARIES}
-    ${ICONV_LIBRARIES}
+    Iconv::Iconv
     ${CMAKE_THREAD_LIBS_INIT}
     ${EXTRA_LIBS}
     ${CLANG_LIBS}
-- 

