--- addon/doxysearch/CMakeLists.txt
+++ addon/doxysearch/CMakeLists.txt
@@ -15,7 +15,7 @@
                doxyindexer.cpp
 )
 target_link_libraries(doxyindexer
-                      ${XAPIAN_LIBRARIES}
+                      CONAN_PKG::xapian-core
                       ${ZLIB_LIBRARIES}
                       ${WIN_EXTRA_LIBS}
                       ${COVERAGE_LINKER_FLAGS}
@@ -28,7 +28,7 @@ add_executable(doxysearch.cgi
 )
 target_link_libraries(doxysearch.cgi
                       doxygen_version
-                      ${XAPIAN_LIBRARIES}
+                      CONAN_PKG::xapian-core
                       ${ZLIB_LIBRARIES}
                       ${WIN_EXTRA_LIBS}
 )
--- libmscgen/mscgen_bool.h
+++ libmscgen/mscgen_bool.h
@@ -23,6 +23,14 @@
 #ifndef MSCGEN_BOOL_H
 #define MSCGEN_BOOL_H
 
+/* Some system headers define TRUE and FALSE so undefine them first */
+#ifdef TRUE
+#undef TRUE
+#endif
+#ifdef FALSE
+#undef FALSE
+#endif
+
 typedef enum
 {
     FALSE = 0,
