diff --git a/CMakeLists.txt b/CMakeLists.txt
index 251e18b77..4df454b39 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -118,6 +118,7 @@ if (sqlite3)
 endif()
 
 find_package(Iconv REQUIRED)
+get_target_property(ICONV_INCLUDE_DIR Iconv::Iconv INTERFACE_INCLUDE_DIRECTORIES)
 include_directories(${ICONV_INCLUDE_DIR})
 
 
diff --git a/addon/doxyapp/CMakeLists.txt b/addon/doxyapp/CMakeLists.txt
index a6fde4285..e19fc3d22 100644
--- a/addon/doxyapp/CMakeLists.txt
+++ b/addon/doxyapp/CMakeLists.txt
@@ -1,4 +1,5 @@
 find_package(Iconv)
+get_target_property(ICONV_INCLUDE_DIR Iconv::Iconv INTERFACE_INCLUDE_DIRECTORIES)
 
 include_directories(
 	${PROJECT_SOURCE_DIR}/src
@@ -42,7 +43,7 @@ mscgen
 doxygen_version
 doxycfg
 vhdlparser
-${ICONV_LIBRARIES}
+Iconv::Iconv
 ${CMAKE_THREAD_LIBS_INIT}
 ${SQLITE3_LIBRARIES}
 ${EXTRA_LIBS}
diff --git a/addon/doxyparse/CMakeLists.txt b/addon/doxyparse/CMakeLists.txt
index f439c29d9..e56b9a1de 100644
--- a/addon/doxyparse/CMakeLists.txt
+++ b/addon/doxyparse/CMakeLists.txt
@@ -1,4 +1,5 @@
 find_package(Iconv)
+get_target_property(ICONV_INCLUDE_DIR Iconv::Iconv INTERFACE_INCLUDE_DIRECTORIES)
 
 include_directories(
 	${PROJECT_SOURCE_DIR}/src
@@ -30,7 +31,7 @@ mscgen
 doxygen_version
 doxycfg
 vhdlparser
-${ICONV_LIBRARIES}
+Iconv::Iconv
 ${CMAKE_THREAD_LIBS_INIT}
 ${SQLITE3_LIBRARIES}
 ${EXTRA_LIBS}
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 18406488d..5c7144582 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -360,7 +360,7 @@ target_link_libraries(doxygen PRIVATE
     doxygen_version
     vhdlparser
     ${SQLITE3_LIBRARIES}
-    ${ICONV_LIBRARIES}
+    Iconv::Iconv
     ${CMAKE_THREAD_LIBS_INIT}
     ${EXTRA_LIBS}
     ${CLANG_LIBS}
-- 

