diff --git a/CMakeLists.txt.old b/CMakeLists.txt
index 119d390..ce2ac92 100644
--- a/CMakeLists.txt.old
+++ b/CMakeLists.txt
@@ -34,7 +34,6 @@ endif()
 
 # set CXX standard
 set(CMAKE_CXX_STANDARD_REQUIRED True)
-set(CMAKE_CXX_STANDARD 11)
 if (${COMPILER_IS_NVCC})
   # GNU CXX extensions are not supported by nvcc
   set(CMAKE_CXX_EXTENSIONS OFF)
@@ -74,6 +73,7 @@ endif()
 add_library(backward ${libtype} backward.cpp)
 target_compile_definitions(backward PUBLIC ${BACKWARD_DEFINITIONS})
 target_include_directories(backward PUBLIC ${BACKWARD_INCLUDE_DIRS})
+conan_target_link_libraries(backward)
 
 ###############################################################################
 # TESTS
@@ -125,3 +125,9 @@ install(
     FILES "BackwardConfig.cmake"
     DESTINATION ${CMAKE_INSTALL_PREFIX}/lib/backward
 )
+install(
+    TARGETS backward
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+)
