--- CMakeLists.txt	2019-12-25 22:05:35.580261859 +0100
+++ CMakeLists.txt.old	2019-12-25 21:49:22.536528035 +0100
@@ -34,6 +34,7 @@
 
 # set CXX standard
 set(CMAKE_CXX_STANDARD_REQUIRED True)
+set(CMAKE_CXX_STANDARD 11)
 if (${COMPILER_IS_NVCC})
   # GNU CXX extensions are not supported by nvcc
   set(CMAKE_CXX_EXTENSIONS OFF)
@@ -73,7 +74,6 @@
 add_library(backward ${libtype} backward.cpp)
 target_compile_definitions(backward PUBLIC ${BACKWARD_DEFINITIONS})
 target_include_directories(backward PUBLIC ${BACKWARD_INCLUDE_DIRS})
-conan_target_link_libraries(backward)
 
 ###############################################################################
 # TESTS
