diff --git src/Imath/ImathFun.cpp src/Imath/ImathFun.cpp
index bfec292..c8c2e7e 100644
--- src/Imath/ImathFun.cpp
+++ src/Imath/ImathFun.cpp
@@ -10,25 +10,25 @@ IMATH_INTERNAL_NAMESPACE_SOURCE_ENTER
 
 float succf(float f) IMATH_NOEXCEPT
 {
-    return isfinite(f) ?
+    return std::isfinite(f) ?
             std::nextafter(f, std::numeric_limits<float>::infinity()) : f;
 }
 
 float predf(float f) IMATH_NOEXCEPT
 {
-    return isfinite(f) ?
+    return std::isfinite(f) ?
             std::nextafter(f, -std::numeric_limits<float>::infinity()) : f;
 }
 
 double succd(double d) IMATH_NOEXCEPT
 {
-    return isfinite(d) ?
+    return std::isfinite(d) ?
             std::nextafter(d, std::numeric_limits<double>::infinity()) : d;
 }
 
 double predd(double d) IMATH_NOEXCEPT
 {
-    return isfinite(d) ?
+    return  std::isfinite(d) ?
             std::nextafter(d, -std::numeric_limits<double>::infinity()) : d;
 }
 
