diff --git a/CMakeLists.txt b/CMakeLists.txt
index faab325a..657abd4c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -53,7 +53,6 @@ elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC")
 endif ()
 
 list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake)
-include(SelectMSVCRuntime)
 
 option(GOOGLE_CLOUD_CPP_ENABLE_MACOS_OPENSSL_CHECK
        "If enabled, check that the user has defined OPENSSL_ROOT_DIR on macOS"
diff --git a/cmake/GoogleCloudCppCommon.cmake b/cmake/GoogleCloudCppCommon.cmake
index b487a1bc..880c98fe 100644
--- a/cmake/GoogleCloudCppCommon.cmake
+++ b/cmake/GoogleCloudCppCommon.cmake
@@ -17,9 +17,6 @@
 # Get the destination directories based on the GNU recommendations.
 include(GNUInstallDirs)
 
-# Pick the right MSVC runtime libraries.
-include(SelectMSVCRuntime)
-
 # Enable Werror
 include(EnableWerror)
 
diff --git a/examples/CMakeLists.txt b/examples/CMakeLists.txt
index 0cb7a9ae..d9016a0b 100644
--- a/examples/CMakeLists.txt
+++ b/examples/CMakeLists.txt
@@ -14,9 +14,6 @@
 # limitations under the License.
 # ~~~
 
-# Pick the right MSVC runtime libraries.
-include(SelectMSVCRuntime)
-
 add_executable(gcs2cbt gcs2cbt.cc)
 target_link_libraries(gcs2cbt google-cloud-cpp::bigtable
                       google-cloud-cpp::storage google-cloud-cpp::grpc_utils)
diff --git a/external/googleapis/CMakeLists.txt b/external/googleapis/CMakeLists.txt
index 03535ff8..2b05c214 100644
--- a/external/googleapis/CMakeLists.txt
+++ b/external/googleapis/CMakeLists.txt
@@ -157,8 +157,6 @@ if (PROTO_INCLUDE_DIR)
     list(INSERT PROTOBUF_IMPORT_DIRS 0 "${PROTO_INCLUDE_DIR}")
 endif ()
 
-include(SelectMSVCRuntime)
-
 google_cloud_cpp_add_protos_property()
 
 function (external_googleapis_short_name var proto)
