diff --git a/CMake/resources.cmake b/CMake/resources.cmake
index bad4279..75e5ff0 100644
--- a/CMake/resources.cmake
+++ b/CMake/resources.cmake
@@ -218,7 +218,8 @@ mark_as_advanced(FREETYPE_PATH)
 # libraries
 find_library(LIB_dl dl)
 if((NOT APPLE) OR FLTK_BACKEND_X11)
-  find_library(LIB_fontconfig fontconfig)
+  find_package(Fontconfig REQUIRED)
+  set(LIB_fontconfig "${Fontconfig_LIBRARIES}" CACHE INTERNAL "")
 endif((NOT APPLE) OR FLTK_BACKEND_X11)
 find_library(LIB_freetype freetype)
 find_library(LIB_GL GL)
