diff --git a/CMakeLists.txt b/CMakeLists.txt
index d4b0a0ce8ff41f7d0539ff4ed925d2588fb8303f..972e3a55c8bd3e508dc097621f4ae92aca3ec1ac 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -195,53 +195,6 @@ else()
 endif()
 set_source_files_properties(${CMAKE_CURRENT_SOURCE_DIR}/src/master.css.inc PROPERTIES GENERATED TRUE)
 
-# Tests
-
-include(FetchContent)
-FetchContent_Declare(
-  googletest
-  URL https://github.com/google/googletest/archive/609281088cfefc76f9d0ce82e1ff6c30cc3591e5.zip
-)
-# For Windows: Prevent overriding the parent project's compiler/linker settings
-set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
-FetchContent_GetProperties(googletest)
-if(NOT googletest_POPULATED)
-  FetchContent_Populate(googletest)
-  add_subdirectory(${googletest_SOURCE_DIR} ${googletest_BINARY_DIR})
-endif()
-
-if (BUILD_TESTING)
-    enable_testing()
-
-    set(TEST_NAME ${PROJECT_NAME}_tests)
-
-    add_executable(
-        ${TEST_NAME}
-        ${TEST_LITEHTML}
-        ${CMAKE_CURRENT_SOURCE_DIR}/src/master.css.inc
-    )
-
-    set_target_properties(${TEST_NAME} PROPERTIES
-        CXX_STANDARD 11
-        C_STANDARD 99
-        PUBLIC_HEADER "${HEADER_LITEHTML}"
-    )
-
-    target_include_directories(
-        ${TEST_NAME}
-        PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/containers
-    )
-
-    target_link_libraries(
-        ${TEST_NAME}
-        ${PROJECT_NAME}
-        gtest_main
-    )
-
-    include(GoogleTest)
-    gtest_discover_tests(${TEST_NAME})
-endif()
-
 #     set(TEST_NAME ${PROJECT_NAME}_tests)
 #     add_executable(${TEST_NAME} ${TEST_LITEHTML} ${CMAKE_CURRENT_SOURCE_DIR}/src/master.css.inc)
 #     set_target_properties(${TEST_NAME} PROPERTIES
