diff --git a/src/soplex/external/fmt/format.h b/src/soplex/external/fmt/format.h
index a51ba231..3b7b4db8 100644
--- a/src/soplex/external/fmt/format.h
+++ b/src/soplex/external/fmt/format.h
@@ -354,7 +354,7 @@ inline typename Container::value_type* get_data(Container& c) {
   return c.data();
 }

-#if defined(_SECURE_SCL) && _SECURE_SCL
+#if defined(_SECURE_SCL) && (!defined(_MSC_VER) || _MSC_VER < 1951)
 // Make a checked iterator to avoid MSVC warnings.
 template <typename T> using checked_ptr = stdext::checked_array_iterator<T*>;
 template <typename T> checked_ptr<T> make_checked(T* p, size_t size) {
