diff --git a/build/make/msvs_common.sh b/build/make/msvs_common.sh
index 27ddf7f..da462ec 100644
--- a/build/make/msvs_common.sh
+++ b/build/make/msvs_common.sh
@@ -9,7 +9,7 @@
 ##  be found in the AUTHORS file in the root of the source tree.
 ##
 
-if [ "$(uname -o 2>/dev/null)" = "Cygwin" ] \
+if [ "$(uname -o 2>/dev/null)" = "Cygwin" ] || [ "$(uname -o 2>/dev/null)" = "Msys" ]\
    && cygpath --help >/dev/null 2>&1; then
     FIXPATH='cygpath -m'
 else
