diff --git a/build/make/gen_msvs_vcxproj.sh b/build/make/gen_msvs_vcxproj.sh
index bb1c31d..36d1dd2 100755
--- a/build/make/gen_msvs_vcxproj.sh
+++ b/build/make/gen_msvs_vcxproj.sh
@@ -140,6 +140,15 @@ for opt in "$@"; do
     optval="${opt#*=}"
     case "$opt" in
         --help|-h) show_help
+        ;;
+        -MD*|-MT*) echo "ignoring $opt (this option is already handled)"
+
+        ;;
+        -Zi) echo "ignoring $opt..."
+
+        ;;
+        -O*) echo "ignoring $opt..."
+
         ;;
         --target=*) target="${optval}"
         ;;
@@ -344,7 +353,7 @@ generate_vcxproj() {
             fi
             tag_content CharacterSet Unicode
             if [ "$config" = "Release" ]; then
-                tag_content WholeProgramOptimization true
+                tag_content WholeProgramOptimization false
             fi
             close_tag PropertyGroup
         done
diff --git a/build/make/msvs_common.sh b/build/make/msvs_common.sh
index 27ddf7f..da462ec 100644
--- a/build/make/msvs_common.sh
+++ b/build/make/msvs_common.sh
@@ -9,7 +9,7 @@
 ##  be found in the AUTHORS file in the root of the source tree.
 ##
 
-if [ "$(uname -o 2>/dev/null)" = "Cygwin" ] \
+if [ "$(uname -o 2>/dev/null)" = "Cygwin" ] || [ "$(uname -o 2>/dev/null)" = "Msys" ]\
    && cygpath --help >/dev/null 2>&1; then
     FIXPATH='cygpath -m'
 else
