From 1ffe4bab11b9fce2db4d4d61e2f838912bc89a67 Mon Sep 17 00:00:00 2001
From: Martin Valgur <martin.valgur@gmail.com>
Date: Fri, 6 Sep 2024 23:34:25 +0300
Subject: [PATCH] Simplify install directories

---
 CMakeLists.txt | 33 ++++-----------------------------
 1 file changed, 4 insertions(+), 29 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 246bf3a6..548ac6aa 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -535,38 +535,12 @@ set_target_properties(connector PROPERTIES
   SOVERSION "${ABI_VERSION_MAJOR}"
 )
 
-
-install(TARGETS connector
-  CONFIGURATIONS Release RelWithDebInfo
-  ARCHIVE DESTINATION "${INSTALL_LIB_DIR_STATIC}" COMPONENT XDevAPIDev
-  RUNTIME DESTINATION "${INSTALL_LIB_DIR}" COMPONENT XDevAPIDll
-  LIBRARY DESTINATION "${INSTALL_LIB_DIR}" COMPONENT XDevAPIDll
-)
-
 install(TARGETS connector
-  CONFIGURATIONS Debug
-  ARCHIVE DESTINATION "${INSTALL_LIB_DIR_STATIC_DEBUG}" COMPONENT XDevAPIDev
-  RUNTIME DESTINATION "${INSTALL_LIB_DIR_DEBUG}" COMPONENT XDevAPIDll
-  LIBRARY DESTINATION "${INSTALL_LIB_DIR_DEBUG}" COMPONENT XDevAPIDll
+  ARCHIVE DESTINATION lib
+  LIBRARY DESTINATION lib
+  RUNTIME DESTINATION bin
 )
 
-if(MSVC AND NOT BUILD_STATIC)
-
-  install(FILES $<TARGET_PDB_FILE:connector>
-    CONFIGURATIONS RelWithDebInfo
-    DESTINATION "${INSTALL_LIB_DIR}"
-    COMPONENT Debuginfo
-  )
-
-  install(FILES $<TARGET_PDB_FILE:connector>
-    CONFIGURATIONS Debug
-    DESTINATION "${INSTALL_LIB_DIR_DEBUG}"
-    COMPONENT Debuginfo
-  )
-
-endif()
-
-
 #
 # Tests
 # =====
