--- a/CMake/AbseilHelpers.cmake
+++ b/CMake/AbseilHelpers.cmake
@@ -206,8 +206,12 @@ function(absl_cc_library)
     endif()
 
     # INTERFACE libraries can't have the CXX_STANDARD property set
+    if(ABSL_PROPAGATE_CXX_STD)
+        target_compile_features(${_NAME} PUBLIC cxx_std_11)
+    else()
     set_property(TARGET ${_NAME} PROPERTY CXX_STANDARD ${ABSL_CXX_STANDARD})
     set_property(TARGET ${_NAME} PROPERTY CXX_STANDARD_REQUIRED ON)
+    endif()
 
     # When being installed, we lose the absl_ prefix.  We want to put it back
     # to have properly named lib files.  This is a no-op when we are not being
