--- src/Makefile.in
+++ src/Makefile.in
@@ -327,7 +327,7 @@
 localstatedir = @localstatedir@
 mandir = @mandir@
 mkdir_p = @mkdir_p@
-my_CFLAGS = @my_CFLAGS@
+my_CFLAGS =
 oldincludedir = @oldincludedir@
 pdfdir = @pdfdir@
 prefix = @prefix@
--- src/modbus-private.h
+++ src/modbus-private.h
@@ -13,7 +13,7 @@
 #else
 # include "stdint.h"
 # include <time.h>
-typedef int ssize_t;
+//typedef int ssize_t;
 #endif
 #include <sys/types.h>
 #include <config.h>
