From 64c7b0a4b3957ab584112f5f733119e79f6323ed Mon Sep 17 00:00:00 2001
Date: Wed, 8 Oct 2025 11:15:20 +0200
Subject: [PATCH] Guard TARGET_PDB_FILE for static builds.

---
 CMakeLists.txt | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d272f0ba3..3cd7ecaeb 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -515,9 +515,9 @@ install(TARGETS datachannel EXPORT LibDataChannelTargets
 	ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
 )
 
-if(MSVC)
-	install(FILES $<TARGET_PDB_FILE:datachannel>
-    DESTINATION ${CMAKE_INSTALL_BINDIR} OPTIONAL)
+if(MSVC AND BUILD_SHARED_LIBS)
+    install(FILES $<TARGET_PDB_FILE:datachannel>
+        DESTINATION ${CMAKE_INSTALL_BINDIR} OPTIONAL)
 endif()
 
 install(FILES ${LIBDATACHANNEL_HEADERS}