--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -447,12 +447,11 @@ if (MSVC)
   if (CMAKE_SYSTEM_VERSION MATCHES "10\\.0.*")
     list(APPEND CRYPTOPP_COMPILE_DEFINITIONS "_WIN32_WINNT=0x0A00")
   endif ()
-  list(APPEND CRYPTOPP_COMPILE_OPTIONS "/FI\"winapifamily.h\"")
+  list(APPEND CRYPTOPP_COMPILE_OPTIONS /FIwinapifamily.h)
 endif ()
 
 # Enable PIC for all target machines except 32-bit i386 due to register pressures.
 if (NOT CRYPTOPP_I386)
-  SET(CMAKE_POSITION_INDEPENDENT_CODE 1)
 endif ()
 
 # IBM XLC compiler options for AIX and Linux.
