From 6a33009da09e903eb55a22b96de169d43c39fe38 Mon Sep 17 00:00:00 2001
From: Tomas Krupa <tomas.krupa@innovatrics.com>
Date: Mon, 24 Feb 2025 11:22:41 +0100
Subject: [PATCH] find OpenCV provided by conan_toolchain.cmake

---
 src/lib/CMakeLists.txt | 22 ++++++++++++----------
 1 file changed, 12 insertions(+), 10 deletions(-)

diff --git a/src/lib/CMakeLists.txt b/src/lib/CMakeLists.txt
index 12b3047..e8f87c6 100644
--- a/src/lib/CMakeLists.txt
+++ b/src/lib/CMakeLists.txt
@@ -14,24 +14,26 @@ add_library( ${PROJECT_NAME} ${sources} )
 #   resample_up.cpp
 # )
 
+find_package(OpenCV REQUIRED CONFIG)
+target_link_libraries(${PROJECT_NAME} PRIVATE opencv::opencv)
 
 if(_WIN32_64)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /WX")
   message(STATUS "LIB: CMAKE_CXX_FLAGS: ${CMAKE_CXX_FLAGS}")
 
   # To access OpenCV, use either these two lines below:
-  include(D:/OpenCV/opencv-4.5.0/build/x64/vc15/lib/OpenCVConfig.cmake)
-  target_link_libraries(${PROJECT_NAME} ${OpenCV_LIBS})
+  # include(D:/OpenCV/opencv-4.5.0/build/x64/vc15/lib/OpenCVConfig.cmake)
+  # target_link_libraries(${PROJECT_NAME} ${OpenCV_LIBS})
 
   # Or to access OpenCV, use these two lines below:
   # set(OpenCV_DIR "D:/OpenCV/opencv-4.5.0/build/x64/vc15/lib/")
   # find_package(OpenCV REQUIRED)
 
-  message(STATUS "OpenCV_INCLUDE_DIRS found as: '${OpenCV_INCLUDE_DIRS}'")
-  message(STATUS "OpenCV_LIBS found as: '${OpenCV_LIBS}'")
+  # message(STATUS "OpenCV_INCLUDE_DIRS found as: '${OpenCV_INCLUDE_DIRS}'")
+  # message(STATUS "OpenCV_LIBS found as: '${OpenCV_LIBS}'")
 
   # line below ALWAYS required
-  target_include_directories(${PROJECT_NAME} PRIVATE ${OpenCV_INCLUDE_DIRS} )
+  # target_include_directories(${PROJECT_NAME} PRIVATE ${OpenCV_INCLUDE_DIRS} )
 
   if(USE_NFIMM)
     target_compile_definitions(${PROJECT_NAME} PUBLIC USE_NFIMM)
@@ -42,11 +44,11 @@ if(_WIN32_64)
   
 else()
   message(STATUS "LIB: CMAKE_CXX_FLAGS: ${CMAKE_CXX_FLAGS}")
-	find_package(PkgConfig REQUIRED)
-	pkg_check_modules(OPENCV REQUIRED opencv4)
-	target_link_libraries(${PROJECT_NAME} ${OPENCV_LINK_LIBRARIES})
-	target_include_directories(${PROJECT_NAME} PUBLIC ${OPENCV_INCLUDE_DIRS})
-	target_compile_options(${PROJECT_NAME} PUBLIC ${OPENCV_CFLAGS_OTHER})
+	# find_package(PkgConfig REQUIRED)
+	# pkg_check_modules(OPENCV REQUIRED opencv4)
+	# target_link_libraries(${PROJECT_NAME} ${OPENCV_LINK_LIBRARIES})
+	# target_include_directories(${PROJECT_NAME} PUBLIC ${OPENCV_INCLUDE_DIRS})
+	# target_compile_options(${PROJECT_NAME} PUBLIC ${OPENCV_CFLAGS_OTHER})
 
   if(USE_NFIMM)
     target_compile_definitions(${PROJECT_NAME} PUBLIC USE_NFIMM)
-- 
2.34.1

