diff --git a/CMakeLists.txt b/CMakeLists.txt
index ca97e64..bdb257e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -22,7 +22,7 @@ if(WIN32)
     cmake_minimum_required(VERSION 3.4)
 
     # c++ 11 support from cmake 3.4 or newer
-    set(CMAKE_CXX_STANDARD 11) # C++11...
+    # set(CMAKE_CXX_STANDARD 11) # C++11...
     set(CMAKE_CXX_STANDARD_REQUIRED ON) #...is required...
     set(CMAKE_CXX_EXTENSIONS OFF) #...without compiler extensions like gnu++11
 
@@ -43,10 +43,10 @@ if(WIN32)
         #
         # for non-windows platform we try to keep cmake 2.8 support
         # since entreprise distribution tends to have 2.8 version.
-        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+        # set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
     else()
         # c++ 11 support from cmake 3.1 or newer
-        set(CMAKE_CXX_STANDARD 11) # C++11...
+        # set(CMAKE_CXX_STANDARD 11) # C++11...
         set(CMAKE_CXX_STANDARD_REQUIRED ON) #...is required...
         set(CMAKE_CXX_EXTENSIONS OFF) #...without compiler extensions like gnu++11
     endif()
