diff --git a/src/inference/src/dev/make_tensor.cpp b/src/inference/src/dev/make_tensor.cpp
index e34497749a..f28c90ccf4 100644
--- a/src/inference/src/dev/make_tensor.cpp
+++ b/src/inference/src/dev/make_tensor.cpp
@@ -5,6 +5,7 @@
 #include "openvino/runtime/make_tensor.hpp"

 #include <memory>
+#include <mutex>

 #include "openvino/runtime/iremote_tensor.hpp"
 #include "openvino/runtime/properties.hpp"

