--- Project/CMake/CMakeLists.txt
+++ Project/CMake/CMakeLists.txt
@@ -397,7 +397,7 @@
 
 target_link_libraries(mediainfo ${ZenLib_LIBRARY} ${ZLIB_LIBRARIES})
 
-if(MSVC AND BUILD_SHARED_LIBS)
+if(0)
   get_target_property(LOCATION mediainfo LOCATION_DEBUG)
   string(REGEX REPLACE "\\.[^.]*$" ".pdb" LOCATION "${LOCATION}")
   install(FILES ${LOCATION} DESTINATION ${BIN_INSTALL_DIR} CONFIGURATIONS Debug)
