diff --git a/libjbig/jbig.h b/libjbig/jbig.h
index 214105a..8bceb3b 100644
--- a/libjbig/jbig.h
+++ b/libjbig/jbig.h
@@ -254,13 +254,12 @@ struct jbg_dec_state {
 #   if defined(_VISUALC_)
 #       pragma warning( disable : 4273 )
 #   endif
-#   if !defined(_JBIGLIB_)
-#       define JBIGEXPORT __declspec(dllimport)
-#   else
-#       define JBIGEXPORT __declspec(dllexport)
-#   endif
+#endif
+
+#ifdef WIN32
+#   define JBIGEXPORT __declspec(dllexport)
 #else
-#   define JBIGEXPORT
+#   define JBIGEXPORT 
 #endif
 
 #if defined(_VISUALC_)
@@ -269,43 +268,51 @@ struct jbg_dec_state {
 #   pragma warning( disable : 4142 )
 #endif
 
-extern JBIGEXPORT void jbg_enc_init(struct jbg_enc_state *s, unsigned long x, unsigned long y,
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+JBIGEXPORT void jbg_enc_init(struct jbg_enc_state *s, unsigned long x, unsigned long y,
                                     int planes, unsigned char **p,
                                     void (*data_out)(unsigned char *start, size_t len,
                                                      void *file),
                                     void *file);
-extern JBIGEXPORT int jbg_enc_lrlmax(struct jbg_enc_state *s, unsigned long mwidth,
+JBIGEXPORT int jbg_enc_lrlmax(struct jbg_enc_state *s, unsigned long mwidth,
                                      unsigned long mheight);
-extern JBIGEXPORT void jbg_enc_layers(struct jbg_enc_state *s, int d);
-extern JBIGEXPORT int  jbg_enc_lrange(struct jbg_enc_state *s, int dl, int dh);
-extern JBIGEXPORT void jbg_enc_options(struct jbg_enc_state *s, int order, int options,
+JBIGEXPORT void jbg_enc_layers(struct jbg_enc_state *s, int d);
+JBIGEXPORT int  jbg_enc_lrange(struct jbg_enc_state *s, int dl, int dh);
+JBIGEXPORT void jbg_enc_options(struct jbg_enc_state *s, int order, int options,
                                        unsigned long l0, int mx, int my);
-extern JBIGEXPORT void jbg_enc_out(struct jbg_enc_state *s);
-extern JBIGEXPORT void jbg_enc_free(struct jbg_enc_state *s);
+JBIGEXPORT void jbg_enc_out(struct jbg_enc_state *s);
+JBIGEXPORT void jbg_enc_free(struct jbg_enc_state *s);
 
-extern JBIGEXPORT void jbg_dec_init(struct jbg_dec_state *s);
-extern JBIGEXPORT void jbg_dec_maxsize(struct jbg_dec_state *s, unsigned long xmax,
+JBIGEXPORT void jbg_dec_init(struct jbg_dec_state *s);
+JBIGEXPORT void jbg_dec_maxsize(struct jbg_dec_state *s, unsigned long xmax,
                                        unsigned long ymax);
-extern JBIGEXPORT int  jbg_dec_in(struct jbg_dec_state *s, unsigned char *data, size_t len,
+JBIGEXPORT int  jbg_dec_in(struct jbg_dec_state *s, unsigned char *data, size_t len,
                                   size_t *cnt);
-extern JBIGEXPORT long jbg_dec_getwidth(const struct jbg_dec_state *s);
-extern JBIGEXPORT long jbg_dec_getheight(const struct jbg_dec_state *s);
-extern JBIGEXPORT unsigned char *jbg_dec_getimage(const struct jbg_dec_state *s, int plane);
-extern JBIGEXPORT long jbg_dec_getsize(const struct jbg_dec_state *s);
-extern JBIGEXPORT void jbg_dec_merge_planes(const struct jbg_dec_state *s, int use_graycode,
+JBIGEXPORT long jbg_dec_getwidth(const struct jbg_dec_state *s);
+JBIGEXPORT long jbg_dec_getheight(const struct jbg_dec_state *s);
+JBIGEXPORT unsigned char *jbg_dec_getimage(const struct jbg_dec_state *s, int plane);
+JBIGEXPORT long jbg_dec_getsize(const struct jbg_dec_state *s);
+JBIGEXPORT void jbg_dec_merge_planes(const struct jbg_dec_state *s, int use_graycode,
                                             void (*data_out)(unsigned char *start, size_t len,
                                                              void *file), void *file);
-extern JBIGEXPORT long jbg_dec_getsize_merged(const struct jbg_dec_state *s);
-extern JBIGEXPORT void jbg_dec_free(struct jbg_dec_state *s);
-
-extern JBIGEXPORT const char *jbg_strerror(int errnum, int language);
-extern JBIGEXPORT void jbg_int2dppriv(unsigned char *dptable, const char *internal);
-extern JBIGEXPORT void jbg_dppriv2int(char *internal, const unsigned char *dptable);
-extern JBIGEXPORT unsigned long jbg_ceil_half(unsigned long x, int n);
-extern JBIGEXPORT void jbg_split_planes(unsigned long x, unsigned long y, int has_planes,
+JBIGEXPORT long jbg_dec_getsize_merged(const struct jbg_dec_state *s);
+JBIGEXPORT void jbg_dec_free(struct jbg_dec_state *s);
+
+JBIGEXPORT const char *jbg_strerror(int errnum, int language);
+JBIGEXPORT void jbg_int2dppriv(unsigned char *dptable, const char *internal);
+JBIGEXPORT void jbg_dppriv2int(char *internal, const unsigned char *dptable);
+JBIGEXPORT unsigned long jbg_ceil_half(unsigned long x, int n);
+JBIGEXPORT void jbg_split_planes(unsigned long x, unsigned long y, int has_planes,
                                         int encode_planes,
                                         const unsigned char *src, unsigned char **dest,
                                         int use_graycode);
-extern JBIGEXPORT int jbg_newlen(unsigned char *bie, size_t len);
+JBIGEXPORT int jbg_newlen(unsigned char *bie, size_t len);
+
+#ifdef __cplusplus
+}
+#endif
 
 #endif /* JBG_H */
