--- CMakeLists.txt
+++ CMakeLists.txt
@@ -161,6 +161,14 @@ if (enable_threads)
   message(STATUS "Thread library: ${CMAKE_THREAD_LIBS_INIT}")
   if (without_libatomic_ops OR BORLAND OR MSVC OR WATCOM)
     include_directories(libatomic_ops/src)
+    # TODO: Rename the following to Atomic_ops
+    find_package(libatomic_ops CONFIG)
+    if (libatomic_ops_FOUND)
+      # TODO: Rename the following to Atomic_ops::atomic_ops
+      get_target_property(AO_INCLUDE_DIRS libatomic_ops::libatomic_ops
+                          INTERFACE_INCLUDE_DIRECTORIES)
+      include_directories(${AO_INCLUDE_DIRS})
+    endif()
     # Note: alternatively, use CFLAGS_EXTRA to pass -I<...>/libatomic_ops/src.
   else()
     # Assume the compiler supports GCC atomic intrinsics.
