diff --git a/NvCloth/include/NvCloth/ps/PsAllocator.h b/NvCloth/include/NvCloth/ps/PsAllocator.h
index 7d61bee..3d42bbc 100644
--- a/NvCloth/include/NvCloth/ps/PsAllocator.h
+++ b/NvCloth/include/NvCloth/ps/PsAllocator.h
@@ -37,8 +37,12 @@
 
 #if(PX_WINDOWS_FAMILY || PX_XBOXONE)
 #include <exception>
+#if(_MSC_VER >= 1923)
+	#include <typeinfo>
+#else
 #include <typeinfo.h>
 #endif
+#endif
 #if(PX_APPLE_FAMILY)
 #include <typeinfo>
 #endif
