diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1c3b2e9..fd31b7f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -20,7 +20,7 @@ if(NOT FUNCHOOK_CPU)
   endif ()
 endif ()

-if (FUNCHOOK_CPU MATCHES "arm64")
+if (1)
   set(FUNCHOOK_DEFAULT_DISASM "capstone")
 else ()
   set(FUNCHOOK_DEFAULT_DISASM "distorm")
@@ -53,7 +53,7 @@ endif ()
 include(CheckCCompilerFlag)
 check_c_compiler_flag(-fvisibility=hidden HAVE_FVISIBILITY_HIDDEN)

-if (FUNCHOOK_DISASM STREQUAL capstone)
+if (1)
   set(DISASM_CAPSTONE 1)
 elseif (FUNCHOOK_DISASM STREQUAL distorm)
   set(DISASM_DISTORM 1)
@@ -85,33 +85,17 @@ endfunction(add_subdirectory_pic)
 #
 # capstone
 #
-if (DISASM_CAPSTONE)
-  configure_file(cmake/capstone.cmake.in capstone-download/CMakeLists.txt)
-  execute_process(COMMAND "${CMAKE_COMMAND}" -G "${CMAKE_GENERATOR}" .
-      WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/capstone-download"
-  )
-  execute_process(COMMAND "${CMAKE_COMMAND}" --build .
-      WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/capstone-download"
-  )
-
-  string(TOUPPER ${FUNCHOOK_CPU} FUNCHOOK_CPU_UPPER)
-
-  set(CAPSTONE_BUILD_SHARED OFF CACHE BOOL "")
-  set(CAPSTONE_BUILD_STATIC_RUNTIME OFF CACHE BOOL "")
-  set(CAPSTONE_BUILD_TESTS OFF CACHE BOOL "")
-  set(CAPSTONE_BUILD_CSTOOL OFF CACHE BOOL "")
-  set(CAPSTONE_ARCHITECTURE_DEFAULT OFF CACHE BOOL "")
-  set(CAPSTONE_${FUNCHOOK_CPU_UPPER}_SUPPORT ON CACHE BOOL "")
-  add_subdirectory_pic(${CMAKE_CURRENT_BINARY_DIR}/capstone-src ${CMAKE_CURRENT_BINARY_DIR}/capstone-build)
+if (1)
+  find_package(capstone CONFIG REQUIRED)

-  list(APPEND FUNCHOOK_DEPS capstone-static)
+  list(APPEND FUNCHOOK_DEPS capstone::capstone)
   set(DISASM capstone)
 endif ()

 #
 # distorm
 #
-if (DISASM_DISTORM)
+if (0)
   set(DISTORM_PATH distorm/)
   set(DISTORM_SRC_DIR ${DISTORM_PATH}/src/)
   set(DISTORM_SOURCES ${DISTORM_SRC_DIR}/decoder.c ${DISTORM_SRC_DIR}/distorm.c ${DISTORM_SRC_DIR}/instructions.c
@@ -139,7 +123,7 @@ endif ()
 #
 # zydis
 #
-if (DISASM_ZYDIS)
+if (0)
   include(ExternalProject)
   ExternalProject_Add(Zydis_src
     GIT_REPOSITORY    https://github.com/zyantific/zydis.git
