see https://github.com/KhronosGroup/MoltenVK/pull/1287

--- a/MoltenVK/MoltenVK/GPUObjects/MVKPixelFormats.h
+++ b/MoltenVK/MoltenVK/GPUObjects/MVKPixelFormats.h
@@ -22,7 +22,7 @@
 #include "MVKEnvironment.h"
 #include "MVKOSExtensions.h"
 #include "MVKBaseObject.h"
-#include <SPIRV-Cross/spirv_msl.hpp>
+#include <spirv_msl.hpp>
 #include <unordered_map>
 
 #import <Metal/Metal.h>
--- a/MoltenVKShaderConverter/Common/SPIRVSupport.cpp
+++ b/MoltenVKShaderConverter/Common/SPIRVSupport.cpp
@@ -18,7 +18,7 @@
 
 #include "SPIRVSupport.h"
 #include "MVKStrings.h"
-#include <SPIRV-Cross/spirv.hpp>
+#include <spirv.hpp>
 #include <ostream>
 
 #import <CoreFoundation/CFByteOrder.h>
--- a/MoltenVKShaderConverter/MoltenVKShaderConverter/SPIRVReflection.h
+++ b/MoltenVKShaderConverter/MoltenVKShaderConverter/SPIRVReflection.h
@@ -19,10 +19,10 @@
 #ifndef __SPIRVReflection_h_
 #define __SPIRVReflection_h_ 1
 
-#include <SPIRV-Cross/spirv.hpp>
-#include <SPIRV-Cross/spirv_common.hpp>
-#include <SPIRV-Cross/spirv_parser.hpp>
-#include <SPIRV-Cross/spirv_reflect.hpp>
+#include <spirv.hpp>
+#include <spirv_common.hpp>
+#include <spirv_parser.hpp>
+#include <spirv_reflect.hpp>
 #include <string>
 #include <vector>
 
--- a/MoltenVKShaderConverter/MoltenVKShaderConverter/SPIRVToMSLConverter.h
+++ b/MoltenVKShaderConverter/MoltenVKShaderConverter/SPIRVToMSLConverter.h
@@ -19,8 +19,8 @@
 #ifndef __SPIRVToMSLConverter_h_
 #define __SPIRVToMSLConverter_h_ 1
 
-#include <SPIRV-Cross/spirv.hpp>
-#include <SPIRV-Cross/spirv_msl.hpp>
+#include <spirv.hpp>
+#include <spirv_msl.hpp>
 #include <string>
 #include <vector>
 #include <unordered_map>
