--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -82,6 +82,11 @@
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake/MoltenVK/")
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake/recipes/")

+find_package(cereal REQUIRED CONFIG)
+find_package(spirv-cross REQUIRED CONFIG)
+find_package(SPIRV-Tools REQUIRED CONFIG)
+find_package(VulkanHeaders REQUIRED CONFIG)
+
 # General CMake utils
 set(CPM_USE_NAMED_CACHE_DIRECTORIES 1)
 include(MoltenVK_CPM_Cache)
