upstream issue: https://github.com/KhronosGroup/MoltenVK/issues/1474

--- a/MoltenVK/MoltenVK/Commands/MVKCommandBuffer.mm
+++ b/MoltenVK/MoltenVK/Commands/MVKCommandBuffer.mm
@@ -772,7 +772,10 @@ void MVKCommandEncoder::markTimestamp(MVKTimestampQueryPool* pQueryPool, uint32_
 		MVKCounterSamplingFlags sampPts = _device->_pMetalFeatures->counterSamplingPoints;
 		for (uint32_t qOfst = 0; qOfst < queryCount; qOfst++) {
 			if (mvkIsAnyFlagEnabled(sampPts, MVK_COUNTER_SAMPLING_AT_PIPELINE_STAGE)) {
-				_timestampStageCounterQueries.push_back({ pQueryPool, query + qOfst });
+				GPUCounterQuery timestampCounterQuery = {};
+				timestampCounterQuery.queryPool = pQueryPool;
+				timestampCounterQuery.query = query + qOfst;
+				_timestampStageCounterQueries.push_back(timestampCounterQuery);
 			} else {
 				encodeGPUCounterSample(pQueryPool, query + qOfst, sampPts);
 			}
