--- src/PE/ResourcesManager.cpp	2019-11-29 15:51:38.000000000 -0500
+++ src/PE/ResourcesManager.cpp	2020-04-30 14:12:47.001136300 -0400
@@ -1174,7 +1174,6 @@

 std::string ResourcesManager::print(uint32_t depth) const {
   std::ostringstream oss;
-  oss << rang::control::forceColor;
   uint32_t current_depth = 0;
   this->print_tree(*this->resources_, oss, current_depth, depth);
   return oss.str();
