diff --git a/CMakeLists.txt b/CMakeLists.txt
index 808330e..9b0f2c4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -469,7 +469,6 @@ add_library (glog
 )
 add_library(glog::glog ALIAS glog)
 
-set_target_properties (glog PROPERTIES POSITION_INDEPENDENT_CODE ON)
 
 if (UNWIND_LIBRARY)
   target_link_libraries (glog PUBLIC ${UNWIND_LIBRARY})
@@ -489,7 +488,7 @@ if (WIN32 AND HAVE_SNPRINTF)
 endif (WIN32 AND HAVE_SNPRINTF)
 
 if (gflags_FOUND)
-  target_link_libraries (glog PUBLIC gflags)
+  target_link_libraries (glog PUBLIC gflags::gflags)
 
   if (NOT BUILD_SHARED_LIBS)
     # Don't use __declspec(dllexport|dllimport) if this is a static build
