--- bin/aclocal.in
+++ bin/aclocal.in
@@ -20,10 +20,13 @@
 # Alexandre Duret-Lutz <adl@gnu.org>.
 
 use 5.006; use strict; use warnings;
+use Cwd 'abs_path';
+use File::Basename;
 
 BEGIN
 {
-  unshift (@INC, '@datadir@/@PACKAGE@-@APIVERSION@')
+  my $scriptpath = abs_path(dirname(__FILE__));
+  unshift (@INC, "$scriptpath/../share/@PACKAGE@-@APIVERSION@")
     unless $ENV{AUTOMAKE_UNINSTALLED};
 }
 
@@ -62,9 +65,15 @@
 # @system_includes can be augmented with the 'dirlist' file or the
 # --aclocal-path option/ACLOCAL_PATH environment variable, and reset
 # with the '--system-acdir' option.
+my $scriptpath = abs_path(dirname(__FILE__));
 my @user_includes = ();
-my @automake_includes = ('@datadir@/aclocal-' . $APIVERSION);
-my @system_includes = ('@datadir@/aclocal');
+my @automake_includes = ("$scriptpath/../share/aclocal-" . $APIVERSION);
+my @system_includes = ("$scriptpath/../share/aclocal");
+my @conan_includes;
+foreach my $conandir (uniq(split(/[:;]/, $ENV{'AUTOMAKE_CONAN_INCLUDES'} || ""))) 
+{
+    push (@conan_includes, $conandir) if $conandir ne '' && -d $conandir;    
+}
 my $aclocal_path = '';
 
 # Whether we should copy M4 file in $user_includes[0].
@@ -447,6 +456,7 @@
     }
   scan_m4_dirs (FT_AUTOMAKE, SCAN_M4_DIRS_ERROR, @automake_includes);
   scan_m4_dirs (FT_SYSTEM, SCAN_M4_DIRS_ERROR, @system_includes);
+  scan_m4_dirs (FT_SYSTEM, SCAN_M4_DIRS_ERROR, @conan_includes);
 
   # Construct a new function that does the searching.  We use a
   # function (instead of just evaluating $search in the loop) so that
@@ -767,7 +777,7 @@
   # aclocal.m4 is not yet available.
   local $ENV{WARNINGS} = 'none';
 
-  my $traces = ($ENV{AUTOM4TE} || '@am_AUTOM4TE@');
+  my $traces = ($ENV{AUTOM4TE} || '/usr/bin/env autom4te');
   $traces .= " --language Autoconf-without-aclocal-m4 ";
 
   # Support AC_CONFIG_MACRO_DIRS also with older autoconf.
--- bin/automake.in
+++ bin/automake.in
@@ -23,10 +23,13 @@
 package Automake;
 
 use 5.006; use strict; use warnings;
+use Cwd 'abs_path';
+use File::Basename;
 
 BEGIN
 {
-  unshift (@INC, '@datadir@/@PACKAGE@-@APIVERSION@')
+  my $scriptpath = abs_path(dirname(__FILE__));
+  unshift (@INC, "$scriptpath/../share/@PACKAGE@-@APIVERSION@")
     unless $ENV{AUTOMAKE_UNINSTALLED};
 
   # Override SHELL.  This is required on DJGPP so that system() uses
@@ -5372,7 +5375,8 @@
   # and will see its warnings then.
   local $ENV{WARNINGS} = 'none';
 
-  my $traces = ($ENV{AUTOCONF} || '@am_AUTOCONF@') . " ";
+  my $traces = ($ENV{AUTOCONF} || '/usr/bin/env autoconf') . " ";
+
 
   # Use a separator unlikely to be used, not ':', the default, which
   # has a precise meaning for AC_CONFIG_FILES and so on.
--- lib/Automake/Config.in
+++ lib/Automake/Config.in
@@ -18,6 +18,8 @@
 package Automake::Config;
 
 use 5.006; use strict; use warnings;
+use Cwd 'abs_path';
+use File::Basename;
 
 use Exporter;
 
@@ -32,7 +34,8 @@
 our $PACKAGE_BUGREPORT = '@PACKAGE_BUGREPORT@';
 our $VERSION = '@VERSION@';
 our $RELEASE_YEAR = '@RELEASE_YEAR@';
-our $libdir = $ENV{"AUTOMAKE_LIBDIR"} || '@datadir@/@PACKAGE@-@APIVERSION@';
+my $scriptpath = abs_path(dirname(__FILE__));
+our $libdir = $ENV{"AUTOMAKE_LIBDIR"} || "$scriptpath/../../@PACKAGE@-@APIVERSION@";
 
 our $perl_threads = 0;
 # We need at least this version for CLONE support.
