--- bin/aclocal.in
+++ bin/aclocal.in
@@ -20,8 +20,13 @@
 # Alexandre Duret-Lutz <adl@gnu.org>.
 
+use Cwd 'abs_path';
+use File::Basename;
+
 BEGIN
 {
-  unshift (@INC, '@datadir@/@PACKAGE@-@APIVERSION@')
+  my $scriptpath = abs_path(dirname(__FILE__));
+  my $pkgdatadir = "$scriptpath/../@datadir@/@PACKAGE@-@APIVERSION@";
+  unshift (@INC, $pkgdatadir)
     unless $ENV{AUTOMAKE_UNINSTALLED};
 }
 
--- bin/automake.in
+++ bin/automake.in
@@ -25,8 +25,14 @@
 use strict;
 
+use Cwd 'abs_path';
+use File::Basename;
+
 BEGIN
 {
-  unshift (@INC, '@datadir@/@PACKAGE@-@APIVERSION@')
+  my $scriptpath = abs_path(dirname(__FILE__));
+  $ENV{'AUTOMAKE_LIBDIR'} = "$scriptpath/../@datadir@/@PACKAGE@-@APIVERSION@" if ! exists $ENV{'AUTOMAKE_LIBDIR'};
+  my $pkgdatadir = "$scriptpath/../@datadir@/@PACKAGE@-@APIVERSION@";
+  unshift (@INC, $pkgdatadir)
     unless $ENV{AUTOMAKE_UNINSTALLED};
 
   # Override SHELL.  This is required on DJGPP so that system() uses
$
