--- bin/aclocal.in
+++ bin/aclocal.in
@@ -64,6 +64,6 @@
 my @user_includes = ();
 my @automake_includes = ('@datadir@/aclocal-' . $APIVERSION);
 my @system_includes = ('@datadir@/aclocal');
+my @conan_includes = uniq(strip(/:/, $ENV{'AUTOMAKE_CONAN_INCLUDES'}))
-
 # Whether we should copy M4 file in $user_includes[0].
 my $install = 0;
@@ -445,6 +445,6 @@ sub scan_m4_files ()
     }
   scan_m4_dirs (FT_AUTOMAKE, SCAN_M4_DIRS_ERROR, @automake_includes);
   scan_m4_dirs (FT_SYSTEM, SCAN_M4_DIRS_ERROR, @system_includes);
+  scan_m4_dirs (FT_SYSTEM, SCAN_M4_DIRS_ERROR, @conan_includes);
-
   # Construct a new function that does the searching.  We use a
   # function (instead of just evaluating $search in the loop) so that
