--- CMakeLists.txt
+++ CMakeLists.txt
@@ -50,9 +50,9 @@ ign_create_packages()
 #============================================================================
 # Configure documentation
 #============================================================================
-configure_file(${CMAKE_SOURCE_DIR}/api.md.in ${CMAKE_BINARY_DIR}/api.md)
-configure_file(${CMAKE_SOURCE_DIR}/tutorials.md.in ${CMAKE_BINARY_DIR}/tutorials.md)
+configure_file(${PROJECT_SOURCE_DIR}/api.md.in ${PROJECT_BINARY_DIR}/api.md)
+configure_file(${PROJECT_SOURCE_DIR}/tutorials.md.in ${PROJECT_BINARY_DIR}/tutorials.md)
 
 ign_create_docs(
-  API_MAINPAGE_MD "${CMAKE_BINARY_DIR}/api.md"
-  TUTORIALS_MAINPAGE_MD "${CMAKE_BINARY_DIR}/tutorials.md")
+    API_MAINPAGE_MD "${PROJECT_BINARY_DIR}/api.md"
+    TUTORIALS_MAINPAGE_MD "${PROJECT_BINARY_DIR}/tutorials.md")
 
