diff --git a/HLSL2GLSLConverter/CMakeLists.txt b/HLSL2GLSLConverter/CMakeLists.txt
index 0b1de80..e69515c 100644
--- a/HLSL2GLSLConverter/CMakeLists.txt
+++ b/HLSL2GLSLConverter/CMakeLists.txt
@@ -33,12 +33,7 @@ target_compile_definitions(HLSL2GLSLConverter PRIVATE UNICODE)
 
 target_link_libraries(HLSL2GLSLConverter 
 PRIVATE 
-    Diligent-BuildSettings 
-    Diligent-TargetPlatform
-    Diligent-Common
-    Diligent-GraphicsTools
-    Diligent-HLSL2GLSLConverterLib 
-    Diligent-GraphicsEngineOpenGL-static
+    CONAN_PKG::diligent-core
 )
 
 source_group("source" FILES ${SOURCE})
diff --git a/NativeApp/Android/ndk_helper/CMakeLists.txt b/NativeApp/Android/ndk_helper/CMakeLists.txt
index 6f23857..3aa07a8 100644
--- a/NativeApp/Android/ndk_helper/CMakeLists.txt
+++ b/NativeApp/Android/ndk_helper/CMakeLists.txt
@@ -29,7 +29,7 @@ PUBLIC
 
 target_link_libraries(NDKHelper 
 PRIVATE 
-	Diligent-BuildSettings
+        CONAN_PKG::diligent-core
 PUBLIC
     native_app_glue
 )
diff --git a/NativeApp/CMakeLists.txt b/NativeApp/CMakeLists.txt
index 366c303..165ba72 100644
--- a/NativeApp/CMakeLists.txt
+++ b/NativeApp/CMakeLists.txt
@@ -432,8 +432,7 @@ PUBLIC
 
 target_link_libraries(Diligent-NativeAppBase 
 PRIVATE 
-    Diligent-BuildSettings
-    Diligent-Common
+    CONAN_PKG::diligent-core
 )
 
 if(PLATFORM_WIN32)
diff --git a/NativeApp/Linux/CMakeLists.txt b/NativeApp/Linux/CMakeLists.txt
index a550fe0..e8f3a87 100644
--- a/NativeApp/Linux/CMakeLists.txt
+++ b/NativeApp/Linux/CMakeLists.txt
@@ -20,7 +20,7 @@ source_group("interface" FILES ${INTERFACE})
 
 target_link_libraries(XCBKeySyms 
 PRIVATE 
-    Diligent-BuildSettings
+    CONAN_PKG::diligent-core
     xcb
 )
 
