diff --git a/CMakeLists.txt b/CMakeLists.txt
index 80becb363..cc1ba753b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -437,7 +437,7 @@ if(DILIGENT_INSTALL_CORE)
         "${DILIGENT_CORE_INSTALL_LIBS_LIST}"
         DiligentCore-static                     # Custom target name
         DiligentCore                            # Folder
-        "${CMAKE_INSTALL_LIBDIR}/${DILIGENT_CORE_DIR}/$<CONFIG>"    # Install destination
+        "${CMAKE_INSTALL_LIBDIR}/DiligentCore"    # Install destination
     )
 
     install(FILES License.txt DESTINATION "Licenses" RENAME DiligentEngine-License.txt)
