diff --git a/AssetLoader/CMakeLists.txt b/AssetLoader/CMakeLists.txt
index 9c9a16d..7479b6c 100644
--- a/AssetLoader/CMakeLists.txt
+++ b/AssetLoader/CMakeLists.txt
@@ -35,13 +35,8 @@ source_group("interface" FILES ${INTERFACE})
 
 target_link_libraries(Diligent-AssetLoader 
 PRIVATE 
-    Diligent-BuildSettings
-    Diligent-Common
-    Diligent-PlatformInterface
-    Diligent-GraphicsEngineInterface
-    Diligent-GraphicsAccessories
-    Diligent-GraphicsTools
     Diligent-TextureLoader
+    diligent-core::diligent-core
 )
 
 if (TARGET draco)
diff --git a/HLSL2GLSLConverter/CMakeLists.txt b/HLSL2GLSLConverter/CMakeLists.txt
index 0b1de80..5f800f3 100644
--- a/HLSL2GLSLConverter/CMakeLists.txt
+++ b/HLSL2GLSLConverter/CMakeLists.txt
@@ -33,12 +33,7 @@ target_compile_definitions(HLSL2GLSLConverter PRIVATE UNICODE)
 
 target_link_libraries(HLSL2GLSLConverter 
 PRIVATE 
-    Diligent-BuildSettings 
-    Diligent-TargetPlatform
-    Diligent-Common
-    Diligent-GraphicsTools
-    Diligent-HLSL2GLSLConverterLib 
-    Diligent-GraphicsEngineOpenGL-static
+    diligent-core::diligent-core
 )
 
 source_group("source" FILES ${SOURCE})
diff --git a/Imgui/CMakeLists.txt b/Imgui/CMakeLists.txt
index 64a2a77..bc0fb39 100644
--- a/Imgui/CMakeLists.txt
+++ b/Imgui/CMakeLists.txt
@@ -116,13 +116,8 @@ source_group("imGuIZMO.quat" FILES ${IMGUIZMO_QUAT_SOURCE})
 
 target_link_libraries(Diligent-Imgui
 PRIVATE
-    Diligent-BuildSettings
-    Diligent-Common
-    Diligent-PlatformInterface
-    Diligent-GraphicsEngineInterface
-    Diligent-GraphicsAccessories
-    Diligent-GraphicsTools
-    imgui::imgui
+     diligent-core::diligent-core
+     imgui::imgui
 )
 
 set_target_properties(Diligent-Imgui PROPERTIES
diff --git a/NativeApp/Android/ndk_helper/CMakeLists.txt b/NativeApp/Android/ndk_helper/CMakeLists.txt
index 6f23857..35d2f02 100644
--- a/NativeApp/Android/ndk_helper/CMakeLists.txt
+++ b/NativeApp/Android/ndk_helper/CMakeLists.txt
@@ -29,7 +29,7 @@ PUBLIC
 
 target_link_libraries(NDKHelper 
 PRIVATE 
-	Diligent-BuildSettings
+        diligent-core::diligent-core
 PUBLIC
     native_app_glue
 )
diff --git a/NativeApp/CMakeLists.txt b/NativeApp/CMakeLists.txt
index 7ea2063..951206b 100644
--- a/NativeApp/CMakeLists.txt
+++ b/NativeApp/CMakeLists.txt
@@ -435,8 +435,7 @@ PUBLIC
 
 target_link_libraries(Diligent-NativeAppBase 
 PRIVATE 
-    Diligent-BuildSettings
-    Diligent-Common
+    diligent-core::diligent-core
 )
 
 if(PLATFORM_WIN32)
diff --git a/NativeApp/Linux/CMakeLists.txt b/NativeApp/Linux/CMakeLists.txt
index a550fe0..e17d01c 100644
--- a/NativeApp/Linux/CMakeLists.txt
+++ b/NativeApp/Linux/CMakeLists.txt
@@ -20,7 +20,7 @@ source_group("interface" FILES ${INTERFACE})
 
 target_link_libraries(XCBKeySyms 
 PRIVATE 
-    Diligent-BuildSettings
+    diligent-core::diligent-core
     xcb
 )
 
