--- a/src/libnova/utility.h
+++ b/src/libnova/utility.h
@@ -247,7 +247,7 @@ double cbrt(double x);
 
 #endif /* __C89_SUB__ */
 
-#if defined(__WIN32__) || defined(sun) || defined(__C89_SUB__)
+#if defined(__C89_SUB__)
 
 /* Not a Number function generator */
 double nan(const char *code);
--- a/src/utility.c
+++ b/src/utility.c
@@ -740,7 +740,7 @@ double cbrt(double x)
 
 #endif /* __C89_SUB__ */
 
-#if defined(__WIN32__) || defined(sun) || defined(__C89_SUB__)
+#if defined(__C89_SUB__)
 
 /* Not a Number function generator */
 double nan(const char *code)
