--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -122,7 +122,7 @@ install-strip : force
 	  */install-reloc) \
 	    $(MAKE) install prefix='$(prefix)' exec_prefix='$(exec_prefix)' libdir='$(libdir)' RELOCATABLE_STRIP='$(STRIP)' ;; \
 	  *) \
-	    $(MAKE) install prefix='$(prefix)' exec_prefix='$(exec_prefix)' libdir='$(libdir)' INSTALL_PROGRAM='$(INSTALL_STRIP_PROGRAM)' ;; \
+	    $(MAKE) install prefix='$(prefix)' exec_prefix='$(exec_prefix)' libdir='$(libdir)' INSTALL_PROGRAM='$(INSTALL_STRIP_PROGRAM)' STRIPPROG='$(STRIP)' ;; \
 	esac
 
 installdirs : force
