--- build/platform-linux.mk
+++ build/platform-linux.mk
@@ -6,1 +6,1 @@
-CFLAGS += -Wall -fno-strict-aliasing -fPIC -MMD -MP
+CFLAGS += -Wall -fno-strict-aliasing -MMD -MP
--- build/platform-bsd.mk
+++ build/platform-bsd.mk
@@ -6,1 +6,1 @@
-CFLAGS += -fPIC
+CFLAGS += 
--- build/platform-darwin.mk
+++ build/platform-darwin.mk
@@ -13,1 +13,1 @@
-CFLAGS += -Wall -fPIC -MMD -MP
+CFLAGS += -Wall  -MMD -MP
