--- build/platform-linux.mk
+++ build/platform-linux.mk
@@ -6,1 +6,1 @@
-CFLAGS += -Wall -fno-strict-aliasing -fPIC -MMD -MP -fstack-protector-all
+CFLAGS += -Wall -fno-strict-aliasing -MMD -MP -fstack-protector-all
--- build/platform-bsd.mk
+++ build/platform-bsd.mk
@@ -6,1 +6,1 @@
-CFLAGS += -fPIC -fstack-protector-all
+CFLAGS += -fstack-protector-all
--- build/platform-darwin.mk
+++ build/platform-darwin.mk
@@ -13,1 +13,1 @@
-CFLAGS += -Wall -fPIC -MMD -MP -fstack-protector-all
+CFLAGS += -Wall  -MMD -MP -fstack-protector-all
