diff --git a/CMakeLists.txt b/CMakeLists.txt
index b8ae7d9..6c55e87 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -72,12 +72,12 @@ if(NOT CMAKE_BUILD_TYPE)
   set(CMAKE_BUILD_TYPE "Release")
 endif()
 
-add_subdirectory(${CMAKE_SOURCE_DIR}/src)
+add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/src)
 
 # 
 # Install Artifacts
 #
-install(FILES ${CMAKE_SOURCE_DIR}/CHANGELOG.md
+install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/CHANGELOG.md
         DESTINATION .)
 
 set(CPACK_PACKAGE_VENDOR "Michele Adduci <adduci@tutanota.com>")
diff --git a/src/tools/CMakeLists.txt b/src/tools/CMakeLists.txt
index baa5281..2d26bcb 100644
--- a/src/tools/CMakeLists.txt
+++ b/src/tools/CMakeLists.txt
@@ -51,7 +51,7 @@ foreach(TOOL ${TOOLS})
   target_include_directories(${TOOL} 
     PRIVATE 
       $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
-      ${CMAKE_SOURCE_DIR}/src/lib/include
+      ${CMAKE_CURRENT_SOURCE_DIR}/src/lib/include
       ${XML_INCLUDE_DIRS}
       ${Boost_INCLUDE_DIR}
   )
@@ -80,7 +80,7 @@ target_link_options(las2e57 PUBLIC ${linker_flags})
 target_include_directories(las2e57
   PRIVATE 
     $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
-    ${CMAKE_SOURCE_DIR}/src/lib/include
+    ${CMAKE_CURRENT_SOURCE_DIR}/src/lib/include
     ${XML_INCLUDE_DIRS}
     ${Boost_INCLUDE_DIR}
 )
