--- a/cmake/SndFileChecks.cmake
+++ b/cmake/SndFileChecks.cmake
@@ -213,7 +213,7 @@ if (MSVC)
 	add_definitions (-D_CRT_SECURE_NO_WARNINGS -D_CRT_NONSTDC_NO_DEPRECATE)
 endif (MSVC)
 
-if (MINGW)
+if (FALSE)
 	if (CMAKE_C_COMPILER_ID STREQUAL GNU)
 		set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -static-libgcc")
 		set (CMAKE_SHARED_LIBRARY_LINK_C_FLAGS "${CMAKE_SHARED_LIBRARY_LINK_C_FLAGS} -static-libgcc -s")
