diff --git a/src/lib/pubkey/ecdh/ecdh.cpp b/src/lib/pubkey/ecdh/ecdh.cpp
index 2995afee4..c0e1d3caf 100644
--- a/src/lib/pubkey/ecdh/ecdh.cpp
+++ b/src/lib/pubkey/ecdh/ecdh.cpp
@@ -33,20 +33,33 @@ class ECDH_KA_Operation final : public PK_Ops::Key_Agreement_with_KDF {
       size_t agreed_value_size() const override { return m_group.get_p_bytes(); }
 
       secure_vector<uint8_t> raw_agree(const uint8_t w[], size_t w_len) override {
-         if(m_group.has_cofactor()) {
+         const auto input_point = [&] {
+            if(m_group.has_cofactor()) {
 #if defined(BOTAN_HAS_LEGACY_EC_POINT)
-            EC_AffinePoint input_point(m_group, m_group.get_cofactor() * m_group.OS2ECP(w, w_len));
-            return input_point.mul_x_only(m_l_times_priv, m_rng, m_ws);
+               return EC_AffinePoint(m_group, m_group.get_cofactor() * m_group.OS2ECP(w, w_len));
 #else
-            throw Not_Implemented("Support for DH with cofactor adjustment not available in this build configuration");
+               throw Not_Implemented(
+                  "Support for DH with cofactor adjustment not available in this build configuration");
 #endif
-         } else {
-            if(auto input_point = EC_AffinePoint::deserialize(m_group, {w, w_len})) {
-               return input_point->mul_x_only(m_l_times_priv, m_rng, m_ws);
             } else {
-               throw Decoding_Error("ECDH - Invalid elliptic curve point");
+               if(auto point = EC_AffinePoint::deserialize(m_group, {w, w_len})) {
+                  return *point;
+               } else {
+                  throw Decoding_Error("ECDH - Invalid elliptic curve point: not on curve");
+               }
             }
+         }();
+
+         // Typical specs (such as BSI's TR-03111 Section 4.3.1) require that
+         // we check the resulting point of the multiplication to not be the
+         // point at infinity. However, since we ensure that our ECC private
+         // scalar can never be zero, checking the peer's input point is
+         // equivalent.
+         if(input_point.is_identity()) {
+            throw Decoding_Error("ECDH - Invalid elliptic curve point: identity");
          }
+
+         return input_point.mul_x_only(m_l_times_priv, m_rng, m_ws);
       }
 
    private:
diff --git a/src/tests/test_ecdh.cpp b/src/tests/test_ecdh.cpp
index d6b64a639..5461735fd 100644
--- a/src/tests/test_ecdh.cpp
+++ b/src/tests/test_ecdh.cpp
@@ -83,6 +83,14 @@ class ECDH_AllGroups_Tests : public Test {
                   Botan::ECDH_PublicKey(group, infinity);
                });
 
+               // Regression test: prohibit ECDH-agreement with all-zero public value
+               result.test_throws<Botan::Decoding_Error>("ECDH public value is point-at-infinity", [&] {
+                  const auto sk = Botan::ECDH_PrivateKey(rng(), group);
+                  Botan::PK_Key_Agreement ka(sk, rng(), kdf);
+                  const auto sec1_infinity = std::array{uint8_t(0x00)};
+                  const auto a_ss = ka.derive_key(0, sec1_infinity);
+               });
+
                for(size_t i = 0; i != 100; ++i) {
                   const Botan::ECDH_PrivateKey a_priv(rng(), group);
                   const auto a_pub = a_priv.public_value();
-- 
2.43.0

