From e64fa20d4f56dc5ab12593f014d033fbdbb04a65 Mon Sep 17 00:00:00 2001
From: Victor Zverovich <victor.zverovich@gmail.com>
Date: Sat, 16 Jan 2021 08:05:02 -0800
Subject: [PATCH] Install fmt/args.h (#2096)

Signed-off-by: Alejandro Colomar <alejandro.colomar@exfo.com>
---
 CMakeLists.txt | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f21cf456..8798614b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -183,8 +183,9 @@ function(add_headers VAR)
 endfunction()
 
 # Define the fmt library, its includes and the needed defines.
-add_headers(FMT_HEADERS chrono.h color.h compile.h core.h format.h format-inl.h
-                        locale.h os.h ostream.h posix.h printf.h ranges.h)
+add_headers(FMT_HEADERS args.h chrono.h color.h compile.h core.h format.h
+                        format-inl.h locale.h os.h ostream.h posix.h printf.h
+                        ranges.h)
 if (FMT_OS)
   set(FMT_SOURCES src/format.cc src/os.cc)
 else()
-- 
2.32.0.rc0

