--- a/include/fmt/core.h
+++ b/include/fmt/core.h
@@ -2783,7 +2783,7 @@ class basic_format_string {
     if constexpr (detail::count_named_args<Args...>() ==
                   detail::count_statically_named_args<Args...>()) {
       using checker =
           detail::format_string_checker<Char, remove_cvref_t<Args>...>;
-      detail::parse_format_string<true>(str_, checker(s));
+      detail::parse_format_string<true>(str_, checker(str_));
     }
 #else
     detail::check_format_string<Args...>(s);
