diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6b39f0e4..76f69653 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -86,8 +86,7 @@ IF(MSVC)
 ENDIF()
 
 #Add custom CMake Module
-SET(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules/"
-  CACHE INTERNAL "Location of our custom CMake modules.")
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules/")
 
 # auto-configure style checks, other CMake modules.
 INCLUDE(CheckLibraryExists)
@@ -645,7 +644,7 @@ IF(USE_HDF5)
     # examples, even though the previous version of what we
     # had worked.
     #####
-    IF(MSVC)
+    IF(0)
       SET(SEARCH_PACKAGE_NAME ${HDF5_PACKAGE_NAME})
       FIND_PACKAGE(HDF5 NAMES ${SEARCH_PACKAGE_NAME} COMPONENTS C HL CONFIG REQUIRED ${NC_HDF5_LINK_TYPE})
     ELSE(MSVC)
@@ -848,7 +847,6 @@ ENDIF(USE_HDF5)
 
 # See if we have libcurl
 FIND_PACKAGE(CURL)
-ADD_DEFINITIONS(-DCURL_STATICLIB=1)
 INCLUDE_DIRECTORIES(${CURL_INCLUDE_DIRS})
 
 # Define a test flag for have curl library
@@ -2028,7 +2026,7 @@ ADD_SUBDIRECTORY(docs)
 # install them in the binary dir. Grab all of the .libs, put them
 # in the libdir.
 ##
-IF(MSVC)
+IF(0)
   FILE(GLOB COPY_FILES ${CMAKE_PREFIX_PATH}/lib/*.lib)
   INSTALL(FILES ${COPY_FILES}
   DESTINATION ${CMAKE_INSTALL_LIBDIR}
diff --git a/liblib/CMakeLists.txt b/liblib/CMakeLists.txt
index be72612c..e0edb4ca 100644
--- a/liblib/CMakeLists.txt
+++ b/liblib/CMakeLists.txt
@@ -90,14 +90,14 @@ IF(USE_HDF5)
     # builds:
     # Make sure that HDF5_C_LIBRARY appears *after*
     # HDF5_HL_LIBRARY.
-    SET(TLL_LIBS ${HDF5_HL_LIBRARIES} ${HDF5_C_LIBRARIES} ${TLL_LIBS} ${SZIP_LIBRARY})
+    SET(TLL_LIBS ${HDF5_HL_LIBRARIES} ${HDF5_LIBRARIES} ${TLL_LIBS} ${SZIP_LIBRARY})
   ELSE() # Windows CMake defines HDF5_LIBRARIES.
     SET(TLL_LIBS ${HDF5_LIBRARIES} ${TLL_LIBS} ${SZIP_LIBRARY})
   ENDIF()
 ENDIF()
 
 IF(FOUND_CURL)
-  SET(TLL_LIBS ${TLL_LIBS} ${CURL_LIBRARY})
+   SET(TLL_LIBS ${TLL_LIBS} ${CURL_LIBRARIES})
 ENDIF()
 
 IF(USE_HDF4)
