diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0c75bbb3..f87d2027 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -75,6 +75,10 @@ target_compile_features(cppserver PUBLIC cxx_std_17)
 list(APPEND INSTALL_TARGETS cppserver)
 list(APPEND LINKLIBS cppserver)
 
+if(WIN32)
+    target_compile_definitions (cppserver PUBLIC -D_WIN32_WINNT=_WIN32_WINNT_WIN7)
+endif()
+
 # Additional module components: benchmarks, examples, plugins, tests, tools and install
 if(NOT CPPSERVER_MODULE)
 
