diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8616cb6..7d26a40 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -45,6 +45,10 @@ target_compile_features(cppserver PUBLIC cxx_std_17)
 list(APPEND INSTALL_TARGETS cppserver)
 list(APPEND LINKLIBS cppserver)
 
+if(WIN32 AND NOT MSVC)
+    target_compile_definitions(cppserver PUBLIC -D_WIN32_WINNT=_WIN32_WINNT_VISTA)
+endif()
+
 # Additional module components: benchmarks, examples, plugins, tests, tools and install
 if(NOT CPPSERVER_MODULE)
 
