diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6bd5cc3..52a7d05 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -48,10 +48,5 @@ endif()

 set_property(GLOBAL PROPERTY USE_FOLDERS ON)

-# Prohibit in-source build
-if("${CMAKE_SOURCE_DIR}" STREQUAL "${CMAKE_BINARY_DIR}")
-  message(FATAL_ERROR "In-source build prohibited.")
-endif("${CMAKE_SOURCE_DIR}" STREQUAL "${CMAKE_BINARY_DIR}")
-
 # Hide some cache variables
 mark_as_advanced(EXECUTABLE_OUTPUT_PATH LIBRARY_OUTPUT_PATH)
