--- a/src/main/include/CMakeLists.txt
+++ b/src/main/include/CMakeLists.txt
@@ -68,16 +68,27 @@ include(CheckIncludeFiles)
 include(CheckIncludeFileCXX)
 include(CheckLibraryExists)
 
-if(WIN32)
-	CHECK_INCLUDE_FILES(sqlext.h HAS_ODBC)
-else()
-	include(FindPkgConfig)
+option(LOG4CXX_ENABLE_ODBC "Support logging via ODBC" OFF)
+if(LOG4CXX_ENABLE_ODBC)
+  if(WIN32)
+    CHECK_INCLUDE_FILES(sqlext.h HAS_ODBC)
+  else()
+    find_package(ODBC)
 
-	pkg_check_modules( odbc QUIET odbc )
-	if(${odbc_FOUND})
-		set(HAS_ODBC 1)
-	endif(${odbc_FOUND})
-endif(WIN32)
+    
+    if(${ODBC_FOUND})
+      set(HAS_ODBC 1)
+    else()
+      set(HAS_ODBC 0)
+    endif(${ODBC_FOUND})
+  endif(WIN32)
+
+  if(NOT ${HAS_ODBC})
+    message(SEND_ERROR "ODBC not found but requested")
+  endif()
+else()
+  set(HAS_ODBC 0)
+endif(LOG4CXX_ENABLE_ODBC)
 
 CHECK_INCLUDE_FILE_CXX(locale HAS_STD_LOCALE)
 CHECK_FUNCTION_EXISTS(mbsrtowcs HAS_MBSRTOWCS)
