diff --git a/src/tools/pgflex b/src/tools/pgflex
index baabe2d..f309c1a 100755
--- a/src/tools/pgflex
+++ b/src/tools/pgflex
@@ -51,7 +51,8 @@ os.chdir(args.privatedir)
 # contents. Set FLEX_TMP_DIR to the target private directory to avoid
 # that. That environment variable isn't consulted on other platforms, so we
 # don't even need to make this conditional.
-env = {'FLEX_TMP_DIR': args.privatedir}
+env = os.environ.copy()
+env['FLEX_TMP_DIR'] =  args.privatedir
 
 # build flex invocation
 command = [args.flex, '-o', args.output_file]
