diff -ur a/gio/meson.build b/gio/meson.build
--- a/gio/meson.build	2020-09-10 12:42:41.546684000 +0200
+++ b/gio/meson.build	2020-09-15 10:45:38.897447913 +0200
@@ -874,14 +874,14 @@
 
 # Dependencies used by executables below
 have_libelf = false
-libelf = dependency('libelf', version : '>= 0.8.12', required : false)
+libelf = dependency('libelf', version : '>= 0.8.12', required : get_option ('libelf'))
 if libelf.found()
   have_libelf = true
 else
   # This fallback is necessary on *BSD. elfutils isn't the only libelf
   # implementation, and *BSD usually includes their own libelf as a system
   # library which doesn't have a corresponding .pc file.
-  libelf = cc.find_library('elf', required : false)
+  libelf = cc.find_library('elf', required : get_option ('libelf'))
   have_libelf = libelf.found()
   have_libelf = have_libelf and cc.has_function('elf_begin', dependencies : libelf)
   have_libelf = have_libelf and cc.has_function('elf_getshdrstrndx', dependencies : libelf)
diff -ur a/meson_options.txt b/meson_options.txt
--- a/meson_options.txt	2020-09-10 12:42:41.673685600 +0200
+++ b/meson_options.txt	2020-09-15 10:44:18.049745441 +0200
@@ -111,3 +111,8 @@
        value : true,
        yield : true,
        description : 'Enable GLib checks such as API guards (see docs/macros.txt)')
+
+option('libelf',
+       type : 'feature',
+       value : 'auto',
+       description : 'Use libelf')
