--- meson_org.build	2023-07-28 11:23:16.976552300 +0200
+++ meson.build	2023-07-28 11:24:04.173300500 +0200
@@ -905,6 +905,11 @@
 
 if host_system == 'qnx'
   glib_conf.set('HAVE_QNX', 1)
+  if meson.get_external_property('qnx_version', false) == '7.0'
+    glib_conf.set('HAVE_QNX700', 1)
+  elif meson.get_external_property('qnx_version', false) == '7.1'
+    glib_conf.set('HAVE_QNX710', 1)
+  endif
 endif
 
 # Check for futex(2)
