--- meson_org.build	2023-07-28 08:17:35.282631200 +0200
+++ meson.build	2023-07-28 08:35:13.238165100 +0200
@@ -891,6 +891,11 @@
 
 if host_system == 'qnx'
   glib_conf.set('HAVE_QNX', 1)
+  if meson.get_external_property('qnx_version', '') == '7.0'
+    glib_conf.set('HAVE_QNX700', 1)
+  elif meson.get_external_property('qnx_version', '') == '7.1'
+    glib_conf.set('HAVE_QNX710', 1)
+  endif
 endif
 
 # Check for futex(2)
