diff --git a/inc/lift/impl/pragma.hpp b/inc/lift/impl/pragma.hpp
index 83d7e11..8521f94 100644
--- a/inc/lift/impl/pragma.hpp
+++ b/inc/lift/impl/pragma.hpp
@@ -5,6 +5,7 @@
     #define DISABLE_WARNING(warningName) DO_PRAGMA(GCC diagnostic ignored #warningName)
 
     #define DISABLE_WARNING_MAYBE_UNINITIALIZED DISABLE_WARNING(-Wmaybe-uninitialized)
+    #define DISABLE_WARNING_SWITCH_ENUM         DISABLE_WARNING(-Wswitch-enum)
 #elif defined(__clang__)
     #define DO_PRAGMA(X) _Pragma(#X)
     #define DISABLE_WARNING_PUSH
@@ -12,9 +13,12 @@
     #define DISABLE_WARNING(warningNumber)
 
     #define DISABLE_WARNING_MAYBE_UNINITIALIZED
+    #define DISABLE_WARNING_SWITCH_ENUM
 #else
     // unknown compiler, ignoring suppression directives
     #define DISABLE_WARNING_PUSH
     #define DISABLE_WARNING_MAYBE_UNINITIALIZED
     #define DISABLE_WARNING_POP
+
+    #define DISABLE_WARNING_SWITCH_ENUM
 #endif
diff --git a/src/executor.cpp b/src/executor.cpp
index ccb9599..dfcfbf0 100644
--- a/src/executor.cpp
+++ b/src/executor.cpp
@@ -486,7 +486,7 @@ auto executor::reset() -> void
 auto executor::convert(CURLcode curl_code) -> lift_status
 {
     DISABLE_WARNING_PUSH
-    DISABLE_WARNING(-Wswitch-enum)
+    DISABLE_WARNING_SWITCH_ENUM
     switch (curl_code)
     {
         case CURLcode::CURLE_OK:
