diff --git a/CMakeLists.txt b/CMakeLists.txt
index a83304e..007cc33 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -59,7 +59,7 @@ set(LIBLIFTHTTP_SOURCE_FILES
     inc/lift/share.hpp src/share.cpp
 )
 
-add_library(${PROJECT_NAME} STATIC ${LIBLIFTHTTP_SOURCE_FILES})
+add_library(${PROJECT_NAME} ${LIBLIFTHTTP_SOURCE_FILES})
 set_target_properties(${PROJECT_NAME} PROPERTIES LINKER_LANGUAGE CXX)
 target_compile_features(${PROJECT_NAME} PUBLIC cxx_std_17)
 
