--- Makefile	2019-10-22 08:11:21.000000000 -0700
+++ Makefile_fix	2019-12-20 10:58:06.255975751 -0800
@@ -1,15 +1,15 @@
 # Variables:
-prefix = /usr/local
+#prefix = /usr/local
 exec_prefix = $(prefix)
 bindir = $(exec_prefix)/bin
 libdir = $(exec_prefix)/lib
 includedir = $(prefix)/include
 pkgconfigdir = $(libdir)/pkgconfig
 
-CFLAGS = -Wall -Wextra
-LDFLAGS_GCSECTIONS = -Wl,--gc-sections
+CFLAGS += -Wall -Wextra
+LDFLAGS_GCSECTIONS += -Wl,--gc-sections
 LDFLAGS += $(if $(gc_sections-pass),$(LDFLAGS_GCSECTIONS))
-LDLIBS = -lm
+LDLIBS += -lm
 
 GIT_ARCHIVE = git archive --prefix="$(PACKAGE)/" -9
 INSTALL = install
@@ -31,8 +31,8 @@
 #
 CBORDUMP_SOURCES = tools/cbordump/cbordump.c
 
-BUILD_SHARED = $(shell file -L /bin/sh 2>/dev/null | grep -q ELF && echo 1)
-BUILD_STATIC = 1
+BUILD_SHARED ?= $(shell file -L /bin/sh 2>/dev/null | grep -q ELF && echo 1)
+BUILD_STATIC ?= 1
 
 ifneq ($(BUILD_STATIC),1)
 ifneq ($(BUILD_SHARED),1)
