diff --git a/src/dawn/native/CMakeLists.txt b/src/dawn/native/CMakeLists.txt
--- a/src/dawn/native/CMakeLists.txt
+++ b/src/dawn/native/CMakeLists.txt
@@ -572,7 +572,7 @@
         SCRIPT "${Dawn_SOURCE_DIR}/generator/opengl_loader_generator.py"
         PRINT_NAME "OpenGL function loader"
         EXTRA_PARAMETERS "--gl-xml"
-             "${Dawn_SOURCE_DIR}/third_party/khronos/OpenGL-Registry/xml/gl.xml"
+             "${OPENGL_REGISTRY_DATA_DIR}/xml/gl.xml"
              "--supported-extensions"
              "${Dawn_SOURCE_DIR}/src/dawn/native/opengl/supported_extensions.json"
         OUTPUT_HEADERS DAWN_NATIVE_OPENGL_AUTOGEN_HEADERS
diff --git a/third_party/CMakeLists.txt b/third_party/CMakeLists.txt
--- a/third_party/CMakeLists.txt
+++ b/third_party/CMakeLists.txt
@@ -25,6 +25,14 @@
 # OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
+find_package(absl REQUIRED CONFIG GLOBAL)
+find_package(SPIRV-Headers REQUIRED CONFIG GLOBAL)
+#find_package(SPIRV-Tools REQUIRED CONFIG GLOBAL)
+find_package(glslang REQUIRED CONFIG GLOBAL)
+find_package(glfw3 REQUIRED CONFIG GLOBAL)
+find_package(Vulkan-Headers REQUIRED CONFIG GLOBAL)
+find_package(VulkanUtilityLibraries REQUIRED CONFIG GLOBAL)
+
 # Don't build testing in third_party dependencies
 set(BUILD_TESTING OFF)
 
@@ -85,7 +93,7 @@
     add_subdirectory(${DAWN_SPIRV_HEADERS_DIR} "${CMAKE_CURRENT_BINARY_DIR}/spirv-headers")
 endif()
 
-if (NOT TARGET SPIRV-Tools AND
+if (
     (((DAWN_ENABLE_OPENGL OR DAWN_ENABLE_VULKAN)
         AND DAWN_ENABLE_SPIRV_VALIDATION)
         OR TINT_BUILD_SPV_READER
@@ -132,8 +140,8 @@
 if (DAWN_ENABLE_DESKTOP_GL OR DAWN_ENABLE_OPENGLES)
     # Header-only library for khrplatform.h
     add_library(dawn_khronos_platform INTERFACE)
-    target_sources(dawn_khronos_platform INTERFACE "${CMAKE_CURRENT_SOURCE_DIR}/khronos/EGL-Registry/api/KHR/khrplatform.h")
-    target_include_directories(dawn_khronos_platform INTERFACE "${CMAKE_CURRENT_SOURCE_DIR}/khronos/EGL-Registry/api")
+    find_package(opengl-registry REQUIRED CONFIG)
+    target_link_libraries(dawn_khronos_platform INTERFACE opengl-registry::opengl-registry)
 endif()
 
 if (NOT TARGET Vulkan-Headers AND DAWN_ENABLE_VULKAN)
