diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7fafe77cc..d5c4d701e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -623,7 +623,7 @@ if (srt_libspec_shared)
 	set_target_properties (${TARGET_srt}_shared PROPERTIES VERSION ${SRT_VERSION} SOVERSION ${SRT_VERSION_MAJOR})
 	list (APPEND INSTALL_TARGETS ${TARGET_srt}_shared)
 	if (ENABLE_ENCRYPTION)
-		target_link_libraries(${TARGET_srt}_shared PRIVATE ${SSL_LIBRARIES})
+		target_link_libraries(${TARGET_srt}_shared PRIVATE ${CONAN_LIBS_OPENSSL})
 	endif()
 	if (MICROSOFT)
 		target_link_libraries(${TARGET_srt}_shared PRIVATE ws2_32.lib)
@@ -657,7 +657,7 @@ if (srt_libspec_static)
 
 	list (APPEND INSTALL_TARGETS ${TARGET_srt}_static)
 	if (ENABLE_ENCRYPTION)
-		target_link_libraries(${TARGET_srt}_static PRIVATE ${SSL_LIBRARIES})
+		target_link_libraries(${TARGET_srt}_static PRIVATE ${CONAN_LIBS_OPENSSL})
 	endif()
 	if (MICROSOFT)
 		target_link_libraries(${TARGET_srt}_static PRIVATE ws2_32.lib)
 
