diff --git a/include/glaze/core/common.hpp b/include/glaze/core/common.hpp
index aed6ce6..6cf4ace 100644
--- a/include/glaze/core/common.hpp
+++ b/include/glaze/core/common.hpp
@@ -12,8 +12,8 @@
 #include <filesystem>
 #include <fstream>
 
-#include "glaze/frozen/string.hpp"
-#include "glaze/frozen/unordered_map.hpp"
+#include "frozen/string.h"
+#include "frozen/unordered_map.h"
 
 #include "glaze/core/context.hpp"
 #include "glaze/core/meta.hpp"
diff --git a/include/glaze/util/hash_map.hpp b/include/glaze/util/hash_map.hpp
index eaaec57..ae75ee2 100644
--- a/include/glaze/util/hash_map.hpp
+++ b/include/glaze/util/hash_map.hpp
@@ -4,7 +4,7 @@
 #include <span>
 #include <vector>
 
-#include "glaze/frozen/random.hpp"
+#include "frozen/random.h"
 #include "glaze/util/string_cmp.hpp"
 
 namespace glz
