diff --git a/CMakeLists.txt b/CMakeLists.txt
index 491ae46..c0a4f3d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,5 +1,5 @@
 cmake_minimum_required(VERSION 3.12)
-list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmakeUtils;")
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmakeUtils;")
 include(setup)
 
 project(nativeipc)
diff --git a/cmakeUtils/setup.cmake b/cmakeUtils/setup.cmake
--- cmakeUtils/setup.cmake
+++ cmakeUtils/setup.cmake
@@ -1,6 +1,4 @@
 include(utilities)
-include(conan)
-include(conanutils)
 
 macro(setup_project)
     if(NOT CMAKE_BUILD_TYPE)
@@ -13,13 +11,9 @@
     option(ENABLE_CODE_DOCUMENTATION "enable code documenation generation" OFF)
     option(MSVC_DYNAMIC_RUNTIME "change default runtime" ON)
 
-    conan_check(VERSION 1.22.2 REQUIRED)
-
     setup_default_cxx_compile_options()
 
     set_output_directories(${CMAKE_BINARY_DIR}/bin)
-
-    run_conan()
 
     include(CTest)
     if(BUILD_TESTING)
diff --git a/cmakeUtils/unix.cmake b/cmakeUtils/unix.cmake
index 5f4bb4c..9b2856a 100644
--- a/cmakeUtils/unix.cmake
+++ b/cmakeUtils/unix.cmake
@@ -1,7 +1,7 @@
 
 macro(setup_default_cxx_compile_options)
     message("Setting up Unix-Like default settings")
-    set(unix_flags "-Wall -Wextra -Werror -Wformat=2 -Wold-style-cast -Wnull-dereference -Wno-missing-field-initializers")
+    set(unix_flags " -Wall -Wextra -Werror -Wformat=2 -Wold-style-cast -Wnull-dereference -Wno-missing-field-initializers")
     string(APPEND CMAKE_CXX_FLAGS ${unix_flags})
     string(APPEND CMAKE_C_FLAGS ${unix_flags})
 endmacro()
diff --git a/libnativeipc/CMakeLists.txt b/libnativeipc/CMakeLists.txt
index fe5cc30..173a156 100644
--- a/libnativeipc/CMakeLists.txt
+++ b/libnativeipc/CMakeLists.txt
@@ -59,8 +59,9 @@
   include/nativeipc
   )
 
+find_package(libuv REQUIRED CONFIG)
 target_link_libraries(nativeipc PRIVATE
-  CONAN_PKG::libuv)
+  uv)
 
 target_compile_definitions(nativeipc PRIVATE)
 
@@ -81,7 +82,9 @@
 endif()
 
 install(TARGETS nativeipc
+  LIBRARY DESTINATION lib
   ARCHIVE DESTINATION lib
+  RUNTIME DESTINATION bin
   )
 
 install(DIRECTORY include
