diff --git a/CMakeLists.txt b/CMakeLists.txt
index 491ae46..c0a4f3d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,5 +1,5 @@
 cmake_minimum_required(VERSION 3.12)
-list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmakeUtils;")
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmakeUtils;")
 include(setup)
 
 project(nativeipc)
diff --git a/cmakeUtils/conanutils.cmake b/cmakeUtils/conanutils.cmake
index cc6c07c..41dc487 100644
--- a/cmakeUtils/conanutils.cmake
+++ b/cmakeUtils/conanutils.cmake
@@ -59,6 +59,10 @@ macro(run_conan)
       set(_keeprpath "KEEP_RPATHS")
     endif()
 
+	if(CONAN_EXPORTED)
+	  set(noload "NO_LOAD")
+	endif()
+
     conan_cmake_run(CONANFILE conan/conanfile.py
         BASIC_SETUP
         CMAKE_TARGETS
@@ -70,5 +74,6 @@ macro(run_conan)
         RELEASE_PROFILE ${_profile}
         RELWITHDEBINFO_PROFILE ${_profile}
         ENV "CONAN_CMAKE_BINARY_DIR_PATH=${CMAKE_BINARY_DIR}"
+		${noload}
         )
 endmacro()
