diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 399b37d..55fee4f 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -108,29 +108,10 @@ INSTALL(FILES MQTTAsync.h MQTTClient.h MQTTClientPersistence.h MQTTProperties.h
 IF (PAHO_WITH_SSL)
     SET(OPENSSL_SEARCH_PATH "" CACHE PATH "Directory containing OpenSSL libraries and includes")
 
-    IF (${CMAKE_SYSTEM_NAME} STREQUAL "Darwin")
-      SET(OPENSSL_SEARCH_PATH "/usr/local/opt/openssl")
-    ENDIF (${CMAKE_SYSTEM_NAME} STREQUAL "Darwin")
-
-    IF (WIN32)
-      SET(OPENSSL_SEARCH_PATH "C:/OpenSSL-Win64")
-      SET(OPENSSL_SEARCH_LIB_PATH "${OPENSSL_SEARCH_PATH}/lib64")
-      IF (DEFINED ENV{PAHO_WINDOWS_BUILD_BIT})
-        IF ($ENV{PAHO_WINDOWS_BUILD_BIT} STREQUAL "x86")
-          SET(OPENSSL_SEARCH_PATH "C:/OpenSSL-Win32")
-          SET(OPENSSL_SEARCH_LIB_PATH "${OPENSSL_SEARCH_PATH}/lib32")
-        ENDIF ()
-      ENDIF ()
-    ELSE()
-      SET(OPENSSL_SEARCH_LIB_PATH "${OPENSSL_SEARCH_PATH}/lib64")
-    ENDIF ()
-
-    FIND_PATH(OPENSSL_INCLUDE_DIR openssl/ssl.h
-        HINTS ${OPENSSL_SEARCH_PATH}/include)
-    FIND_LIBRARY(OPENSSL_LIB NAMES ssl libssl ssleay32
-        HINTS ${OPENSSL_SEARCH_PATH}/lib ${OPENSSL_SEARCH_LIB_PATH})
-    FIND_LIBRARY(OPENSSLCRYPTO_LIB NAMES crypto libcrypto libeay32
-      	HINTS ${OPENSSL_SEARCH_PATH}/lib ${OPENSSL_SEARCH_LIB_PATH})
+    find_package(OpenSSL REQUIRED CONFIG)
+    
+    SET(OPENSSL_LIB ${OPENSSL_SSL_LIBRARY})
+    SET(OPENSSLCRYPTO_LIB ${OPENSSL_CRYPTO_LIBRARY})
 
     MESSAGE(STATUS "OpenSSL hints: ${OPENSSL_SEARCH_PATH}")
     MESSAGE(STATUS "OpenSSL headers found at ${OPENSSL_INCLUDE_DIR}")
@@ -151,6 +132,8 @@ IF (PAHO_WITH_SSL)
 
     TARGET_LINK_LIBRARIES(paho-mqtt3cs ${OPENSSL_LIB} ${OPENSSLCRYPTO_LIB} ${LIBS_SYSTEM})
     TARGET_LINK_LIBRARIES(paho-mqtt3as ${OPENSSL_LIB} ${OPENSSLCRYPTO_LIB} ${LIBS_SYSTEM})
+    target_link_directories(paho-mqtt3cs PRIVATE "${OPENSSL_ROOT_DIR}/lib")
+    target_link_directories(paho-mqtt3as PRIVATE "${OPENSSL_ROOT_DIR}/lib")
     SET_TARGET_PROPERTIES(
         paho-mqtt3cs paho-mqtt3as PROPERTIES
         VERSION ${CLIENT_VERSION}
