diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 52cea7c..f798247 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -177,7 +177,7 @@ IF (NOT PAHO_WITH_SSL)
             TARGET_LINK_LIBRARIES(paho-mqtt3a-static ${LIBS_SYSTEM})
             
             IF (NOT WIN32)
-                SET_TARGET_PROPERTIES(paho-mqtt3a-static PROPERTIES OUTPUT_NAME paho-mqtt3a)
+                SET_TARGET_PROPERTIES(paho-mqtt3a-static PROPERTIES OUTPUT_NAME paho-mqtt3a-static)
             ENDIF()       
             SET_TARGET_PROPERTIES(
                 paho-mqtt3a-static PROPERTIES
@@ -210,7 +210,7 @@ IF (NOT PAHO_WITH_SSL)
             TARGET_LINK_LIBRARIES(paho-mqtt3c-static ${LIBS_SYSTEM})
     
             IF (NOT WIN32)
-                SET_TARGET_PROPERTIES(paho-mqtt3c-static PROPERTIES OUTPUT_NAME paho-mqtt3c)
+                SET_TARGET_PROPERTIES(paho-mqtt3c-static PROPERTIES OUTPUT_NAME paho-mqtt3c-static)
             ENDIF()       
             SET_TARGET_PROPERTIES(
                 paho-mqtt3c-static PROPERTIES
@@ -351,7 +351,7 @@ IF (PAHO_WITH_SSL)
                 SOVERSION ${PAHO_VERSION_MAJOR}
                 COMPILE_DEFINITIONS "OPENSSL=1;PAHO_MQTT_STATIC=1")
             IF (NOT WIN32)
-                SET_TARGET_PROPERTIES(paho-mqtt3as-static PROPERTIES OUTPUT_NAME paho-mqtt3as)
+                SET_TARGET_PROPERTIES(paho-mqtt3as-static PROPERTIES OUTPUT_NAME paho-mqtt3as-static)
             ENDIF()
     
     	    IF(${CMAKE_SYSTEM_NAME} STREQUAL "Darwin")
@@ -396,7 +396,7 @@ IF (PAHO_WITH_SSL)
                 SOVERSION ${PAHO_VERSION_MAJOR}
                 COMPILE_DEFINITIONS "OPENSSL=1;PAHO_MQTT_STATIC=1")
             IF (NOT WIN32)
-                SET_TARGET_PROPERTIES(paho-mqtt3cs-static PROPERTIES OUTPUT_NAME paho-mqtt3cs)
+                SET_TARGET_PROPERTIES(paho-mqtt3cs-static PROPERTIES OUTPUT_NAME paho-mqtt3cs-static)
             ENDIF()
     
             IF(${CMAKE_SYSTEM_NAME} STREQUAL "Darwin")
