--- src/pwl.c
+++ src/pwl.c
@@ -59,8 +59,10 @@
 #include <string.h>
 #include <sys/types.h>
 #include <sys/stat.h>
-#include <sys/file.h>
 #include <fcntl.h>
+#ifndef _WIN32
+# include <sys/file.h>
+#endif
 
 #include <glib.h>
 #include <glib/gstdio.h>
@@ -182,8 +184,13 @@
 
 static int edit_dist(const char* word1, const char* word2);
 
-#define enchant_lock_file(f) flock (fileno (f), LOCK_EX)
-#define enchant_unlock_file(f) flock (fileno (f), LOCK_UN)
+#ifdef HAVE_FLOCK
+# define enchant_lock_file(f) flock (fileno (f), LOCK_EX)
+# define enchant_unlock_file(f) flock (fileno (f), LOCK_UN)
+#else
+# define enchant_lock_file(f)
+# define enchant_unlock_file(f)
+#endif
 
 /**
  * enchant_pwl_init
