diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2ceabf7..0c05298 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -446,7 +446,7 @@ install(
 ################################################################################
 # Asmjit Target
 ################################################################################
-
+find_package(asmjit CONFIG REQUIRED)
 if(NOT TARGET asmjit)
   # Download asmjit from github if ASMJIT_SRC_DIR is not specified.
   if(NOT DEFINED ASMJIT_SRC_DIR)
@@ -468,7 +468,7 @@ endif()
 ################################################################################
 # Cpuinfo Target
 ################################################################################
-
+find_package(cpuinfo CONFIG REQUIRED)
 if(NOT TARGET cpuinfo)
   #Download cpuinfo from github if CPUINFO_SOURCE_DIR is not specified.
   if(NOT DEFINED CPUINFO_SOURCE_DIR)
diff --git a/cmake/modules/CppLibrary.cmake b/cmake/modules/CppLibrary.cmake
index 92a93a6..abd08a2 100644
--- a/cmake/modules/CppLibrary.cmake
+++ b/cmake/modules/CppLibrary.cmake
@@ -96,7 +96,6 @@ function(cpp_library)
             -Wno-deprecated-declarations
             -Wall
             -Wextra
-            -Werror
             -Wunknown-pragmas
             -Wimplicit-fallthrough
             -Wno-strict-aliasing
