--- a/src/FbgemmSparseDense.cc
+++ b/src/FbgemmSparseDense.cc
@@ -72,6 +72,9 @@ fbgemmDenseToBCSR(int R, int C, const T* inp) {
   return fbgemmDenseToBCSR<T, RB, CB>(R, C, inp, C);
 }
 
+template <typename T, int RB, int CB>
+constexpr int BCSRMatrix<T, RB, CB>::COLTILE;
+
 template <typename T, int RB, int CB>
 void BCSRMatrix<T, RB, CB>::pack(const DTYPE* src, size_t ld) {
   rowBPtr.push_back(0);
