diff --git a/CMakeLists.txt b/CMakeLists.txt
index c45e569..5a1d800 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -12,23 +12,24 @@ else()
     link_libraries(pthread)
 endif()
 
-link_directories(.)
-include_directories(fmt/include)
+find_package(fmt REQUIRED CONFIG)
+
+if(BUILD_SHARED_LIBS)
 
 add_library(fmtlog-shared SHARED fmtlog.cc)
-if(MSVC)
-  target_link_libraries(fmtlog-shared fmt)
+if(1)
+  target_link_libraries(fmtlog-shared PUBLIC fmt::fmt)
 endif()
 install(TARGETS fmtlog-shared)
 
+else()
+
 add_library(fmtlog-static fmtlog.cc)
-if(MSVC)
-  target_link_libraries(fmtlog-static fmt)
+if(1)
+  target_link_libraries(fmtlog-static PUBLIC fmt::fmt)
 endif()
 install(TARGETS fmtlog-static)
 
-add_subdirectory(fmt)
-add_subdirectory(test)
-if(NOT MSVC)
-  add_subdirectory(bench)
 endif()
+
+install(FILES fmtlog.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/fmtlog)
