diff --git a/src/paths.cpp b/src/paths.cpp
index d0bc7f5..59aa70e 100644
--- a/src/paths.cpp
+++ b/src/paths.cpp
@@ -19,7 +19,7 @@ std::vector<std::string> Path::listFiles(std::string basePath, std::vector<std::
     std::vector<std::string> filesDirs;
     for (const auto &dirEntry: fs::recursive_directory_iterator(basePath)) {
         std::filesystem::path file = dirEntry.path();
-        if (contains != "" and ((std::string) file.filename()).find(contains) == std::string::npos) {
+        if (contains != "" and file.filename().string().find(contains) == std::string::npos) {
             continue;
         }
         if (validExts.empty()) {
