--- a/cmake/Findwebp.cmake
+++ b/cmake/Findwebp.cmake
@@ -43,7 +43,7 @@ if (webp_FOUND)
             INTERFACE_LINK_FLAGS "${webp_LINK_FLAGS}"
         )
     endif()
-    if (NOT TARGET WEBP::webpdemux)
+    if (NOT TARGET WebP::webpdemux)
         add_library(WebP::webpdemux UNKNOWN IMPORTED)
         set_target_properties(WebP::webpdemux PROPERTIES
             IMPORTED_LOCATION "${webpdemux_LIBRARY}"
