diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9b6d1eb..fd0e84f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -39,6 +39,7 @@ endif()
 
 #---- Find other packages ----
 find_package(Threads REQUIRED)
+find_package(minitrace REQUIRED)
 
 
 set(BEHAVIOR_TREE_LIBRARY ${PROJECT_NAME})
@@ -80,10 +81,6 @@ endif()
 
 add_subdirectory(3rdparty/lexy)
 
-add_library(minitrace STATIC 3rdparty/minitrace/minitrace.cpp)
-target_compile_definitions(minitrace PRIVATE MTR_ENABLED=True)
-set_property(TARGET minitrace PROPERTY POSITION_INDEPENDENT_CODE ON)
-
 list(APPEND BT_SOURCE
     src/action_node.cpp
     src/basic_types.cpp
@@ -166,7 +163,7 @@ target_link_libraries(${BTCPP_LIBRARY}
         Threads::Threads
         ${CMAKE_DL_LIBS}
         $<BUILD_INTERFACE:foonathan::lexy>
-        minitrace
+        minitrace::minitrace
     PUBLIC
         ${BTCPP_EXTRA_LIBRARIES}
 )
