diff --git a/tools/bt_recorder.cpp b/tools/bt_recorder.cpp
index 3aa6740..4b36414 100644
--- a/tools/bt_recorder.cpp
+++ b/tools/bt_recorder.cpp
@@ -16,12 +16,17 @@ static void s_signal_handler(int)
 
 static void CatchSignals(void)
 {
+#ifdef _WIN32
+    signal(SIGINT, s_signal_handler);
+    signal(SIGTERM, s_signal_handler);
+#else
     struct sigaction action;
     action.sa_handler = s_signal_handler;
     action.sa_flags = 0;
     sigemptyset(&action.sa_mask);
     sigaction(SIGINT, &action, NULL);
     sigaction(SIGTERM, &action, NULL);
+#endif
 }
 
 int main(int argc, char* argv[])
