diff --git a/CMakeLists.txt b/CMakeLists.txt
index a2f23cf..00cf414 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -67,14 +67,14 @@ endif()
 
 #---- Find other packages ----
 find_package(Threads)
-find_package(ZMQ)
+find_package(ZeroMQ)
 
 list(APPEND BEHAVIOR_TREE_PUBLIC_LIBRARIES
     ${CMAKE_THREAD_LIBS_INIT}
     ${CMAKE_DL_LIBS}
 )
 
-if( ZMQ_FOUND )
+if( ZeroMQ_FOUND )
     message(STATUS "ZeroMQ found.")
     add_definitions( -DZMQ_FOUND )
     list(APPEND BT_SOURCE src/loggers/bt_zmq_publisher.cpp)
@@ -140,7 +140,9 @@ endif()
 #############################################################
 # LIBRARY
 
-add_subdirectory(3rdparty/lexy)
+find_package(cppzmq REQUIRED CONFIG)
+find_package(lexy REQUIRED CONFIG)
+find_package(tinyxml2 REQUIRED CONFIG)
 
 list(APPEND BT_SOURCE
     src/action_node.cpp
@@ -175,9 +177,6 @@ list(APPEND BT_SOURCE
     src/loggers/bt_cout_logger.cpp
     src/loggers/bt_file_logger.cpp
     src/loggers/bt_minitrace_logger.cpp
-
-    3rdparty/tinyxml2/tinyxml2.cpp
-    3rdparty/minitrace/minitrace.cpp
     )
 
 if(BTCPP_MANUAL_SELECTOR)
@@ -221,8 +223,10 @@ target_link_libraries(${BEHAVIOR_TREE_LIBRARY}
         ${BEHAVIOR_TREE_PUBLIC_LIBRARIES}
     PRIVATE
         ${Boost_LIBRARIES}
-        ${ZMQ_LIBRARIES}
-        $<BUILD_INTERFACE:foonathan::lexy>
+        ${ZeroMQ_LIBRARIES}
+        cppzmq
+        foonathan::lexy
+        tinyxml2::tinyxml2
 )
 
 #get_target_property(my_libs ${BEHAVIOR_TREE_LIBRARY} INTERFACE_LINK_LIBRARIES)
@@ -239,12 +243`,10 @@ target_include_directories(${BEHAVIOR_TREE_LIBRARY} PUBLIC
     ${BUILD_TOOL_INCLUDE_DIRS})
 
 target_include_directories(${BEHAVIOR_TREE_LIBRARY} PRIVATE
-    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/3rdparty>
-    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/3rdparty/lexy/include>
     )
 
-if( ZMQ_FOUND )
-    target_compile_definitions(${BEHAVIOR_TREE_LIBRARY} PUBLIC ZMQ_FOUND)
+if( ZeroMQ_FOUND )
+    target_compile_definitions(${BEHAVIOR_TREE_LIBRARY} PUBLIC ZeroMQ_FOUND)
 endif()
 
 if(MSVC)
diff --git a/tools/CMakeLists.txt b/tools/CMakeLists.txt
index 163e703..b98f525 100644
--- a/tools/CMakeLists.txt
+++ b/tools/CMakeLists.txt
@@ -7,9 +7,9 @@ target_link_libraries(bt3_log_cat  ${BEHAVIOR_TREE_LIBRARY} )
 install(TARGETS bt3_log_cat
         DESTINATION ${BEHAVIOR_TREE_BIN_DESTINATION} )
 
-if( ZMQ_FOUND )
+if( ZeroMQ_FOUND )
     add_executable(bt3_recorder         bt_recorder.cpp )
-    target_link_libraries(bt3_recorder  ${BEHAVIOR_TREE_LIBRARY} ${ZMQ_LIBRARIES})
+    target_link_libraries(bt3_recorder  ${BEHAVIOR_TREE_LIBRARY} ${ZeroMQ_LIBRARIES})
     install(TARGETS bt3_recorder
             DESTINATION ${BEHAVIOR_TREE_BIN_DESTINATION} )
 endif()
diff --git a/src/loggers/bt_zmq_publisher.cpp b/src/loggers/bt_zmq_publisher.cpp
index 1325e0b..b269c2f 100644
--- a/src/loggers/bt_zmq_publisher.cpp
+++ b/src/loggers/bt_zmq_publisher.cpp
@@ -1,7 +1,7 @@
 #include <future>
 #include "behaviortree_cpp/loggers/bt_zmq_publisher.h"
 #include "behaviortree_cpp/flatbuffers/bt_flatbuffer_helper.h"
-#include "cppzmq/zmq.hpp"
+#include "zmq.hpp"

 namespace BT
 {
diff --git a/src/xml_parsing.cpp b/src/xml_parsing.cpp
index 18c2ed2..da71f11 100644
--- a/src/xml_parsing.cpp
+++ b/src/xml_parsing.cpp
@@ -24,7 +24,7 @@

 #include <map>
 #include "behaviortree_cpp/xml_parsing.h"
-#include "tinyxml2/tinyxml2.h"
+#include "tinyxml2.h"
 #include <filesystem>

 #ifdef USING_ROS
