From 78b159dfa617d86e107f08f116750d190ae0ee3c Mon Sep 17 00:00:00 2001
From: Uilian Ries <uilianries@gmail.com>
Date: Mon, 15 Dec 2025 18:31:41 +0100
Subject: [PATCH] Turn cppzmq dependency public

Signed-off-by: Uilian Ries <uilianries@gmail.com>
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c9528bda6..d44131e03 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -240,7 +240,7 @@ target_link_libraries(${BTCPP_LIBRARY}
 
 if(BTCPP_GROOT_INTERFACE)
     target_link_libraries(${BTCPP_LIBRARY}
-        PRIVATE
+        PUBLIC
             $<BUILD_INTERFACE:cppzmq>
     )
 endif()
