--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -43,11 +43,9 @@ target_compile_options(vk-bootstrap-compiler-warnings
         $<$<OR:$<CXX_COMPILER_ID:AppleClang>,$<CXX_COMPILER_ID:GNU>,${VK_BOOTSTRAP_COMPILER_CLANGPP}>:
         -Wall
         -Wextra
-        -pedantic-errors
         -Wconversion
         -Wsign-conversion>
         $<$<CXX_COMPILER_ID:MSVC>:
-        /WX
         /W4>
         )
 
@@ -71,7 +69,7 @@ install(TARGETS vk-bootstrap
 
 option(VK_BOOTSTRAP_TEST "Test Vk-Bootstrap with glfw and Catch2" OFF)
 
-if (CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME OR VK_BOOTSTRAP_TEST)
+if (VK_BOOTSTRAP_TEST)
 
     add_subdirectory(ext)
     add_subdirectory(tests)
