diff --git a/CMakeLists.txt b/CMakeLists.txt
index f1e0181..4d154e0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -83,7 +83,7 @@ install(TARGETS vk-bootstrap vk-bootstrap-compiler-warnings vk-bootstrap-vulkan-
 
 option(VK_BOOTSTRAP_TEST "Test Vk-Bootstrap with glfw and Catch2" OFF)
 
-if (CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME OR VK_BOOTSTRAP_TEST)
+if (VK_BOOTSTRAP_TEST)
 
     add_subdirectory(ext)
     add_subdirectory(tests)
