diff --git a/CMakeLists.txt b/CMakeLists.txt
index e4f3119..e712f6c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -9,6 +9,9 @@ if(NOT DEFINED CMAKE_FIRST_RUN)
     SET(CMAKE_FIRST_RUN 1 CACHE INTERNAL "")
 endif()
 
+include("${CMAKE_CURRENT_BINARY_DIR}/conanbuildinfo.cmake")
+conan_basic_setup()
+
 ###############################################################################
 ### GLOBAL ###
 
@@ -142,7 +145,7 @@ endif(CMAKE_COMPILER_IS_GNUCXX)
 ###############################################################################
 ### Python ###
 
-OCIOFindPython()
+#OCIOFindPython()
 
 # Find Python, used for (possibly) building pyglue, and now to
 # construct the external project path
@@ -204,6 +207,7 @@ else(USE_EXTERNAL_TINYXML)
         set(tinyxml_sources ${TINYXML_SOURCE_DIR}/tinyxml/tinystr.cpp       ${TINYXML_SOURCE_DIR}/tinyxml/tinyxml.cpp 
                             ${TINYXML_SOURCE_DIR}/tinyxml/tinyxmlerror.cpp  ${TINYXML_SOURCE_DIR}/tinyxml/tinyxmlparser.cpp )
         set(tinyxml_headers ${TINYXML_SOURCE_DIR}/tinyxml/tinystr.h         ${TINYXML_SOURCE_DIR}/tinyxml/tinyxml.h )
+        file(MAKE_DIRECTORY "${TINYXML_SOURCE_DIR}")
         add_custom_command(                                 ## will be done at build time
             OUTPUT ${tinyxml_sources} ${tinyxml_headers}    ## expected output files
             COMMAND ${CMAKE_COMMAND} -E tar xzf ${TINYXML_ZIPFILE}
@@ -248,6 +252,10 @@ endif(USE_EXTERNAL_TINYXML)
 ###############################################################################
 ### YAML ###
 
+set(YAML_CPP_INCLUDE_DIRS ${CONAN_INCLUDE_DIRS_YAML-CPP})
+set(YAML_CPP_LIBRARIES ${CONAN_LIBS_YAML-CPP})
+
+if(FALSE)
 if(USE_EXTERNAL_YAML)
     # Set minimum yaml version for non-patched sources.
     set(YAML_VERSION_MIN "0.3.0")
@@ -366,6 +374,7 @@ endif(USE_EXTERNAL_YAML)
 if(YAML_CPP_VERSION VERSION_LESS "0.5.0")
     set(YAML_CPP_COMPILE_FLAGS "-DOLDYAML")
 endif()
+endif(FALSE)
 
 ###############################################################################
 ### Externals ###
diff --git a/src/core/CMakeLists.txt b/src/core/CMakeLists.txt
index 1eb691b..2d4f1d6 100644
--- a/src/core/CMakeLists.txt
+++ b/src/core/CMakeLists.txt
@@ -21,10 +21,10 @@ list(APPEND core_export_headers ${CMAKE_BINARY_DIR}/export/OpenColorABI.h)
 if(WIN32)
     # On debug mode there are other kinds of warning...
     if("${CMAKE_BUILD_TYPE}" STREQUAL "Release")
-        set(EXTERNAL_COMPILE_FLAGS "${EXTERNAL_COMPILE_FLAGS} /WX")
+        # set(EXTERNAL_COMPILE_FLAGS "${EXTERNAL_COMPILE_FLAGS} /WX")
     endif()
 else()
-    set(EXTERNAL_COMPILE_FLAGS "${EXTERNAL_COMPILE_FLAGS} -Werror")
+    # set(EXTERNAL_COMPILE_FLAGS "${EXTERNAL_COMPILE_FLAGS} -Werror")
 endif()
 
 # SHARED
diff --git a/src/pyglue/CMakeLists.txt b/src/pyglue/CMakeLists.txt
index a90ca1c..a156dac 100644
--- a/src/pyglue/CMakeLists.txt
+++ b/src/pyglue/CMakeLists.txt
@@ -23,7 +23,7 @@ endif()
 # Process all warnings as errors
 # Unfortunately Windows still has a warning
 if(UNIX)
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror")
+    # set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror")
 endif()
 
 find_package(PythonLibs)
