--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -119,7 +119,6 @@ message(STATUS "Building ${CMAKE_BUILD_TYPE}")
 
 ## C++ version and flags
 # C++14 is needed for gtest, otherwise, C++11 is sufficient for the library
-set(CMAKE_CXX_STANDARD 14)
 if (MSVC)
   add_compile_definitions(_CRT_SECURE_NO_WARNINGS)
 endif()
--- a/src/apps/ojph_stream_expand/CMakeLists.txt
+++ b/src/apps/ojph_stream_expand/CMakeLists.txt
@@ -1,7 +1,6 @@
 ## building ojph_stream_expand
 ##############################
 
-set(CMAKE_CXX_STANDARD 14)
 
 file(GLOB OJPH_STREAM_EXPAND  "*.cpp" "*.h")
 file(GLOB OJPH_SOCKETS         "../others/ojph_sockets.cpp")
--- a/src/core/CMakeLists.txt
+++ b/src/core/CMakeLists.txt
@@ -135,9 +135,6 @@ if (BUILD_SHARED_LIBS AND WIN32)
 endif()
 
 ## include library version/name
-if (NOT MSVC)
-  target_compile_options(openjph PRIVATE -fPIC)
-endif()
 target_compile_definitions(openjph PUBLIC _FILE_OFFSET_BITS=64)
 target_include_directories(openjph PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/common> $<INSTALL_INTERFACE:include/openjph>)
 
