diff --git src/core/CMakeLists.txt src/core/CMakeLists.txt
index ea19aea..132a619 100644
--- src/core/CMakeLists.txt
+++ src/core/CMakeLists.txt
@@ -135,7 +135,7 @@ if (BUILD_SHARED_LIBS AND WIN32)
 endif()
 
 ## include library version/name
-set_target_properties(openjph PROPERTIES POSITION_INDEPENDENT_CODE ON)
+#set_target_properties(openjph PROPERTIES POSITION_INDEPENDENT_CODE ON)
 target_compile_definitions(openjph PUBLIC _FILE_OFFSET_BITS=64)
 target_include_directories(openjph PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/common> $<INSTALL_INTERFACE:include>)
 
