commit 9750ebd6681495f2594b3e5c90053e89f17f1ef9
Author: Martin Delille <martin@delille.org>
Date:   Sat Jul 17 18:20:56 2021 +0200

    Simpler CMakeLists.txt

diff --git a/CMakeLists.txt b/CMakeLists.txt
index e77cb93cc..fcad50524 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -58,25 +58,15 @@ set(AAFSDK_OUT_DIR "${AAFSDK_ROOT}/out" CACHE STRING "Path to AAFSDK output dire
 
 project(AAF-SDK LANGUAGES CXX)
 
-enable_testing()
-
 include(pdefs)
 include(cdefs)
 include(common)
 include(ss)
 
 add_subdirectory(DevUtils)
-add_subdirectory(examples)
-add_subdirectory(examples2)
 add_subdirectory(ref-impl)
-add_subdirectory(test)
-add_subdirectory(Utilities)
 
 add_custom_target(Everything)
 add_dependencies(Everything
     MakeSDK
-    MakeSDKExamples
-    MakeSDKExamples2
-    MakeSDKTests
-    MakeSDKUtilities
 )
