commit 6b2c4ece07218499ff50ceecb009f152a3102ced
Author: Martin Delille <martin@delille.org>
Date:   Tue Jul 27 16:54:26 2021 +0200

    Remove expat

diff --git a/ref-impl/CMakeLists.txt b/ref-impl/CMakeLists.txt
index b1fe34c52..09550edf6 100644
--- a/ref-impl/CMakeLists.txt
+++ b/ref-impl/CMakeLists.txt
@@ -37,9 +37,11 @@ project(MakeSDK LANGUAGES CXX)
 
 add_subdirectory(aafiid)
 add_subdirectory(aaflib)
-add_subdirectory(expat)
 add_subdirectory(libjpeg)
 add_subdirectory(plugins)
+
+find_package(expat REQUIRED)
+
 add_subdirectory(src/com-api)
 add_subdirectory(src/impl)
 add_subdirectory(src/OM)
diff --git a/ref-impl/src/OM/CMakeLists.txt b/ref-impl/src/OM/CMakeLists.txt
index e0c5be195..571caa598 100644
--- a/ref-impl/src/OM/CMakeLists.txt
+++ b/ref-impl/src/OM/CMakeLists.txt
@@ -158,7 +158,9 @@ target_include_directories(OM PRIVATE
     # HACK (see OMStructuredStorage.h): Required with OM_USE_SCHEMASOFT_SS and OM_USE_GSF_SS
     ${AAFSDK_ROOT}/ss-impl/ref
     # AAF-XML stored format support requires the expat library which is bundled with the AAF SDK.
-    ${AAFSDK_ROOT}/ref-impl/expat
+    #${AAFSDK_ROOT}/ref-impl/expat
+    # => using expat from conan
+    expat
 )
 
 target_compile_definitions(OM PUBLIC
