commit 2909f77ad421951cb3abf49da9af9b6317691c1a
Author: Martin Delille <martin@delille.org>
Date:   Tue Sep 7 16:21:39 2021 +0200

    Define interface for Windows

diff --git a/ref-impl/include/ref-api/AAF.h b/ref-impl/include/ref-api/AAF.h
index 058e325cc..6d93d0f97 100644
--- a/ref-impl/include/ref-api/AAF.h
+++ b/ref-impl/include/ref-api/AAF.h
@@ -49,6 +49,11 @@
 #endif
 
 #ifdef __cplusplus
+
+#if defined (_WINDOWS_) && !defined (interface)
+#define interface struct
+#endif
+
 interface IAAFAES3PCMDescriptor;
 interface IAAFAIFCDescriptor;
 interface IAAFAsyncStreamRawStorage;
