diff --git a/ref-impl/src/impl/aafErr.cpp b/ref-impl/src/impl/aafErr.cpp
index 79e4c5e..fe7a287 100644
--- a/ref-impl/src/impl/aafErr.cpp
+++ b/ref-impl/src/impl/aafErr.cpp
@@ -178,7 +178,7 @@ static char    *localErrorStrings[300];
 const char *aafGetErrorString(
 		aafErr_t code)
 {
-	if (code < 300 && code >= 0 && localErrorStrings[0] != '\0')
+	if (code < 300 && code >= 0 && localErrorStrings[0] != ((void*)0)) /* or nullptr if c++11 */
 		return (localErrorStrings[code]);
 	else
 		return ("AAF_ERR: Unknown error code");
