diff --git a/ref-impl/plugins/CAAFDNxHDCodec.cpp b/ref-impl/plugins/CAAFDNxHDCodec.cpp
index 42f11b6..be9b11a 100644
--- a/ref-impl/plugins/CAAFDNxHDCodec.cpp
+++ b/ref-impl/plugins/CAAFDNxHDCodec.cpp
@@ -727,7 +727,7 @@ HRESULT STDMETHODCALLTYPE
 
 	// Loop over all flavours recording the largest name's buffer size
 	aafUInt32 max = 0;
-	for (size_t i = 0; i < sizeof(kSupportedFlavours); i++)
+	for (size_t i = 0; i < sizeof(kSupportedFlavours) / sizeof(kSupportedFlavours[0]); i++)
 	{
 		// compute length of name in bytes including terminating NUL
 		aafUInt32 len = (wcsu8slen(kSupportedFlavours[i].name) + 1) *
diff --git a/ref-impl/plugins/CAAFVC3Codec.cpp b/ref-impl/plugins/CAAFVC3Codec.cpp
index 170fc2a..26b05f6 100644
--- a/ref-impl/plugins/CAAFVC3Codec.cpp
+++ b/ref-impl/plugins/CAAFVC3Codec.cpp
@@ -716,7 +716,7 @@ HRESULT STDMETHODCALLTYPE
 
 	// Loop over all flavours recording the largest name's buffer size
 	aafUInt32 max = 0;
-	for (size_t i = 0; i < sizeof(kSupportedFlavours); i++)
+	for (size_t i = 0; i < sizeof(kSupportedFlavours) / sizeof(kSupportedFlavours[0]); i++)
 	{
 		// compute length of name in bytes including terminating NUL
 		aafUInt32 len = (wcsu8slen(kSupportedFlavours[i].name) + 1) *
@@ -740,7 +740,7 @@ HRESULT STDMETHODCALLTYPE
 	if (0 >= bufSize)
 		return AAFRESULT_INVALID_PARAM;
 
-	for (size_t i = 0; i < sizeof(kSupportedFlavours); i++)
+	for (size_t i = 0; i < sizeof(kSupportedFlavours) / sizeof(kSupportedFlavours[0]); i++)
 	{
 		if (flavour == kSupportedFlavours[i].flavour)
 		{
