diff --git a/src/s2/s2lax_polygon_shape.cc b/src/s2/s2lax_polygon_shape.cc
index e71476a..7bfe874 100644
--- a/src/s2/s2lax_polygon_shape.cc
+++ b/src/s2/s2lax_polygon_shape.cc
@@ -33,7 +33,7 @@ namespace {
 template <typename T>
 std::unique_ptr<T> make_unique_for_overwrite(size_t n) {
   // We only need to support this one variant.
-  static_assert(std::is_array<T>::value);
+  static_assert(std::is_array<T>::value, "");
   return std::unique_ptr<T>(new typename absl::remove_extent_t<T>[n]);
 }
 }  // namespace
