--- a/DevIL/src-IL/src/il_manip.cpp
+++ b/DevIL/src-IL/src/il_manip.cpp
@@ -37,9 +37,9 @@ ILushort ILAPIENTRY ilFloatToHalf(ILuint i) {
 	// of float and half (127 versus 15).
 	//
 
-	register int s =  (i >> 16) & 0x00008000;
-	register int e = ((i >> 23) & 0x000000ff) - (127 - 15);
-	register int m =   i        & 0x007fffff;
+	int s =  (i >> 16) & 0x00008000;
+	int e = ((i >> 23) & 0x000000ff) - (127 - 15);
+	int m =   i        & 0x007fffff;
 
 	//
 	// Now reassemble s, e and m into a half:
