--- a/lib/CMakeLists.txt
+++ b/lib/CMakeLists.txt
@@ -4,7 +4,6 @@ message(STATUS "GAINPUT version ${GAINPUT_VERSION}")
 set(CMAKE_MACOSX_RPATH 1)
 
 if(CMAKE_COMPILER_IS_GNUCXX)
-	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++98 -Wall -Wextra -pedantic -Wshadow -Wno-variadic-macros")
 endif()
 
 include_directories (include/)
@@ -36,12 +35,11 @@ if (GAINPUT_BUILD_STATIC)
   set(install_libs ${install_libs} gainputstatic)
 endif (GAINPUT_BUILD_STATIC)
 
+if(GAINPUT_BUILD_SHARED)
 if(WIN32)
 	target_link_libraries(gainput ${XINPUT} ws2_32)
-	target_link_libraries(gainputstatic ${XINPUT} ws2_32)
 	add_definitions(-DGAINPUT_LIB_DYNAMIC=1)
 elseif(ANDROID)
-	target_link_libraries(gainputstatic native_app_glue log android)
 	target_link_libraries(gainput native_app_glue log android)
 elseif(APPLE)
 	find_library(FOUNDATION Foundation)
@@ -58,6 +56,7 @@ elseif(APPLE)
       target_link_libraries(gainput ${APPKIT})
   endif()
 endif()
+endif()
 
 # Library installation directory
 if(NOT DEFINED CMAKE_INSTALL_LIBDIR)
