diff --git a/main.mk b/main.mk
index 2f02f87..078e8d6 100644
--- a/main.mk
+++ b/main.mk
@@ -582,7 +582,7 @@ LIBOBJS0 = alter.o analyze.o attach.o auth.o \
          vdbe.o vdbeapi.o vdbeaux.o vdbeblob.o vdbemem.o vdbesort.o \
          vdbetrace.o vdbevtab.o vtab.o \
          wal.o walker.o where.o wherecode.o whereexpr.o \
-         window.o $(SQLCIPHER_OBJ) 
+         window.o $(SQLCIPHER_OBJ)
 LIBOBJS = $(LIBOBJS0)
 
 #
@@ -1130,7 +1130,7 @@ T.link.tcl = $(T.tcl.env.source); $(T.link)
 # of the static library ($(libsqlite3.LIB)) is constant on all tested
 # platforms.
 #
-libsqlite3.DLL.basename ?= libsqlite3
+libsqlite3.DLL.basename ?= libsqlcipher
 #
 # libsqlite3.DLL => the DLL library
 #
@@ -1146,7 +1146,7 @@ libsqlite3.out.implib ?=
 #
 # libsqlite3.LIB => the static library
 #
-libsqlite3.LIB = libsqlite3$(T.lib)
+libsqlite3.LIB = libsqlcipher$(T.lib)
 
 #
 # libsqlite3.DLL.install-rules => the suffix of the symoblic name of
@@ -1630,12 +1630,12 @@ install-dll-darwin: $(install-dir.lib) $(libsqlite3.DLL)
 	$(INSTALL) $(libsqlite3.DLL) "$(install-dir.lib)"
 	@echo "Setting up $(libsqlite3.DLL) version symlinks..."; \
 	cd "$(install-dir.lib)" || exit $$?; \
-	rm -f libsqlite3.0$(T.dll) libsqlite3.$(PACKAGE_VERSION)$(T.dll) || exit $$?; \
-	dllname=libsqlite3.$(PACKAGE_VERSION)$(T.dll); \
+	rm -f libsqlcipher.0$(T.dll) libsqlcipher.$(PACKAGE_VERSION)$(T.dll) || exit $$?; \
+	dllname=libsqlcipher.$(PACKAGE_VERSION)$(T.dll); \
 	mv $(libsqlite3.DLL) $$dllname || exit $$?; \
 	ln -s $$dllname $(libsqlite3.DLL) || exit $$?; \
-	ln -s $$dllname libsqlite3.0$(T.dll) || exit $$?; \
-	ls -la $$dllname $(libsqlite3.DLL) libsqlite3.0$(T.dll)
+	ln -s $$dllname libsqlcipher.0$(T.dll) || exit $$?; \
+	ls -la $$dllname $(libsqlite3.DLL) libsqlcipher.0$(T.dll)
 
 install-dll-1: install-dll-$(libsqlite3.DLL.install-rules)
 install-dll-0 install-dll-:
@@ -2518,7 +2518,7 @@ tidy:
 	rm -f lemon$(B.exe) sqlite*.tar.gz
 	rm -f mkkeywordhash$(B.exe) mksourceid$(B.exe)
 	rm -f parse.* fts5parse.*
-	rm -f $(libsqlite3.DLL) $(libsqlite3.LIB) $(libtclsqlite3.DLL) libsqlite3$(T.dll).a
+	rm -f $(libsqlite3.DLL) $(libsqlite3.LIB) $(libtclsqlite3.DLL) libsqlcipher$(T.dll).a
 	rm -f tclsqlite3$(T.exe) $(TESTPROGS)
 	rm -f LogEst$(T.exe) fts3view$(T.exe) rollback-test$(T.exe) showdb$(T.exe)
 	rm -f showjournal$(T.exe) showstat4$(T.exe) showwal$(T.exe) speedtest1$(T.exe)
