diff --git a/Makefile.in b/Makefile.in
index d8f5a62..57b54d5 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -647,7 +647,7 @@ DBFUZZ_OPT =
 # This is the default Makefile target.  The objects listed here
 # are what get build when you type just "make" with no arguments.
 #
-all:	sqlite3.h libsqlcipher.la sqlcipher$(TEXE) $(HAVE_TCL:1=libtclsqlite3.la)
+all:	sqlite3.h libsqlcipher.la
 
 Makefile: $(TOP)/Makefile.in
 	./config.status
@@ -1445,9 +1445,8 @@ lib_install:	libsqlcipher.la
 	$(INSTALL) -d $(DESTDIR)$(libdir)
 	$(LTINSTALL) libsqlcipher.la $(DESTDIR)$(libdir)
 
-install:	sqlcipher$(TEXE) lib_install sqlite3.h sqlcipher.pc ${HAVE_TCL:1=tcl_install}
+install:	lib_install sqlite3.h sqlcipher.pc
 	$(INSTALL) -d $(DESTDIR)$(bindir)
-	$(LTINSTALL) sqlcipher$(TEXE) $(DESTDIR)$(bindir)
 	$(INSTALL) -d $(DESTDIR)$(includedir)
 	$(INSTALL) -m 0644 sqlite3.h $(DESTDIR)$(includedir)
 	$(INSTALL) -m 0644 $(TOP)/src/sqlite3ext.h $(DESTDIR)$(includedir)
