diff --git a/CMakeLists.txt b/CMakeLists.txt
index de9cd40..e6d4e44 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -96,14 +96,13 @@ target_include_directories(oboe
 #     Enable -Ofast
 target_compile_options(oboe
         PRIVATE
-        -std=c++17
         -Wall
         -Wextra-semi
         -Wshadow
         -Wshadow-field
         "$<$<CONFIG:RELEASE>:-Ofast>"
         "$<$<CONFIG:DEBUG>:-O3>"
-        "$<$<CONFIG:DEBUG>:-Werror>")
+)
 
 target_compile_definitions(oboe PRIVATE $<$<BOOL:${OBOE_DISABLE_CONVERSION}>:DISABLE_CONVERSION=1>)
 
@@ -118,8 +117,8 @@ target_link_options(oboe PRIVATE "-Wl,-z,max-page-size=16384")
 
 # When installing oboe put the libraries in the lib/<ABI> folder e.g. lib/arm64-v8a
 install(TARGETS oboe
-        LIBRARY DESTINATION lib/${ANDROID_ABI}
-        ARCHIVE DESTINATION lib/${ANDROID_ABI})
+        LIBRARY DESTINATION lib
+        ARCHIVE DESTINATION lib)
 
 # Also install the headers
 install(DIRECTORY include/oboe DESTINATION include)
