diff --git a/CMakeLists.txt b/CMakeLists.txt
index a75251a..48e35c8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -15,10 +15,9 @@ option(RYML_BUILD_API "Enable API generation (python, etc)" OFF)
 option(RYML_DBG "Enable (very verbose) ryml debug prints." OFF)
 
 
-#-------------------------------------------------------
+find_package(c4core REQUIRED CONFIG)
 
-c4_require_subproject(c4core INCORPORATE
-    SUBDIRECTORY ${RYML_EXT_DIR}/c4core)
+#-------------------------------------------------------
 
 c4_add_library(ryml
     SOURCES
@@ -50,10 +49,10 @@ c4_add_library(ryml
         ryml.natvis
     SOURCE_ROOT ${RYML_SRC_DIR}
     INC_DIRS
+        $<BUILD_INTERFACE:${C4CORE_INCLUDE_DIR}>
         $<BUILD_INTERFACE:${RYML_SRC_DIR}>
         $<INSTALL_INTERFACE:include>
-    LIBS c4core
-    INCORPORATE c4core
+    LIBS c4core::c4core
     )
 
 if(NOT RYML_DEFAULT_CALLBACKS)
@@ -68,7 +67,6 @@ endif()
 #-------------------------------------------------------
 
 c4_install_target(ryml)
-c4_install_exports(DEPENDENCIES c4core)
 c4_pack_project()
 
 
