--- win32/net-snmp/net-snmp-config.h.in
+++ win32/net-snmp/net-snmp-config.h.in
@@ -1777,59 +1777,7 @@
 /* MSVC OpenSSL linker settings. */
 #if defined(_MSC_VER)
 #  if defined(NETSNMP_USE_OPENSSL)
-#    ifdef _DLL
-#      ifdef _DEBUG
-#        ifdef OPENSSL_BEFORE_1_1_0
-#          pragma comment(lib, "libeay32MDd.lib")
-#        else
-#          ifdef _M_X64
-#            pragma comment(lib, "libcrypto64MDd.lib")
-#            pragma comment(lib, "libssl64MDd.lib")
-#          else
-#            pragma comment(lib, "libcrypto32MDd.lib")
-#            pragma comment(lib, "libssl32MDd.lib")
-#          endif
-#        endif
-#      else
-#        ifdef OPENSSL_BEFORE_1_1_0
-#          pragma comment(lib, "libeay32MD.lib")
-#        else
-#          ifdef _M_X64
-#            pragma comment(lib, "libcrypto64MD.lib")
-#            pragma comment(lib, "libssl64MD.lib")
-#          else
-#            pragma comment(lib, "libcrypto32MD.lib")
-#            pragma comment(lib, "libssl32MD.lib")
-#          endif
-#        endif
-#      endif
-#    else
-#      ifdef _DEBUG
-#        ifdef OPENSSL_BEFORE_1_1_0
-#          pragma comment(lib, "libeay32MTd.lib")
-#        else
-#          ifdef _M_X64
-#            pragma comment(lib, "libcrypto64MTd.lib")
-#            pragma comment(lib, "libssl64MTd.lib")
-#          else
-#            pragma comment(lib, "libcrypto32MTd.lib")
-#            pragma comment(lib, "libssl32MTd.lib")
-#          endif
-#        endif
-#      else
-#        ifdef OPENSSL_BEFORE_1_1_0
-#          pragma comment(lib, "libeay32MT.lib")
-#        else
-#          ifdef _M_X64
-#            pragma comment(lib, "libcrypto64MT.lib")
-#            pragma comment(lib, "libssl64MT.lib")
-#          else
-#            pragma comment(lib, "libcrypto32MT.lib")
-#            pragma comment(lib, "libssl32MT.lib")
-#          endif
-#        endif
-#      endif
-#    endif
+/* Conan: system_libs */
 #    pragma comment(lib, "gdi32.lib")
 #    pragma comment(lib, "user32.lib")
 #  endif
