diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4e7f0ff..fd1cee8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -54,6 +54,7 @@ function(libenvpp_set_compiler_parameters TARGET)
 endfunction()
 
 # External dependencies.
+if(0)
 if(LIBENVPP_INSTALL)
 	set(FMT_INSTALL ON CACHE BOOL "" FORCE)
 endif()
@@ -63,7 +64,8 @@ FetchContent_Declare(fmt
 	GIT_TAG 11.0.2
 )
 FetchContent_MakeAvailable(fmt)
-
+endif()
+find_package(fmt REQUIRED CONFIG)
 if(LIBENVPP_TESTS)
 	FetchContent_Declare(Catch2
 		GIT_REPOSITORY https://github.com/catchorg/Catch2.git
@@ -133,7 +135,9 @@ endif()
 
 # Installation target.
 if(LIBENVPP_INSTALL)
-	# Libenvpp installation.
+  include(CMakePackageConfigHelpers)
+
+  # Libenvpp installation.
 	set(LIBENVPP_PROJECT_CONFIG_OUT "${CMAKE_CURRENT_BINARY_DIR}/libenvpp-config.cmake")
 	set(LIBENVPP_VERSION_CONFIG_FILE "${CMAKE_CURRENT_BINARY_DIR}/libenvpp-config-version.cmake")
 	install(DIRECTORY ${PROJECT_SOURCE_DIR}/include/
