diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8729676..cc58853 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -55,7 +55,7 @@ endfunction()
 
 # External dependencies.
 include(FetchContent)
-
+if(0)
 if(LIBENVPP_INSTALL)
 	set(FMT_INSTALL ON CACHE BOOL "" FORCE)
 endif()
@@ -78,7 +78,8 @@ macro(fetch_content_from_submodule DEPNAME RELPATH)
 endmacro()
 
 fetch_content_from_submodule(fmt external/fmt)
-
+endif()
+find_package(fmt REQUIRED CONFIG)
 if(LIBENVPP_TESTS)
 	fetch_content_from_submodule(Catch2 external/Catch2)
 	list(APPEND CMAKE_MODULE_PATH ${catch2_SOURCE_DIR}/extras)
