Index: all/src/configure.ac
===================================================================
--- a/configure.ac
+++ b/configure.ac
@@ -119,50 +119,8 @@ AC_CONFIG_FILES([
 	gl/Makefile
 	src/Makefile
 	src/skels/Makefile
-	src/tests/Makefile
-	tests/Makefile
-	tests/test_conf.conf
-	tests/test_conf_inc.conf
-	tests/test_conf_parser_ov2.c
-	tests/test_conf_parser_ov3.c
-	tests/test_conf_parser_ov4.c
-])
 
-AC_DEFUN([AC_CONFIG_TEST_SH], [AC_CONFIG_FILES([$1], [chmod -- +x $1])])
+])
 
-AC_CONFIG_TEST_SH([tests/more_than_once.sh])
-AC_CONFIG_TEST_SH([tests/no_optgiven.sh])
-AC_CONFIG_TEST_SH([tests/test_all_opts.sh])
-AC_CONFIG_TEST_SH([tests/test_all_opts_file_save.sh])
-AC_CONFIG_TEST_SH([tests/test_conf_parser.sh])
-AC_CONFIG_TEST_SH([tests/test_conf_parser_err.sh])
-AC_CONFIG_TEST_SH([tests/test_conf_parser_err_string.sh])
-AC_CONFIG_TEST_SH([tests/test_conf_parser_grp.sh])
-AC_CONFIG_TEST_SH([tests/test_conf_parser_ov.sh])
-AC_CONFIG_TEST_SH([tests/test_conf_parser_ov2.sh])
-AC_CONFIG_TEST_SH([tests/test_conf_parser_ov3.sh])
-AC_CONFIG_TEST_SH([tests/test_conf_parser_ov4.sh])
-AC_CONFIG_TEST_SH([tests/test_conf_parser_save.sh])
-AC_CONFIG_TEST_SH([tests/test_default_values.sh])
-AC_CONFIG_TEST_SH([tests/test_dep.sh])
-AC_CONFIG_TEST_SH([tests/test_groups.sh])
-AC_CONFIG_TEST_SH([tests/test_manual_help.sh])
-AC_CONFIG_TEST_SH([tests/test_manual_help_args.sh])
-AC_CONFIG_TEST_SH([tests/test_modes.sh])
-AC_CONFIG_TEST_SH([tests/test_multiple.sh])
-AC_CONFIG_TEST_SH([tests/test_multiple_err.sh])
-AC_CONFIG_TEST_SH([tests/test_multiple_parsers.sh])
-AC_CONFIG_TEST_SH([tests/test_output_dir.sh])
-AC_CONFIG_TEST_SH([tests/test_output_dirs.sh])
-AC_CONFIG_TEST_SH([tests/test_sections.sh])
-AC_CONFIG_TEST_SH([tests/test_show_help.sh])
-AC_CONFIG_TEST_SH([tests/test_simple_multiple.sh])
-AC_CONFIG_TEST_SH([tests/test_simple_multiple_default.sh])
-AC_CONFIG_TEST_SH([tests/test_unamed.sh])
-AC_CONFIG_TEST_SH([tests/test_unnamed.sh])
-AC_CONFIG_TEST_SH([tests/test_values.sh])
-AC_CONFIG_TEST_SH([tests/test_values_err.sh])
-AC_CONFIG_TEST_SH([tests/test_wrong_ggo.sh])
-AC_CONFIG_TEST_SH([tests/valgrind_tests.sh])
 
 AC_OUTPUT
Index: all/src/Makefile.am
===================================================================
--- a/Makefile.am
+++ b/Makefile.am
@@ -18,7 +18,7 @@
 ACLOCAL_AMFLAGS = -I m4 -I gl/m4
 
 EXTRA_DIST = configure TODO LICENSE gl/m4/gnulib-cache.m4
-SUBDIRS = gl src doc tests
+SUBDIRS = gl src
 
 gengetoptdoc_DATA = ChangeLog COPYING NEWS THANKS INSTALL README LICENSE
 
Index: all/src/src/Makefile.am
===================================================================
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -16,7 +16,7 @@
 # with gengetopt; see the file COPYING. If not, write to the Free Software 
 # Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA. 
 
-SUBDIRS = skels tests
+SUBDIRS = skels
 
 bin_PROGRAMS = gengetopt
 
