--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -39,8 +39,6 @@ file(GLOB_RECURSE msdfgen-ext_PRIVATE_HEADERS RELATIVE ${CMAKE_CURRENT_SOURCE_DI
 
 file(GLOB_RECURSE msdfgen-ext_SOURCES RELATIVE ${CMAKE_CURRENT_SOURCE_DIR}
 	"ext/*.cpp"
-	"lib/*.cpp"
-	"lib/*.cpp"
 )
 
 # Build the library (aliased name because it's the same target name the exe)
@@ -86,14 +84,14 @@ add_library(msdfgen::msdfgen-ext ALIAS msdfgen-ext)
 set_target_properties(msdfgen-ext PROPERTIES
 	PUBLIC_HEADER "${msdfgen-ext_PUBLIC_HEADERS}"
 )
-target_link_libraries(msdfgen-ext PUBLIC msdfgen::msdfgen Freetype::Freetype)
+find_package(lodepng REQUIRED CONFIG)
+find_package(tinyxml2 REQUIRED CONFIG)
+target_link_libraries(msdfgen-ext PUBLIC msdfgen::msdfgen Freetype::Freetype lodepng::lodepng tinyxml2::tinyxml2)
 target_include_directories(msdfgen-ext
 PUBLIC
 	$<INSTALL_INTERFACE:include>
 	$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../>
 	
-PRIVATE
-	${CMAKE_CURRENT_SOURCE_DIR}/include
 )
 
 target_compile_definitions(msdfgen-ext PUBLIC MSDFGEN_CMAKE_BUILD)
