--- core/src/DecodeHints.h
+++ core/src/DecodeHints.h
@@ -16,6 +16,7 @@
 * limitations under the License.
 */
 
+#include <cstdint>
 #include <vector>
 #include <sstream>
 #include <string>
--- core/src/pdf417/PDFCodewordDecoder.cpp
+++ core/src/pdf417/PDFCodewordDecoder.cpp
@@ -17,6 +17,7 @@
 
 #include "PDFCodewordDecoder.h"
 
+#include <cstdint>
 #include <vector>
 #include <numeric>
 #include <limits>
--- core/src/textcodec/JPTextEncoder.cpp
+++ core/src/textcodec/JPTextEncoder.cpp
@@ -37,6 +37,8 @@
 
 #include "JPTextEncoder.h"
 
+#include <cstdint>
+
 /*
 * This data is derived from Unicode 1.1,
 * JIS X 0208 (1990) to Unicode mapping table version 0.9 .
