--- a/glslang/CMakeLists.txt
+++ b/glslang/CMakeLists.txt
@@ -167,7 +167,6 @@
 add_library(glslang ${LIB_TYPE} ${BISON_GLSLParser_OUTPUT_SOURCE} ${GLSLANG_SOURCES} ${GLSLANG_HEADERS})
 set_target_properties(glslang PROPERTIES
     FOLDER glslang
-    POSITION_INDEPENDENT_CODE ON
     VERSION   "${GLSLANG_VERSION}"
     SOVERSION "${GLSLANG_VERSION_MAJOR}")
 target_link_libraries(glslang PRIVATE OGLCompiler OSDependent MachineIndependent)
