--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -306,10 +306,6 @@
     add_subdirectory(External)
 endif()
 
-if(NOT TARGET SPIRV-Tools-opt)
-    set(ENABLE_OPT OFF)
-endif()
-
 if(ENABLE_OPT)
     message(STATUS "optimizer enabled")
     add_definitions(-DENABLE_OPT=1)
--- a/SPIRV/CMakeLists.txt
+++ b/SPIRV/CMakeLists.txt
@@ -91,14 +91,7 @@
 endif()
 
 if(ENABLE_OPT)
-    target_include_directories(SPIRV
-        PRIVATE ${spirv-tools_SOURCE_DIR}/include
-        PRIVATE ${spirv-tools_SOURCE_DIR}/source
-    )
-    target_link_libraries(SPIRV PRIVATE MachineIndependent SPIRV-Tools-opt)
-    target_include_directories(SPIRV PUBLIC
-        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../External>
-        $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/External>)
+    target_link_libraries(SPIRV glslang ${CONAN_LIBS})
 else()
     target_link_libraries(SPIRV PRIVATE MachineIndependent)
 endif(ENABLE_OPT)
--- a/glslang/CMakeLists.txt
+++ b/glslang/CMakeLists.txt
@@ -167,7 +167,6 @@
 add_library(glslang ${LIB_TYPE} ${BISON_GLSLParser_OUTPUT_SOURCE} ${GLSLANG_SOURCES} ${GLSLANG_HEADERS})
 set_target_properties(glslang PROPERTIES
     FOLDER glslang
-    POSITION_INDEPENDENT_CODE ON
     VERSION   "${GLSLANG_VERSION}"
     SOVERSION "${GLSLANG_VERSION_MAJOR}")
 target_link_libraries(glslang PRIVATE OGLCompiler OSDependent MachineIndependent)
@@ -206,6 +205,9 @@
                 ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
                 LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
                 RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
+        install(TARGETS MachineIndependent GenericCodeGen
+                EXPORT  glslangTargets
+                ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
     else()
         install(TARGETS glslang MachineIndependent GenericCodeGen
                 EXPORT  glslangTargets
