--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -303,9 +303,6 @@
     add_subdirectory(External)
 endif()

-if(NOT TARGET SPIRV-Tools-opt)
-    set(ENABLE_OPT OFF)
-endif()

 if(ENABLE_OPT)
     message(STATUS "optimizer enabled")
--- a/SPIRV/CMakeLists.txt
+++ b/SPIRV/CMakeLists.txt
@@ -91,14 +91,7 @@
 endif()

 if(ENABLE_OPT)
-    target_include_directories(SPIRV
-        PRIVATE ${spirv-tools_SOURCE_DIR}/include
-        PRIVATE ${spirv-tools_SOURCE_DIR}/source
-    )
-    target_link_libraries(SPIRV PRIVATE MachineIndependent SPIRV-Tools-opt)
-    target_include_directories(SPIRV PUBLIC
-        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../External>
-        $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/External>)
+    target_link_libraries(SPIRV glslang ${CONAN_LIBS})
 else()
     target_link_libraries(SPIRV PRIVATE MachineIndependent)
 endif(ENABLE_OPT)
