diff -Naur a/crypto/os_int.h b/crypto/os_int.h
--- a/crypto/os_int.h	2015-03-09 07:32:51.000000000 +0300
+++ b/crypto/os_int.h	2021-06-29 18:10:14.553713945 +0300
@@ -41,25 +41,17 @@
 extern "C" {
 #endif
 
-#if defined(WIN32)
-typedef UINT8 uint8_t;
-typedef INT8 int8_t;
-typedef UINT16 uint16_t;
-typedef INT16 int16_t;
-typedef UINT32 uint32_t;
-typedef INT32 int32_t;
-typedef UINT64 uint64_t;
-typedef INT64 int64_t;
-#else   /* Not Win32 */
 
 #ifdef CONFIG_PLATFORM_SOLARIS
 #include <inttypes.h>
 #else
 #include <stdint.h>
-#include <endian.h>
-#endif /* Not Solaris */
 
+#if !defined(WIN32)
+#include <endian.h>
 #endif /* Not Win32 */
+    
+#endif /* Not Solaris */
 
 #ifdef __cplusplus
 }
diff -Naur a/ssl/os_port.h b/ssl/os_port.h
--- a/ssl/os_port.h	2015-04-30 08:53:33.000000000 +0300
+++ b/ssl/os_port.h	2021-06-28 17:47:49.818656716 +0300
@@ -122,6 +122,10 @@ EXP_FUNC void STDCALL gettimeofday(struc
 EXP_FUNC int STDCALL strcasecmp(const char *s1, const char *s2);
 EXP_FUNC int STDCALL getdomainname(char *buf, int buf_size);
 
+#ifndef be64toh
+#define be64toh(x) _byteswap_uint64(x) 
+#endif
+
 #else   /* Not Win32 */
 
 #include <unistd.h>
