--- CMakeLists.txt
+++ CMakeLists.txt
@@ -107,10 +107,10 @@ set(OPTIONAL_LIBRARIES)
 ########################################################################
 # LIBZMQ dependency
 ########################################################################
-find_package(libzmq REQUIRED)
+find_package(ZeroMQ REQUIRED)
-IF (LIBZMQ_FOUND)
+IF (ZEROMQ_FOUND)
-    include_directories(${LIBZMQ_INCLUDE_DIRS})
-    list(APPEND MORE_LIBRARIES ${LIBZMQ_LIBRARIES})
+    include_directories(${CONAN_INCLUDE_DIRS_LIBZMQ})
+    list(APPEND MORE_LIBRARIES ZeroMQ::ZeroMQ)
     set(pkg_config_libs_private "${pkg_config_libs_private} -lzmq")
 ELSE (LIBZMQ_FOUND)
     message( FATAL_ERROR "libzmq not found." )
@@ -121,8 +121,8 @@ ENDIF (LIBZMQ_FOUND)
 ########################################################################
 find_package(czmq REQUIRED)
 IF (CZMQ_FOUND)
-    include_directories(${CZMQ_INCLUDE_DIRS})
-    list(APPEND MORE_LIBRARIES ${CZMQ_LIBRARIES})
+    include_directories(${CONAN_INCLUDE_DIRS_CZMQ})
+    list(APPEND MORE_LIBRARIES czmq::czmq)
     set(pkg_config_libs_private "${pkg_config_libs_private} -lczmq")
 ELSE (CZMQ_FOUND)
     message( FATAL_ERROR "czmq not found." )
@@ -168,7 +168,7 @@
 endif()
 add_library(zyre ${zyre_sources})
 set_target_properties(zyre
-    PROPERTIES DEFINE_SYMBOL "ZYRE_EXPORTS"
+    PROPERTIES DEFINE_SYMBOL "ZYRE_EXPORTS" LINKER_LANGUAGE CXX
 )
 set_target_properties (zyre
     PROPERTIES SOVERSION "2.0.0"
@@ -182,9 +182,9 @@ install(TARGETS zyre
     ARCHIVE DESTINATION "lib${LIB_SUFFIX}" # .lib file
     RUNTIME DESTINATION bin              # .dll file
 )
-
+if(BUILD_EXECUTABLES)
 include(${CMAKE_CURRENT_SOURCE_DIR}/src/CMakeLists-local.txt) # Optional project-local hook
-
+endif()
 ########################################################################
 # pkgconfig
 ########################################################################
@@ -211,6 +211,7 @@ install(
 ########################################################################
 # executables
 ########################################################################
+if(BUILD_EXECUTABLES)
 add_executable(
     perf_local
     "${SOURCE_DIR}/src/perf_local.c"
@@ -304,7 +305,7 @@ set_target_properties(
     zyre_selftest
     PROPERTIES RUNTIME_OUTPUT_DIRECTORY "${SOURCE_DIR}/src"
 )
-
+endif()
 ########################################################################
 # tests
 ########################################################################
