--- CMakeLists.txt
+++ CMakeLists.txt
@@ -22,7 +22,7 @@
 ########################################################################
 # options
 ########################################################################
-if (NOT CMAKE_BUILD_TYPE)
+if (0)
     if (EXISTS "${SOURCE_DIR}/.git")
         set (CMAKE_BUILD_TYPE Debug)
     else ()
@@ -43,7 +43,7 @@
         endif ()
     endif ()
 endif ()
-if (${CMAKE_BUILD_TYPE} STREQUAL "Debug")
+if ("${CMAKE_BUILD_TYPE}" STREQUAL "Debug")
     OPTION (ENABLE_DRAFTS "Build and install draft classes and methods" ON)
 else ()
     OPTION (ENABLE_DRAFTS "Build and install draft classes and methods" OFF)
