--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,9 +1,6 @@
 cmake_minimum_required(VERSION 3.12)
 project(vault VERSION 0.52.0 DESCRIPTION "Vault library for C++")
 
-set(CMAKE_CXX_STANDARD 17)
-set(CMAKE_CXX_STANDARD_REQUIRED ON)
-set(CMAKE_CXX_EXTENSIONS OFF)
 set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
 
 option(ENABLE_TEST "Enable tests?" ON)
@@ -13,12 +10,7 @@ option(LINK_CURL "Link curl library for vault" OFF)
 option(BUILD_SHARED_LIBS "Build vault as a shared library" ON)
 option(INSTALL "Run install targets" ON)
 
-find_package(CURL)
-if (CURL_FOUND)
-  include_directories(${CURL_INCLUDE_DIR})
-else (CURL_FOUND)
-  message(FATAL_ERROR "CURL not found")
-endif (CURL_FOUND)
+find_package(CURL REQUIRED)
 
 include(GNUInstallDirs)
 include_directories("${PROJECT_SOURCE_DIR}/lib")
@@ -119,9 +111,11 @@ set_target_properties(vault PROPERTIES
 
 target_include_directories(vault PRIVATE src)
 
-if(LINK_CURL)
-  target_link_libraries(vault curl)
-endif(LINK_CURL)
+target_link_libraries(vault PUBLIC CURL::libcurl)
+target_compile_features(vault PUBLIC cxx_std_17)
+if(WIN32)
+    set_target_properties(vault PROPERTIES RUNTIME_OUTPUT_NAME "VaultClient")
+endif()
 
 if (ENABLE_COVERAGE)
   target_link_libraries(vault gcov)
