--- Source/Checks/cm_cxx_features.cmake
+++ Source/Checks/cm_cxx_features.cmake
@@ -21,6 +21,8 @@
     set(check_output "${OUTPUT}")
     # Filter out MSBuild output that looks like a warning.
     string(REGEX REPLACE " +0 Warning\\(s\\)" "" check_output "${check_output}")
+    # Filter out MSBuild output that looks like a warning.
+    string(REGEX REPLACE " warning MSB[0-9]{4}" "" check_output "${check_output}")
     # Filter out warnings caused by user flags.
     string(REGEX REPLACE "[^\n]*warning:[^\n]*-Winvalid-command-line-argument[^\n]*" "" check_output "${check_output}")
     # Filter out warnings caused by local configuration.
