From c4f9194f9b31f9747297c1593d45c6bdbf8eb774 Mon Sep 17 00:00:00 2001
From: Tarcisio Rodrigues <tarcisio.genaro@gmail.com>
Date: Sun, 14 Jun 2020 19:24:26 -0300
Subject: [PATCH 6/8] opt for unique_ptr vs auto_ptr through CMake option

---
 CMakeLists.txt | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index bc0a0ef8..5690742a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -66,9 +66,10 @@ else()
 message(FATAL_ERROR "Must set SHARED_PTR")
 endif()
 
-include(FindUniquePtr)
-FIND_UNIQUE_PTR()
-if (HAVE_UNIQUE_PTR_IN_STD_NAMESPACE)
+set(UNIQUE_PTR AUTO CACHE STRING "Define unique pointer implementation")
+set_property(CACHE UNIQUE_PTR PROPERTY STRINGS UNIQUE AUTO)
+
+if ("${UNIQUE_PTR}" STREQUAL "UNIQUE")
 message("-- set HAVE_STD_UNIQUE_PTR")
 add_definitions("-DHAVE_STD_UNIQUE_PTR=1")
 endif()
-- 
2.25.0

