From 5ca88b3e944fe351821be9240b32a4c2e2a64685 Mon Sep 17 00:00:00 2001
From: Tarcisio Rodrigues <tarcisio.genaro@gmail.com>
Date: Sun, 14 Jun 2020 22:44:09 -0300
Subject: [PATCH 7/8] manage postgres through conan

---
 CMakeLists.txt | 2 --
 1 file changed, 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5690742a..dbb1099d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -46,8 +46,6 @@ add_definitions("-DHAVE_MYSQL=1")
 endif()
 
 if(HAVE_POSTGRESQL)
-find_package(PostgreSQL REQUIRED)
-include_directories(${PostgreSQL_INCLUDE_DIRS})
 
 message("-- Building with POSTGRESQL")
 add_definitions("-DHAVE_POSTGRESQL=1")
-- 
2.25.0

