diff --git a/Foundation/CMakeLists.txt b/Foundation/CMakeLists.txt
index 41ba99936..1062f196d 100644
--- a/Foundation/CMakeLists.txt
+++ b/Foundation/CMakeLists.txt
@@ -17,8 +17,11 @@ POCO_SOURCES_AUTO_PLAT(SRCS WIN32
 
 # Version Resource
 if(MSVC AND BUILD_SHARED_LIBS)
+	# Empty File needed otherwise cmake does not find it.
+	file(GENERATE OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/pocomsg.rc CONTENT "")
 	source_group("Resources" FILES ${PROJECT_SOURCE_DIR}/DLLVersion.rc)
 	list(APPEND SRCS ${PROJECT_SOURCE_DIR}/DLLVersion.rc)
+	list(APPEND SRCS ${CMAKE_CURRENT_BINARY_DIR}/pocomsg.rc)
 endif()
 
 # Messages
