diff --git a/googletest/cmake/internal_utils.cmake b/googletest/cmake/internal_utils.cmake
index 8c1f9ba..2203d3c 100644
--- a/googletest/cmake/internal_utils.cmake
+++ b/googletest/cmake/internal_utils.cmake
@@ -166,7 +166,7 @@ function(cxx_library_with_type name type cxx_flags)
   # Generate debug library name with a postfix.
   set_target_properties(${name}
     PROPERTIES
-    DEBUG_POSTFIX "d")
+    DEBUG_POSTFIX "${CUSTOM_DEBUG_POSTFIX}")
   if (BUILD_SHARED_LIBS OR type STREQUAL "SHARED")
     set_target_properties(${name}
       PROPERTIES
