--- configure.ac
+++ configure.ac
@@ -27,6 +27,7 @@

 dnl Add check-news when it checks for more than 15 lines
 AC_INIT([MPFR],[4.0.2])
+AC_CANONICAL_TARGET

 dnl Older Automake versions than 1.13 may still be supported, but no longer
 dnl tested, and many things have changed in 1.13. Moreover the INSTALL file

--- acinclude.m4
+++ acinclude.m4
@@ -41,7 +41,6 @@
 AC_REQUIRE([AC_OBJEXT])
 AC_REQUIRE([MPFR_CHECK_LIBM])
 AC_REQUIRE([MPFR_CHECK_LIBQUADMATH])
-AC_REQUIRE([AC_HEADER_TIME])
 AC_REQUIRE([AC_CANONICAL_HOST])

 dnl Features for the MPFR shared cache. This needs to be done
