--- CMakeLists.txt
+++ CMakeLists.txt
@@ -273,12 +273,12 @@
        FORCE)
 endif (NOT CMAKE_BUILD_TYPE)
 
-set (CMAKE_CXX_STANDARD 98 CACHE STRING
-     "C++ standard to build all targets. Supported values are 98, 11, and 14.")
-set (CMAKE_CXX_STANDARD_REQUIRED ON CACHE BOOL
-     "The with CMAKE_CXX_STANDARD selected C++ standard is a requirement.")
-mark_as_advanced (CMAKE_CXX_STANDARD_REQUIRED)
-
+#set (CMAKE_CXX_STANDARD 98 CACHE STRING
+#     "C++ standard to build all targets. Supported values are 98, 11, and 14.")
+#set (CMAKE_CXX_STANDARD_REQUIRED ON CACHE BOOL
+#     "The with CMAKE_CXX_STANDARD selected C++ standard is a requirement.")
+#mark_as_advanced (CMAKE_CXX_STANDARD_REQUIRED)
+#
 if (NOT (WIN32 OR CYGWIN))
   option (BUILD_SHARED_LIBS "Build shared libraries." ON)
 else (NOT (WIN32 OR CYGWIN))
