diff --git a/cmake/project-defaults.cmake b/cmake/project-defaults.cmake
index c262bac8..f6604316 100644
--- a/cmake/project-defaults.cmake
+++ b/cmake/project-defaults.cmake
@@ -18,7 +18,6 @@ add_compile_options(
     /EHsc
     # Enable warnings and warnings as errors
     /W4
-    /WX
     # Disable RTTI
     $<$<COMPILE_LANGUAGE:CXX>:/GR->
     # Use /O2 (Maximize Speed)
@@ -34,7 +33,6 @@ add_compile_options(
     # Enable warnings and warnings as errors
     -Wall
     -Wextra
-    -Werror
     -Wconversion
     # Disable RTTI
     $<$<COMPILE_LANGUAGE:CXX>:-fno-rtti>
