diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1e29cfc1d..94e9daf53 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -62,6 +62,10 @@ endif()
 # include_directories(SYSTEM "${CMAKE_CURRENT_SOURCE_DIR}/modules")
 include_directories(SYSTEM "${CMAKE_CURRENT_SOURCE_DIR}/plugins")
 
+if(CYGWIN OR MINGW)
+  add_definitions(-Wa,-mbig-obj)
+endif()
+
 # Library
 file(GLOB_RECURSE LIB_HEADER_FILES "include/*.h")
 file(GLOB_RECURSE LIB_INLINE_FILES "include/*.inl")
