diff --git a/CMakeLists.txt b/CMakeLists.txt
index dcd57785b..7e3a1c756 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -17,49 +17,6 @@ if(DOXYGEN_FOUND)
   endif()
 endif()
 
-# CMake module path
-set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
-
-# Compiler features
-include(SetCompilerFeatures)
-include(SetCompilerWarnings)
-include(SetPlatformFeatures)
-include(SystemInformation)
-
-# External packages
-find_package(Threads REQUIRED)
-if(UNIX AND NOT APPLE AND NOT MSYS)
-  find_package(LibBFD)
-  find_package(LibDL)
-  find_package(LibRT)
-  find_package(LibUUID)
-endif()
-if(WIN32 OR MSYS)
-  find_package(DbgHelp)
-  find_package(RPC)
-  find_package(Userenv)
-endif()
-
-# Modules
-add_subdirectory("modules")
-
-# Link libraries
-list(APPEND LINKLIBS Threads::Threads)
-if(UNIX AND NOT APPLE AND NOT MSYS)
-  list(APPEND LINKLIBS ${LIBBFD_LIBRARIES})
-  list(APPEND LINKLIBS ${LIBDL_LIBRARIES})
-  list(APPEND LINKLIBS ${LIBRT_LIBRARIES})
-  list(APPEND LINKLIBS ${LIBUUID_LIBRARIES})
-endif()
-if(WIN32 OR MSYS)
-  list(APPEND LINKLIBS ${DBGHELP_LIBRARIES})
-  list(APPEND LINKLIBS ${RPC_LIBRARIES})
-  list(APPEND LINKLIBS ${USERENV_LIBRARIES})
-  list(APPEND LINKLIBS ${LIBVLD_LIBRARIES})
-endif()
-
-# System directories
-include_directories(SYSTEM "${CMAKE_CURRENT_SOURCE_DIR}/modules")
 include_directories(SYSTEM "${CMAKE_CURRENT_SOURCE_DIR}/plugins")
 
 # Library
@@ -67,9 +24,8 @@ file(GLOB_RECURSE LIB_HEADER_FILES "include/*.h")
 file(GLOB_RECURSE LIB_INLINE_FILES "include/*.inl")
 file(GLOB_RECURSE LIB_SOURCE_FILES "source/*.cpp")
 add_library(cppcommon ${LIB_HEADER_FILES} ${LIB_INLINE_FILES} ${LIB_SOURCE_FILES})
-set_target_properties(cppcommon PROPERTIES COMPILE_FLAGS "${PEDANTIC_COMPILE_FLAGS}" FOLDER "libraries")
-target_include_directories(cppcommon PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}/include" PUBLIC ${vld})
-target_link_libraries(cppcommon ${LINKLIBS} fmt)
+target_include_directories(cppcommon PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}/include")
+set_target_properties(cppcommon PROPERTIES FOLDER libraries)
 list(APPEND INSTALL_TARGETS cppcommon)
 list(APPEND LINKLIBS cppcommon)
 
