diff --git a/CMakeLists.txt b/CMakeLists.txt
index ad3d53aa8..d4dcbf8b5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -74,6 +74,10 @@ target_link_libraries(cppcommon PUBLIC ${LINKLIBS} CONAN_PKG::fmt)
 list(APPEND INSTALL_TARGETS cppcommon)
 list(APPEND LINKLIBS cppcommon)
 
+if(WIN32 AND NOT MSVC)
+  target_compile_definitions (cppcommon PUBLIC -D_WIN32_WINNT=_WIN32_WINNT_VISTA)
+endif()
+
 # Additional module components: benchmarks, examples, plugins, tests, tools and install
 if(NOT CPPCOMMON_MODULE)
 
