commit 6381672c1c540c5132d9e13b04cbb87d973fa537 (HEAD -> master, origin/master, origin/HEAD)
Author: emeryberger <emery.berger@gmail.com>
Date:   Sat Oct 24 17:50:39 2020 -0400

    Changed to 3.

diff --git a/coz b/coz
index eb7f29b..db10b5b 100755
--- a/coz
+++ b/coz
@@ -1,4 +1,4 @@
-#!/usr/bin/env python2.7
+#!/usr/bin/env python3
 
 # Copyright (c) 2019, Charlie Curtsinger and Emery Berger,
 #                     University of Massachusetts Amherst
