--- a/include/freetype/internal/compiler-macros.h
+++ b/include/freetype/internal/compiler-macros.h
@@ -42,8 +42,8 @@ FT_BEGIN_HEADER
       ( defined( __cplusplus ) && __cplusplus > 201402L )
 #    define FALL_THROUGH  [[__fallthrough__]]
 #  elif ( defined( __GNUC__ ) && __GNUC__ >= 7 )          || \
-        ( defined( __clang__ ) && __clang_major__ >= 10 )
-#    define FALL_THROUGH  __attribute__(( __fallthrough__ ))
+        ( defined( __clang__ ) && __clang_major__ >= 12 )
+#    define FALL_THROUGH  __attribute__((fallthrough))
 #  else
 #    define FALL_THROUGH  ( (void)0 )
 #  endif
