--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -23,7 +23,7 @@ add_custom_command(OUTPUT dict-crc.h zxcvbn.dict COMMAND dictgen -b -o ${CMAKE_C
 add_library(zxcvbn-shared SHARED zxcvbn.c dict-crc.h dict-src.h)
 target_include_directories(zxcvbn-shared PRIVATE ${CMAKE_CURRENT_BINARY_DIR})
 target_include_directories(zxcvbn-shared PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
-target_link_libraries(zxcvbn-shared PUBLIC ${LIBM})
+
 set_target_properties(zxcvbn-shared PROPERTIES 
 	POSITION_INDEPENDENT_CODE ON
 	OUTPUT_NAME zxcvbn
@@ -34,7 +34,7 @@ set_target_properties(zxcvbn-shared PROPERTIES
 # build static version of library
 add_library(zxcvbn-static STATIC zxcvbn.c dict-crc.h dict-src.h)
 target_include_directories(zxcvbn-static PRIVATE ${CMAKE_CURRENT_BINARY_DIR} PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
-target_link_libraries(zxcvbn-static PUBLIC ${LIBM})
+
 set_target_properties(zxcvbn-static PROPERTIES OUTPUT_NAME zxcvbn)
 
 # in root projects we also want to build/run tests...
