--- CMakeLists.txt
+++ CMakeLists.txt
@@ -27,23 +27,6 @@ set(ENTITYX_BUILD_SHARED true CACHE BOOL "Build shared libraries?")
 include(${CMAKE_ROOT}/Modules/CheckIncludeFile.cmake)
 include(CheckCXXSourceCompiles)

-# Default compiler args
-if (CMAKE_CXX_COMPILER_ID MATCHES "(GNU|.*Clang)")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pedantic -Werror -Wall -Wextra -Wno-unused-parameter -Wno-error=unused-variable -Wno-error=sign-compare")
-    set(CMAKE_CXX_FLAGS_DEBUG "-O0 -g")
-    set(CMAKE_CXX_FLAGS_MINSIZEREL "-g -Os -DNDEBUG")
-    set(CMAKE_CXX_FLAGS_RELEASE "-g -O2 -DNDEBUG")
-    set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "-O2 -g")
-elseif(CMAKE_CXX_COMPILER_ID STREQUAL 'MSVC')
-    # /Zi - Produces a program database (PDB) that contains type information and symbolic debugging information for use with the debugger.
-    # /FS - Allows multiple cl.exe processes to write to the same .pdb file
-    # /DEBUG - Enable debug during linking
-    # /Od - Disables optimization
-    set(CMAKE_CXX_FLAGS_DEBUG "/Zi /FS /DEBUG /Od /MDd")
-    # /Ox - Full optimization
-    set(CMAKE_CXX_FLAGS_RELEASE "/Ox -DNDEBUG")
-    set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "/Ox /Zi /FS /DEBUG")
-endif()

 # if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
 #     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Weverything -Wno-c++98-compat -Wno-shadow -Wno-padded -Wno-missing-noreturn -Wno-global-constructors")
@@ -55,28 +38,6 @@ if(NOT DEFINED CMAKE_INSTALL_LIBDIR)
 endif(NOT DEFINED CMAKE_INSTALL_LIBDIR)
 set(libdir ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR})

-set(OLD_CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
-if ((MAKE_CXX_COMPILER_ID STREQUAL "GNU") OR (CMAKE_CXX_COMPILER_ID STREQUAL "Clang"))
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -stdlib=libc++")
-endif()
-check_cxx_source_compiles(
-"
-#include <memory>
-
-int main() {
-    std::shared_ptr<int>();
-}
-"
-ENTITYX_HAVE_CXX11_STDLIB
-)
-
-if (NOT ENTITYX_HAVE_CXX11_STDLIB)
-    message(STATUS "-- Not using -stdlib=libc++ (test failed to build)")
-    set(CMAKE_CXX_FLAGS "${OLD_CMAKE_CXX_FLAGS}")
-else ()
-    message(STATUS "-- Using -stdlib=libc++")
-endif ()
-

 # Misc features
 check_include_file("stdint.h" HAVE_STDINT_H)
