--- a/libheif/plugins/CMakeLists.txt
+++ b/libheif/plugins/CMakeLists.txt
@@ -44,23 +44,23 @@
 
 set(X265_sources encoder_x265.h encoder_x265.cc)
 set(X265_extra_plugin_sources)
-plugin_compilation(x265 X265 X265 X265)
+plugin_compilation(x265 x265 X265 X265)
 
 set(LIBDE265_sources decoder_libde265.cc decoder_libde265.h)
 set(LIBDE265_extra_plugin_sources ../error.cc)
-plugin_compilation(libde265 LIBDE265 LIBDE265 LIBDE265)
+plugin_compilation(libde265 libde265 LIBDE265 LIBDE265)
 
 set(DAV1D_sources decoder_dav1d.cc decoder_dav1d.h)
 set(DAV1D_extra_plugin_sources ../common_utils.cc ../common_utils.h)
-plugin_compilation(dav1d DAV1D DAV1D DAV1D)
+plugin_compilation(dav1d Dav1d DAV1D DAV1D)
 
 set(AOM_DECODER_sources decoder_aom.cc decoder_aom.h)
 set(AOM_DECODER_extra_plugin_sources)
-plugin_compilation(aomdec AOM AOM_DECODER AOM_DECODER)
+plugin_compilation(aomdec libaom-av1 AOM_DECODER AOM_DECODER)
 
 set(AOM_ENCODER_sources encoder_aom.cc encoder_aom.h)
 set(AOM_ENCODER_extra_plugin_sources ../error.cc ../common_utils.cc ../common_utils.h)
-plugin_compilation(aomenc AOM AOM_ENCODER AOM_ENCODER)
+plugin_compilation(aomenc libaom-av1 AOM_ENCODER AOM_ENCODER)
 
 set(SvtEnc_sources encoder_svt.cc encoder_svt.h)
 set(SvtEnc_extra_plugin_sources)
