Index: CMakeLists.txt
===================================================================
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -341,7 +341,7 @@ add_custom_target(PACKAGE_SRC_TAR
                    COMMENT "Creating ${PROJECT_BINARY_DIR}/vigra-${vigra_version}-src.tar.gz")
 
 ADD_DEPENDENCIES(PACKAGE_SRC_TAR check)
-ADD_DEPENDENCIES(PACKAGE_SRC_TAR doc_cpp)
+#ADD_DEPENDENCIES(PACKAGE_SRC_TAR doc_cpp)
 IF(WITH_VIGRANUMPY AND PYTHON_SPHINX)
     ADD_DEPENDENCIES(PACKAGE_SRC_TAR doc_python)
 ENDIF()
