--- CMakeLists.txt
+++ CMakeLists.txt
@@ -484,8 +484,8 @@
 if(static_runtime)
 	include(ucm_flags)
 	ucm_set_runtime(STATIC)
-	set(Boost_USE_MULTITHREADED ON)
-	set(Boost_USE_STATIC_RUNTIME ON)
+	# set(Boost_USE_MULTITHREADED ON)
+	# set(Boost_USE_STATIC_RUNTIME ON)
 	set(OPENSSL_USE_STATIC_LIBS TRUE)
 	set(OPENSSL_MSVC_STATIC_RT TRUE)
 endif()
@@ -630,7 +630,7 @@
 )

 if(OPENSSL_FOUND)
-	target_link_libraries(torrent-rasterbar PUBLIC OpenSSL::SSL)
+	target_link_libraries(torrent-rasterbar PUBLIC OpenSSL::SSL ${CONAN_SYSTEM_LIBS_OPENSSL})
 	target_compile_definitions(torrent-rasterbar PUBLIC TORRENT_USE_OPENSSL)
 	target_sources(torrent-rasterbar PRIVATE src/pe_crypto)
 endif()
