--- libmpdec++/Makefile.vc
+++ libmpdec++/Makefile.vc
@@ -16,10 +16,10 @@
 # LIBSHARED_CXX = libmpdec++-2.5.0.dll
 
 !if "$(DEBUG)" == "1"
-OPT = /MTd /Od /Zi /EHsc
-OPT_SHARED = /MDd /Od /Zi /EHsc
+OPT = /Od /Zi /EHsc
+OPT_SHARED = /Od /Zi /EHsc
 !else
-OPT = /MT /Ox /GS /EHsc
-OPT_SHARED = /MD /Ox /GS /EHsc
+OPT = /Ox /GS /EHsc
+OPT_SHARED = /Ox /GS /EHsc
 !endif
 
--- libmpdec/Makefile.vc
+++ libmpdec/Makefile.vc
@@ -36,10 +36,10 @@
 !endif
 
 !if "$(DEBUG)" == "1"
-OPT = /MTd /Od /Zi /EHsc
-OPT_SHARED = /MDd /Od /Zi /EHsc
+OPT = /Od /Zi /EHsc
+OPT_SHARED = /Od /Zi /EHsc
 !else
-OPT = /MT /Ox /GS /EHsc
-OPT_SHARED = /MD /Ox /GS /EHsc
+OPT = /Ox /GS /EHsc
+OPT_SHARED = /Ox /GS /EHsc
 !endif
 
