diff --git a/libmpdec++/Makefile.vc b/libmpdec++/Makefile.vc
index 5985c33..bc2bdac 100644
--- a/libmpdec++/Makefile.vc
+++ b/libmpdec++/Makefile.vc
@@ -14,11 +14,11 @@ LIBIMPORT_CXX = libmpdec++-4.0.0.dll.lib
 LIBSHARED_CXX = libmpdec++-4.0.0.dll
 
 !if "$(DEBUG)" == "1"
-OPT = /MTd /Od /Zi /EHsc
-OPT_SHARED = /MDd /Od /Zi /EHsc
+OPT = /Od /Zi /EHsc
+OPT_SHARED = /Od /Zi /EHsc
 !else
-OPT = /MT /O2 /GS /EHsc /DNDEBUG
-OPT_SHARED = /MD /O2 /GS /EHsc /DNDEBUG
+OPT = /O2 /GS /EHsc /DNDEBUG
+OPT_SHARED = /O2 /GS /EHsc /DNDEBUG
 !endif
 
 !if "$(CC)" == "clang-cl"
diff --git a/libmpdec/Makefile.vc b/libmpdec/Makefile.vc
index 7f72a4b..2ee3ddb 100644
--- a/libmpdec/Makefile.vc
+++ b/libmpdec/Makefile.vc
@@ -31,11 +31,11 @@ CONFIG = /DCONFIG_32 /DANSI
 !endif
 
 !if "$(DEBUG)" == "1"
-OPT = /MTd /Od /Zi /EHsc
-OPT_SHARED = /MDd /Od /Zi /EHsc
+OPT = /Od /Zi /EHsc
+OPT_SHARED = /Od /Zi /EHsc
 !else
-OPT = /MT /O2 /GS /EHsc /DNDEBUG
-OPT_SHARED = /MD /O2 /GS /EHsc /DNDEBUG
+OPT = /O2 /GS /EHsc /DNDEBUG
+OPT_SHARED = /O2 /GS /EHsc /DNDEBUG
 !endif
 
 !if "$(CC)" == "clang-cl"
