--- external/crashpad/third_party/mini_chromium/CMakeLists.txt
+++ external/crashpad/third_party/mini_chromium/CMakeLists.txt
@@ -204,3 +204,7 @@
     DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/crashpad"
     FILES_MATCHING PATTERN "*.h"
 )
+crashpad_install_dev(DIRECTORY build
+    DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/crashpad/mini_chromium"
+    FILES_MATCHING PATTERN "*.h"
+)
--- external/crashpad/util/CMakeLists.txt
+++ external/crashpad/util/CMakeLists.txt
@@ -438,3 +438,7 @@
 add_library(crashpad::util ALIAS crashpad_util)
 
 crashpad_install_target(crashpad_util)
+crashpad_install_dev(DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/"
+    DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/crashpad/util"
+    FILES_MATCHING PATTERN "*.h"
+)
