--- a/external/crashpad/handler/CMakeLists.txt
+++ b/external/crashpad/handler/CMakeLists.txt
@@ -60,6 +60,7 @@ target_link_libraries(crashpad_handler_lib
 
 if(WIN32)
     if(MSVC)
+        target_link_libraries(crashpad_handler_lib PUBLIC crashpad_getopt)
         target_compile_options(crashpad_handler_lib PRIVATE "/wd4201")
     endif()
 endif()
--- a/external/crashpad/third_party/getopt/CMakeLists.txt
+++ b/external/crashpad/third_party/getopt/CMakeLists.txt
@@ -9,6 +9,7 @@ if(MSVC)
     target_link_libraries(crashpad_getopt PRIVATE
         $<BUILD_INTERFACE:crashpad_interface>
     )
+    crashpad_install_target(crashpad_getopt)
 else()
     add_library(crashpad_getopt INTERFACE)
 endif()
