diff --git a/configure b/configure
index babbae1..a89b829 100755
--- a/configure
+++ b/configure
@@ -207,6 +223,28 @@ if compile_prog "" "" "__kernel_timespec"; then
 fi
 print_config "__kernel_timespec" "$__kernel_timespec"
 
+##########################################
+# check for statx
+statx="no"
+cat > $TMPC << EOF
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include <string.h>
+#include <linux/stat.h>
+int main(int argc, char **argv)
+{
+  struct statx x;
+
+  return memset(&x, 0, sizeof(x)) != NULL;
+}
+EOF
+if compile_prog "" "" "statx"; then
+  statx="yes"
+fi
+print_config "statx" "$statx"
+
 #############################################################################
 
 if test "$__kernel_rwf_t" = "yes"; then
@@ -215,3 +253,7 @@ fi
 if test "$__kernel_timespec" = "yes"; then
   output_sym "CONFIG_HAVE_KERNEL_TIMESPEC"
 fi
+if test "$statx" = "yes"; then
+  output_sym "CONFIG_HAVE_STATX"
+fi
+

