diff --git a/test/Makefile b/test/Makefile
index 6cbdb4a..ff7ff0f 100644
--- a/test/Makefile
+++ b/test/Makefile
@@ -18,10 +18,14 @@ all_targets += poll poll-cancel ring-leak fsync io_uring_setup io_uring_register
 		poll-link accept-link fixed-link poll-cancel-ton teardowns \
 		poll-many b5837bd5311d-test accept-test d77a67ed5f27-test \
 		connect 7ad0e4b2f83c-test submit-reuse fallocate open-close \
-		file-update statx accept-reuse
+		file-update accept-reuse
 
 include ../Makefile.quiet
 
+ifdef CONFIG_HAVE_STATX
+all_targets += statx
+endif
+
 all: $(all_targets)
 
 %: %.c
@@ -39,7 +43,11 @@ test_srcs := poll.c poll-cancel.c ring-leak.c fsync.c io_uring_setup.c \
 	accept-link.c fixed-link.c poll-cancel-ton.c teardowns.c poll-many.c \
 	b5837bd5311d-test.c accept-test.c d77a67ed5f27-test.c connect.c \
 	7ad0e4b2f83c-test.c submit-reuse.c fallocate.c open-close.c \
-	file-update.c statx.c accept-reuse.c
+	file-update.c accept-reuse.c
+
+ifdef CONFIG_HAVE_STATX
+test_srcs += statx.c
+endif
 
 test_objs := $(patsubst %.c,%.ol,$(test_srcs))
 
