diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1c87994..78bfad0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -130,6 +130,7 @@ endif ()
 
 include (${SHARED_DIR}/utils/uncommon/shared_uncommon.cmake)
 
+if (NOT BUILD_SHARED_LIBS)
 # STATIC LIBRARY
 add_library(iwasm_static STATIC ${WAMR_RUNTIME_LIB_SOURCE})
 set_target_properties (iwasm_static PROPERTIES OUTPUT_NAME vmlib)
@@ -140,7 +141,7 @@ if (WAMR_BUILD_WASM_CACHE EQUAL 1)
 endif ()
 
 install (TARGETS iwasm_static ARCHIVE DESTINATION lib)
-
+else()
 # SHARED LIBRARY
 add_library (iwasm_shared SHARED ${WAMR_RUNTIME_LIB_SOURCE})
 set_target_properties (iwasm_shared PROPERTIES OUTPUT_NAME iwasm)
@@ -155,6 +156,7 @@ if (MINGW)
 endif ()
 
 install (TARGETS iwasm_shared LIBRARY DESTINATION lib)
+endif()
 
 # HEADERS
 install (FILES
