From b20b15f99a0067c5b3ba6343ed863b53c75a267b Mon Sep 17 00:00:00 2001
From: Lucas K Dal Castel <lucas.castel@ael.com.br>
Date: Fri, 10 Nov 2023 13:21:17 -0300
Subject: [PATCH] Harmonize Static Lib build

---
 CMakeLists.txt    | 27 +++++++--------------------
 include/AL/alut.h | 26 +++++++++++++++-----------
 2 files changed, 22 insertions(+), 31 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7aa9687..8dd6344 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -45,7 +45,6 @@ SET(ADD_CFLAGS "")
 SET(ADD_LDFLAGS "")
 SET(ADD_LIBS "")
 
-OPTION(BUILD_STATIC    "build static library too"   ON)
 OPTION(PROFILE         "enable profile"            OFF)
 OPTION(OPTIMIZATION    "enable optimization"        ON)
 OPTION(WARNINGS        "enable warnings"            ON)
@@ -118,27 +117,15 @@ SET(CMAKE_CXX_FLAGS "${ADD_CFLAGS} ${CMAKE_CXX_FLAGS}")
 SET(CMAKE_SHARED_LINKER_FLAGS "${ADD_LDFLAGS} ${CMAKE_SHARED_LINKER_FLAGS}")
 SET(CMAKE_MODULE_LINKER_FLAGS "${ADD_LDFLAGS} ${CMAKE_MODULE_LINKER_FLAGS}")
 
-IF(BUILD_STATIC)
-  # we can't create a static library with the same name
-  # as the shared one, so we copy it over after creation
-  ADD_LIBRARY(alut_static STATIC ${ALUT_SOURCES})
-  TARGET_LINK_LIBRARIES(alut_static ${OPENAL_LIB} ${ADD_LIBS})
-  IF(NOT WIN32)
-    ADD_CUSTOM_COMMAND(
-      TARGET alut_static
-      POST_BUILD
-      COMMAND ${CMAKE_COMMAND}
-      ARGS -E copy
-        ${CMAKE_BINARY_DIR}/${CMAKE_STATIC_LIBRARY_PREFIX}alut_static${CMAKE_STATIC_LIBRARY_SUFFIX}
-        ${CMAKE_BINARY_DIR}/${CMAKE_STATIC_LIBRARY_PREFIX}alut${CMAKE_STATIC_LIBRARY_SUFFIX})
-    INSTALL_FILES(/lib FILES ${CMAKE_STATIC_LIBRARY_PREFIX}alut${CMAKE_STATIC_LIBRARY_SUFFIX})
-  ENDIF(NOT WIN32)
-ENDIF(BUILD_STATIC)
-
-
-ADD_LIBRARY(alut SHARED ${ALUT_SOURCES})
+ADD_LIBRARY(alut ${ALUT_SOURCES})
 SET_TARGET_PROPERTIES(alut PROPERTIES VERSION ${VERSION} SOVERSION ${MAJOR_VERSION})
 set_target_properties(alut PROPERTIES PUBLIC_HEADER "include/AL/alut.h")
+
+get_target_property(target_type alut TYPE)
+if (target_type STREQUAL "STATIC_LIBRARY")
+  target_compile_definitions(alut PUBLIC ALUT_LIBTYPE_STATIC)
+endif ()
+
 TARGET_LINK_LIBRARIES(alut ${OPENAL_LIB} ${ADD_LIBS})
 
 install(TARGETS alut
diff --git a/include/AL/alut.h b/include/AL/alut.h
index 4b05a3c..5772c6c 100644
--- a/include/AL/alut.h
+++ b/include/AL/alut.h
@@ -16,18 +16,22 @@
 extern "C" {
 #endif
 
-#if defined(_WIN32) && !defined(_XBOX)
- #if defined (ALUT_BUILD_LIBRARY)
-  #define ALUT_API __declspec(dllexport)
- #else
-  #define ALUT_API __declspec(dllimport)
- #endif
+#ifndef ALUT_LIBTYPE_STATIC
+  #if defined(_WIN32) && !defined(_XBOX)
+   #if defined (ALUT_BUILD_LIBRARY)
+    #define ALUT_API __declspec(dllexport)
+   #else
+    #define ALUT_API __declspec(dllimport)
+   #endif
+  #else
+   #if defined(ALUT_BUILD_LIBRARY) && defined(HAVE_GCC_VISIBILITY)
+    #define ALUT_API __attribute__((visibility("default")))
+   #else
+    #define ALUT_API extern
+   #endif
+  #endif
 #else
- #if defined(ALUT_BUILD_LIBRARY) && defined(HAVE_GCC_VISIBILITY)
-  #define ALUT_API __attribute__((visibility("default")))
- #else
-  #define ALUT_API extern
- #endif
+  #define ALUT_API
 #endif
 
 #if defined(_WIN32)
-- 
2.36.1.windows.1

