diff --git a/CMakeLists.txt b/CMakeLists.txt
index 696d270..006bf68 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -462,17 +462,10 @@ endif()
 
 option(SENTRY_INTEGRATION_QT "Build Qt integration")
 if(SENTRY_INTEGRATION_QT)
-	if(QT_DEFAULT_MAJOR_VERSION)
-		# Let user choose major version
-		set(Qt_VERSION_MAJOR ${QT_DEFAULT_MAJOR_VERSION})
-	else()
-		# Find best match, prioritizing Qt 6 if available
-		find_package(Qt NAMES Qt6 Qt5 COMPONENTS Core REQUIRED)
-	endif()
-	find_package(Qt${Qt_VERSION_MAJOR} COMPONENTS Core REQUIRED)
+	find_package(qt REQUIRED)
 	message(STATUS "Found Qt: ${Qt${Qt_VERSION_MAJOR}_DIR} "
 		"(found version \"${Qt${Qt_VERSION_MAJOR}_VERSION}\")")
-	target_link_libraries(sentry PRIVATE Qt${Qt_VERSION_MAJOR}::Core)
+	target_link_libraries(sentry PRIVATE qt::qt)
 endif()
 
 include(CMakePackageConfigHelpers)
