diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2407fca..89416c7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -460,8 +460,8 @@ endif()
 
 option(SENTRY_INTEGRATION_QT "Build Qt integration")
 if(SENTRY_INTEGRATION_QT)
-	find_package(Qt5 COMPONENTS Core REQUIRED)
-	target_link_libraries(sentry PRIVATE Qt5::Core)
+	find_package(qt REQUIRED)
+	target_link_libraries(sentry PRIVATE qt::qt)
 endif()
 
 include(CMakePackageConfigHelpers)
