From 714d98a15f5cacb9fbbb55b11760c0746cd53d43 Mon Sep 17 00:00:00 2001
From: Carlos Gomes Martinho <kmartinho8@gmail.com>
Date: Mon, 27 Apr 2020 18:58:57 +0200
Subject: [PATCH] build: Set C standard if not set (#247)

---
 CMakeLists.txt | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 74e8666..c41d902 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,6 +6,10 @@ if(CMAKE_SOURCE_DIR STREQUAL PROJECT_SOURCE_DIR)
 	set(SENTRY_MAIN_PROJECT ON)
 endif()
 
+if(NOT CMAKE_C_STANDARD)
+  set(CMAKE_C_STANDARD 11)
+endif()
+
 include(GNUInstallDirs)
 set(CMAKE_INSTALL_CMAKEDIR "${CMAKE_INSTALL_LIBDIR}/cmake/sentry")
 
