From 903c17ae20888679caab1871cc74de577509452e Mon Sep 17 00:00:00 2001
From: Cyriuz <Cyriuz@users.noreply.github.com>
Date: Mon, 13 Mar 2023 14:55:26 +0100
Subject: [PATCH] Allow setting CRASHPAD_WER_ENABLED when using system crashpad
 (#816)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 831e8e1..aeea09f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -426,8 +426,9 @@ if(SENTRY_BACKEND_CRASHPAD)
 			set(CRASHPAD_ENABLE_INSTALL ON CACHE BOOL "Enable crashpad installation" FORCE)
 		endif()
 		add_subdirectory(external/crashpad crashpad_build)
+
 		if(CRASHPAD_WER_ENABLED)
-			add_compile_definitions(CRASHPAD_WER_ENABLED)
+			add_dependencies(sentry crashpad::wer)
 		endif()
 
 		# set static runtime if enabled
@@ -482,8 +483,9 @@ if(SENTRY_BACKEND_CRASHPAD)
 		endif()
 	endif()
 	add_dependencies(sentry crashpad::handler)
+
 	if(CRASHPAD_WER_ENABLED)
-		add_dependencies(sentry crashpad::wer)
+		add_compile_definitions(CRASHPAD_WER_ENABLED)
 	endif()
 elseif(SENTRY_BACKEND_BREAKPAD)
 	option(SENTRY_BREAKPAD_SYSTEM "Use system breakpad" OFF)
-- 
2.39.1.windows.1
