From b2af8551957abaa3b25c720fccef5fc77d14fd77 Mon Sep 17 00:00:00 2001
Date: Mon, 1 Dec 2025 15:54:44 +0100
Subject: [PATCH] [CPU][ARM] Define HWCAP2_I8MM constant properly (#33060)
URL: https://github.com/openvinotoolkit/openvino/pull/33060

diff --git a/src/inference/src/system_conf.cpp b/src/inference/src/system_conf.cpp
index dd47185947..fbfd82e4ef 100644
--- a/src/inference/src/system_conf.cpp
+++ b/src/inference/src/system_conf.cpp
@@ -22,6 +22,7 @@
 #    include <sys/auxv.h>
 #    define ARM_COMPUTE_CPU_FEATURE_HWCAP_FPHP    (1 << 9)
 #    define ARM_COMPUTE_CPU_FEATURE_HWCAP_ASIMDHP (1 << 10)
+#    define ARM_COMPUTE_CPU_FEATURE_HWCAP2_I8MM   (1 << 13)
 #    define ARM_COMPUTE_CPU_FEATURE_HWCAP_SVE     (1 << 24)
 #elif defined(__APPLE__) && defined(__aarch64__)
 #    include <sys/sysctl.h>
@@ -224,7 +225,7 @@ bool with_cpu_arm_i8mm() {
 #    if !defined(_WIN64) && !defined(BARE_METAL) && !defined(__APPLE__) && !defined(__OpenBSD__) && \
         !defined(__arm__) && defined(__aarch64__)
     const uint32_t hwcaps = getauxval(AT_HWCAP);
-    return hwcaps & HWCAP2_I8MM;
+    return hwcaps & ARM_COMPUTE_CPU_FEATURE_HWCAP2_I8MM;
 #    elif !defined(_WIN64) && !defined(BARE_METAL) && !defined(__APPLE__) && !defined(__OpenBSD__) && \
         !defined(__aarch64__) && defined(__arm__)
     return false;
-- 
2.51.2.windows.1

