--- platform-android.mk	2017-06-16 12:20:14.000000000 +0800
+++ platform-android_.mk	2019-11-04 11:20:37.849441400 +0800
@@ -38,23 +38,12 @@
 
 TOOLCHAINPREFIX = $(shell NDK_PROJECT_PATH=$(SRC_PATH)/codec/build/android/dec make --no-print-dir -f $(NDKROOT)/build/core/build-local.mk DUMP_TOOLCHAIN_PREFIX APP_ABI=$(APP_ABI))
 
-SYSROOT = $(NDKROOT)/platforms/android-$(NDKLEVEL)/arch-$(ARCH)
-CXX = $(TOOLCHAINPREFIX)g++
-CC = $(TOOLCHAINPREFIX)gcc
-AR = $(TOOLCHAINPREFIX)ar
-CFLAGS += -DANDROID_NDK -fpic --sysroot=$(SYSROOT) -MMD -MP
+CFLAGS += -DANDROID_NDK -fpic -MMD -MP
 CXXFLAGS += -fno-rtti -fno-exceptions
-LDFLAGS += --sysroot=$(SYSROOT)
 SHLDFLAGS = -Wl,--no-undefined -Wl,-z,relro -Wl,-z,now -Wl,-soname,lib$(PROJECT_NAME).so
 
-ifneq ($(CXX),$(wildcard $(CXX)))
-ifneq ($(CXX).exe,$(wildcard $(CXX).exe))
-$(error Compiler not found, bad NDKROOT or ARCH?)
-endif
-endif
-
 STL_INCLUDES = \
-    -I$(NDKROOT)/sources/cxx-stl/stlport/stlport
+    -I$(NDKROOT)/sources/cxx-stl/llvm-libc++/include
 STL_LIB = \
     $(NDKROOT)/sources/cxx-stl/stlport/libs/$(APP_ABI)/libstlport_static.a
 
