diff --git a/cmake/modules/LLVMExternalProjectUtils.cmake b/cmake/modules/LLVMExternalProjectUtils.cmake
index caccb62d666c..d53e9c78262a 100644
--- a/cmake/modules/LLVMExternalProjectUtils.cmake
+++ b/cmake/modules/LLVMExternalProjectUtils.cmake
@@ -173,7 +173,7 @@ function(llvm_ExternalProject_Add name source_dir)
     LibEdit_INCLUDE_DIRS
     LibEdit_LIBRARIES
     ZLIB_INCLUDE_DIR
-    ZLIB_LIBRARY
+    ZLIB_LIBRARIES
     zstd_INCLUDE_DIR
     zstd_LIBRARY
     LIBXML2_LIBRARY
diff --git a/lib/Support/CMakeLists.txt b/lib/Support/CMakeLists.txt
index a579eaf7d953..82f8d7fe494f 100644
--- a/lib/Support/CMakeLists.txt
+++ b/lib/Support/CMakeLists.txt
@@ -331,32 +331,12 @@ set(llvm_system_libs ${system_libs})
 # move to using CMake export, this block can be removed.
 if(LLVM_ENABLE_ZLIB)
   # CMAKE_BUILD_TYPE is only meaningful to single-configuration generators.
-  if(CMAKE_BUILD_TYPE)
-    string(TOUPPER ${CMAKE_BUILD_TYPE} build_type)
-    get_property(zlib_library TARGET ZLIB::ZLIB PROPERTY LOCATION_${build_type})
-  endif()
-  if(NOT zlib_library)
-    get_property(zlib_library TARGET ZLIB::ZLIB PROPERTY LOCATION)
-  endif()
-  get_library_name(${zlib_library} zlib_library)
-  set(llvm_system_libs ${llvm_system_libs} "${zlib_library}")
+  set(llvm_system_libs ${llvm_system_libs} ZLIB::ZLIB)
 endif()
 
 if(LLVM_ENABLE_ZSTD)
   # CMAKE_BUILD_TYPE is only meaningful to single-configuration generators.
-  if(CMAKE_BUILD_TYPE)
-    string(TOUPPER ${CMAKE_BUILD_TYPE} build_type)
-    get_property(zstd_library TARGET ${zstd_target} PROPERTY LOCATION_${build_type})
-  endif()
-  if(NOT zstd_library)
-    get_property(zstd_library TARGET ${zstd_target} PROPERTY LOCATION)
-  endif()
-  if (zstd_target STREQUAL zstd::libzstd_shared)
-    get_library_name(${zstd_library} zstd_library)
-    set(llvm_system_libs ${llvm_system_libs} "${zstd_library}")
-  else()
-    set(llvm_system_libs ${llvm_system_libs} "${zstd_STATIC_LIBRARY}")
-  endif()
+    set(llvm_system_libs ${llvm_system_libs} zstd::libzstd_static)
 endif()
 
 set_property(TARGET LLVMSupport PROPERTY LLVM_SYSTEM_LIBS "${llvm_system_libs}")
diff --git a/lib/WindowsManifest/CMakeLists.txt b/lib/WindowsManifest/CMakeLists.txt
index 910132a4c7de..cfbd11c1db8c 100644
--- a/lib/WindowsManifest/CMakeLists.txt
+++ b/lib/WindowsManifest/CMakeLists.txt
@@ -22,13 +22,5 @@ add_llvm_component_library(LLVMWindowsManifest
 # move to using CMake export, this block can be removed.
 if(LLVM_ENABLE_LIBXML2)
   # CMAKE_BUILD_TYPE is only meaningful to single-configuration generators.
-  if(CMAKE_BUILD_TYPE)
-    string(TOUPPER ${CMAKE_BUILD_TYPE} build_type)
-    get_property(libxml2_library TARGET LibXml2::LibXml2 PROPERTY LOCATION_${build_type})
-  endif()
-  if(NOT libxml2_library)
-    get_property(libxml2_library TARGET LibXml2::LibXml2 PROPERTY LOCATION)
-  endif()
-  get_library_name(${libxml2_library} libxml2_library)
-  set_property(TARGET LLVMWindowsManifest PROPERTY LLVM_SYSTEM_LIBS ${libxml2_library})
+  set_property(TARGET LLVMWindowsManifest PROPERTY LLVM_SYSTEM_LIBS LibXml2::LibXml2)
 endif()
