diff --git a/vpython3.bat b/vpython3.bat
index d40cca5..5173dc1 100644
--- a/vpython3.bat
+++ b/vpython3.bat
@@ -3,10 +3,4 @@
 :: Use of this source code is governed by a BSD-style license that can be
 :: found in the LICENSE file.
 
-:: TODO(crbug.com/1003139): Remove.
-:: Add Python 3 to PATH to work around crbug.com/1003139.
-for /f %%i in (%~dp0python3_bin_reldir.txt) do set PYTHON3_BIN_RELDIR=%%i
-set PATH=%~dp0%PYTHON3_BIN_RELDIR%;%~dp0%PYTHON3_BIN_RELDIR%\Scripts;%~dp0%PYTHON3_BIN_RELDIR%\DLLs;%PATH%
-
-call "%~dp0\cipd_bin_setup.bat" > nul 2>&1
-"%~dp0\.cipd_bin\vpython3.exe" -vpython-interpreter "%~dp0\%PYTHON3_BIN_RELDIR%\python3.exe" %*
+call python.exe %*
