diff --git a/CMakeLists.txt b/CMakeLists.txt
index 13c28b2..d58e394 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -14,6 +14,12 @@ endif ()
 
 if (YACLIB_CXX_STANDARD)
   set(CMAKE_CXX_STANDARD ${YACLIB_CXX_STANDARD})
+elseif (NOT CMAKE_CXX_STANDARD AND NOT CORO IN_LIST YACLIB_FLAGS)
+  message("SETTING DEFAULT 17")
+  set(CMAKE_CXX_STANDARD 17)
+else ()
+  message("SETTING DEFAULT 20")
+  set(CMAKE_CXX_STANDARD 20)
 endif ()
 
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
