diff --git a/ext/sctp/meson.build b/ext/sctp/meson.build
--- a/ext/sctp/meson.build
+++ b/ext/sctp/meson.build
@@ -12,9 +12,9 @@
 
 sctp_platform_deps = []
 
-found_system_usrsctp = false
+found_system_usrsctp = true
 
-if not get_option('sctp-internal-usrsctp').enabled()
+if false
   sctp_dep = cc.find_library('usrsctp', required: false)
   sctp_header = cc.has_header('usrsctp.h')
   if host_system == 'windows'
@@ -49,7 +49,8 @@
   sctp_args = []
 endif
 
-if sctp_dep.found() and sctp_header
+sctp_dep = dependency('usrsctp', required: true)
+if sctp_dep.found()
   gstsctp = library('gstsctp',
     sctp_sources,
     c_args : gst_plugins_bad_args + sctp_args,
diff --git a/ext/bz2/meson.build b/ext/bz2/meson.build
--- a/ext/bz2/meson.build
+++ b/ext/bz2/meson.build
@@ -4,7 +4,7 @@
   'gstbz2enc.c',
 ]
 
-bz2_dep = cc.find_library('bz2', required : get_option('bz2'))
+bz2_dep = dependency('bzip2', required : get_option('bz2'))
 
 if bz2_dep.found() and cc.has_header_symbol('bzlib.h', 'BZ2_bzlibVersion')
   gstbz2 = library('gstbz2',
diff --git a/ext/zxing/gstzxing.cpp b/ext/zxing/gstzxing.cpp
--- a/ext/zxing/gstzxing.cpp
+++ b/ext/zxing/gstzxing.cpp
@@ -60,9 +60,9 @@
 
 #include <gst/video/video.h>
 
-#include "ReadBarcode.h"
-#include "TextUtfEncoding.h"
-#include "ZXVersion.h"
+#include "ZXing/ReadBarcode.h"
+#include "ZXing/TextUtfEncoding.h"
+#include "ZXing/ZXVersion.h"
 
 using namespace ZXing;
 
diff --git a/ext/faac/meson.build b/ext/faac/meson.build
--- a/ext/faac/meson.build
+++ b/ext/faac/meson.build
@@ -1,8 +1,5 @@
-faac_dep = cc.find_library('faac', required : get_option('faac'))
-have_faac_h = cc.has_header_symbol('faac.h', 'faacEncOpen')
-if not have_faac_h and get_option('faac').enabled()
-  error('faac plugin enabled: found libfaac, but not faac.h')
-endif
+faac_dep = dependency('faac', required : get_option('faac'))
+have_faac_h = true
 
 if faac_dep.found() and have_faac_h
   gstfaac = library('gstfaac',
