--- a/src/osgEarthDrivers/lerc/CMakeLists.txt
+++ b/src/osgEarthDrivers/lerc/CMakeLists.txt
@@ -1,15 +1,6 @@
-include_directories(BEFORE ../../third_party/lerc/include )
-
 add_osgearth_plugin(
     TARGET osgdb_lerc
     SOURCES
-        ReaderWriterLERC.cpp
-        ../../third_party/lerc/src/LercLib/BitMask.cpp
-        ../../third_party/lerc/src/LercLib/BitStuffer2.cpp
-        ../../third_party/lerc/src/LercLib/Huffman.cpp
-        ../../third_party/lerc/src/LercLib/Lerc.cpp
-        ../../third_party/lerc/src/LercLib/Lerc_c_api_impl.cpp
-        ../../third_party/lerc/src/LercLib/Lerc2.cpp
-        ../../third_party/lerc/src/LercLib/RLE.cpp
-        ../../third_party/lerc/src/LercLib/Lerc1Decode/BitStuffer.cpp
-        ../../third_party/lerc/src/LercLib/Lerc1Decode/CntZImage.cpp )
\ No newline at end of file
+        ReaderWriterLERC.cpp)
+find_package(lerc REQUIRED CONFIG)
+target_link_libraries(osgdb_lerc PRIVATE lerc::lerc)
--- src/osgEarthDrivers/lerc/ReaderWriterLERC.cpp
+++ src/osgEarthDrivers/lerc/ReaderWriterLERC.cpp
@@ -66,7 +66,7 @@
         char* data = new char[length];
         fin.read(data, length);
 
-        uint32 infoArr[8];
+        uint32 infoArr[11];
 
         lerc_status hr(0);
 
@@ -84,6 +84,9 @@
         unsigned int numBands = infoArr[5];
         unsigned int nValidPixels = infoArr[6];
         unsigned int blobSize = infoArr[7];
+        unsigned int nMasks = infoArr[8];
+        unsigned int nDepth = infoArr[9];
+        unsigned int nUsesNoDataValue = infoArr[10];
 
         GLenum glDataType;
         int    sampleSize;
@@ -175,7 +178,7 @@
 
         // Decode the image
         unsigned int bandOffset = 0;
-        hr = lerc_decode((const unsigned char*)(data), length, 0, numDims, width, height, numBands, dataType, (void*)output);
+        hr = lerc_decode((const unsigned char*)(data), length, nMasks, 0, numDims, width, height, numBands, dataType, (void*)output);
         if (hr)
         {
             delete[]output;
@@ -267,6 +270,7 @@
 
         unsigned int numDims = 1;
         unsigned int numBands = 1;
+        unsigned int numMasks = 0;
         uint32 dataType;
         unsigned int sampleSize;
 
@@ -348,7 +352,7 @@
 
 
         hr = lerc_computeCompressedSize((void*)imageData,    // raw image data, row by row, band by band
-            dataType, numDims, width, height, numBands,
+            dataType, numDims, width, height, numBands, numMasks,
             0,
             maxZError,           // max coding error per pixel, or precision
             &numBytesNeeded);    // size of outgoing Lerc blob
@@ -362,7 +366,7 @@
         Byte* pLercBlob = new Byte[numBytesBlob];
 
         hr = lerc_encode((void*)imageData,    // raw image data, row by row, band by band
-            dataType, numDims, width, height, numBands,
+            dataType, numDims, width, height, numBands, numMasks,
             0,         // can give nullptr if all pixels are valid
             maxZError,           // max coding error per pixel, or precision
             pLercBlob,           // buffer to write to, function will fail if buffer too small
