diff --git a/src/osgEarthDrivers/lerc/CMakeLists.txt b/src/osgEarthDrivers/lerc/CMakeLists.txt
index e8d0f8c..44ffdb3 100644
--- a/src/osgEarthDrivers/lerc/CMakeLists.txt
+++ b/src/osgEarthDrivers/lerc/CMakeLists.txt
@@ -1,15 +1,12 @@
-include_directories(BEFORE ../../third_party/lerc/include )
+find_package(lerc REQUIRED)
 
 add_osgearth_plugin(
     TARGET osgdb_lerc
+
     SOURCES
         ReaderWriterLERC.cpp
-        ../../third_party/lerc/src/LercLib/BitMask.cpp
-        ../../third_party/lerc/src/LercLib/BitStuffer2.cpp
-        ../../third_party/lerc/src/LercLib/Huffman.cpp
-        ../../third_party/lerc/src/LercLib/Lerc.cpp
-        ../../third_party/lerc/src/LercLib/Lerc_c_api_impl.cpp
-        ../../third_party/lerc/src/LercLib/Lerc2.cpp
-        ../../third_party/lerc/src/LercLib/RLE.cpp
-        ../../third_party/lerc/src/LercLib/Lerc1Decode/BitStuffer.cpp
-        ../../third_party/lerc/src/LercLib/Lerc1Decode/CntZImage.cpp )
\ No newline at end of file
+
+    LIBRARIES
+        PRIVATE
+            lerc::lerc
+)
\ No newline at end of file
