--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -46,7 +46,7 @@ set(EVENT__LIBRARY_TYPE DEFAULT CACHE STRING
 
 project(libevent C)
 
-set(CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake/")
+list(APPEND CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake/")
 string(REGEX MATCH "SunOS" SOLARIS "${CMAKE_SYSTEM_NAME}")
 
 
@@ -823,14 +823,14 @@ if (NOT EVENT__DISABLE_OPENSSL)
 
     set(EVENT__HAVE_OPENSSL 1)
 
-    message(STATUS "OpenSSL include: ${OPENSSL_INCLUDE_DIR}")
-    message(STATUS "OpenSSL lib: ${OPENSSL_LIBRARIES}")
+    message(STATUS "OpenSSL include: ${OpenSSL_INCLUDE_DIRS}")
+    message(STATUS "OpenSSL lib: ${OpenSSL_LIBRARIES}")
 
-    include_directories(${OPENSSL_INCLUDE_DIR})
+    include_directories(${OpenSSL_INCLUDE_DIRS})
 
     list(APPEND SRC_OPENSSL bufferevent_openssl.c)
     list(APPEND HDR_PUBLIC include/event2/bufferevent_ssl.h)
-    list(APPEND LIB_APPS ${OPENSSL_LIBRARIES})
+    list(APPEND LIB_APPS ${OpenSSL_LIBRARIES})
 endif()
 
 if (NOT EVENT__DISABLE_THREAD_SUPPORT)
@@ -932,7 +932,7 @@ add_event_library(event_extra
 
 if (NOT EVENT__DISABLE_OPENSSL)
     add_event_library(event_openssl
-        LIBRARIES event_core_shared ${OPENSSL_LIBRARIES}
+        LIBRARIES event_core_shared ${OpenSSL_LIBRARIES}
         SOURCES ${SRC_OPENSSL})
 endif()
 
@@ -946,7 +946,6 @@ endif()
 # library exists for historical reasons; it contains the contents of
 # both libevent_core and libevent_extra. You shouldn’t use it; it may
 # go away in a future version of Libevent.
-add_event_library(event SOURCES ${SRC_CORE} ${SRC_EXTRA})
 
 set(WIN32_GETOPT)
 if (WIN32)
--- a/cmake/AddEventLibrary.cmake
+++ b/cmake/AddEventLibrary.cmake
@@ -127,7 +127,7 @@ macro(add_event_library LIB_NAME)
                 COMMAND ${CMAKE_COMMAND} -E create_symlink
                     "$<TARGET_FILE_NAME:${LIB_NAME}_shared>"
                     "${LIB_LINK_NAME}"
-                WORKING_DIRECTORY "lib")
+                WORKING_DIRECTORY "${CMAKE_BINARY_DIR}/lib")
         endif()
 
         list(APPEND LIBEVENT_SHARED_LIBRARIES "${LIB_NAME}_shared")
@@ -145,7 +145,7 @@ macro(add_event_library LIB_NAME)
         EXPORT LibeventTargets
         LIBRARY DESTINATION "lib" COMPONENT lib
         ARCHIVE DESTINATION "lib" COMPONENT lib
-        RUNTIME DESTINATION "lib" COMPONENT lib
+        RUNTIME DESTINATION "bin" COMPONENT lib
         PUBLIC_HEADER DESTINATION "include/event2"
         COMPONENT dev
     )
