diff --git a/external/googleapis/CMakeLists.txt b/external/googleapis/CMakeLists.txt
index 05d90c8..3059d5c 100644
--- a/external/googleapis/CMakeLists.txt
+++ b/external/googleapis/CMakeLists.txt
@@ -104,29 +104,30 @@ endforeach ()
 
 include(ExternalProject)
 
-externalproject_add(
-    googleapis_download
-    EXCLUDE_FROM_ALL ON
-    PREFIX "${PROJECT_BINARY_DIR}/external/googleapis"
-    URL ${GOOGLE_CLOUD_CPP_GOOGLEAPIS_URL}
-    URL_HASH SHA256=${GOOGLE_CLOUD_CPP_GOOGLEAPIS_SHA256}
-    PATCH_COMMAND
-        ""
-        # ~~~
-        # Scaffolding for patching googleapis after download. For example:
-        #   PATCH_COMMAND
-        #       patch
-        #       -p1
-        #       --input=/workspace/external/googleapis.patch
-        # NOTE: This should only be used while developing with a new
-        # protobuf message. No changes to `PATCH_COMMAND` should ever be
-        # committed to the main branch.
-        # ~~~
-    CONFIGURE_COMMAND ""
-    BUILD_COMMAND ""
-    INSTALL_COMMAND ""
-    BUILD_BYPRODUCTS ${EXTERNAL_GOOGLEAPIS_BYPRODUCTS}
-    LOG_DOWNLOAD OFF)
+# Provided by Conan as a requirement
+# externalproject_add(
+#     googleapis_download
+#     EXCLUDE_FROM_ALL ON
+#     PREFIX "${PROJECT_BINARY_DIR}/external/googleapis"
+#     URL ${GOOGLE_CLOUD_CPP_GOOGLEAPIS_URL}
+#     URL_HASH SHA256=${GOOGLE_CLOUD_CPP_GOOGLEAPIS_SHA256}
+#     PATCH_COMMAND
+#         ""
+#         # ~~~
+#         # Scaffolding for patching googleapis after download. For example:
+#         #   PATCH_COMMAND
+#         #       patch
+#         #       -p1
+#         #       --input=/workspace/external/googleapis.patch
+#         # NOTE: This should only be used while developing with a new
+#         # protobuf message. No changes to `PATCH_COMMAND` should ever be
+#         # committed to the main branch.
+#         # ~~~
+#     CONFIGURE_COMMAND ""
+#     BUILD_COMMAND ""
+#     INSTALL_COMMAND ""
+#     BUILD_BYPRODUCTS ${EXTERNAL_GOOGLEAPIS_BYPRODUCTS}
+#     LOG_DOWNLOAD OFF)
 
 # Sometimes (this happens often with vcpkg) protobuf is installed in a non-
 # standard directory. We need to find out where, and then add that directory to
@@ -176,7 +177,6 @@ function (external_googleapis_add_library proto)
 endfunction ()
 
 function (external_googleapis_set_version_and_alias short_name)
-    add_dependencies("google_cloud_cpp_${short_name}" googleapis_download)
     set_target_properties(
         "google_cloud_cpp_${short_name}"
         PROPERTIES EXPORT_NAME google-cloud-cpp::${short_name}
