diff --git a/include/h5pp/details/h5ppHid.h b/include/h5pp/details/h5ppHid.h
index b1f4989..8af0eb3 100644
--- a/include/h5pp/details/h5ppHid.h
+++ b/include/h5pp/details/h5ppHid.h
@@ -15,7 +15,7 @@ namespace h5pp::hid {
         virtual ~hid_base() = default;
         hid_base()          = default;
         // Use enable_if to avoid implicit conversion from hid_h5x and still have a non-explicit hid_t constructor
-        template<typename T, typename = std::enable_if_t<std::is_same_v<T, hid_t>>>
+        template<typename T, typename = std::enable_if_t<std::is_integral_v<T>>>
         hid_base(const T &other) {
             // constructor from hid_t
             if constexpr(zeroValueIsOK) {
@@ -66,7 +66,7 @@ namespace h5pp::hid {
             return *this;
         }
 
-        template<typename T, typename = std::enable_if_t<std::is_same_v<T, hid_t>>>
+        template<typename T, typename = std::enable_if_t<std::is_integral_v<T>>>
         hid_base &operator=(const T &rhs) {
             // Copy assignment from hid_t
             if constexpr(zeroValueIsOK) {

