diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9ec2b4a..877baa4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -71,8 +71,6 @@ enable_testing()
 
 
 add_subdirectory (sigc++)
-add_subdirectory (examples)
-add_subdirectory (tests)
 
 
 set (PROJECT_CMAKE_NAME		"${PROJECT_NAME}-3")
diff --git a/sigc++/CMakeLists.txt b/sigc++/CMakeLists.txt
index fa5a91a..a71ee6f 100644
--- a/sigc++/CMakeLists.txt
+++ b/sigc++/CMakeLists.txt
@@ -42,6 +42,8 @@ set (PROJECT_CMAKE_NAME ${PROJECT_NAME}-3)
 install (
     TARGETS ${SIGCPP_LIB_NAME}
     EXPORT "${PROJECT_CMAKE_NAME}Targets"
+    ARCHIVE DESTINATION "lib"
     LIBRARY DESTINATION "lib"
+    RUNTIME DESTINATION "bin"
     INCLUDES DESTINATION "${INCLUDE_INSTALL_DIR}"
 )
