diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7f72825..d9db6de 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,19 +1,10 @@
-project (kaitai_struct_cpp_stl_runtime CXX)
 cmake_minimum_required (VERSION 3.3)
+project (kaitai_struct_cpp_stl_runtime CXX)
 
 set (CMAKE_INCLUDE_CURRENT_DIR ON)
 
 find_package(ZLIB)
-
-find_path(ICONV_INCLUDE_DIRS iconv.h)
-mark_as_advanced(ICONV_INCLUDE_DIRS)
-find_library(ICONV_LIBRARIES NAMES libiconv libiconv-2 iconv)
-mark_as_advanced(ICONV_LIBRARIES)
-
-set(ICONV_FOUND FALSE)
-if(ICONV_INCLUDE_DIRS AND ICONV_LIBRARIES)
-    SET(ICONV_FOUND TRUE)
-endif()
+find_package(Iconv)
 
 set (HEADERS
     kaitai/kaitaistream.h
@@ -38,9 +29,8 @@ if (ZLIB_FOUND)
     target_compile_definitions(${PROJECT_NAME} PRIVATE -DKS_ZLIB)
 endif()
 
-if(ICONV_FOUND)
-    target_include_directories(${PROJECT_NAME} PRIVATE ${ICONV_INCLUDE_DIRS})
-    target_link_libraries(${PROJECT_NAME} PRIVATE ${ICONV_LIBRARIES})
+if(Iconv_FOUND)
+    target_link_libraries(${PROJECT_NAME} PRIVATE Iconv::Iconv)
 endif()
 
 if (STRING_ENCODING_TYPE STREQUAL "ICONV")
