diff --git a/CMakeLists.txt b/CMakeLists.txt
index 533815d..debc90e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -191,8 +191,6 @@ if(JPEGXL_STATIC)
   # so just linking all other libraries is fine.
   if (NOT MSVC AND NOT APPLE)
     set(CMAKE_FIND_LIBRARY_SUFFIXES .a)
-    set(CMAKE_EXE_LINKER_FLAGS
-        "${CMAKE_EXE_LINKER_FLAGS} -static -static-libgcc -static-libstdc++")
   endif()
 endif()  # JPEGXL_STATIC
 
@@ -331,7 +329,6 @@ include(GNUInstallDirs)
 # Separately build/configure testing frameworks and other third_party libraries
 # to allow disabling tests in those libraries.
 include(third_party/testing.cmake)
-add_subdirectory(third_party)
 # Copy the JXL license file to the output build directory.
 configure_file("${CMAKE_CURRENT_SOURCE_DIR}/LICENSE"
                ${PROJECT_BINARY_DIR}/LICENSE.jpeg-xl COPYONLY)
@@ -468,5 +465,3 @@ if (JPEGXL_ENABLE_PLUGINS)
 add_subdirectory(plugins)
 endif ()
 
-# Binary tools
-add_subdirectory(tools)
