diff --git a/lib/jpegli.cmake b/lib/jpegli.cmake
index 2dcbb8c..9c72e7e 100644
--- a/lib/jpegli.cmake
+++ b/lib/jpegli.cmake
@@ -41,7 +41,7 @@ set(JPEGLI_INTERNAL_SOURCES
 )
 
 set(JPEGLI_INTERNAL_LIBS
-  hwy
+  highway::highway
   jxl-static
   Threads::Threads
   ${ATOMICS_LIBRARIES}
@@ -55,7 +55,7 @@ target_include_directories(jpegli-static PUBLIC
   "$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}>"
   "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
   "$<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/include>"
-  "$<BUILD_INTERFACE:$<TARGET_PROPERTY:hwy,INTERFACE_INCLUDE_DIRECTORIES>>"
+  "$<BUILD_INTERFACE:$<TARGET_PROPERTY:highway::highway,INTERFACE_INCLUDE_DIRECTORIES>>"
 )
 target_include_directories(jpegli-static PUBLIC "${JPEG_INCLUDE_DIRS}")
 target_link_libraries(jpegli-static PUBLIC ${JPEGLI_INTERNAL_LIBS})
diff --git a/lib/jxl.cmake b/lib/jxl.cmake
index a8a96e1..f69384a 100644
--- a/lib/jxl.cmake
+++ b/lib/jxl.cmake
@@ -364,23 +364,22 @@ set(JPEGXL_INTERNAL_SOURCES_ENC
 
 set_source_files_properties(jxl/enc_fast_lossless.cc PROPERTIES COMPILE_FLAGS -O3)
 
+find_package(highway REQUIRED CONFIG)
+find_package(brotli REQUIRED CONFIG)
+
 set(JPEGXL_DEC_INTERNAL_LIBS
-  hwy
+  brotli::brotli
+  highway::highway
   Threads::Threads
   ${ATOMICS_LIBRARIES}
 )
 
-if (JPEGXL_ENABLE_TRANSCODE_JPEG OR JPEGXL_ENABLE_BOXES)
-list(APPEND JPEGXL_DEC_INTERNAL_LIBS brotlidec-static brotlicommon-static)
-endif()
-
 if(JPEGXL_ENABLE_PROFILER)
 list(APPEND JPEGXL_DEC_INTERNAL_LIBS jxl_profiler)
 endif()
 
 set(JPEGXL_INTERNAL_LIBS
   ${JPEGXL_DEC_INTERNAL_LIBS}
-  brotlienc-static
 )
 
 if (JPEGXL_ENABLE_SKCMS)
@@ -392,7 +391,8 @@ if (JPEGXL_ENABLE_SKCMS)
     list(APPEND JPEGXL_INTERNAL_LIBS skcms)
   endif ()
 else ()
-  list(APPEND JPEGXL_INTERNAL_LIBS lcms2)
+  find_package(lcms REQUIRED CONFIG)
+  list(APPEND JPEGXL_INTERNAL_LIBS lcms::lcms)
 endif ()
 
 if (JPEGXL_ENABLE_TRANSCODE_JPEG)
@@ -425,9 +425,9 @@ set_property(TARGET jxl_dec-obj PROPERTY POSITION_INDEPENDENT_CODE ON)
 target_include_directories(jxl_dec-obj PUBLIC
   "$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}>"
   "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
-  "$<BUILD_INTERFACE:$<TARGET_PROPERTY:hwy,INTERFACE_INCLUDE_DIRECTORIES>>"
-  "$<BUILD_INTERFACE:$<TARGET_PROPERTY:brotlicommon-static,INTERFACE_INCLUDE_DIRECTORIES>>"
-)
+  "$<BUILD_INTERFACE:$<TARGET_PROPERTY:highway::highway,INTERFACE_INCLUDE_DIRECTORIES>>"
+  "$<BUILD_INTERFACE:$<TARGET_PROPERTY:brotli::brotli,INTERFACE_INCLUDE_DIRECTORIES>>"
+  )
 target_compile_definitions(jxl_dec-obj PUBLIC
   ${OBJ_COMPILE_DEFINITIONS}
 )
@@ -443,9 +443,10 @@ set_property(TARGET jxl_enc-obj PROPERTY POSITION_INDEPENDENT_CODE ON)
 target_include_directories(jxl_enc-obj PUBLIC
   ${PROJECT_SOURCE_DIR}
   ${CMAKE_CURRENT_SOURCE_DIR}/include
-  $<TARGET_PROPERTY:hwy,INTERFACE_INCLUDE_DIRECTORIES>
-  $<TARGET_PROPERTY:brotlicommon-static,INTERFACE_INCLUDE_DIRECTORIES>
-)
+  "$<BUILD_INTERFACE:$<TARGET_PROPERTY:highway::highway,INTERFACE_INCLUDE_DIRECTORIES>>"
+  "$<BUILD_INTERFACE:$<TARGET_PROPERTY:lcms::lcms,INTERFACE_INCLUDE_DIRECTORIES>>"
+  "$<BUILD_INTERFACE:$<TARGET_PROPERTY:brotli::brotli,INTERFACE_INCLUDE_DIRECTORIES>>"
+  )
 target_compile_definitions(jxl_enc-obj PUBLIC
   ${OBJ_COMPILE_DEFINITIONS}
 )
@@ -453,17 +454,6 @@ if (JPEGXL_ENABLE_PROFILER)
 target_link_libraries(jxl_enc-obj PUBLIC jxl_profiler)
 endif()
 
-#TODO(lode): don't depend on CMS for the core library
-if (JPEGXL_ENABLE_SKCMS)
-  target_include_directories(jxl_enc-obj PRIVATE
-    $<TARGET_PROPERTY:skcms,INCLUDE_DIRECTORIES>
-  )
-else ()
-  target_include_directories(jxl_enc-obj PRIVATE
-    $<TARGET_PROPERTY:lcms2,INCLUDE_DIRECTORIES>
-  )
-endif ()
-
 # Generate version.h
 configure_file("jxl/version.h.in" "include/jxl/version.h")
 
@@ -482,9 +472,6 @@ set_target_properties(jxl_enc-obj PROPERTIES
   VISIBILITY_INLINES_HIDDEN 1
   DEFINE_SYMBOL JXL_INTERNAL_LIBRARY_BUILD
 )
-generate_export_header(jxl_enc-obj
-  BASE_NAME JXL
-  EXPORT_FILE_NAME include/jxl/jxl_export.h)
 target_include_directories(jxl_enc-obj PUBLIC
     ${CMAKE_CURRENT_BINARY_DIR}/include)
 
@@ -563,9 +550,7 @@ if (BUILD_SHARED_LIBS)
 
 # Public shared library.
 add_library(jxl SHARED ${JPEGXL_INTERNAL_OBJECTS})
-strip_static(JPEGXL_INTERNAL_SHARED_LIBS JPEGXL_INTERNAL_LIBS)
-target_link_libraries(jxl PUBLIC ${JPEGXL_COVERAGE_FLAGS})
-target_link_libraries(jxl PRIVATE ${JPEGXL_INTERNAL_SHARED_LIBS})
+target_link_libraries(jxl PUBLIC ${JPEGXL_COVERAGE_FLAGS} ${JPEGXL_INTERNAL_LIBS})
 # Shared library include path contains only the "include/" paths.
 target_include_directories(jxl PUBLIC
   "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
@@ -574,22 +559,8 @@ set_target_properties(jxl PROPERTIES
   VERSION ${JPEGXL_LIBRARY_VERSION}
   SOVERSION ${JPEGXL_LIBRARY_SOVERSION}
   LIBRARY_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}"
-  RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}")
-
-# Public shared decoder library.
-add_library(jxl_dec SHARED $<TARGET_OBJECTS:jxl_dec-obj>)
-strip_static(JPEGXL_DEC_INTERNAL_SHARED_LIBS JPEGXL_DEC_INTERNAL_LIBS)
-target_link_libraries(jxl_dec PUBLIC ${JPEGXL_COVERAGE_FLAGS})
-target_link_libraries(jxl_dec PRIVATE ${JPEGXL_DEC_INTERNAL_SHARED_LIBS})
-# Shared library include path contains only the "include/" paths.
-target_include_directories(jxl_dec PUBLIC
-  "${CMAKE_CURRENT_SOURCE_DIR}/include"
-  "${CMAKE_CURRENT_BINARY_DIR}/include")
-set_target_properties(jxl_dec PROPERTIES
-  VERSION ${JPEGXL_LIBRARY_VERSION}
-  SOVERSION ${JPEGXL_LIBRARY_SOVERSION}
-  LIBRARY_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}"
-  RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}")
+  RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}"
+  DEFINE_SYMBOL JXL_INTERNAL_LIBRARY_BUILD)
 
 # Check whether the linker support excluding libs
 set(LINKER_EXCLUDE_LIBS_FLAG "-Wl,--exclude-libs=ALL")
@@ -601,7 +572,7 @@ list(REMOVE_ITEM CMAKE_EXE_LINKER_FLAGS ${LINKER_EXCLUDE_LIBS_FLAG})
 # Add a jxl.version file as a version script to tag symbols with the
 # appropriate version number. This script is also used to limit what's exposed
 # in the shared library from the static dependencies bundled here.
-foreach(target IN ITEMS jxl jxl_dec)
+foreach(target IN ITEMS jxl)
   set_target_properties(${target} PROPERTIES
       LINK_DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/jxl/jxl.version)
   if(APPLE)
@@ -632,8 +603,14 @@ install(TARGETS jxl
 else()
 add_library(jxl ALIAS jxl-static)
 add_library(jxl_dec ALIAS jxl_dec-static)
+set_target_properties(jxl-static PROPERTIES
+  DEFINE_SYMBOL JXL_INTERNAL_LIBRARY_BUILD)
 endif()  # BUILD_SHARED_LIBS
 
+generate_export_header(jxl
+  BASE_NAME JXL
+  EXPORT_FILE_NAME include/jxl/jxl_export.h)
+
 # Add a pkg-config file for libjxl.
 set(JPEGXL_LIBRARY_REQUIRES
     "libhwy libbrotlicommon libbrotlienc libbrotlidec")
diff --git a/lib/jxl_profiler.cmake b/lib/jxl_profiler.cmake
index 8faa626..3bfa950 100644
--- a/lib/jxl_profiler.cmake
+++ b/lib/jxl_profiler.cmake
@@ -11,7 +11,7 @@ set(JPEGXL_PROFILER_SOURCES
 
 ### Static library.
 add_library(jxl_profiler STATIC ${JPEGXL_PROFILER_SOURCES})
-target_link_libraries(jxl_profiler PUBLIC hwy)
+target_link_libraries(jxl_profiler PUBLIC highway::highway)
 
 target_compile_options(jxl_profiler PRIVATE ${JPEGXL_INTERNAL_FLAGS})
 target_compile_options(jxl_profiler PUBLIC ${JPEGXL_COVERAGE_FLAGS})
