diff --git a/lib/jxl.cmake b/lib/jxl.cmake
index 72c07f4..0120b16 100644
--- a/lib/jxl.cmake
+++ b/lib/jxl.cmake
@@ -348,10 +348,12 @@ set(JPEGXL_INTERNAL_SOURCES_ENC
   jxl/progressive_split.h
 )
 
+find_package(highway REQUIRED CONFIG)
+find_package(brotli REQUIRED CONFIG)
+
 set(JPEGXL_DEC_INTERNAL_LIBS
-  brotlidec-static
-  brotlicommon-static
-  hwy
+  brotli::brotli
+  highway::highway
   Threads::Threads
   ${ATOMICS_LIBRARIES}
 )
@@ -362,7 +364,6 @@ endif()
 
 set(JPEGXL_INTERNAL_LIBS
   ${JPEGXL_DEC_INTERNAL_LIBS}
-  brotlienc-static
 )
 
 # strips the -static suffix from all the elements in LIST
@@ -383,7 +384,10 @@ if (JPEGXL_ENABLE_SKCMS)
     list(APPEND JPEGXL_INTERNAL_LIBS skcms)
   endif ()
 else ()
-  list(APPEND JPEGXL_INTERNAL_LIBS lcms2)
+  # pkg_check_modules(LIB_LCMS2 REQUIRED lcms2)
+  # list(APPEND JPEGXL_INTERNAL_LIBS ${LIB_LCMS2_LIBRARIES})
+  find_package(lcms REQUIRED CONFIG)
+  list(APPEND JPEGXL_INTERNAL_LIBS lcms::lcms)
 endif ()
 
 if (NOT JPEGXL_ENABLE_TRANSCODE_JPEG)
@@ -408,8 +412,8 @@ set_property(TARGET jxl_dec-obj PROPERTY POSITION_INDEPENDENT_CODE ON)
 target_include_directories(jxl_dec-obj PUBLIC
   "$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}>"
   "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
-  "$<BUILD_INTERFACE:$<TARGET_PROPERTY:hwy,INTERFACE_INCLUDE_DIRECTORIES>>"
-  "$<BUILD_INTERFACE:$<TARGET_PROPERTY:brotlicommon-static,INTERFACE_INCLUDE_DIRECTORIES>>"
+  "$<BUILD_INTERFACE:$<TARGET_PROPERTY:highway::highway,INTERFACE_INCLUDE_DIRECTORIES>>"
+  "$<BUILD_INTERFACE:$<TARGET_PROPERTY:brotli::brotli,INTERFACE_INCLUDE_DIRECTORIES>>"
 )
 target_compile_definitions(jxl_dec-obj PUBLIC
   ${OBJ_COMPILE_DEFINITIONS}
@@ -426,8 +430,8 @@ set_property(TARGET jxl_enc-obj PROPERTY POSITION_INDEPENDENT_CODE ON)
 target_include_directories(jxl_enc-obj PUBLIC
   ${PROJECT_SOURCE_DIR}
   ${CMAKE_CURRENT_SOURCE_DIR}/include
-  $<TARGET_PROPERTY:hwy,INTERFACE_INCLUDE_DIRECTORIES>
-  $<TARGET_PROPERTY:brotlicommon-static,INTERFACE_INCLUDE_DIRECTORIES>
+  "$<BUILD_INTERFACE:$<TARGET_PROPERTY:highway::highway,INTERFACE_INCLUDE_DIRECTORIES>>"
+  "$<BUILD_INTERFACE:$<TARGET_PROPERTY:lcms::lcms,INTERFACE_INCLUDE_DIRECTORIES>>"
 )
 target_compile_definitions(jxl_enc-obj PUBLIC
   ${OBJ_COMPILE_DEFINITIONS}
@@ -436,17 +440,6 @@ if (JPEGXL_ENABLE_PROFILER)
 target_link_libraries(jxl_enc-obj PUBLIC jxl_profiler)
 endif()
 
-#TODO(lode): don't depend on CMS for the core library
-if (JPEGXL_ENABLE_SKCMS)
-  target_include_directories(jxl_enc-obj PRIVATE
-    $<TARGET_PROPERTY:skcms,INCLUDE_DIRECTORIES>
-  )
-else ()
-  target_include_directories(jxl_enc-obj PRIVATE
-    $<TARGET_PROPERTY:lcms2,INCLUDE_DIRECTORIES>
-  )
-endif ()
-
 # Generate version.h
 configure_file("jxl/version.h.in" "include/jxl/version.h")
 
@@ -465,9 +458,6 @@ set_target_properties(jxl_enc-obj PROPERTIES
   VISIBILITY_INLINES_HIDDEN 1
   DEFINE_SYMBOL JXL_INTERNAL_LIBRARY_BUILD
 )
-generate_export_header(jxl_enc-obj
-  BASE_NAME JXL
-  EXPORT_FILE_NAME include/jxl/jxl_export.h)
 target_include_directories(jxl_enc-obj PUBLIC
     ${CMAKE_CURRENT_BINARY_DIR}/include)
 
@@ -546,9 +536,7 @@ if (BUILD_SHARED_LIBS)
 
 # Public shared library.
 add_library(jxl SHARED ${JPEGXL_INTERNAL_OBJECTS})
-strip_static(JPEGXL_INTERNAL_SHARED_LIBS JPEGXL_INTERNAL_LIBS)
 target_link_libraries(jxl PUBLIC ${JPEGXL_COVERAGE_FLAGS})
-target_link_libraries(jxl PRIVATE ${JPEGXL_INTERNAL_SHARED_LIBS})
 # Shared library include path contains only the "include/" paths.
 target_include_directories(jxl PUBLIC
   "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
@@ -557,13 +545,13 @@ set_target_properties(jxl PROPERTIES
   VERSION ${JPEGXL_LIBRARY_VERSION}
   SOVERSION ${JPEGXL_LIBRARY_SOVERSION}
   LIBRARY_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}"
-  RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}")
+  RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}"
+  DEFINE_SYMBOL JXL_INTERNAL_LIBRARY_BUILD)
 
 # Public shared decoder library.
 add_library(jxl_dec SHARED $<TARGET_OBJECTS:jxl_dec-obj>)
-strip_static(JPEGXL_DEC_INTERNAL_SHARED_LIBS JPEGXL_DEC_INTERNAL_LIBS)
 target_link_libraries(jxl_dec PUBLIC ${JPEGXL_COVERAGE_FLAGS})
-target_link_libraries(jxl_dec PRIVATE ${JPEGXL_DEC_INTERNAL_SHARED_LIBS})
+target_link_libraries(jxl_dec PRIVATE ${JPEGXL_DEC_INTERNAL_LIBS})
 # Shared library include path contains only the "include/" paths.
 target_include_directories(jxl_dec PUBLIC
   "${CMAKE_CURRENT_SOURCE_DIR}/include"
@@ -615,8 +603,14 @@ install(TARGETS jxl
 else()
 add_library(jxl ALIAS jxl-static)
 add_library(jxl_dec ALIAS jxl_dec-static)
+set_target_properties(jxl-static PROPERTIES
+  DEFINE_SYMBOL JXL_INTERNAL_LIBRARY_BUILD)
 endif()  # BUILD_SHARED_LIBS
 
+generate_export_header(jxl
+  BASE_NAME JXL
+  EXPORT_FILE_NAME include/jxl/jxl_export.h)
+
 # Add a pkg-config file for libjxl.
 set(JPEGXL_LIBRARY_REQUIRES
     "libhwy libbrotlicommon libbrotlienc libbrotlidec")
diff --git a/lib/jxl_profiler.cmake b/lib/jxl_profiler.cmake
index 8faa626..3bfa950 100644
--- a/lib/jxl_profiler.cmake
+++ b/lib/jxl_profiler.cmake
@@ -11,7 +11,7 @@ set(JPEGXL_PROFILER_SOURCES
 
 ### Static library.
 add_library(jxl_profiler STATIC ${JPEGXL_PROFILER_SOURCES})
-target_link_libraries(jxl_profiler PUBLIC hwy)
+target_link_libraries(jxl_profiler PUBLIC highway::highway)
 
 target_compile_options(jxl_profiler PRIVATE ${JPEGXL_INTERNAL_FLAGS})
 target_compile_options(jxl_profiler PUBLIC ${JPEGXL_COVERAGE_FLAGS})
