diff --git a/a/lib/CMakeLists.txt b/b/lib/CMakeLists.txt
index 2c98293..54ddc12 100644
--- a/a/lib/CMakeLists.txt
+++ b/b/lib/CMakeLists.txt
@@ -115,6 +115,10 @@ set(JPEGXL_COVERAGE_FLAGS
 endif()  # JPEGXL_ENABLE_COVERAGE
 endif()  #!MSVC
 
+find_package(brotli REQUIRED)
+find_package(highway REQUIRED)
+find_package(lcms REQUIRED)
+
 # The jxl library definition.
 include(jxl.cmake)
 
diff --git a/a/lib/jxl.cmake b/b/lib/jxl.cmake
index c95be2d..0a5fd8c 100644
--- a/a/lib/jxl.cmake
+++ b/b/lib/jxl.cmake
@@ -300,9 +300,10 @@ set(JPEGXL_INTERNAL_SOURCES_ENC
 )
 
 set(JPEGXL_DEC_INTERNAL_LIBS
-  brotlidec-static
-  brotlicommon-static
-  hwy
+  brotli::brotlidec
+  brotli::brotlicommon
+  highway::highway
+  lcms::lcms
 )
 
 if(JPEGXL_ENABLE_PROFILER)
@@ -311,7 +312,7 @@ endif()
 
 set(JPEGXL_INTERNAL_LIBS
   ${JPEGXL_DEC_INTERNAL_LIBS}
-  brotlienc-static
+  brotli::brotlienc
   Threads::Threads
 )
 
@@ -376,6 +377,7 @@ target_compile_definitions(jxl_enc-obj PUBLIC
 if (JPEGXL_ENABLE_PROFILER)
 target_link_libraries(jxl_enc-obj PUBLIC jxl_profiler)
 endif()
+target_link_libraries(jxl_enc-obj PUBLIC ${JPEGXL_INTERNAL_LIBS})
 
 # Headers for exporting/importing public headers
 include(GenerateExportHeader)
@@ -406,7 +408,7 @@ add_library(jxl_dec-static STATIC
   $<TARGET_OBJECTS:jxl_dec-obj>
 )
 target_link_libraries(jxl_dec-static
-  PUBLIC ${JPEGXL_COVERAGE_FLAGS} ${JPEGXL_DEC_INTERNAL_LIBS} hwy)
+  PUBLIC ${JPEGXL_COVERAGE_FLAGS} ${JPEGXL_DEC_INTERNAL_LIBS} highway::highway)
 target_include_directories(jxl_dec-static PUBLIC
   "${PROJECT_SOURCE_DIR}"
   "${CMAKE_CURRENT_SOURCE_DIR}/include"
@@ -421,7 +423,7 @@ add_library(jxl-static STATIC
   $<TARGET_OBJECTS:jxl_dec-obj>
 )
 target_link_libraries(jxl-static
-  PUBLIC ${JPEGXL_COVERAGE_FLAGS} ${JPEGXL_INTERNAL_LIBS} hwy)
+  PUBLIC ${JPEGXL_COVERAGE_FLAGS} ${JPEGXL_INTERNAL_LIBS} highway::highway)
 target_include_directories(jxl-static PUBLIC
   "${PROJECT_SOURCE_DIR}"
   "${CMAKE_CURRENT_SOURCE_DIR}/include"
diff --git a/a/lib/jxl_profiler.cmake b/b/lib/jxl_profiler.cmake
index 8faa626..3bfa950 100644
--- a/a/lib/jxl_profiler.cmake
+++ b/b/lib/jxl_profiler.cmake
@@ -11,7 +11,7 @@ set(JPEGXL_PROFILER_SOURCES
 
 ### Static library.
 add_library(jxl_profiler STATIC ${JPEGXL_PROFILER_SOURCES})
-target_link_libraries(jxl_profiler PUBLIC hwy)
+target_link_libraries(jxl_profiler PUBLIC highway::highway)
 
 target_compile_options(jxl_profiler PRIVATE ${JPEGXL_INTERNAL_FLAGS})
 target_compile_options(jxl_profiler PUBLIC ${JPEGXL_COVERAGE_FLAGS})
