diff --git a/CMakeLists.txt b/CMakeLists.txt
index 146d2b8..f83e119 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -52,7 +52,6 @@ if (CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
 endif ()
 
 if (BUILD_DROGON_SHARED)
-    set(BUILD_TRANTOR_SHARED TRUE)
     set(CMAKE_POSITION_INDEPENDENT_CODE TRUE)
     find_package(Threads)
     # set(BUILD_EXAMPLES FALSE)
@@ -110,7 +109,6 @@ target_include_directories(
     $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/orm_lib/inc>
     $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/nosql_lib/redis/inc>
     $<BUILD_INTERFACE:${PROJECT_BINARY_DIR}>
-    $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/trantor>
     $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/exports>
     $<INSTALL_INTERFACE:${INSTALL_INCLUDE_DIR}>)
 
@@ -120,8 +118,6 @@ if (WIN32)
         PRIVATE $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/third_party/mman-win32>)
 endif (WIN32)
 
-add_subdirectory(trantor)
-
 target_link_libraries(${PROJECT_NAME} PUBLIC trantor)
 
 if(${CMAKE_SYSTEM_NAME} STREQUAL "Haiku")
