diff --git a/CMakeLists.txt b/CMakeLists.txt
index ca6dff2..712e2a2 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -177,7 +177,8 @@ if (${CMAKE_SYSTEM_NAME} STREQUAL "Android")
 endif ()
 
 if(NEED_BOOST_FS)
-    find_package(Boost 1.49.0 COMPONENTS filesystem system REQUIRED)
+    # TODO: component specified find_package is always failed. Need to fix it.
+    find_package(Boost 1.49.0 REQUIRED)
     message(STATUS "Using Boost filesytem::path")
     message(STATUS "Boost include dir: " ${Boost_INCLUDE_DIR})
     include_directories(${BOOST_INCLUDE_DIRS})
