diff --git a/CMakeLists.txt b/CMakeLists.txt
index ca6dff2..df13d78 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -118,7 +118,8 @@ endif (WIN32)
 
 add_subdirectory(trantor)
 
-target_link_libraries(${PROJECT_NAME} PUBLIC trantor)
+find_package(Trantor REQUIRED)
+target_link_libraries(${PROJECT_NAME} PUBLIC Trantor::Trantor)
 
 if(${CMAKE_SYSTEM_NAME} STREQUAL "Haiku")
     target_link_libraries(${PROJECT_NAME} PRIVATE network)
