diff --git a/CMakeLists.txt b/CMakeLists.txt
index c346e60..e561e5c 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -206,9 +206,9 @@ else()
 endif()
 
 # jsoncpp
-find_package(Jsoncpp REQUIRED)
-target_link_libraries(${PROJECT_NAME} PUBLIC Jsoncpp_lib)
-list(APPEND INCLUDE_DIRS_FOR_DYNAMIC_VIEW ${JSONCPP_INCLUDE_DIRS})
+find_package(jsoncpp REQUIRED)
+target_link_libraries(${PROJECT_NAME} PUBLIC jsoncpp_lib)
+list(APPEND INCLUDE_DIRS_FOR_DYNAMIC_VIEW ${jsoncpp_INCLUDE_DIRS})
 
 # yamlcpp
 if(BUILD_YAML_CONFIG)
