diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5d92323..b971267 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -394,7 +394,7 @@     # Find sqlite3.
     find_package(SQLite3 QUIET)
     find_package(unofficial-sqlite3 QUIET)
     if (SQLite3_FOUND)
-        target_link_libraries(${PROJECT_NAME} PRIVATE SQLite3_lib)
+        target_link_libraries(${PROJECT_NAME} PRIVATE SQLite::SQLite3)
         set(DROGON_FOUND_SQLite3 TRUE)
     elseif (unofficial-sqlite3_FOUND)
         target_link_libraries(${PROJECT_NAME} PRIVATE unofficial::sqlite3::sqlite3)
