diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3158fa64c..fcb59d999 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -412,8 +412,6 @@ if (BUILD_MYSQL)
 
     if (DROGON_FOUND_MYSQL)
         message(STATUS "Ok! We find mariadb!")
-        include(CheckLibraryExists)
-        check_library_exists(${MYSQL_LIB_NAME} mysql_optionsv "" HAS_MYSQL_OPTIONSV)
         if (HAS_MYSQL_OPTIONSV)
             message(STATUS "Mariadb support mysql_optionsv")
             add_definitions(-DHAS_MYSQL_OPTIONSV)
