diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3158fa6..cb5004b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -398,17 +398,10 @@ endif (BUILD_POSTGRESQL)
 
 if (BUILD_MYSQL)
     # Find mysql, only mariadb client library is supported
-    find_package(MySQL QUIET)
-    find_package(unofficial-libmariadb QUIET)
-    if (MySQL_FOUND)
-        target_link_libraries(${PROJECT_NAME} PRIVATE MySQL_lib)
-        set(DROGON_FOUND_MYSQL TRUE)
-        set(MYSQL_LIB_NAME MySQL_lib)
-    elseif (unofficial-libmariadb_FOUND)
-        target_link_libraries(${PROJECT_NAME} PRIVATE unofficial::libmariadb)
-        set(DROGON_FOUND_MYSQL TRUE)
-        set(MYSQL_LIB_NAME unofficial::libmariadb)
-    endif ()
+    find_package(mariadb-connector-c QUIET)
+    set(DROGON_FOUND_MYSQL TRUE)
+    set(MYSQL_LIB_NAME mariadb-connector-c::mariadb-connector-c)
+    target_link_libraries(${PROJECT_NAME} PRIVATE mariadb-connector-c::mariadb-connector-c)
 
     if (DROGON_FOUND_MYSQL)
         message(STATUS "Ok! We find mariadb!")
