--- CMakeLists.txt	2019-12-05 00:18:57.000000000 +0100
+++ CMakeLists.txt	2019-12-10 17:54:40.000000000 +0100
@@ -52,14 +52,14 @@
 if (ENABLE_TESTS)
     find_package(Boost 1.59.0 COMPONENTS system unit_test_framework REQUIRED)
 else()
-    find_package(Boost 1.59.0 COMPONENTS system REQUIRED)
+    find_package(Boost)
 endif()
 include_directories(${Boost_INCLUDE_DIR})
 
 set(LIB_LIST
     ${CMAKE_THREAD_LIBS_INIT}
     ${ZLIB_LIBRARIES}
-    ${OPENSSL_LIBRARIES}
+    ${OpenSSL_LIBRARIES}
     ${Boost_LIBRARIES}
     ${CURL_LIBRARIES}
 )