diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index c8584f8..ad0ce59 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -21,3 +21,91 @@ target_link_libraries(co
     PUBLIC Threads::Threads
     PRIVATE ${CMAKE_DL_LIBS}
 )
+
+# Installation
+
+## library & public headers
+install(
+        TARGETS co
+        EXPORT coostExport
+        LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+        ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+        RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+)
+
+install(
+        DIRECTORY ${PROJECT_SOURCE_DIR}/include/co
+        DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
+)
+
+## license
+install(
+        FILES ${PROJECT_SOURCE_DIR}/LICENSE.md
+        DESTINATION ${CMAKE_INSTALL_DOCDIR}
+)
+
+## pkgconfig file
+if(WIN32)
+    string(APPEND CO_PKG_EXTRA_LIBS " -lws2_32")
+else()
+    if(UNIX AND NOT APPLE)
+        string(APPEND CO_PKG_EXTRA_LIBS " -lpthread")
+        if(NOT BUILD_SHARED_LIBS)
+            string(APPEND CO_PKG_EXTRA_LIBS " -ldl")
+        endif()
+    endif()
+endif()
+
+configure_file(
+        ${PROJECT_SOURCE_DIR}/cmake/coost.pc.in
+        coost.pc
+        @ONLY
+)
+
+install(
+        FILES ${CMAKE_CURRENT_BINARY_DIR}/coost.pc
+        DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig
+)
+
+## CMake config file
+set(CO_CMAKE_CONFIG_DEPS "find_dependency(Threads)\n")
+if(WITH_LIBCURL)
+    string(APPEND CO_CMAKE_CONFIG_DEPS "find_dependency(CURL)\n")
+endif()
+if(WITH_LIBCURL OR WITH_OPENSSL)
+    string(APPEND CO_CMAKE_CONFIG_DEPS "find_dependency(OpenSSL 1.1.0)\n")
+endif()
+
+configure_package_config_file(
+        ${PROJECT_SOURCE_DIR}/cmake/coostConfig.cmake.in
+        coostConfig.cmake
+        INSTALL_DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/coost
+)
+
+write_basic_package_version_file(
+        coostConfigVersion.cmake
+        VERSION ${PROJECT_VERSION}
+        COMPATIBILITY AnyNewerVersion
+)
+
+export(
+        EXPORT coostExport
+        NAMESPACE coost::
+        FILE coostTargets.cmake
+)
+
+install(
+        EXPORT coostExport
+        NAMESPACE coost::
+        FILE coostTargets.cmake
+        DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/coost
+)
+
+install(
+        FILES
+        ${CMAKE_CURRENT_BINARY_DIR}/coostConfig.cmake
+        ${CMAKE_CURRENT_BINARY_DIR}/coostConfigVersion.cmake
+        DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/coost
+)
+
+export(PACKAGE coost)
