--- a/tensorpipe/util/ringbuffer/consumer.h
+++ b/tensorpipe/util/ringbuffer/consumer.h
@@ -10,6 +10,11 @@
 
 #include <tensorpipe/util/ringbuffer/ringbuffer.h>
 
+#include <array>
+#include <cstring>
+#include <tuple>
+#include <utility>
+
 namespace tensorpipe {
 namespace util {
 namespace ringbuffer {
--- a/tensorpipe/util/ringbuffer/producer.h
+++ b/tensorpipe/util/ringbuffer/producer.h
@@ -10,6 +10,12 @@
 
 #include <tensorpipe/util/ringbuffer/ringbuffer.h>
 
+#include <algorithm>
+#include <array>
+#include <cstring>
+#include <tuple>
+#include <utility>
+
 namespace tensorpipe {
 namespace util {
 namespace ringbuffer {
