diff --git a/a/include/libhal/percentage.hpp b/b/include/libhal/percentage.hpp
index e570ee7..291a06c 100644
--- a/a/include/libhal/percentage.hpp
+++ b/b/include/libhal/percentage.hpp
@@ -73,7 +73,7 @@ public:
   {
     return hal::equals(m_value, p_other.value(), p_epsilon);
   }
-  constexpr bool operator<=>(const percentage_t& p_percent) const = default;
+  constexpr auto operator<=>(const percentage_t& p_percent) const = default;
   constexpr bool operator==(const percentage_t& p_percent) const
   {
     return equals(p_percent);
