--- a/urdf_parser/test/CMakeLists.txt
+++ b/urdf_parser/test/CMakeLists.txt
@@ -1,18 +1,8 @@
-include_directories(
-  ${CMAKE_CURRENT_SOURCE_DIR}/gtest/include
-  ${CMAKE_CURRENT_SOURCE_DIR}/gtest
-  ${CMAKE_CURRENT_SOURCE_DIR}
-)
-
-# Build gtest
-add_library(gtest STATIC gtest/src/gtest-all.cc)
-add_library(gtest_main STATIC gtest/src/gtest_main.cc)
-target_link_libraries(gtest_main gtest)
-target_compile_features(gtest PUBLIC cxx_std_11)
-
 execute_process(COMMAND cmake -E remove_directory ${CMAKE_BINARY_DIR}/test_results)
 execute_process(COMMAND cmake -E make_directory ${CMAKE_BINARY_DIR}/test_results)
 
+find_package(GTest REQUIRED)
+
 # unit test to fix geometry problems
 set(tests
      urdf_double_convert.cpp
@@ -27,8 +17,8 @@
   add_executable(${BINARY_NAME} ${GTEST_SOURCE_file})
 
   target_link_libraries(${BINARY_NAME}
-    gtest_main
-    gtest
+    GTest::gtest
+    GTest::gtest_main
     urdfdom_model
   )
   if (UNIX)
