--- Modules/ThirdParty/GDCM/src/gdcm/Source/MediaStorageAndFileFormat/CMakeLists.txt
+++ Modules/ThirdParty/GDCM/src/gdcm/Source/MediaStorageAndFileFormat/CMakeLists.txt
@@ -149,7 +149,8 @@
     )
 endif()
 if(GDCM_USE_SYSTEM_OPENJPEG)
-  include_directories(${OPENJPEG_INCLUDE_DIRS} )
+  include_directories(${OpenJPEG_INCLUDE_DIRS} )
+  link_libraries(${OpenJPEG_LIBRARIES})
 else()
   include_directories(
   "${GDCM_BINARY_DIR}/Utilities/gdcmopenjpeg"
--- Modules/ThirdParty/GDCM/src/gdcm/Utilities/CMakeLists.txt
+++ Modules/ThirdParty/GDCM/src/gdcm/Utilities/CMakeLists.txt
@@ -30,6 +30,9 @@
   set(OPENJPEG_INSTALL_INCLUDE_DIR  ${GDCM_INSTALL_INCLUDE_DIR}/gdcmopenjpeg)
   add_subdirectory(gdcmopenjpeg)
   mark_as_advanced(BUILD_JPIP BUILD_JPWL BUILD_CODEC BUILD_DOC BUILD_MJ2 BUILD_PKGCONFIG_FILES BUILD_SHARED_LIBS BUILD_THIRDPARTY OPJ_DATA_ROOT OPJ_DISABLE_TPSOT_FIX OPJ_USE_THREAD)
+else()
+  find_package(OpenJPEG REQUIRED)
+
 endif()
 
 # Do jpegls (JPEG-LS aka near lossless implementation)
