diff --git a/cmake/CompileProtos.cmake b/cmake/CompileProtos.cmake
index 4eb5ea3..9de47cd 100644
--- a/cmake/CompileProtos.cmake
+++ b/cmake/CompileProtos.cmake
@@ -315,6 +315,10 @@ include(GNUInstallDirs)
 
 # Install headers for a C++ proto library.
 function (google_cloud_cpp_install_proto_library_headers target)
+    get_target_property(type ${target} TYPE)
+    if ("${type}" STREQUAL "INTERFACE_LIBRARY")
+        return()
+    endif ()
     get_target_property(target_sources ${target} SOURCES)
     foreach (header ${target_sources})
         # Skip anything that is not a header file.
@@ -332,6 +336,10 @@ endfunction ()
 
 # Install protos for a C++ proto library.
 function (google_cloud_cpp_install_proto_library_protos target source_dir)
+    get_target_property(type ${target} TYPE)
+    if ("${type}" STREQUAL "INTERFACE_LIBRARY")
+        return()
+    endif ()
     get_target_property(target_protos ${target} PROTO_SOURCES)
     foreach (header ${target_protos})
         # Skip anything that is not a header file.
diff --git a/google/cloud/dialogflow_es/CMakeLists.txt b/google/cloud/dialogflow_es/CMakeLists.txt
index fad8716..cc44723 100644
--- a/google/cloud/dialogflow_es/CMakeLists.txt
+++ b/google/cloud/dialogflow_es/CMakeLists.txt
@@ -35,7 +35,12 @@ endif ()
 
 include(CompileProtos)
 add_library(google_cloud_cpp_dialogflow_es_protos INTERFACE)
-external_googleapis_set_version_and_alias(dialogflow_es_protos)
+set_target_properties(
+    google_cloud_cpp_dialogflow_es_protos
+    PROPERTIES EXPORT_NAME google-cloud-cpp::dialogflow_es_protos)
+add_library(google-cloud-cpp::dialogflow_es_protos ALIAS
+            google_cloud_cpp_dialogflow_es_protos)
+
 target_link_libraries(
     google_cloud_cpp_dialogflow_es_protos
     PUBLIC
