diff --git a/CMakeLists.txt b/CMakeLists.txt
index 236c6e1..f961398 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -68,7 +68,6 @@ elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC")
 endif ()
 
 list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake)
-include(SelectMSVCRuntime)
 
 option(GOOGLE_CLOUD_CPP_ENABLE_MACOS_OPENSSL_CHECK
        "If enabled, check that the user has defined OPENSSL_ROOT_DIR on macOS"
diff --git a/cmake/GoogleCloudCppCommon.cmake b/cmake/GoogleCloudCppCommon.cmake
index 057fe6b..dbffd7b 100644
--- a/cmake/GoogleCloudCppCommon.cmake
+++ b/cmake/GoogleCloudCppCommon.cmake
@@ -21,9 +21,6 @@ get_filename_component(GOOGLE_CLOUD_CPP_SUBPROJECT
 # Get the destination directories based on the GNU recommendations.
 include(GNUInstallDirs)
 
-# Pick the right MSVC runtime libraries.
-include(SelectMSVCRuntime)
-
 # Enable Werror
 include(EnableWerror)
 
diff --git a/examples/CMakeLists.txt b/examples/CMakeLists.txt
index 0cb7a9a..d9016a0 100644
--- a/examples/CMakeLists.txt
+++ b/examples/CMakeLists.txt
@@ -14,9 +14,6 @@
 # limitations under the License.
 # ~~~
 
-# Pick the right MSVC runtime libraries.
-include(SelectMSVCRuntime)
-
 add_executable(gcs2cbt gcs2cbt.cc)
 target_link_libraries(gcs2cbt google-cloud-cpp::bigtable
                       google-cloud-cpp::storage google-cloud-cpp::grpc_utils)
diff --git a/external/googleapis/CMakeLists.txt b/external/googleapis/CMakeLists.txt
index 28bf016..c281ce8 100644
--- a/external/googleapis/CMakeLists.txt
+++ b/external/googleapis/CMakeLists.txt
@@ -103,8 +103,6 @@ if (PROTO_INCLUDE_DIR)
     list(INSERT PROTOBUF_IMPORT_DIRS 0 "${PROTO_INCLUDE_DIR}")
 endif ()
 
-include(SelectMSVCRuntime)
-
 google_cloud_cpp_add_protos_property()
 
 function (external_googleapis_short_name var proto)
