diff --git a/CMakeLists.txt b/CMakeLists.txt
index f3b6cbd..f3e8993 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -100,7 +100,7 @@ IF(MSVC)
   ADD_DEFINITIONS(-DWIN32_LEAN_AND_MEAN)
   IF (MSVC)
     # Treat warnings as errors
-    SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -WX")
+    # SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -WX")
     SET(CONFIG_TYPES "DEBUG" "RELEASE" "RELWITHDEBINFO")
     FOREACH(BUILD_TYPE ${CONFIG_TYPES})
       FOREACH(COMPILER CXX C)
