diff --git a/cmake/InstallDependencies.cmake b/cmake/InstallDependencies.cmake
index 8171db3..33b0a00 100644
--- a/cmake/InstallDependencies.cmake
+++ b/cmake/InstallDependencies.cmake
@@ -1,12 +1,4 @@
 if(REDUCT_CPP_USE_CONAN)
-    message(STATUS "Using Conan to fetch dependencies")
-    find_program(CONAN conan)
-    if(NOT CONAN)
-        message(
-            FATAL_ERROR
-            "Conan not found. Please install Conan and run cmake again"
-        )
-    endif()
 
     find_package(fmt REQUIRED)
     find_package(nlohmann_json REQUIRED)
