diff --git a/src/reduct/client.cc b/src/reduct/client.cc
index 5153d84..2c81cf2 100644
--- a/src/reduct/client.cc
+++ b/src/reduct/client.cc
@@ -2,7 +2,6 @@
 
 #include "reduct/client.h"
 
-#include <date/date.h>
 #include <fmt/core.h>
 #include <nlohmann/json.hpp>
 
@@ -57,7 +56,7 @@ class Client : public IClient {
             .fingerprint = license.at("fingerprint"),
         };
         std::istringstream(license.at("expiry_date").get<std::string>()) >>
-            date::parse("%FT%TZ", server_info.license->expiry_date);
+            std::chrono::parse("%FT%TZ", server_info.license->expiry_date);
       }
 
       return {
@@ -142,7 +141,7 @@ class Client : public IClient {
       token_list.reserve(json_tokens.size());
       for (const auto& token : json_tokens) {
         Time created_at;
-        std::istringstream(token.at("created_at").get<std::string>()) >> date::parse("%FT%TZ", created_at);
+        std::istringstream(token.at("created_at").get<std::string>()) >> std::chrono::parse("%FT%TZ", created_at);
 
         token_list.push_back(Token{
             .name = token.at("name"),
diff --git a/src/reduct/internal/serialisation.cc b/src/reduct/internal/serialisation.cc
index 01f19e1..f9ef21c 100644
--- a/src/reduct/internal/serialisation.cc
+++ b/src/reduct/internal/serialisation.cc
@@ -1,8 +1,9 @@
 // Copyright 2022 Alexey Timin
 
-#include "reduct/internal/serialisation.h"
-
 #include <date/date.h>
+using date::from_stream;
+
+#include "reduct/internal/serialisation.h"
 
 namespace reduct::internal {
 
@@ -65,7 +66,7 @@ Result<IBucket::Settings> ParseBucketSettings(const nlohmann::json& json) {
 
 Result<IClient::FullTokenInfo> ParseTokenInfo(const nlohmann::json& json) {
   IClient::Time created_at;
-  std::istringstream(json.at("created_at").get<std::string>()) >> date::parse("%FT%TZ", created_at);
+  std::istringstream(json.at("created_at").get<std::string>()) >> std::chrono::parse("%FT%TZ", created_at);
 
   return {
       IClient::FullTokenInfo{
