--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -168,7 +168,7 @@ endif()
 include_directories(${OPENGL_INCLUDE_DIR} ${OPENAL_INCLUDE_DIR})
 set(LIBS_PRIVATE ${LIBS_PRIVATE} ${OPENAL_LIBRARY})
 include(LibraryPathToLinkerFlags)
-library_path_to_linker_flags(__PKG_CONFIG_LIBS_PRIVATE "${LIBS_PRIVATE}")
+# library_path_to_linker_flags(__PKG_CONFIG_LIBS_PRIVATE "${LIBS_PRIVATE}")
 
 if(STATIC)
   MESSAGE(STATUS "Building raylib static library")
