diff --git a/cmake/LibraryConfigurations.cmake b/cmake/LibraryConfigurations.cmake
index c156d6b3..608fe531 100644
--- a/cmake/LibraryConfigurations.cmake
+++ b/cmake/LibraryConfigurations.cmake
@@ -42,7 +42,7 @@ elseif (${PLATFORM} MATCHES "Android")
     set(CMAKE_POSITION_INDEPENDENT_CODE ON)
     
     add_definitions(-DANDROID -D__ANDROID_API__=21)
-    include_directories(external/android/native_app_glue)
+    include_directories(${ANDROID_NDK}/sources/android/native_app_glue)
     set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--exclude-libs,libatomic.a -Wl,--build-id -Wl,-z,noexecstack -Wl,-z,relro -Wl,-z,now -Wl,--warn-shared-textrel -Wl,--fatal-warnings -uANativeActivity_onCreate")
     
     find_library(OPENGL_LIBRARY OpenGL)
