diff --git a/CMakeLists.txt b/CMakeLists.txt
index ed62385..acdb38c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -10,6 +10,7 @@ set(MD_VERSION "${MD_VERSION_MAJOR}.${MD_VERSION_MINOR}.${MD_VERSION_RELEASE}")
 set(PROJECT_VERSION "${MD_VERSION}")
 set(PROJECT_URL "https://github.com/mity/md4c")
 
+if(0)
 if(WIN32)
     # On Windows, given there is no standard lib install dir etc., we rather
     # by default build static lib.
@@ -19,6 +20,7 @@ else()
     # shared lib.
     option(BUILD_SHARED_LIBS "help string describing option" ON)
 endif()
+endif()
 
 add_definitions(
     -DMD_VERSION_MAJOR=${MD_VERSION_MAJOR}
