diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3d4418ad61..6d3d81c07b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -40,7 +40,6 @@ project(${PACKAGE_NAME} LANGUAGES C CXX)
 set(gRPC_INSTALL_BINDIR "bin" CACHE STRING "Installation directory for executables")
 set(gRPC_INSTALL_LIBDIR "lib" CACHE STRING "Installation directory for libraries")
 set(gRPC_INSTALL_INCLUDEDIR "include" CACHE STRING "Installation directory for headers")
-set(gRPC_INSTALL_CMAKEDIR "lib/cmake/${PACKAGE_NAME}" CACHE STRING "Installation directory for cmake config files")
 set(gRPC_INSTALL_SHAREDIR "share/grpc" CACHE STRING "Installation directory for root certificates")
 
 # Options
@@ -18334,100 +18333,3 @@ target_link_libraries(uri_fuzzer_test_one_entry
 
 
 endif()
-
-
-
-
-
-
-
-if(gRPC_INSTALL)
-  install(EXPORT gRPCTargets
-    DESTINATION ${gRPC_INSTALL_CMAKEDIR}
-    NAMESPACE gRPC::
-  )
-endif()
-
-include(CMakePackageConfigHelpers)
-
-configure_file(cmake/gRPCConfig.cmake.in
-  gRPCConfig.cmake @ONLY)
-write_basic_package_version_file(${CMAKE_CURRENT_BINARY_DIR}/gRPCConfigVersion.cmake
-  VERSION ${PACKAGE_VERSION}
-  COMPATIBILITY AnyNewerVersion)
-install(FILES
-    ${CMAKE_CURRENT_BINARY_DIR}/gRPCConfig.cmake
-    ${CMAKE_CURRENT_BINARY_DIR}/gRPCConfigVersion.cmake
-  DESTINATION ${gRPC_INSTALL_CMAKEDIR}
-)
-
-install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/etc/roots.pem
-  DESTINATION ${gRPC_INSTALL_SHAREDIR})
-
-# Function to generate pkg-config files.
-function(generate_pkgconfig name description version requires
-                            libs libs_private output_filename)
-  set(PC_NAME "${name}")
-  set(PC_DESCRIPTION "${description}")
-  set(PC_VERSION "${version}")
-  set(PC_REQUIRES "${requires}")
-  set(PC_LIB "${libs}")
-  set(PC_LIBS_PRIVATE "${libs_private}")
-  set(output_filepath "${grpc_BINARY_DIR}/libs/opt/pkgconfig/${output_filename}")
-  configure_file(
-    "${grpc_SOURCE_DIR}/cmake/pkg-config-template.pc.in"
-    "${output_filepath}"
-    @ONLY)
-  install(FILES "${output_filepath}"
-    DESTINATION "lib/pkgconfig/")
-endfunction()
-
-# gpr .pc file
-generate_pkgconfig(
-  "gpr"
-  "gRPC platform support library"
-  "${gRPC_CORE_VERSION}"
-  ""
-  "-lgpr"
-  ""
-  "gpr.pc")
-
-# grpc .pc file
-generate_pkgconfig(
-  "gRPC"
-  "high performance general RPC framework"
-  "${gRPC_CORE_VERSION}"
-  "gpr openssl"
-  "-lgrpc -laddress_sorting -lupb -lcares -lz"
-  ""
-  "grpc.pc")
-
-# grpc_unsecure .pc file
-generate_pkgconfig(
-  "gRPC unsecure"
-  "high performance general RPC framework without SSL"
-  "${gRPC_CORE_VERSION}"
-  "gpr"
-  "-lgrpc_unsecure"
-  ""
-  "grpc_unsecure.pc")
-
-# grpc++ .pc file
-generate_pkgconfig(
-  "gRPC++"
-  "C++ wrapper for gRPC"
-  "${PACKAGE_VERSION}"
-  "grpc"
-  "-lgrpc++"
-  ""
-  "grpc++.pc")
-
-# grpc++_unsecure .pc file
-generate_pkgconfig(
-  "gRPC++ unsecure"
-  "C++ wrapper for gRPC without SSL"
-  "${PACKAGE_VERSION}"
-  "grpc_unsecure"
-  "-lgrpc++_unsecure"
-  ""
-  "grpc++_unsecure.pc")
-- 
2.28.0

