diff -Naur a/Makefile b/Makefile
--- a/Makefile	2014-10-31 13:51:20.000000000 +0300
+++ b/Makefile	2019-05-22 11:28:06.000000000 +0300
@@ -89,7 +89,7 @@
 	$(MAKE) win32releaseconf
 
 install: $(PREFIX) all
-	cp --no-dereference $(STAGE)/libax* $(PREFIX)/lib
+	cp -RP $(STAGE)/libax* $(PREFIX)/lib
 	chmod 755 $(PREFIX)/lib/libax* 
 ifdef CONFIG_SAMPLES
 	install -m 755 $(STAGE)/ax* $(PREFIX)/bin 
diff -Naur a/axtlswrap/Makefile b/axtlswrap/Makefile
--- a/axtlswrap/Makefile	2014-10-27 13:30:46.000000000 +0300
+++ b/axtlswrap/Makefile	2019-05-22 11:28:06.000000000 +0300
@@ -58,7 +58,7 @@
 $(TARGET): $(OBJ) $(AXTLS_HOME)/$(STAGE)/libaxtls.a
 	$(LD) $(LDFLAGS) -o $@ $(OBJ) $(LIBS)
 ifdef CONFIG_STRIP_UNWANTED_SECTIONS
-	$(STRIP) --remove-section=.comment $(TARGET)
+	$(STRIP)  $(TARGET)
 endif
 
 endif   # CONFIG_AXTLSWRAP
diff -Naur a/config/Rules.mak b/config/Rules.mak
--- a/config/Rules.mak	2014-10-27 13:30:48.000000000 +0300
+++ b/config/Rules.mak	2019-05-22 11:28:06.000000000 +0300
@@ -174,7 +174,7 @@
 else
     CFLAGS+=$(WARNINGS) $(OPTIMIZATIONS) -D_GNU_SOURCE -DNDEBUG
     LDFLAGS += -Wl,-warn-common
-    STRIPCMD:=$(STRIP) -s --remove-section=.note --remove-section=.comment
+    STRIPCMD:=$(STRIP) -s --remove-section=.note 
 endif
 ifeq ($(strip $(CONFIG_STATIC)),y)
     LDFLAGS += --static
diff -Naur a/crypto/os_int.h b/crypto/os_int.h
--- a/crypto/os_int.h	2015-03-09 07:32:51.000000000 +0300
+++ b/crypto/os_int.h	2019-05-22 11:28:06.000000000 +0300
@@ -56,7 +56,7 @@
 #include <inttypes.h>
 #else
 #include <stdint.h>
-#include <endian.h>
+#include <machine/endian.h>
 #endif /* Not Solaris */
 
 #endif /* Not Win32 */
diff -Naur a/httpd/Makefile b/httpd/Makefile
--- a/httpd/Makefile	2014-10-27 13:30:44.000000000 +0300
+++ b/httpd/Makefile	2019-05-22 11:28:06.000000000 +0300
@@ -97,13 +97,13 @@
 $(TARGET): $(OBJ) $(AXTLS_HOME)/$(STAGE)/libaxtls.a
 	$(LD) $(LDFLAGS) -o $@ $(OBJ) $(LIBS)
 ifdef CONFIG_STRIP_UNWANTED_SECTIONS
-	$(STRIP) --remove-section=.comment $(TARGET)
+	$(STRIP)  $(TARGET)
 endif
 
 $(TARGET2): htpasswd.o $(AXTLS_HOME)/$(STAGE)/libaxtls.a
 	$(LD) $(LDFLAGS) -o $@ htpasswd.o $(LIBS)
 ifdef CONFIG_STRIP_UNWANTED_SECTIONS
-	$(STRIP) --remove-section=.comment $(TARGET2)
+	$(STRIP)  $(TARGET2)
 endif
 
 else    # Win32
diff -Naur a/samples/c/Makefile b/samples/c/Makefile
--- a/samples/c/Makefile	2014-10-27 13:30:44.000000000 +0300
+++ b/samples/c/Makefile	2019-05-22 11:28:06.000000000 +0300
@@ -61,7 +61,7 @@
 $(TARGET): $(OBJ) $(LIBS)/libaxtls.a
 	$(LD) $(LDFLAGS) -o $@ $(OBJ) -L$(LIBS) -laxtls 
 ifdef CONFIG_STRIP_UNWANTED_SECTIONS
-	$(STRIP) --remove-section=.comment $(TARGET)
+	$(STRIP)  $(TARGET)
 endif   # use strip
 else    # Win32
 
diff -Naur a/ssl/Makefile b/ssl/Makefile
--- a/ssl/Makefile	2015-03-10 04:20:25.000000000 +0300
+++ b/ssl/Makefile	2019-05-22 11:28:06.000000000 +0300
@@ -44,7 +44,7 @@
 
 ifndef CONFIG_PLATFORM_WIN32
 TARGET1=$(AXTLS_HOME)/$(STAGE)/libaxtls.a
-BASETARGET=libaxtls.so
+BASETARGET=libaxtls.dylib
 CRYPTO_PATH=$(AXTLS_HOME)/crypto/
 ifdef CONFIG_PLATFORM_CYGWIN
 TARGET2=$(AXTLS_HOME)/$(STAGE)/libaxtls.dll.a
@@ -53,8 +53,8 @@
 endif
 
 # shared library major/minor numbers
-LIBMAJOR=$(BASETARGET).1
-LIBMINOR=$(BASETARGET).1.2
+LIBMAJOR=libaxtls.1.dylib
+LIBMINOR=libaxtls.1.2.dylib
 else
 TARGET1=$(AXTLS_HOME)/$(STAGE)/axtls.lib
 TARGET2=$(AXTLS_HOME)/$(STAGE)/axtls.dll
@@ -98,7 +98,7 @@
 
 $(TARGET2) : $(CRYPTO_OBJ) $(OBJ)
 ifndef CONFIG_PLATFORM_CYGWIN
-	$(LD) $(LDFLAGS) $(LDSHARED) -Wl,-soname,$(LIBMAJOR) -o $(AXTLS_HOME)/$(STAGE)/$(LIBMINOR) $(CRYPTO_OBJ) $(OBJ)
+	$(LD) $(LDFLAGS) $(LDSHARED) -Wl,-install_name,$(LIBMAJOR) -o $(AXTLS_HOME)/$(STAGE)/$(LIBMINOR) $(CRYPTO_OBJ) $(OBJ)
 	cd $(AXTLS_HOME)/$(STAGE); ln -sf $(LIBMINOR) $(LIBMAJOR); ln -sf $(LIBMAJOR) $(BASETARGET); cd -
 else
 	$(LD) $(LDFLAGS) $(LDSHARED) -o $(AXTLS_HOME)/$(STAGE)/cygaxtls.dll \
diff -Naur a/ssl/os_port.h b/ssl/os_port.h
--- a/ssl/os_port.h	2015-04-30 08:53:33.000000000 +0300
+++ b/ssl/os_port.h	2019-05-22 11:28:06.000000000 +0300
@@ -136,7 +136,7 @@
 #include <sys/wait.h>
 #include <netinet/in.h>
 #include <arpa/inet.h>
-#include <asm/byteorder.h>
+#include <architecture/byte_order.h>
 
 #define SOCKET_READ(A,B,C)      read(A,B,C)
 #define SOCKET_WRITE(A,B,C)     write(A,B,C)
@@ -144,7 +144,7 @@
 #define TTY_FLUSH()
 
 #ifndef be64toh
-#define be64toh(x) __be64_to_cpu(x)
+#define be64toh(x) OSSwapBigToHostInt64(x)
 #endif
 
 #endif  /* Not Win32 */
