diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4a38212..bb851de 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -152,9 +152,7 @@ if (NOT MSVC)
   set (libdir ${INSTALL_LIB_DIR})
   set (includedir ${INSTALL_INCLUDE_DIR})
   set (VERSION ${LIBCMAES_VERSION_STRING})
-  configure_file (libcmaes.pc.in libcmaes.pc @ONLY)
-  install (FILES ${CMAKE_CURRENT_BINARY_DIR}/libcmaes.pc
-           DESTINATION ${RELATIVE_INSTALL_LIB_DIR}/pkgconfig)
+  
 endif ()
 
 # ---------- subdirectories ----------
@@ -174,23 +172,4 @@ endif ()
 # ------------ export ------------
 add_library (libcmaes::cmaes ALIAS cmaes)
 
-include (CMakePackageConfigHelpers)
-write_basic_package_version_file (
-  libcmaesConfigVersion.cmake VERSION ${PACKAGE_VERSION}
-  COMPATIBILITY AnyNewerVersion)
-
-# export target to build directory
-export (TARGETS cmaes NAMESPACE libcmaes:: FILE libcmaesTargets.cmake)
-# export target on install
-install (
-  EXPORT libcmaesTargets
-  FILE libcmaesTargets.cmake
-  NAMESPACE libcmaes::
-  DESTINATION ${RELATIVE_INSTALL_CMAKE_DIR})
-
-configure_file (libcmaesConfig.cmake.in libcmaesConfig.cmake @ONLY)
-install (
-  FILES "${CMAKE_CURRENT_BINARY_DIR}/libcmaesConfig.cmake"
-        "${CMAKE_CURRENT_BINARY_DIR}/libcmaesConfigVersion.cmake"
-        "${CMAKE_CURRENT_SOURCE_DIR}/cmake/FindEigen3.cmake"
-  DESTINATION ${RELATIVE_INSTALL_CMAKE_DIR})
+
