diff --git a/cmake/utilities.cmake b/cmake/utilities.cmake
index 9b78002..9c7bf65 100644
--- a/cmake/utilities.cmake
+++ b/cmake/utilities.cmake
@@ -148,7 +148,7 @@ macro(do_packaging)
             Devel)
     endif()
     if (MSVC AND BUILD_SHARED_LIBS)
-        install (FILES $<TARGET_PDB_FILE:${PROJECT_NAME}> DESTINATION ${CMAKE_INSTALL_BINDIR} OPTIONAL)
+        # install (FILES $<TARGET_PDB_FILE:${PROJECT_NAME}> DESTINATION ${CMAKE_INSTALL_BINDIR} OPTIONAL)
     endif()
 endmacro()
 
