diff --git a/cmake/sdks.cmake b/cmake/sdks.cmake
index 975281faaa..ebedbcc6e5 100644
--- a/cmake/sdks.cmake
+++ b/cmake/sdks.cmake
@@ -236,7 +236,7 @@ function(add_sdks)
 
     # the catch-all config needs to list all the targets in a dependency-sorted order
     include(dependencies)
-    sort_links(EXPORTS)
+    # sort_links(EXPORTS)
 
     # make an everything config by just including all the individual configs
     file(WRITE ${CMAKE_BINARY_DIR}/aws-sdk-cpp-config.cmake "")
