diff --git a/aws-cpp-sdk-core/CMakeLists.txt b/aws-cpp-sdk-core/CMakeLists.txt
index 17b4e46442..a15e30c980 100644
--- a/aws-cpp-sdk-core/CMakeLists.txt
+++ b/aws-cpp-sdk-core/CMakeLists.txt
@@ -63,12 +63,8 @@ file(GLOB UTILS_STL_HEADERS "include/aws/core/utils/memory/stl/*.h")
 file(GLOB UTILS_LOGGING_HEADERS "include/aws/core/utils/logging/*.h")
 file(GLOB UTILS_RATE_LIMITER_HEADERS "include/aws/core/utils/ratelimiter/*.h")
 file(GLOB UTILS_STREAM_HEADERS "include/aws/core/utils/stream/*.h")
-file(GLOB CJSON_HEADERS "include/aws/core/external/cjson/*.h")
-file(GLOB TINYXML2_HEADERS "include/aws/core/external/tinyxml2/tinyxml2.h")
 
 file(GLOB AWS_SOURCE "${CMAKE_CURRENT_SOURCE_DIR}/source/*.cpp")
-file(GLOB AWS_TINYXML2_SOURCE "${CMAKE_CURRENT_SOURCE_DIR}/source/external/tinyxml2/*.cpp")
-file(GLOB CJSON_SOURCE "${CMAKE_CURRENT_SOURCE_DIR}/source/external/cjson/*.cpp")
 file(GLOB AWS_AUTH_SOURCE "${CMAKE_CURRENT_SOURCE_DIR}/source/auth/*.cpp")
 file(GLOB AWS_CLIENT_SOURCE "${CMAKE_CURRENT_SOURCE_DIR}/source/client/*.cpp")
 file(GLOB AWS_INTERNAL_SOURCE "${CMAKE_CURRENT_SOURCE_DIR}/source/internal/*.cpp")
@@ -362,8 +358,6 @@ if(MSVC)
     source_group("Header Files\\aws\\core\\utils\\memory\\stl" FILES ${UTILS_STL_HEADERS})
     source_group("Header Files\\aws\\core\\utils\\stream" FILES ${UTILS_STREAM_HEADERS})
     source_group("Header Files\\aws\\core\\utils\\ratelimiter" FILES ${UTILS_RATE_LIMITER_HEADERS})
-    source_group("Header Files\\aws\\core\\external\\cjson" FILES ${CJSON_HEADERS})
-    source_group("Header Files\\aws\\core\\external\\tinyxml2" FILES ${TINYXML2_HEADERS})
 
     # http client conditional headers
     if(ENABLE_CURL_CLIENT)
@@ -382,8 +376,6 @@ if(MSVC)
     endif()
 
     source_group("Source Files" FILES ${AWS_SOURCE})
-    source_group("Source Files\\external\\cjson" FILES ${CJSON_SOURCE})
-    source_group("Source Files\\external\\tinyxml2" FILES ${AWS_TINYXML2_SOURCE})
     source_group("Source Files\\auth" FILES ${AWS_AUTH_SOURCE})
     source_group("Source Files\\client" FILES ${AWS_CLIENT_SOURCE})
     source_group("Source Files\\internal" FILES ${AWS_INTERNAL_SOURCE})
@@ -572,8 +564,6 @@ install (FILES ${UTILS_STL_HEADERS} DESTINATION ${INCLUDE_DIRECTORY}/aws/core/ut
 install (FILES ${UTILS_RATE_LIMITER_HEADERS} DESTINATION ${INCLUDE_DIRECTORY}/aws/core/utils/ratelimiter)
 install (FILES ${UTILS_STREAM_HEADERS} DESTINATION ${INCLUDE_DIRECTORY}/aws/core/utils/stream)
 install (FILES ${UTILS_THREADING_HEADERS} DESTINATION ${INCLUDE_DIRECTORY}/aws/core/utils/threading)
-install (FILES ${CJSON_HEADERS} DESTINATION ${INCLUDE_DIRECTORY}/aws/core/external/cjson)
-install (FILES ${TINYXML2_HEADERS} DESTINATION ${INCLUDE_DIRECTORY}/aws/core/external/tinyxml2)
 
 # android logcat headers
 if(PLATFORM_ANDROID)
diff --git a/aws-cpp-sdk-core/include/aws/core/utils/json/JsonSerializer.h b/aws-cpp-sdk-core/include/aws/core/utils/json/JsonSerializer.h
index 657ee6eff4..d41bb814f9 100644
--- a/aws-cpp-sdk-core/include/aws/core/utils/json/JsonSerializer.h
+++ b/aws-cpp-sdk-core/include/aws/core/utils/json/JsonSerializer.h
@@ -11,7 +11,7 @@
 #include <aws/core/utils/memory/stl/AWSStreamFwd.h>
 #include <aws/core/utils/memory/stl/AWSString.h>
 #include <aws/core/utils/memory/stl/AWSMap.h>
-#include <aws/core/external/cjson/cJSON.h>
+#include <cjson/cJSON.h>
 
 #include <utility>
 
diff --git a/aws-cpp-sdk-core/source/Aws.cpp b/aws-cpp-sdk-core/source/Aws.cpp
index b367feed6d..e8496b2277 100644
--- a/aws-cpp-sdk-core/source/Aws.cpp
+++ b/aws-cpp-sdk-core/source/Aws.cpp
@@ -10,7 +10,7 @@
 #include <aws/core/utils/logging/CRTLogging.h>
 #include <aws/core/utils/logging/DefaultLogSystem.h>
 #include <aws/core/Globals.h>
-#include <aws/core/external/cjson/cJSON.h>
+#include <cjson/cJSON.h>
 #include <aws/core/monitoring/MonitoringManager.h>
 #include <aws/core/net/Net.h>
 #include <aws/core/config/AWSProfileConfigLoader.h>
diff --git a/aws-cpp-sdk-core/source/external/cjson/cJSON.cpp b/aws-cpp-sdk-core/source/external/cjson/cJSON.cpp
index cd51ae9745..a8cef01cfa 100644
--- a/aws-cpp-sdk-core/source/external/cjson/cJSON.cpp
+++ b/aws-cpp-sdk-core/source/external/cjson/cJSON.cpp
@@ -56,7 +56,7 @@
 #pragma GCC visibility pop
 #endif
 
-#include <aws/core/external/cjson/cJSON.h>
+#include <cjson/cJSON.h>
 
 /* define our own boolean type */
 // #ifdef true
diff --git a/aws-cpp-sdk-core/source/external/tinyxml2/tinyxml2.cpp b/aws-cpp-sdk-core/source/external/tinyxml2/tinyxml2.cpp
index 151a368676..9ebdeffa40 100644
--- a/aws-cpp-sdk-core/source/external/tinyxml2/tinyxml2.cpp
+++ b/aws-cpp-sdk-core/source/external/tinyxml2/tinyxml2.cpp
@@ -27,7 +27,7 @@ This file has been modified from its original version by Amazon:
   (2) #includes all use <>
 */
 
-#include <aws/core/external/tinyxml2/tinyxml2.h>
+#include <tinyxml2/tinyxml2.h>
 
 #include <new>		// yes, this one new style header, is in the Android SDK.
 #if defined(ANDROID_NDK) || defined(__BORLANDC__) || defined(__QNXNTO__)
@@ -2803,4 +2803,4 @@ bool XMLPrinter::Visit( const XMLUnknown& unknown )
 
 }   // namespace tinyxml2
 }   // namespace External
-}   // namespace Aws
\ No newline at end of file
+}   // namespace Aws
diff --git a/aws-cpp-sdk-core/source/utils/xml/XmlSerializer.cpp b/aws-cpp-sdk-core/source/utils/xml/XmlSerializer.cpp
index 2d91f70000..10a468d95f 100644
--- a/aws-cpp-sdk-core/source/utils/xml/XmlSerializer.cpp
+++ b/aws-cpp-sdk-core/source/utils/xml/XmlSerializer.cpp
@@ -6,7 +6,7 @@
 #include <aws/core/utils/xml/XmlSerializer.h>
 
 #include <aws/core/utils/StringUtils.h>
-#include <aws/core/external/tinyxml2/tinyxml2.h>
+#include <tinyxml2/tinyxml2.h>
 
 #include <utility>
 #include <algorithm>
