diff --git a/cmake/yacma/YACMACompilerLinkerSettings.cmake b/cmake/yacma/YACMACompilerLinkerSettings.cmake
index edafe6d..10f40ed 100644
--- a/cmake/yacma/YACMACompilerLinkerSettings.cmake
+++ b/cmake/yacma/YACMACompilerLinkerSettings.cmake
@@ -91,11 +91,6 @@ if(NOT _YACMACompilerLinkerSettingsRun)
         # For now it seems like -Wshadow from clang behaves better than GCC's, just enable it here
         # for the time being.
         _YACMA_CHECK_ENABLE_DEBUG_CXX_FLAG(-Wshadow)
-        # Clang is better at this flag than GCC.
-        # NOTE: enable unconditionally, as it seems like the CMake
-        # machinery for detecting this fails. Perhaps the source code
-        # used for checking the flag emits warnings?
-        list(APPEND _YACMA_CXX_FLAGS_DEBUG "-Werror")
         # New warnings in clang 8.
         # NOTE: a few issues with macros here, let's disable for now.
         # _YACMA_CHECK_ENABLE_DEBUG_CXX_FLAG(-Wextra-semi-stmt)
@@ -190,8 +185,6 @@ if(NOT _YACMACompilerLinkerSettingsRun)
     if(YACMA_COMPILER_IS_MSVC AND NOT YACMA_COMPILER_IS_CLANGXX)
         # Enable higher warning level than usual.
         _YACMA_CHECK_ENABLE_DEBUG_CXX_FLAG(/W4)
-        # Treat warnings as errors.
-        _YACMA_CHECK_ENABLE_DEBUG_CXX_FLAG(/WX)
     endif()
 
     # Set the cache variables.
