From c19a6b34a9f406e17da642f30e2507a5cc6b1519 Mon Sep 17 00:00:00 2001
From: Joakim Haugen <joakim.haugen@sintef.no>
Date: Wed, 8 Nov 2023 12:16:23 +0100
Subject: [PATCH] Do not add expat as external project

---
 Config.cmake/fmixml.cmake | 81 ++-------------------------------------
 1 file changed, 3 insertions(+), 78 deletions(-)

diff --git a/Config.cmake/fmixml.cmake b/Config.cmake/fmixml.cmake
index ddd3c1f..7aa7ffa 100644
--- a/Config.cmake/fmixml.cmake
+++ b/Config.cmake/fmixml.cmake
@@ -137,83 +137,8 @@ set(FMIXMLSOURCE
 	src/FMI2/fmi2_xml_variable.c
 )
 
-include(ExternalProject)
-
-# The *_POSTFIX variables are set because it makes it easier to determine the name of
-# the lib expat will produce at configure time. Note that Expat has some special handling
-# for it for MSVC which this in effect negates. https://github.com/libexpat/libexpat/pull/316
-set(EXPAT_SETTINGS
-    -DEXPAT_BUILD_TOOLS:BOOLEAN=OFF
-    -DEXPAT_BUILD_EXAMPLES:BOOLEAN=OFF
-    -DEXPAT_BUILD_TESTS:BOOLEAN=OFF
-    -DEXPAT_SHARED_LIBS:BOOLEAN=OFF
-    -DEXPAT_DTD:BOOLEAN=OFF
-    -DEXPAT_NS:BOOLEAN=OFF
-    -DEXPAT_MSVC_STATIC_CRT:BOOLEAN=${FMILIB_BUILD_WITH_STATIC_RTLIB}
-    -DCMAKE_DEBUG_POSTFIX:STRING=
-    -DCMAKE_RELEASE_POSTFIX:STRING=
-    -DCMAKE_MINSIZEREL_POSTFIX:STRING=
-    -DCMAKE_RELWITHDEBINFO_POSTFIX:STRING=
-    -DCMAKE_POSITION_INDEPENDENT_CODE:BOOLEAN=ON
-    -DCMAKE_BUILD_TYPE:STRING=${CMAKE_BUILD_TYPE}
-    -DCMAKE_EXE_LINKER_FLAGS:STRING=${CMAKE_EXE_LINKER_FLAGS}
-    -DCMAKE_LINK_LIBRARY_FLAG:STRING=${CMAKE_LINK_LIBRARY_FLAG}
-    -DCMAKE_MODULE_LINKER_FLAGS:STRING=${CMAKE_MODULE_LINKER_FLAGS}
-    -DCMAKE_SHARED_LINKER_FLAGS:STRING=${CMAKE_SHARED_LINKER_FLAGS}	
-    -DCMAKE_INSTALL_PREFIX:PATH=${CMAKE_BINARY_DIR}/ExpatEx/install
-)
-
-ExternalProject_Add(
-	expatex
-	PREFIX "${FMIXML_EXPAT_DIR}"
-	SOURCE_DIR "${FMIXML_EXPAT_DIR}"
-	CMAKE_CACHE_ARGS ${EXPAT_SETTINGS}
-	BINARY_DIR ${CMAKE_BINARY_DIR}/ExpatEx
-	INSTALL_DIR ${CMAKE_BINARY_DIR}/ExpatEx/install
-	TMP_DIR     ${CMAKE_BINARY_DIR}/ExpatEx/tmp
-    STAMP_DIR   ${CMAKE_BINARY_DIR}/ExpatEx/stamp
-)
-
-ExternalProject_Add_Step(
-	expatex dependent_reconfigure
-	DEPENDEES configure
-	DEPENDERS build
-	COMMAND ${CMAKE_COMMAND} -E echo "Running:  ${CMAKE_COMMAND} -G "${CMAKE_GENERATOR}"  ${EXPAT_SETTINGS} ${FMIXML_EXPAT_DIR}"
-	COMMAND ${CMAKE_COMMAND} -G "${CMAKE_GENERATOR}" ${EXPAT_SETTINGS} "${FMIXML_EXPAT_DIR}"
-	DEPENDS ${CMAKE_BINARY_DIR}/CMakeCache.txt
-	WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/ExpatEx
-)
-
-add_dependencies(expatex ${CMAKE_BINARY_DIR}/CMakeCache.txt ${FMILIBRARYHOME}/CMakeLists.txt)
-
-
-if(MSVC)
-    # Expat uses special naming with MSVC, which is mirrored here.
-    set(EXPAT_LIB_PREFIX lib)
-else()
-    set(EXPAT_LIB_PREFIX ${CMAKE_STATIC_LIBRARY_PREFIX})
-endif()
-  
-set(expatlib "${CMAKE_BINARY_DIR}/ExpatEx/${CMAKE_CFG_INTDIR}/${EXPAT_LIB_PREFIX}expat${CMAKE_STATIC_LIBRARY_SUFFIX}")
-  
-add_library(expat STATIC IMPORTED)
-
-set_target_properties(
-	expat PROPERTIES 
-		IMPORTED_LOCATION "${expatlib}"
-)
-
-add_dependencies(expat expatex)
-
-if(FMILIB_INSTALL_SUBLIBS)
-	install(FILES 
-	"${CMAKE_BINARY_DIR}/ExpatEx/install/lib/${CMAKE_STATIC_LIBRARY_PREFIX}expat${CMAKE_STATIC_LIBRARY_SUFFIX}"
-	DESTINATION lib)
-endif()
-
-set(EXPAT_INCLUDE_DIRS ${CMAKE_BINARY_DIR}/ExpatEx/install/include)
-
-include_directories("${EXPAT_INCLUDE_DIRS}" "${FMILIB_THIRDPARTYLIBS}/FMI/" "${FMIXMLGENDIR}/FMI1" "${FMIXMLGENDIR}/FMI2")
+find_package(expat CONFIG REQUIRED)
+include_directories("${FMILIB_THIRDPARTYLIBS}/FMI/" "${FMIXMLGENDIR}/FMI1" "${FMIXMLGENDIR}/FMI2")
 
 PREFIXLIST(FMIXMLSOURCE  ${FMIXMLDIR}/)
 PREFIXLIST(FMIXMLHEADERS ${FMIXMLDIR}/)
@@ -232,6 +157,6 @@ if(MSVC)
     target_compile_definitions(fmixml PUBLIC XML_STATIC)
 endif()
 
-target_link_libraries(fmixml ${JMUTIL_LIBRARIES} expat)
+target_link_libraries(fmixml ${JMUTIL_LIBRARIES} expat::expat)
 
 endif(NOT FMIXMLDIR)
-- 
2.30.2

