--- PCbuild/_freeze_importlib.vcxproj
+++ PCbuild/_freeze_importlib.vcxproj
@@ -151,1 +151,1 @@
-  <Target Name="RebuildImportLib" AfterTargets="AfterBuild" Condition="$(Configuration) == 'Debug' or $(Configuration) == 'Release'"
+  <!--<Target Name="RebuildImportLib" AfterTargets="AfterBuild" Condition="$(Configuration) == 'Debug' or $(Configuration) == 'Release'"
@@ -153,1 +153,1 @@
-  </Target>
+  </Target>-->
--- PCbuild/_ssl.vcxproj
+++ PCbuild/_ssl.vcxproj
@@ -99,9 +99,9 @@
   </ItemDefinitionGroup>
   <ItemGroup>
     <ClCompile Include="..\Modules\_ssl.c" />
-    <ClCompile Include="$(opensslIncludeDir)\applink.c">
+    <!--<ClCompile Include="$(opensslIncludeDir)\applink.c">
       <PreprocessorDefinitions>_CRT_SECURE_NO_WARNINGS;$(PreprocessorDefinitions)</PreprocessorDefinitions>
-    </ClCompile>
+    </ClCompile>-->
   </ItemGroup>
   <ItemGroup>
     <ResourceCompile Include="..\PC\python_nt.rc" />
