--- CMakeLists.txt	2020-10-05 16:38:06.000000000 -0300
+++ CMakeLists.txt	2020-10-05 16:37:56.000000000 -0300
@@ -226,11 +226,7 @@
 
 option(SQLITECPP_INTERNAL_SQLITE "Add the internal SQLite3 source to the project." ON)
 if (SQLITECPP_INTERNAL_SQLITE)
-    message(STATUS "Compile sqlite3 from source in subdirectory")
-    option(SQLITE_ENABLE_JSON1 "Enable JSON1 extension when building internal sqlite3 library." ON)
-    # build the SQLite3 C library (for ease of use/compatibility) versus Linux sqlite3-dev package
-    add_subdirectory(sqlite3)
-    target_link_libraries(SQLiteCpp PUBLIC sqlite3)
+    target_link_libraries(SQLiteCpp PUBLIC CONAN_PKG::sqlite3)
 else (SQLITECPP_INTERNAL_SQLITE)
     find_package (SQLite3 REQUIRED)
     message(STATUS "Link to sqlite3 system library")
