--- a/cmake/ssl.cmake
+++ b/cmake/ssl.cmake
@@ -195,10 +195,10 @@ MACRO (MYSQL_CHECK_SSL)
     ENDIF()
 
     FIND_LIBRARY(OPENSSL_LIBRARY
-                 NAMES ssl libssl ssleay32 ssleay32MD
+                 NAMES ssl libssl ssleay32 ssleay32MD ssld libssld
                  HINTS ${OPENSSL_ROOT_DIR}/lib)
     FIND_LIBRARY(CRYPTO_LIBRARY
-                 NAMES crypto libcrypto libeay32
+                 NAMES crypto libcrypto libeay32 cryptod libcryptod
                  HINTS ${OPENSSL_ROOT_DIR}/lib)
 
     IF(OPENSSL_INCLUDE_DIR)

