diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 935d585..23a5997 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -56,7 +56,7 @@ target_include_directories(libJSBSim PUBLIC
 
 if(EXPAT_FOUND)
   target_include_directories(libJSBSim PRIVATE ${EXPAT_INCLUDE_DIRS})
-  if (PKG_CONFIG_FOUND)
+  if (0)
     target_link_libraries(libJSBSim ${PC_EXPAT_LIBRARIES})
   else()
     target_link_libraries(libJSBSim ${EXPAT_LIBRARIES})
diff --git a/src/simgear/xml/CMakeLists.txt b/src/simgear/xml/CMakeLists.txt
index f772a03..947a477 100644
--- a/src/simgear/xml/CMakeLists.txt
+++ b/src/simgear/xml/CMakeLists.txt
@@ -34,4 +34,8 @@ if(WIN32 AND NOT EXPAT_FOUND)
   target_compile_definitions(Xml PRIVATE XML_STATIC)
 endif()
 
+if (EXPAT_FOUND)
+    target_link_libraries(Xml ${EXPAT_LIBRARIES})
+endif()
+
 install(FILES ${HEADERS} DESTINATION include/JSBSim/simgear/xml COMPONENT devel)
