diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 935d585..23a5997 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -56,7 +56,7 @@ target_include_directories(libJSBSim PUBLIC
 
 if(EXPAT_FOUND)
   target_include_directories(libJSBSim PRIVATE ${EXPAT_INCLUDE_DIRS})
-  if (PKG_CONFIG_FOUND)
+  if (0)
     target_link_libraries(libJSBSim ${PC_EXPAT_LIBRARIES})
   else()
     target_link_libraries(libJSBSim ${EXPAT_LIBRARIES})
