From 0294761dcf6682dd0784b88130d580a344eaa0f5 Mon Sep 17 00:00:00 2001
From: Martin Valgur <martin.valgur@gmail.com>
Date: Fri, 22 Mar 2024 20:06:59 +0200
Subject: [PATCH] Fix 'cvd/convert_image.h' not being installed

---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f34f8ea2..9d8481d4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -94,6 +94,7 @@ set(HEADERS
 	cvd/colourspace_convert.h
 	cvd/colourspace_frame.h
 	cvd/connected_components.h
+	cvd/convert_image.h
 	cvd/convolution.h
 	cvd/deinterlacebuffer.h
 	cvd/deinterlaceframe.h
