diff --git a/CMakeLists.txt b/CMakeLists.txt
index b8171cd..b793784 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -21,6 +21,7 @@ include(CommonCppFlags)
 #-------------------------------------------------------------------------------------------
 file(GLOB EASTL_SOURCES "source/*.cpp")
 add_library(EASTL ${EASTL_SOURCES})
+set(CMAKE_CXX_STANDARD 14)
 
 if(EASTL_BUILD_BENCHMARK)
     add_subdirectory(benchmark)
@@ -46,5 +47,19 @@ target_include_directories(EASTL PUBLIC include)
 #-------------------------------------------------------------------------------------------
 # Dependencies
 #-------------------------------------------------------------------------------------------
-target_link_libraries(EASTL EABase)
+find_package(EABase REQUIRED CONFIG)
+target_link_libraries(EASTL PUBLIC EABase::EABase)
 
+if(BUILD_SHARED_LIBS)
+  target_compile_definitions(EASTL PUBLIC EASTL_DLL)
+  if(MSVC OR CYGWIN)
+    target_compile_definitions(EASTL PRIVATE "EASTL_API=__declspec(dllexport)")
+  endif()
+endif()
+
+include(GNUInstallDirs)
+install(TARGETS EASTL
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(DIRECTORY ${CMAKE_INSTALL_INCLUDEDIR}/EASTL DESTINATION include)
