--- CMakeLists.txt
+++ CMakeLists.txt
@@ -47,8 +47,22 @@
 # Dependencies
 #-------------------------------------------------------------------------------------------
 if (NOT TARGET EABase)
-    add_subdirectory(test/packages/EABase)
+    #add_subdirectory(test/packages/EABase)
 endif()
 
-target_link_libraries(EASTL EABase)
+find_package(EABase REQUIRED CONFIG)
+target_link_libraries(EASTL EABase::EABase)
 
+if(BUILD_SHARED_LIBS)
+  target_compile_definitions(EASTL PUBLIC EASTL_DLL)
+  if(MSVC OR CYGWIN)
+    target_compile_definitions(EASTL PRIVATE "EASTL_API=__declspec(dllexport)")
+  endif()
+endif()
+
+include(GNUInstallDirs)
+install(TARGETS EASTL
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(DIRECTORY ${CMAKE_INSTALL_INCLUDEDIR}/EASTL DESTINATION include)
