diff --git a/CMakeLists.txt b/CMakeLists.txt
index 24aa14e..f8329ec 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -153,13 +153,13 @@ endif(NOT USE_BUNDLED)

 ADD_SUBDIRECTORY(PDFWriter)

-if(PROJECT_IS_TOP_LEVEL)
+if(0)
     # avoid installing the testing lib altogether when included in another project.
     # it's annoying when in parent all, and more annoying to then get the tests added
     # to the parent project ctest.
     enable_testing()
     ADD_SUBDIRECTORY(PDFWriterTesting)
-endif(PROJECT_IS_TOP_LEVEL)
+endif()

 include(InstallRequiredSystemLibraries)
 set(CPACK_RESOURCE_FILE_LICENSE "${CMAKE_CURRENT_SOURCE_DIR}/LICENSE")
