diff --git a/CMakeLists.txt b/CMakeLists.txt
index e4f3119..c09d7c6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -9,6 +9,9 @@ if(NOT DEFINED CMAKE_FIRST_RUN)
     SET(CMAKE_FIRST_RUN 1 CACHE INTERNAL "")
 endif()
 
+include("${CMAKE_CURRENT_BINARY_DIR}/conanbuildinfo.cmake")
+conan_basic_setup()
+
 ###############################################################################
 ### GLOBAL ###
 
@@ -142,7 +145,7 @@ endif(CMAKE_COMPILER_IS_GNUCXX)
 ###############################################################################
 ### Python ###
 
-OCIOFindPython()
+#OCIOFindPython()
 
 # Find Python, used for (possibly) building pyglue, and now to
 # construct the external project path
@@ -248,6 +251,10 @@ endif(USE_EXTERNAL_TINYXML)
 ###############################################################################
 ### YAML ###
 
+set(YAML_CPP_INCLUDE_DIRS ${CONAN_INCLUDE_DIRS_YAML-CPP})
+set(YAML_CPP_LIBRARIES ${CONAN_LIBS_YAML-CPP})
+
+if(FALSE)
 if(USE_EXTERNAL_YAML)
     # Set minimum yaml version for non-patched sources.
     set(YAML_VERSION_MIN "0.3.0")
@@ -366,6 +373,7 @@ endif(USE_EXTERNAL_YAML)
 if(YAML_CPP_VERSION VERSION_LESS "0.5.0")
     set(YAML_CPP_COMPILE_FLAGS "-DOLDYAML")
 endif()
+endif(FALSE)
 
 ###############################################################################
 ### Externals ###
