diff --git a/CMakeLists.txt b/CMakeLists.txt
index b0840ac..e4875ab 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -9,9 +9,9 @@ cmake_minimum_required(VERSION 3.12)
 
 set(CMAKE_MODULE_PATH
     ${CMAKE_MODULE_PATH}
-    ${CMAKE_SOURCE_DIR}/share/cmake/utils
-    ${CMAKE_SOURCE_DIR}/share/cmake/macros
-    ${CMAKE_SOURCE_DIR}/share/cmake/modules
+    ${CMAKE_CURRENT_SOURCE_DIR}/share/cmake/utils
+    ${CMAKE_CURRENT_SOURCE_DIR}/share/cmake/macros
+    ${CMAKE_CURRENT_SOURCE_DIR}/share/cmake/modules
 )
 
 set(CMAKE_WARN_DEPRECATED ON)
@@ -272,7 +272,7 @@ else()
     set(OCIO_SETUP_NAME setup_ocio.sh)
 endif()
 
-configure_file(${CMAKE_SOURCE_DIR}/share/ocio/${OCIO_SETUP_NAME}.in
+configure_file(${PROJECT_SOURCE_DIR}/share/ocio/${OCIO_SETUP_NAME}.in
     ${CMAKE_CURRENT_BINARY_DIR}/share/ocio/${OCIO_SETUP_NAME} @ONLY)
 
 INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/share/ocio/${OCIO_SETUP_NAME} DESTINATION share/ocio/)
diff --git a/share/cmake/modules/FindExtPackages.cmake b/share/cmake/modules/FindExtPackages.cmake
index 86a0225..73d8599 100644
--- a/share/cmake/modules/FindExtPackages.cmake
+++ b/share/cmake/modules/FindExtPackages.cmake
@@ -50,7 +50,7 @@ else()
 
     # OpenEXR/IlmBase (<=2.5)
     # https://github.com/AcademySoftwareFoundation/openexr
-    find_package(Half 2.4.0 REQUIRED)
+    find_package(OpenEXR 2.4.0 REQUIRED)
 
     set(OCIO_HALF_LIB IlmBase::Half CACHE STRING "Half library target" FORCE)
     set(OCIO_USE_IMATH_HALF "0" CACHE STRING "Whether 'half' type will be sourced from the Imath library (>=v3.0)" FORCE)
@@ -65,7 +65,7 @@ if(OCIO_BUILD_APPS)
 
     # lcms2
     # https://github.com/mm2/Little-CMS
-    find_package(lcms2 2.2 REQUIRED)
+    find_package(lcms 2.2 REQUIRED)
 endif()
 
 if(OCIO_BUILD_OPENFX)
diff --git a/src/apps/ociobakelut/CMakeLists.txt b/src/apps/ociobakelut/CMakeLists.txt
index 7eb1cd8..de13607 100755
--- a/src/apps/ociobakelut/CMakeLists.txt
+++ b/src/apps/ociobakelut/CMakeLists.txt
@@ -35,7 +35,7 @@ set_target_properties(ociobakelut
 target_link_libraries(ociobakelut 
     PRIVATE 
         apputils
-        lcms2::lcms2
+        lcms::lcms
         OpenColorIO
 )
 
