diff --git a/share/cmake/utils/CppVersion.cmake b/share/cmake/utils/CppVersion.cmake
index aeca6c0..6b4dc4e 100644
--- a/share/cmake/utils/CppVersion.cmake
+++ b/share/cmake/utils/CppVersion.cmake
@@ -12,11 +12,11 @@ if(NOT DEFINED CMAKE_CXX_STANDARD)
     message(STATUS "Setting C++ version to '11' as none was specified.")
     set(CMAKE_CXX_STANDARD 11 CACHE STRING "C++ standard to compile against")
 elseif(NOT CMAKE_CXX_STANDARD IN_LIST SUPPORTED_CXX_STANDARDS)
-    message(FATAL_ERROR 
+    message(WARNING
             "CMAKE_CXX_STANDARD=${CMAKE_CXX_STANDARD} is unsupported. Supported standards are: ${SUPPORTED_CXX_STANDARDS_STR}.")
 endif()
 
-set_property(CACHE CMAKE_CXX_STANDARD PROPERTY STRINGS "${SUPPORTED_CXX_STANDARDS}")
+#set_property(CACHE CMAKE_CXX_STANDARD PROPERTY STRINGS "${SUPPORTED_CXX_STANDARDS}")
 
 include(CheckCXXCompilerFlag)
 
