diff --git src/OpenColorIO/ConfigUtils.cpp src/OpenColorIO/ConfigUtils.cpp
index 2e774726..b4228ff7 100644
--- src/OpenColorIO/ConfigUtils.cpp
+++ src/OpenColorIO/ConfigUtils.cpp
@@ -3,7 +3,7 @@
 
 #include "ConfigUtils.h"
 #include "MathUtils.h"
-#include "pystring/pystring.h"
+#include "pystring.h"
 #include "utils/StringUtils.h"
 
 namespace OCIO_NAMESPACE
diff --git src/OpenColorIO/FileRules.cpp src/OpenColorIO/FileRules.cpp
index 61a5e0f1..e0df0d02 100644
--- src/OpenColorIO/FileRules.cpp
+++ src/OpenColorIO/FileRules.cpp
@@ -62,7 +62,7 @@ std::string ConvertToRegularExpression(const char * globPattern, bool ignoreCase
 
     if (ignoreCase)
     {
-        const size_t length = strlen(globPattern);
+        const size_t length = std::strlen(globPattern);
         bool respectCase = false;
         for (size_t i = 0; i < length; ++i)
         {
