diff --git a/src/OpenColorIO/Config.cpp b/src/OpenColorIO/Config.cpp
index f09b9ee..7f66ab1 100644
--- a/src/OpenColorIO/Config.cpp
+++ b/src/OpenColorIO/Config.cpp
@@ -34,7 +34,7 @@
 #include "Platform.h"
 #include "PrivateTypes.h"
 #include "Processor.h"
-#include "pystring/pystring.h"
+#include "pystring.h"
 #include "transforms/FileTransform.h"
 #include "utils/StringUtils.h"
 #include "ViewingRules.h"
diff --git a/src/OpenColorIO/Context.cpp b/src/OpenColorIO/Context.cpp
index bb6fb07..a8890ed 100644
--- a/src/OpenColorIO/Context.cpp
+++ b/src/OpenColorIO/Context.cpp
@@ -15,7 +15,7 @@
 #include "OCIOZArchive.h"
 #include "PathUtils.h"
 #include "PrivateTypes.h"
-#include "pystring/pystring.h"
+#include "pystring.h"
 #include "utils/StringUtils.h"
 
 namespace OCIO_NAMESPACE
diff --git a/src/OpenColorIO/OCIOYaml.cpp b/src/OpenColorIO/OCIOYaml.cpp
index 62cbb0d..59c1564 100644
--- a/src/OpenColorIO/OCIOYaml.cpp
+++ b/src/OpenColorIO/OCIOYaml.cpp
@@ -19,7 +19,7 @@
 #include "ParseUtils.h"
 #include "PathUtils.h"
 #include "Platform.h"
-#include "pystring/pystring.h"
+#include "pystring.h"
 #include "utils/StringUtils.h"
 #include "ViewingRules.h"
 #include "yaml-cpp/yaml.h"
diff --git a/src/OpenColorIO/OCIOZArchive.cpp b/src/OpenColorIO/OCIOZArchive.cpp
index 24d836e..a9177eb 100644
--- a/src/OpenColorIO/OCIOZArchive.cpp
+++ b/src/OpenColorIO/OCIOZArchive.cpp
@@ -11,7 +11,7 @@
 #include <OpenColorIO/OpenColorIO.h>
 #include "Mutex.h"
 #include "Platform.h"
-#include "pystring/pystring.h"
+#include "pystring.h"
 #include "utils/StringUtils.h"
 #include "transforms/FileTransform.h"
 
diff --git a/src/OpenColorIO/Op.cpp b/src/OpenColorIO/Op.cpp
index e1bd564..cf05c37 100755
--- a/src/OpenColorIO/Op.cpp
+++ b/src/OpenColorIO/Op.cpp
@@ -20,7 +20,7 @@
 #include "ops/lut1d/Lut1DOp.h"
 #include "ops/lut3d/Lut3DOp.h"
 #include "ops/range/RangeOp.h"
-#include "pystring/pystring.h"
+#include "pystring.h"
 
 namespace OCIO_NAMESPACE
 {
diff --git a/src/OpenColorIO/PathUtils.cpp b/src/OpenColorIO/PathUtils.cpp
index 9dc8c6b..4a1096d 100644
--- a/src/OpenColorIO/PathUtils.cpp
+++ b/src/OpenColorIO/PathUtils.cpp
@@ -10,7 +10,7 @@
 #include "Mutex.h"
 #include "PathUtils.h"
 #include "Platform.h"
-#include "pystring/pystring.h"
+#include "pystring.h"
 #include "utils/StringUtils.h"
 #include "OCIOZArchive.h"
 
diff --git a/src/OpenColorIO/fileformats/FileFormatCTF.cpp b/src/OpenColorIO/fileformats/FileFormatCTF.cpp
index ebed326..9f70ff8 100644
--- a/src/OpenColorIO/fileformats/FileFormatCTF.cpp
+++ b/src/OpenColorIO/fileformats/FileFormatCTF.cpp
@@ -23,7 +23,7 @@
 #include "OpBuilders.h"
 #include "ops/noop/NoOps.h"
 #include "Platform.h"
-#include "pystring/pystring.h"
+#include "pystring.h"
 #include "TransformBuilder.h"
 #include "transforms/FileTransform.h"
 #include "utils/StringUtils.h"
diff --git a/src/OpenColorIO/fileformats/FileFormatDiscreet1DL.cpp b/src/OpenColorIO/fileformats/FileFormatDiscreet1DL.cpp
index a52bc72..bd827f0 100755
--- a/src/OpenColorIO/fileformats/FileFormatDiscreet1DL.cpp
+++ b/src/OpenColorIO/fileformats/FileFormatDiscreet1DL.cpp
@@ -16,7 +16,7 @@
 #include "ops/lut1d/Lut1DOp.h"
 #include "ops/lut3d/Lut3DOp.h"
 #include "ParseUtils.h"
-#include "pystring/pystring.h"
+#include "pystring.h"
 #include "Platform.h"
 #include "transforms/FileTransform.h"
 #include "utils/StringUtils.h"
diff --git a/src/OpenColorIO/fileformats/FileFormatICC.cpp b/src/OpenColorIO/fileformats/FileFormatICC.cpp
index 1fcfd9a..ca5d9b4 100755
--- a/src/OpenColorIO/fileformats/FileFormatICC.cpp
+++ b/src/OpenColorIO/fileformats/FileFormatICC.cpp
@@ -15,7 +15,7 @@
 #include "ops/matrix/MatrixOp.h"
 #include "ops/range/RangeOp.h"
 #include "Platform.h"
-#include "pystring/pystring.h"
+#include "pystring.h"
 #include "transforms/FileTransform.h"
 
 
diff --git a/src/OpenColorIO/fileformats/FileFormatIridasLook.cpp b/src/OpenColorIO/fileformats/FileFormatIridasLook.cpp
index 7402efd..cc3acb4 100755
--- a/src/OpenColorIO/fileformats/FileFormatIridasLook.cpp
+++ b/src/OpenColorIO/fileformats/FileFormatIridasLook.cpp
@@ -13,7 +13,7 @@
 #include "ops/lut3d/Lut3DOp.h"
 #include "ParseUtils.h"
 #include "Platform.h"
-#include "pystring/pystring.h"
+#include "pystring.h"
 #include "transforms/FileTransform.h"
 #include "utils/StringUtils.h"
 #include "utils/NumberUtils.h"
diff --git a/src/OpenColorIO/transforms/FileTransform.cpp b/src/OpenColorIO/transforms/FileTransform.cpp
index 4fd4d5d..dc5eb3c 100755
--- a/src/OpenColorIO/transforms/FileTransform.cpp
+++ b/src/OpenColorIO/transforms/FileTransform.cpp
@@ -19,7 +19,7 @@
 #include "ops/noop/NoOps.h"
 #include "PathUtils.h"
 #include "Platform.h"
-#include "pystring/pystring.h"
+#include "pystring.h"
 #include "utils/StringUtils.h"
 
 namespace OCIO_NAMESPACE
diff --git a/tests/cpu/ColorSpace_tests.cpp b/tests/cpu/ColorSpace_tests.cpp
index 8c37c37..9791e05 100644
--- a/tests/cpu/ColorSpace_tests.cpp
+++ b/tests/cpu/ColorSpace_tests.cpp
@@ -6,7 +6,7 @@
 
 #include "ColorSpace.cpp"
 
-#include <pystring/pystring.h>
+#include <pystring.h>
 #include "testutils/UnitTest.h"
 #include "UnitTestUtils.h"
 
diff --git a/tests/cpu/Config_tests.cpp b/tests/cpu/Config_tests.cpp
index 5d30fda..6d4b70d 100644
--- a/tests/cpu/Config_tests.cpp
+++ b/tests/cpu/Config_tests.cpp
@@ -7,7 +7,7 @@
 #include "Config.cpp"
 #include "utils/StringUtils.h"
 
-#include <pystring/pystring.h>
+#include <pystring.h>
 #include "testutils/UnitTest.h"
 #include "UnitTestLogUtils.h"
 #include "UnitTestUtils.h"
diff --git a/tests/cpu/Context_tests.cpp b/tests/cpu/Context_tests.cpp
index f78bad8..86d6b62 100644
--- a/tests/cpu/Context_tests.cpp
+++ b/tests/cpu/Context_tests.cpp
@@ -8,7 +8,7 @@
 
 #include "PathUtils.h"
 #include "Platform.h"
-#include "pystring/pystring.h"
+#include "pystring.h"
 #include "testutils/UnitTest.h"
 
 namespace OCIO = OCIO_NAMESPACE;
diff --git a/vendor/openfx/OCIOUtils.cpp b/vendor/openfx/OCIOUtils.cpp
index ca44905..469ed35 100644
--- a/vendor/openfx/OCIOUtils.cpp
+++ b/vendor/openfx/OCIOUtils.cpp
@@ -9,7 +9,7 @@ namespace OCIO = OCIO_NAMESPACE;
 #include <vector>
 
 #include "ofxsLog.h"
-#include "pystring/pystring.h"
+#include "pystring.h"
 
 namespace
 {
