--- xorg-macros.m4.in
+++ xorg-macros.m4.in
@@ -36,7 +36,7 @@
 #
 # See the "minimum version" comment for each macro you use to see what
 # version you require.
-m4_defun([XORG_MACROS_VERSION],[
+AC_DEFUN([XORG_MACROS_VERSION],[
 m4_define([vers_have], [@VERSION@])
 m4_define([maj_have], m4_substr(vers_have, 0, m4_index(vers_have, [.])))
 m4_define([maj_needed], m4_substr([$1], 0, m4_index([$1], [.])))

