--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -339,28 +339,28 @@ set(
 
 include_directories(${CMAKE_CURRENT_SOURCE_DIR}/src)
 
-if(CMAKE_BUILD_TYPE MATCHES "[dD]ebug")
-    set(qhull_CPP qhullcpp_d)
-    set(qhull_SHARED qhull_d) 
-    set(qhull_SHAREDP qhull_pd)
-    set(qhull_SHAREDR qhull_rd)
-    set(qhull_STATIC qhullstatic_d)
-    set(qhull_STATICR qhullstatic_rd)
+set(qhull_CPP qhullcpp)
+set(qhull_SHARED libqhull)  # Temporarily avoid name conflict with qhull executable
+set(qhull_SHAREDP qhull_p)
+set(qhull_SHAREDR qhull_r)
+set(qhull_STATIC qhullstatic)
+set(qhull_STATICR qhullstatic_r)
+
+if(BUILD_SHARED_LIBS)
+set(
+    qhull_TARGETS_INSTALL
+        ${qhull_SHAREDR}
+        qhull rbox qconvex qdelaunay qvoronoi qhalf
+        ${qhull_SHARED}
+)
 else()
-    set(qhull_CPP qhullcpp)
-    set(qhull_SHARED libqhull)  # Temporarily avoid name conflict with qhull executable
-    set(qhull_SHAREDP qhull_p)
-    set(qhull_SHAREDR qhull_r)
-    set(qhull_STATIC qhullstatic)
-    set(qhull_STATICR qhullstatic_r)
-endif()
 
 set(
     qhull_TARGETS_INSTALL
-        ${qhull_CPP} ${qhull_STATIC} ${qhull_STATICR} ${qhull_SHAREDR}
+        ${qhull_STATIC} ${qhull_STATICR}
         qhull rbox qconvex qdelaunay qvoronoi qhalf
-        ${qhull_SHARED} ${qhull_SHAREDP}  # Deprecated, use qhull_r instead
 )
+endif()
 set(
     qhull_TARGETS_TEST   # Unused
         user_eg user_eg2 user_eg3 user_egp testqset testqset_r
@@ -375,10 +375,12 @@ add_library(${qhull_SHAREDR} SHARED
         src/libqhull_r/qhull_r-exports.def)
 set_target_properties(${qhull_SHAREDR} PROPERTIES
     SOVERSION ${qhull_SOVERSION}
-    VERSION ${qhull_VERSION})
+    VERSION ${qhull_VERSION}
+    OUTPUT_NAME "${qhull_SHAREDR}$<$<CONFIG:Debug>:d>")
 
 if(UNIX)
     target_link_libraries(${qhull_SHAREDR} m)
+    if(0)
     if(APPLE)
         set_target_properties(${qhull_SHAREDR} PROPERTIES 
             INSTALL_NAME_DIR "${LIB_INSTALL_DIR}")
@@ -388,6 +390,7 @@ if(UNIX)
             INSTALL_RPATH_USE_LINK_PATH TRUE
             BUILD_WITH_INSTALL_RPATH FALSE)
     endif()
+    endif()
 endif(UNIX)
 
 # ---------------------------------------
@@ -398,18 +401,15 @@ add_library(${qhull_SHARED} SHARED
         ${libqhull_SOURCES}
         src/libqhull/qhull-exports.def)
         
-if(qhull_SHARED MATCHES "libqhull")
-   set(qhull_OUTPUT_NAME qhull)
-   set_target_properties(${qhull_SHARED} PROPERTIES
-        OUTPUT_NAME "${qhull_OUTPUT_NAME}" )
-endif()
 
 set_target_properties(${qhull_SHARED} PROPERTIES
     SOVERSION ${qhull_SOVERSION}
-    VERSION ${qhull_VERSION})
+    VERSION ${qhull_VERSION}
+    OUTPUT_NAME "qhull$<$<CONFIG:Debug>:_d>")
 
 if(UNIX)
     target_link_libraries(${qhull_SHARED} m)
+    if(0)
     if(APPLE)
         set_target_properties(${qhull_SHARED} PROPERTIES 
             INSTALL_NAME_DIR "${LIB_INSTALL_DIR}")
@@ -419,6 +419,7 @@ if(UNIX)
             INSTALL_RPATH_USE_LINK_PATH TRUE
             BUILD_WITH_INSTALL_RPATH FALSE)
     endif()
+    endif()
 endif(UNIX)
 
 # ---------------------------------------
@@ -431,10 +432,12 @@ add_library(${qhull_SHAREDP} SHARED
 set_target_properties(${qhull_SHAREDP} PROPERTIES
     COMPILE_DEFINITIONS "qh_QHpointer"
     SOVERSION ${qhull_SOVERSION}
-    VERSION ${qhull_VERSION})
+    VERSION ${qhull_VERSION}
+    OUTPUT_NAME "${qhull_SHAREDP}$<$<CONFIG:Debug>:d>")
 
 if(UNIX)
     target_link_libraries(${qhull_SHAREDP} m)
+    if(0)
     if(APPLE)
         set_target_properties(${qhull_SHAREDP} PROPERTIES 
             INSTALL_NAME_DIR "${LIB_INSTALL_DIR}")
@@ -444,6 +447,7 @@ if(UNIX)
             INSTALL_RPATH_USE_LINK_PATH TRUE
             BUILD_WITH_INSTALL_RPATH FALSE)
     endif()
+    endif()
 endif(UNIX)
 
 # ---------------------------------------
@@ -452,11 +456,13 @@ endif(UNIX)
 
 add_library(${qhull_STATIC} STATIC ${libqhull_SOURCES})
 set_target_properties(${qhull_STATIC} PROPERTIES
-    VERSION ${qhull_VERSION})
+    VERSION ${qhull_VERSION}
+    OUTPUT_NAME "${qhull_STATIC}$<$<CONFIG:Debug>:_d>")
 
 add_library(${qhull_STATICR} STATIC ${libqhullr_SOURCES})
 set_target_properties(${qhull_STATICR} PROPERTIES
-    VERSION ${qhull_VERSION})
+    VERSION ${qhull_VERSION}
+    OUTPUT_NAME "${qhull_STATICR}$<$<CONFIG:Debug>:d>")
 
 if(UNIX)
     target_link_libraries(${qhull_STATIC} m)
@@ -471,6 +477,7 @@ endif(UNIX)
 add_library(${qhull_CPP} STATIC ${libqhullcpp_SOURCES})
 set_target_properties(${qhull_CPP} PROPERTIES
     VERSION ${qhull_VERSION}
+    OUTPUT_NAME "${qhull_CPP}$<$<CONFIG:Debug>:_d>"
     POSITION_INDEPENDENT_CODE "TRUE")
 
 # ---------------------------------------
@@ -561,10 +568,8 @@ target_link_libraries(user_eg2 ${qhull_STATICR})
 
 set(user_eg3_SOURCES    src/user_eg3/user_eg3_r.cpp)
 
-add_executable(user_eg3 ${user_eg3_SOURCES})
 # qhull_STATICR must be last, otherwise qh_fprintf,etc. are not loaded from qhull_CPP
 # user_eg3 may be linked to qhull_SHAREDR if user_eg3_DEFINES is added
-target_link_libraries(user_eg3 ${qhull_CPP} ${qhull_STATICR})
 
 # ---------------------------------------
 # qhullp is qhull/unix.c linked to deprecated qh_QHpointer libqhull_p
@@ -615,8 +620,6 @@ add_test(NAME user_eg
    COMMAND sh -c "./user_eg")
 add_test(NAME user_eg2
    COMMAND sh -c "./user_eg2")
-add_test(NAME user_eg3
-   COMMAND sh -c "./user_eg3 rbox '10 D2' '2 D2' qhull 's p' facets")
 
 # ---------------------------------------
 # Define install
@@ -624,6 +627,7 @@ add_test(NAME user_eg3
 
 install(TARGETS ${qhull_TARGETS_INSTALL} EXPORT QhullTargets
         RUNTIME DESTINATION ${BIN_INSTALL_DIR}
+        BUNDLE DESTINATION ${BIN_INSTALL_DIR}
         LIBRARY DESTINATION ${LIB_INSTALL_DIR}
         ARCHIVE DESTINATION ${LIB_INSTALL_DIR}
         INCLUDES DESTINATION include)
@@ -681,10 +685,7 @@ foreach(pkgconfig IN ITEMS "${qhull_SHARED};Qhull shared library"
 endforeach()
 
 install(FILES ${libqhull_HEADERS}    DESTINATION ${INCLUDE_INSTALL_DIR}/libqhull)
-install(FILES ${libqhull_DOC}        DESTINATION ${INCLUDE_INSTALL_DIR}/libqhull)
 install(FILES ${libqhullr_HEADERS}    DESTINATION ${INCLUDE_INSTALL_DIR}/libqhull_r)
-install(FILES ${libqhullr_DOC}        DESTINATION ${INCLUDE_INSTALL_DIR}/libqhull_r)
-install(FILES ${libqhullcpp_HEADERS} DESTINATION ${INCLUDE_INSTALL_DIR}/libqhullcpp)
 install(FILES html/qhull.man         DESTINATION ${MAN_INSTALL_DIR} RENAME qhull.1)
 install(FILES html/rbox.man          DESTINATION ${MAN_INSTALL_DIR} RENAME rbox.1)
 install(FILES ${doc_FILES}           DESTINATION ${DOC_INSTALL_DIR})
