--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -355,12 +355,20 @@ else()
     set(qhull_STATICR qhullstatic_r)
 endif()
 
+if(BUILD_SHARED_LIBS)
 set(
     qhull_TARGETS_INSTALL
-        ${qhull_CPP} ${qhull_STATIC} ${qhull_STATICR} ${qhull_SHAREDR}
+        ${qhull_SHAREDR}
         qhull rbox qconvex qdelaunay qvoronoi qhalf
-        ${qhull_SHARED} ${qhull_SHAREDP}  # Deprecated, use qhull_r instead
+        ${qhull_SHARED}
 )
+else()
+set(
+    qhull_TARGETS_INSTALL
+        ${qhull_STATIC} ${qhull_STATICR}
+        qhull rbox qconvex qdelaunay qvoronoi qhalf
+)
+endif()
 set(
     qhull_TARGETS_TEST   # Unused
         user_eg user_eg2 user_eg3 user_egp testqset testqset_r
@@ -561,10 +569,8 @@ target_link_libraries(user_eg2 ${qhull_STATICR})
 
 set(user_eg3_SOURCES    src/user_eg3/user_eg3_r.cpp)
 
-add_executable(user_eg3 ${user_eg3_SOURCES})
 # qhull_STATICR must be last, otherwise qh_fprintf,etc. are not loaded from qhull_CPP
 # user_eg3 may be linked to qhull_SHAREDR if user_eg3_DEFINES is added
-target_link_libraries(user_eg3 ${qhull_CPP} ${qhull_STATICR})
 
 # ---------------------------------------
 # qhullp is qhull/unix.c linked to deprecated qh_QHpointer libqhull_p
@@ -615,8 +621,6 @@ add_test(NAME user_eg
    COMMAND sh -c "./user_eg")
 add_test(NAME user_eg2
    COMMAND sh -c "./user_eg2")
-add_test(NAME user_eg3
-   COMMAND sh -c "./user_eg3 rbox '10 D2' '2 D2' qhull 's p' facets")
 
 # ---------------------------------------
 # Define install
@@ -681,10 +685,7 @@ foreach(pkgconfig IN ITEMS "${qhull_SHARED};Qhull shared library"
 endforeach()
 
 install(FILES ${libqhull_HEADERS}    DESTINATION ${INCLUDE_INSTALL_DIR}/libqhull)
-install(FILES ${libqhull_DOC}        DESTINATION ${INCLUDE_INSTALL_DIR}/libqhull)
 install(FILES ${libqhullr_HEADERS}    DESTINATION ${INCLUDE_INSTALL_DIR}/libqhull_r)
-install(FILES ${libqhullr_DOC}        DESTINATION ${INCLUDE_INSTALL_DIR}/libqhull_r)
-install(FILES ${libqhullcpp_HEADERS} DESTINATION ${INCLUDE_INSTALL_DIR}/libqhullcpp)
 install(FILES html/qhull.man         DESTINATION ${MAN_INSTALL_DIR} RENAME qhull.1)
 install(FILES html/rbox.man          DESTINATION ${MAN_INSTALL_DIR} RENAME rbox.1)
 install(FILES ${doc_FILES}           DESTINATION ${DOC_INSTALL_DIR})
