diff --git a/src/simdutf/icelake/intrinsics.h b/src/simdutf/icelake/intrinsics.h
index c71a085..03701be 100644
--- a/src/simdutf/icelake/intrinsics.h
+++ b/src/simdutf/icelake/intrinsics.h
@@ -62,6 +62,6 @@
 
 
 #if defined(__GNUC__) && !defined(__clang__)
-#if __GNUC__ == 8
+#if __GNUC__ == 8 || __GNUC__ == 9
 #define SIMDUTF_GCC8 1
 #endif //  __GNUC__ == 8
 #endif // defined(__GNUC__) && !defined(__clang__)
@@ -83,4 +83,4 @@ inline __m512i _mm512_set_epi8(uint8_t a0, uint8_t a1, uint8_t a2, uint8_t a3, u
 }
 #endif // SIMDUTF_GCC8
 
