diff --git a/src/icelake/icelake_utf8_common.inl.cpp b/src/icelake/icelake_utf8_common.inl.cpp
index 962700b..a192a18 100644
--- a/src/icelake/icelake_utf8_common.inl.cpp
+++ b/src/icelake/icelake_utf8_common.inl.cpp
@@ -448,12 +448,12 @@ __m512i prev(__m512i input, __m512i previous) {
     static_assert(N<=32, "N must be no larger than 32");
     const __m512i movemask = _mm512_setr_epi32(28,29,30,31,0,1,2,3,4,5,6,7,8,9,10,11);
     const __m512i rotated = _mm512_permutex2var_epi32(input, movemask, previous);
-#if SIMDUTF_GCC8
-    constexpr int shift = 16-N; // workaround for GCC8
+#if SIMDUTF_GCC8 || SIMDUTF_GCC9
+    constexpr int shift = 16-N; // workaround for GCC8,9
     return _mm512_alignr_epi8(input, rotated, shift);
 #else
     return _mm512_alignr_epi8(input, rotated, 16-N);
-#endif // SIMDUTF_GCC8
+#endif // SIMDUTF_GCC8 || SIMDUTF_GCC9
 }
 
 template <unsigned idx0, unsigned idx1, unsigned idx2, unsigned idx3>
diff --git a/src/simdutf/icelake/intrinsics.h b/src/simdutf/icelake/intrinsics.h
index c71a085..edcd289 100644
--- a/src/simdutf/icelake/intrinsics.h
+++ b/src/simdutf/icelake/intrinsics.h
@@ -64,7 +64,9 @@
 #if defined(__GNUC__) && !defined(__clang__)
 #if __GNUC__ == 8
 #define SIMDUTF_GCC8 1
-#endif //  __GNUC__ == 8
+#elif __GNUC__ == 9
+#define SIMDUTF_GCC9 1
+#endif //  __GNUC__ == 8 || __GNUC__ == 9
 #endif // defined(__GNUC__) && !defined(__clang__)
 
 #if SIMDUTF_GCC8
@@ -83,4 +85,4 @@ inline __m512i _mm512_set_epi8(uint8_t a0, uint8_t a1, uint8_t a2, uint8_t a3, u
 }
 #endif // SIMDUTF_GCC8
 
-#endif // SIMDUTF_HASWELL_INTRINSICS_H
\ No newline at end of file
+#endif // SIMDUTF_HASWELL_INTRINSICS_H
