diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index c52b352f..92fcd847 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -142,11 +142,6 @@ if (MSVC)
     set_target_properties(exiv2lib PROPERTIES LINK_FLAGS "/ignore:4099")
 endif()
 
-set_target_properties( exiv2lib_int PROPERTIES
-    POSITION_INDEPENDENT_CODE ON
-    COMPILE_DEFINITIONS exiv2lib_EXPORTS
-)
-
 target_include_directories(exiv2lib_int PRIVATE ${ZLIB_INCLUDE_DIR})
 target_include_directories(exiv2lib PRIVATE 
     $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/xmpsdk/include>
