diff --git a/CMakeLists.txt b/CMakeLists.txt
index 87cb1f7..4524490 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -87,7 +87,7 @@ function(generate_library_target target_name)
         target_include_directories(${target_name} PRIVATE
                 $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/src/deps/fast_float/single_include>)
     else ()
-        target_link_libraries(${target_name} INTERFACE
+        target_link_libraries(${target_name} PRIVATE
                 FastFloat::fast_float)
     endif ()
 endfunction()
@@ -104,7 +104,7 @@ function(generate_header_only_target target_name)
         target_include_directories(${target_name} INTERFACE
                 $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/src/deps/fast_float/single_include>)
     else ()
-        target_link_libraries(${target_name} INTERFACE
+        target_link_libraries(${target_name} PRIVATE
                 FastFloat::fast_float)
     endif ()
 endfunction()
