diff --git mDNSPosix/Makefile mDNSPosix/Makefile
index 7faa242..e398d35 100755
--- mDNSPosix/Makefile
+++ mDNSPosix/Makefile
@@ -78,6 +78,10 @@ LINKOPTS_PTHREAD = -lpthread
 LDSUFFIX = so
 JAVACFLAGS_OS = -fPIC -shared -ldns_sd
 
+ifeq ($(unicast_disabled), y)
+CFLAGS_COMMON += -DUNICAST_DISABLED
+endif
+
 # Set up diverging paths for debug vs. prod builds
 ifeq "$(DEBUG)" "1"
 CFLAGS_DEBUGGING = -g -DMDNS_DEBUGMSGS=2
