--- a/google/cloud/functions/CMakeLists.txt
+++ b/google/cloud/functions/CMakeLists.txt
@@ -132,7 +132,6 @@ install(
     RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} COMPONENT runtime
     LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
             COMPONENT runtime
-            NAMELINK_SKIP
     LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
             COMPONENT development
             NAMELINK_COMPONENT development
