From 891c7d9e56f1bfc42597fc803e3ca062875398d8 Mon Sep 17 00:00:00 2001
From: Rui Oliveira <ruimail24@gmail.com>
Date: Wed, 16 Jun 2021 13:22:17 +0100
Subject: [PATCH] Remove header and libc/libm checks

---
 configure.ac | 18 +++++-------------
 1 file changed, 5 insertions(+), 13 deletions(-)

diff --git a/configure.ac b/configure.ac
index e6bc297e..53aca562 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1,15 +1,15 @@
 # Copyright (c) 2007 - 2019 Joseph Gaeddert
-# 
+#
 # Permission is hereby granted, free of charge, to any person obtaining a copy
 # of this software and associated documentation files (the "Software"), to deal
 # in the Software without restriction, including without limitation the rights
 # to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 # copies of the Software, and to permit persons to whom the Software is
 # furnished to do so, subject to the following conditions:
-# 
+#
 # The above copyright notice and this permission notice shall be included in
 # all copies or substantial portions of the Software.
-# 
+#
 # THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 # IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 # FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
@@ -18,7 +18,7 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 # THE SOFTWARE.
 
-# 
+#
 # liquid-dsp library configure
 # Process with autoconf to generate configure script
 #
@@ -81,20 +81,12 @@ AC_FUNC_ERROR_AT_LINE
 AC_FUNC_MALLOC
 AC_FUNC_REALLOC
 
-# AC_CHECK_LIB (library, function, [action-if-found], [action-if-not-found], [other-libraries])
-AC_CHECK_LIB([c],[main],[],[AC_MSG_ERROR(Could not use standard C library)],   [])
-AC_CHECK_LIB([m],[main],[],[AC_MSG_ERROR(Could not use standard math library)],[])
-
 # AC_CHECK_FUNC(function, [action-if-found], [action-if-not-found])
 AC_CHECK_FUNCS([malloc realloc free memset memmove],[],
                [AC_MSG_ERROR(Could not use memory allocation functions)])
 AC_CHECK_FUNCS([sinf cosf expf cargf cexpf crealf cimagf sqrtf],[],
                [AC_MSG_ERROR(Could not use special math functions)],)
 
-# Check for necessary header files
-AC_CHECK_HEADERS([stdio.h stdlib.h complex.h string.h getopt.h sys/resource.h float.h inttypes.h limits.h stdlib.h string.h unistd.h],[],
-                 [AC_MSG_ERROR(Could not use standard headers)])
-
 # Check for optional header files, libraries, programs
 AC_CHECK_HEADERS(fec.h fftw3.h)
 AC_CHECK_LIB([fftw3f], [fftwf_plan_dft_1d], [],
@@ -248,7 +240,7 @@ esac
 # autoconf variable substitutions
 #
 AC_SUBST(LIBS)                      # shared libraries (-lc, -lm, etc.)
-AC_SUBST(MLIBS_DOTPROD)             # 
+AC_SUBST(MLIBS_DOTPROD)             #
 AC_SUBST(MLIBS_VECTOR)              #
 
 AC_SUBST(AR_LIB)                    # archive library
-- 
2.32.0.windows.1

