From 8a99b1c6853193c1f518bc81b705d7094225e114 Mon Sep 17 00:00:00 2001
From: Rui Oliveira <ruimail24@gmail.com>
Date: Fri, 18 Jun 2021 09:37:46 +0100
Subject: [PATCH] Remove headers check

---
 configure.ac | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/configure.ac b/configure.ac
index e6bc297e..5b591c87 100644
--- a/configure.ac
+++ b/configure.ac
@@ -81,20 +81,12 @@ AC_FUNC_ERROR_AT_LINE
 AC_FUNC_MALLOC
 AC_FUNC_REALLOC
 
-# AC_CHECK_LIB (library, function, [action-if-found], [action-if-not-found], [other-libraries])
-AC_CHECK_LIB([c],[main],[],[AC_MSG_ERROR(Could not use standard C library)],   [])
-AC_CHECK_LIB([m],[main],[],[AC_MSG_ERROR(Could not use standard math library)],[])
-
 # AC_CHECK_FUNC(function, [action-if-found], [action-if-not-found])
 AC_CHECK_FUNCS([malloc realloc free memset memmove],[],
                [AC_MSG_ERROR(Could not use memory allocation functions)])
 AC_CHECK_FUNCS([sinf cosf expf cargf cexpf crealf cimagf sqrtf],[],
                [AC_MSG_ERROR(Could not use special math functions)],)
 
-# Check for necessary header files
-AC_CHECK_HEADERS([stdio.h stdlib.h complex.h string.h getopt.h sys/resource.h float.h inttypes.h limits.h stdlib.h string.h unistd.h],[],
-                 [AC_MSG_ERROR(Could not use standard headers)])
-
 # Check for optional header files, libraries, programs
 AC_CHECK_HEADERS(fec.h fftw3.h)
 AC_CHECK_LIB([fftw3f], [fftwf_plan_dft_1d], [],
-- 
2.32.0.windows.1

