diff --git a/NvCloth/compiler/cmake/windows/NvCloth.cmake b/NvCloth/compiler/cmake/windows/NvCloth.cmake
index 065ea18..c2cb1a7 100644
--- a/NvCloth/compiler/cmake/windows/NvCloth.cmake
+++ b/NvCloth/compiler/cmake/windows/NvCloth.cmake
@@ -162,7 +162,11 @@ LIST(APPEND NVCLOTH_COMPILE_DEFS
 )
 ENDIF()
 
-SET(NVCLOTH_LIBTYPE SHARED)
+IF(DEFINED PX_STATIC_LIBRARIES)
+	SET(NVCLOTH_LIBTYPE STATIC)
+ELSE()
+	SET(NVCLOTH_LIBTYPE SHARED)
+ENDIF()
 
 # include common NvCloth settings
 INCLUDE(../common/NvCloth.cmake)
