--- a/tensorflow/lite/CMakeLists.txt	2023-04-12 18:04:31
+++ b/tensorflow/lite/CMakeLists.txt	2023-04-13 08:40:25
@@ -138,12 +138,11 @@
 endmacro()
 # Find TensorFlow Lite dependencies.
 find_package(absl REQUIRED)
-find_package(eigen REQUIRED)
+find_package(Eigen3 REQUIRED)
 find_package(farmhash REQUIRED)
-find_package(fft2d REQUIRED)
+find_package(fft REQUIRED)
 find_package(flatbuffers REQUIRED)
 find_package(gemmlowp REQUIRED)
-find_package(neon2sse REQUIRED)
 find_package(clog REQUIRED)
 find_package(cpuinfo REQUIRED)  #CPUINFO is used by XNNPACK and RUY library
 find_package(ruy REQUIRED)
@@ -151,6 +150,11 @@
 # We used to have an actual compilation logic with flatc but decided to use
 # schema_generated.h since flatc doesn't work with cross compilation.
 set(TFLITE_FLATBUFFERS_SCHEMA_DIR "${TFLITE_SOURCE_DIR}/schema")
+if(TARGET flatbuffers::flatbuffers_shared)
+    set(FLATBUFFERS_TARGET flatbuffers::flatbuffers_shared)
+else()
+    set(FLATBUFFERS_TARGET flatbuffers::flatbuffers)
+endif()
 set(TF_TARGET_PRIVATE_OPTIONS "")
 if(CMAKE_CXX_COMPILER_ID MATCHES "Clang$")
   # TensorFlow uses a heap of deprecated proto fields so surpress these
@@ -162,6 +166,10 @@
 set(TFLITE_TARGET_PRIVATE_OPTIONS "")
 # Additional library dependencies based upon enabled features.
 set(TFLITE_TARGET_DEPENDENCIES "")
+if (NOT CMAKE_SYSTEM_PROCESSOR OR CMAKE_SYSTEM_PROCESSOR MATCHES "x86")
+        find_package(NEON_2_SSE REQUIRED)
+        list(APPEND TFLITE_TARGET_DEPENDENCIES NEON_2_SSE::NEON_2_SSE)
+endif()
 if(CMAKE_CXX_COMPILER_ID MATCHES "Clang$")
   # TFLite uses deprecated methods in neon2sse which generates a huge number of
   # warnings so surpress these until they're fixed.
@@ -406,12 +414,14 @@
 if(TFLITE_ENABLE_XNNPACK)
   find_package(fp16_headers REQUIRED)
   find_package(xnnpack REQUIRED)
+  find_package(pthreadpool REQUIRED)
   populate_tflite_source_vars("delegates/xnnpack"
     TFLITE_DELEGATES_XNNPACK_SRCS
     FILTER ".*(_test|_tester)\\.(cc|h)"
   )
   list(APPEND TFLITE_TARGET_DEPENDENCIES
-    XNNPACK
+    xnnpack::xnnpack
+    pthreadpool::pthreadpool
   )
   list(APPEND TFLITE_TARGET_PUBLIC_OPTIONS "-DTFLITE_BUILD_WITH_XNNPACK_DELEGATE")
 endif()
@@ -519,17 +529,16 @@
 target_link_libraries(tensorflow-lite
   PUBLIC
     Eigen3::Eigen
-    NEON_2_SSE
     absl::flags
     absl::hash
     absl::status
     absl::strings
     absl::synchronization
     absl::variant
-    farmhash
-    fft2d_fftsg2d
-    flatbuffers
-    gemmlowp
+    farmhash::farmhash
+    fft::fft
+    ${FLATBUFFERS_TARGET}
+    gemmlowp::eight_bit_int_gemm
     ruy::ruy
     ${CMAKE_DL_LIBS}
     ${TFLITE_TARGET_DEPENDENCIES}
