diff --git a/tensorflow/lite/CMakeLists.txt b/tensorflow/lite/CMakeLists.txt
--- tensorflow/lite/CMakeLists.txt
+++ tensorflow/lite/CMakeLists.txt
@@ -142,9 +142,9 @@
 find_package(farmhash REQUIRED)
 find_package(fft2d REQUIRED)
 find_package(flatbuffers REQUIRED)
 find_package(gemmlowp REQUIRED)
-find_package(neon2sse REQUIRED)
+
 find_package(clog REQUIRED)
 find_package(cpuinfo REQUIRED)  #CPUINFO is used by XNNPACK and RUY library
 find_package(ruy REQUIRED)
 # Generate TensorFlow Lite FlatBuffer code.
@@ -156,16 +156,16 @@
   # TensorFlow uses a heap of deprecated proto fields so surpress these
   # warnings until they're fixed.
   list(APPEND TF_TARGET_PRIVATE_OPTIONS "-Wno-deprecated-declarations")
 endif()
-# Additional compiler flags used when compiling TF Lite.
 set(TFLITE_TARGET_PUBLIC_OPTIONS "")
 set(TFLITE_TARGET_PRIVATE_OPTIONS "")
-# Additional library dependencies based upon enabled features.
 set(TFLITE_TARGET_DEPENDENCIES "")
+if (NOT CMAKE_SYSTEM_PROCESSOR OR CMAKE_SYSTEM_PROCESSOR MATCHES "x86")
+  find_package(neon2sse REQUIRED)
+  list(APPEND TFLITE_TARGET_DEPENDENCIES NEON_2_SSE)
+endif()
 if(CMAKE_CXX_COMPILER_ID MATCHES "Clang$")
-  # TFLite uses deprecated methods in neon2sse which generates a huge number of
-  # warnings so surpress these until they're fixed.
   list(APPEND TFLITE_TARGET_PRIVATE_OPTIONS "-Wno-deprecated-declarations")
 endif()
 if(CMAKE_SYSTEM_NAME MATCHES "Windows")
   # Use NOMINMAX to disable the min / max macros in windows.h as they break
@@ -498,9 +498,9 @@
 )
 target_link_libraries(tensorflow-lite
   PUBLIC
     Eigen3::Eigen
-    NEON_2_SSE
+
     absl::flags
     absl::hash
     absl::status
     absl::strings
