diff --git a/CMakeLists.txt b/CMakeLists.txt
index d72f102..7bf1655 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -3,7 +3,7 @@ cmake_minimum_required(VERSION 3.9)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_EXTENSIONS OFF)
 set(CMAKE_CXX_STANDARD 11)
-set(CMAKE_POSITION_INDEPENDENT_CODE ON)
+# set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 
 # The project definition
 project(erkir VERSION 2.1.1
@@ -14,7 +14,7 @@ project(erkir VERSION 2.1.1
 include(GNUInstallDirs)
 
 # General options
-option(BUILD_SHARED_LIBS "Build using shared libraries" ON)
+# option(BUILD_SHARED_LIBS "Build using shared libraries" ON)
 option(ENABLE_TESTING "Enable unit test build" OFF)
 
 if (0)
