--- ogr/ogrsf_frmts/parquet/CMakeLists.txt
+++ ogr/ogrsf_frmts/parquet/CMakeLists.txt
@@ -9,7 +9,7 @@ add_gdal_driver(TARGET ogr_Parquet
 gdal_standard_includes(ogr_Parquet)
 target_include_directories(ogr_Parquet PRIVATE $<TARGET_PROPERTY:ogrsf_generic,SOURCE_DIR>
                                                $<TARGET_PROPERTY:ogr_MEM,SOURCE_DIR>)
-if(${Parquet_VERSION} VERSION_GREATER_EQUAL 10.0)
+if(${ARROW_VERSION} VERSION_GREATER_EQUAL 10.0)
     if(ARROW_USE_STATIC_LIBRARIES)
         gdal_target_link_libraries(ogr_Parquet PRIVATE Arrow::arrow_static Parquet::parquet_static)
     else()
@@ -26,7 +26,7 @@ endif()

 if (GDAL_USE_ARROWDATASET)
     target_compile_definitions(ogr_Parquet PRIVATE -DGDAL_USE_ARROWDATASET)
-    if(${ArrowDataset_VERSION} VERSION_GREATER_EQUAL 10.0)
+    if(${ARROW_VERSION} VERSION_GREATER_EQUAL 10.0)
         if(ARROW_USE_STATIC_LIBRARIES)
             gdal_target_link_libraries(ogr_Parquet PRIVATE ArrowDataset::arrow_dataset_static)
         else()