--- a/vcg/math/eigen.h
+++ b/vcg/math/eigen.h
@@ -53,10 +53,10 @@ struct ei_to_vcgtype;
 }
 
 #include "base.h"
-#include "../../eigenlib/Eigen/LU"
-#include "../../eigenlib/Eigen/Geometry"
-#include "../../eigenlib/Eigen/Array"
-#include "../../eigenlib/Eigen/Core"
+#include <Eigen/LU>
+#include <Eigen/Geometry>
+#include <Eigen/Array>
+#include <Eigen/Core>
 
 // add support for unsigned char and short int
 namespace Eigen {
