--- Source/C/NCSUtil/NCSPrefsXML.cpp.ori	2012-10-17 12:40:01.000000000 +0200
+++ Source/C/NCSUtil/NCSPrefsXML.cpp	2012-10-17 12:40:30.854812252 +0200
@@ -32,7 +32,11 @@
 	} else {
 		char *pHomeDir = getenv("HOME");
 		if( pHomeDir ) {
+#ifdef NCS_BUILD_UNICODE
+			sUserPrefsFile.Format( NCS_T("%s%ls"), pHomeDir, NCS_T("/.erm/ncsuserprefs.xml") );
+#else
 			sUserPrefsFile.Format( NCS_T("%s%s"), pHomeDir, NCS_T("/.erm/ncsuserprefs.xml") );
+#endif
 		} else {
 			sUserPrefsFile = NCS_T("/etc/erm/ncsuserprefs.xml");
 		}
