diff --git a/cmake/Dependencies.cmake b/cmake/Dependencies.cmake
index 06464e799a..72b7d2809f 100644
--- a/cmake/Dependencies.cmake
+++ b/cmake/Dependencies.cmake
@@ -992,6 +992,7 @@ if(BUILD_PYTHON)
 endif()
 
 # ---[ pybind11
+if(BUILD_PYTHON)
 if(NOT pybind11_PREFER_third_party)
   find_package(pybind11 CONFIG)
   if(NOT pybind11_FOUND)
@@ -1013,6 +1014,10 @@ else()
 endif()
 message(STATUS "pybind11 include dirs: " "${pybind11_INCLUDE_DIRS}")
 include_directories(SYSTEM ${pybind11_INCLUDE_DIRS})
+else()
+  find_package(Python 3.8.0 REQUIRED)
+  set(PYTHON_EXECUTABLE ${Python_EXECUTABLE})
+endif()
 
 # ---[ MPI
 if(USE_MPI)
