diff --git a/tools/codegen/dest/register_dispatch_key.py b/tools/codegen/dest/register_dispatch_key.py
index 6ab7f58843..87e762116b 100644
--- a/tools/codegen/dest/register_dispatch_key.py
+++ b/tools/codegen/dest/register_dispatch_key.py
@@ -1,6 +1,6 @@
 from typing import List, Optional, Union
 import itertools
-from typing_extensions import Literal
+from typing import Literal
 from dataclasses import dataclass
 
 from tools.codegen.context import *
diff --git a/tools/codegen/gen.py b/tools/codegen/gen.py
index 4b8e59d2b2..ab4c17a461 100644
--- a/tools/codegen/gen.py
+++ b/tools/codegen/gen.py
@@ -1,6 +1,6 @@
 import os
 from typing import List, Dict, Optional, Tuple, Set, Callable, Any, Union, Sequence
-from typing_extensions import Literal
+from typing import Literal
 import yaml
 from collections import OrderedDict, defaultdict
 import argparse
