diff --git a/source/CMakeLists.txt b/source/CMakeLists.txt
index fe6b710..30bd4aa 100644
--- a/source/CMakeLists.txt
+++ b/source/CMakeLists.txt
@@ -6,10 +6,13 @@ project(BearSource
         LANGUAGES C CXX
         )
 
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-exceptions")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DSPDLOG_NO_EXCEPTIONS")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DGOOGLE_PROTOBUF_NO_RTTI")
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra")
+
+if (CMAKE_CXX_COMPILER_FRONTEND_VARIANT STREQUAL "GNU")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-exceptions")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra")
+endif()
 
 if (ENABLE_UNIT_TESTS)
     find_program(MEMORYCHECK_COMMAND NAMES valgrind)
