diff --git a/source/CMakeLists.txt b/source/CMakeLists.txt
index fe6b710..8dc8655 100644
--- a/source/CMakeLists.txt
+++ b/source/CMakeLists.txt
@@ -26,8 +26,8 @@ find_package(Threads REQUIRED)
 find_package(nlohmann_json REQUIRED)
 find_package(fmt REQUIRED)
 find_package(spdlog REQUIRED)
-find_package(PkgConfig REQUIRED)
-pkg_check_modules(gRPC REQUIRED IMPORTED_TARGET protobuf grpc++)
+find_package(gRPC CONFIG REQUIRED)
+
 
 if (fmt_VERSION_MAJOR GREATER_EQUAL 9)
     set(FMT_NEEDS_OSTREAM_FORMATTER 1)
diff --git a/source/intercept/proto/CMakeLists.txt b/source/intercept/proto/CMakeLists.txt
index e4fb8c1..7fbdb18 100644
--- a/source/intercept/proto/CMakeLists.txt
+++ b/source/intercept/proto/CMakeLists.txt
@@ -62,4 +62,4 @@ target_sources(rpc_a
         INTERFACE
             $<TARGET_OBJECTS:rpc_a>
         )
-target_link_libraries(rpc_a PUBLIC PkgConfig::gRPC)
+target_link_libraries(rpc_a PUBLIC gRPC::grpc++)
