diff --git a/CMakeLists.txt b/CMakeLists.txt
index b776cb7..8bbe6a6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.8 FATAL_ERROR)
+cmake_minimum_required(VERSION 3.5 FATAL_ERROR)
 set(CMAKE_SUPPRESS_REGENERATION true)
 project("LAStools")
 
@@ -10,12 +10,5 @@ endif()
 
 option(BUILD_SHARED_LIBS "Build LASlib as DLL" OFF)
 
-if (BUILD_SHARED_LIBS AND UNIX AND NOT APPLE)
-	set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib/LASlib")
-endif()
 
 add_subdirectory(LASlib/src)
-add_subdirectory(src)
-if (EXISTS src_full)
-	add_subdirectory(src_full)
-endif()
diff --git a/LASlib/src/CMakeLists.txt b/LASlib/src/CMakeLists.txt
index 1b170bf..6a114eb 100644
--- a/LASlib/src/CMakeLists.txt
+++ b/LASlib/src/CMakeLists.txt
@@ -101,15 +101,7 @@ endif()
 
 install(FILES ${LAS_INCLUDES} DESTINATION include/LASlib)
 
-if (MSVC)
-	foreach( OUTPUTCONFIG ${CMAKE_CONFIGURATION_TYPES} )
-		install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/../lib/${OUTPUTCONFIG} DESTINATION lib/LASlib)
-	endforeach( OUTPUTCONFIG CMAKE_CONFIGURATION_TYPES )
-else()
-	install(TARGETS LASlib EXPORT laslib-targets
-		ARCHIVE DESTINATION lib/LASlib
-		LIBRARY DESTINATION lib/LASlib
-		RUNTIME DESTINATION lib/LASlib)
-	install(EXPORT laslib-targets DESTINATION lib/cmake/LASlib)
-	install(FILES ${CMAKE_SOURCE_DIR}/LASlib/src/laslib-config.cmake DESTINATION lib/cmake/LASlib)
-endif(MSVC)
+install(TARGETS LASlib EXPORT laslib-targets
+	ARCHIVE DESTINATION lib/LASlib
+	LIBRARY DESTINATION lib/LASlib
+	RUNTIME DESTINATION lib/LASlib)
