diff --git a/CMakeLists.txt b/CMakeLists.txt
index a7c70d1..3cf0915 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -112,10 +112,12 @@ target_include_directories(
 
 if (APPLE OR (CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND CMAKE_CXX_COMPILER_VERSION
     VERSION_LESS 12))
-
-    add_subdirectory(externals/fmt)
-    set(FMT_INCLUDE_DIR "${CMAKE_SOURCE_DIR}/externals/fmt/include")
-    target_link_libraries(${LIBRARY_NAME} PRIVATE fmt)
+    find_package(fmt REQUIRED CONFIG)
+    if (TARGET fmt::fmt)
+        target_link_libraries(${LIBRARY_NAME} PRIVATE fmt::fmt)
+    else()
+        target_link_libraries(${LIBRARY_NAME} PRIVATE fmt::fmt-header-only)
+    endif()
 endif ()
 
 if (BUILD_FAKER_TESTS)
