--- build/cmake/aom_install.cmake
+++ build/cmake/aom_install.cmake
@@ -93,4 +93,20 @@
                       "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_BINDIR}")
     endif()
   endif()
+
+  if (MSVC)
+    if(BUILD_SHARED_LIBS)
+      set_target_properties(aom_static PROPERTIES OUTPUT_NAME aom_static)
+    endif()
+    set(AOM_INSTALL_LIBS aom)
+    install(
+      FILES ${AOM_INSTALL_INCS}
+      DESTINATION "include/aom")
+    install(
+      TARGETS ${AOM_INSTALL_LIBS}
+      RUNTIME DESTINATION bin
+      LIBRARY DESTINATION lib
+      ARCHIVE DESTINATION lib)
+  endif()
+
 endmacro()
