diff --git a/CMakeLists.txt b/CMakeLists.txt
index 86e9eef..be3c79b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -16,18 +16,6 @@ set(CMAKE_CXX_STANDARD_REQUIRED ON)
 find_package(QT NAMES Qt6 Qt5 COMPONENTS Core Gui REQUIRED)
 find_package(Qt${QT_VERSION_MAJOR} COMPONENTS Core Gui REQUIRED)
 
-if(NOT DEFINED ${QXLSX_PARENTPATH})
-	set(QXLSX_PARENTPATH ${CMAKE_CURRENT_SOURCE_DIR}/../)
-endif(NOT DEFINED ${QXLSX_PARENTPATH}) 
-	
-if(NOT DEFINED ${QXLSX_HEADERPATH})	
-	set(QXLSX_HEADERPATH ${CMAKE_CURRENT_SOURCE_DIR}/../QXlsx/header/)
-endif(NOT DEFINED ${QXLSX_HEADERPATH})		
-
-if(NOT DEFINED ${QXLSX_SOURCEPATH})
-	set(QXLSX_SOURCEPATH ${CMAKE_CURRENT_SOURCE_DIR}/../QXlsx/source/)
-endif(NOT DEFINED ${QXLSX_SOURCEPATH})	
-
 message("Current Path of QXlsx")
 message(${QXLSX_PARENTPATH})
 message(${QXLSX_HEADERPATH})
@@ -39,7 +27,7 @@ file(GLOB QXLSX_H "${QXLSX_HEADERPATH}/*.h")
 set(SRC_FILES ${QXLSX_CPP})
 list(APPEND SRC_FILES ${QXLSX_H})
 
-add_library(QXlsx STATIC
+add_library(QXlsx
  ${SRC_FILES} )
 
 target_include_directories(QXlsx PRIVATE ${QXLSX_HEADERPATH})
