diff --git a/QXlsx/CMakeLists.txt b/QXlsx/CMakeLists.txt
index 86e9eef..d369bd2 100644
--- a/QXlsx/CMakeLists.txt
+++ b/QXlsx/CMakeLists.txt
@@ -1,21 +1,26 @@
 # CMakeLists.txt for QXlsx Library
 
-cmake_minimum_required(VERSION 3.14)
+cmake_minimum_required(VERSION 3.15.6)
 
-project(QXlsx LANGUAGES CXX)
+project(QXlsx
+    VERSION 1.4.4
+    LANGUAGES CXX
+)
 
 set(CMAKE_INCLUDE_CURRENT_DIR ON)
 
-set(CMAKE_AUTOUIC ON)
 set(CMAKE_AUTOMOC ON)
-set(CMAKE_AUTORCC ON)
 
 set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
+include(GNUInstallDirs)
+
 find_package(QT NAMES Qt6 Qt5 COMPONENTS Core Gui REQUIRED)
 find_package(Qt${QT_VERSION_MAJOR} COMPONENTS Core Gui REQUIRED)
 
+set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules)
+
 if(NOT DEFINED ${QXLSX_PARENTPATH})
 	set(QXLSX_PARENTPATH ${CMAKE_CURRENT_SOURCE_DIR}/../)
 endif(NOT DEFINED ${QXLSX_PARENTPATH}) 
@@ -33,14 +38,43 @@ message(${QXLSX_PARENTPATH})
 message(${QXLSX_HEADERPATH})
 message(${QXLSX_SOURCEPATH})
 
+# Due historical reasons this value is kept off
+option(BUILD_SHARED_LIBS "Build in shared lib mode" OFF)
+
 file(GLOB QXLSX_CPP "${QXLSX_SOURCEPATH}/*.cpp")
 file(GLOB QXLSX_H "${QXLSX_HEADERPATH}/*.h")
 
 set(SRC_FILES ${QXLSX_CPP})
 list(APPEND SRC_FILES ${QXLSX_H})
 
-add_library(QXlsx STATIC
- ${SRC_FILES} )
+set(QXLSX_PUBLIC_HEADERS
+    header/xlsxdocument.h
+    header/xlsxabstractooxmlfile.h
+    header/xlsxabstractsheet.h
+    header/xlsxabstractsheet_p.h
+    header/xlsxcellformula.h
+    header/xlsxcell.h
+    header/xlsxcelllocation.h
+    header/xlsxcellrange.h
+    header/xlsxcellreference.h
+    header/xlsxchart.h
+    header/xlsxchartsheet.h
+    header/xlsxconditionalformatting.h
+    header/xlsxdatavalidation.h
+    header/xlsxdatetype.h
+    header/xlsxdocument.h
+    header/xlsxformat.h
+    header/xlsxglobal.h
+    header/xlsxrichstring.h
+    header/xlsxworkbook.h
+    header/xlsxworksheet.h
+)
+
+add_library(QXlsx
+    ${SRC_FILES}
+    ${QXLSX_PUBLIC_HEADERS}
+)
+add_library(QXlsx::Core ALIAS QXlsx)
 
 target_include_directories(QXlsx PRIVATE ${QXLSX_HEADERPATH})
 
@@ -49,4 +83,48 @@ target_link_libraries(${PROJECT_NAME}
  Qt${QT_VERSION_MAJOR}::GuiPrivate
  )
 
+set_property(TARGET QXlsx PROPERTY PUBLIC_HEADER ${QXLSX_PUBLIC_HEADERS})
+
+target_include_directories(QXlsx PUBLIC
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/header>
+  $<INSTALL_INTERFACE:include/QXlsx>
+)
+
+set_target_properties(QXlsx PROPERTIES
+    EXPORT_NAME Core
+    VERSION ${PROJECT_VERSION}
+    SOVERSION 0
+)
+
+install(TARGETS QXlsx
+    EXPORT QXlsxTargets DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    RUNTIME DESTINATION bin COMPONENT runtime
+    ARCHIVE DESTINATION lib COMPONENT devel
+    PUBLIC_HEADER DESTINATION include/QXlsx COMPONENT devel
+)
+
+install(EXPORT QXlsxTargets
+    DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/QXlsx/
+    FILE QXlsxTargets.cmake
+    NAMESPACE QXlsx::
+    COMPONENT Devel
+)
+
+configure_file(${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules/qxlsx-config.cmake.in
+    ${CMAKE_CURRENT_BINARY_DIR}/QXlsxConfig.cmake
+    @ONLY
+)
+configure_file(${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules/qxlsx-config-version.cmake.in
+    ${CMAKE_CURRENT_BINARY_DIR}/QXlsxConfigVersion.cmake
+    @ONLY
+)
+install(
+    FILES
+        ${CMAKE_CURRENT_BINARY_DIR}/QXlsxConfig.cmake
+        ${CMAKE_CURRENT_BINARY_DIR}/QXlsxConfigVersion.cmake
+    DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/QXlsx/
+)
+
 target_compile_definitions(QXlsx PRIVATE QXLSX_LIBRARY)
+
+include(CPackConfig)
diff --git a/QXlsx/cmake/modules/CPackConfig.cmake b/QXlsx/cmake/modules/CPackConfig.cmake
new file mode 100644
index 0000000..07dd1ab
--- /dev/null
+++ b/QXlsx/cmake/modules/CPackConfig.cmake
@@ -0,0 +1,21 @@
+# SPDX-FileCopyrightText: (C) 2021 Daniel Nicoletti <dantti12@gmail.com>
+# SPDX-License-Identifier: MIT
+
+set(CPACK_PACKAGE_VENDOR "Daniel Nicoletti")
+set(CPACK_PACKAGE_DESCRIPTION_SUMMARY "QXlsx library.")
+set(CPACK_RESOURCE_FILE_README "${CMAKE_CURRENT_SOURCE_DIR}/../README.md")
+set(CPACK_PACKAGE_CONTACT "Daniel Nicoletti <dantti12@gmail.com>")
+
+if(UNIX)
+  if(NOT CPACK_GENERATOR)
+    set(CPACK_GENERATOR "DEB")
+  endif()
+
+  set(CPACK_DEBIAN_PACKAGE_SHLIBDEPS ON)
+  set(CPACK_STRIP_FILES 1)
+  if(${CMAKE_VERSION} VERSION_GREATER "3.5")
+    set(CPACK_DEBIAN_FILE_NAME DEB-DEFAULT)
+  endif()
+endif()
+
+include(CPack)
diff --git a/QXlsx/cmake/modules/qxlsx-config-version.cmake.in b/QXlsx/cmake/modules/qxlsx-config-version.cmake.in
new file mode 100644
index 0000000..68cbf8f
--- /dev/null
+++ b/QXlsx/cmake/modules/qxlsx-config-version.cmake.in
@@ -0,0 +1,15 @@
+# SPDX-FileCopyrightText: (C) 2021 Daniel Nicoletti <dantti12@gmail.com>
+# SPDX-License-Identifier: MIT
+
+SET(PACKAGE_VERSION @PROJECT_VERSION@)
+IF (PACKAGE_FIND_VERSION VERSION_EQUAL PACKAGE_VERSION)
+  SET(PACKAGE_VERSION_EXACT "true")
+ENDIF (PACKAGE_FIND_VERSION VERSION_EQUAL PACKAGE_VERSION)
+IF (NOT PACKAGE_FIND_VERSION VERSION_GREATER PACKAGE_VERSION)
+  SET(PACKAGE_VERSION_COMPATIBLE "true")
+ELSE (NOT PACKAGE_FIND_VERSION VERSION_GREATER PACKAGE_VERSION)
+  SET(PACKAGE_VERSION_UNSUITABLE "true")
+ENDIF (NOT PACKAGE_FIND_VERSION VERSION_GREATER PACKAGE_VERSION)
+IF (PACKAGE_VERSION_UNSUITABLE)
+  MESSAGE("VERSION CHECK FAILED FOR ${PACKAGE_FIND_NAME}. WANTED ${PACKAGE_FIND_VERSION}, HAVE ${PACKAGE_VERSION}")
+ENDIF(PACKAGE_VERSION_UNSUITABLE)
diff --git a/QXlsx/cmake/modules/qxlsx-config.cmake.in b/QXlsx/cmake/modules/qxlsx-config.cmake.in
new file mode 100644
index 0000000..1aab813
--- /dev/null
+++ b/QXlsx/cmake/modules/qxlsx-config.cmake.in
@@ -0,0 +1,14 @@
+# SPDX-FileCopyrightText: (C) 2021 Daniel Nicoletti <dantti12@gmail.com>
+# SPDX-License-Identifier: MIT
+
+# - Config information for QXlsx
+# This file defines:
+#
+#  QXlsx_INCLUDE_DIR - the QXlsx include directory
+#  QXlsx_LIBRARY - Link these to use QXlsx
+
+SET(prefix "@CMAKE_INSTALL_PREFIX@")
+SET(exec_prefix "@CMAKE_INSTALL_PREFIX@")
+SET(QXlsx_FOUND "TRUE")
+
+include("${CMAKE_CURRENT_LIST_DIR}/QXlsxTargets.cmake")
