diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2bfab61..4873040 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -105,28 +105,28 @@ list(APPEND CMAKE_MODULE_PATH "${SOURCE_DIR}")
 set(OPTIONAL_LIBRARIES)
 
 ########################################################################
-# LIBZMQ dependency
+# ZEROMQ dependency
 ########################################################################
-find_package(libzmq REQUIRED)
-IF (LIBZMQ_FOUND)
-    include_directories(${LIBZMQ_INCLUDE_DIRS})
-    list(APPEND MORE_LIBRARIES ${LIBZMQ_LIBRARIES})
+find_package(ZeroMQ REQUIRED)
+IF (ZEROMQ_FOUND)
+    include_directories(${ZEROMQ_INCLUDE_DIRS})
+    list(APPEND MORE_LIBRARIES ${ZEROMQ_LIBRARIES})
     set(pkg_config_libs_private "${pkg_config_libs_private} -lzmq")
-ELSE (LIBZMQ_FOUND)
-    message( FATAL_ERROR "libzmq not found." )
-ENDIF (LIBZMQ_FOUND)
+ELSE (ZEROMQ_FOUND)
+    message( FATAL_ERROR "ZEROMQ not found." )
+ENDIF (ZEROMQ_FOUND)
 
 ########################################################################
 # CZMQ dependency
 ########################################################################
 find_package(czmq REQUIRED)
-IF (CZMQ_FOUND)
-    include_directories(${CZMQ_INCLUDE_DIRS})
-    list(APPEND MORE_LIBRARIES ${CZMQ_LIBRARIES})
+IF (czmq_FOUND)
+    include_directories(${czmq_INCLUDE_DIRS})
+    list(APPEND MORE_LIBRARIES ${czmq_LIBRARIES})
     set(pkg_config_libs_private "${pkg_config_libs_private} -lczmq")
-ELSE (CZMQ_FOUND)
+ELSE (czmq_FOUND)
     message( FATAL_ERROR "czmq not found." )
-ENDIF (CZMQ_FOUND)
+ENDIF (czmq_FOUND)
 
 ########################################################################
 # includes
@@ -218,8 +218,8 @@ add_executable(
 target_link_libraries(
     perf_local
     zyre
-    ${LIBZMQ_LIBRARIES}
-    ${CZMQ_LIBRARIES}
+    ${ZEROMQ_LIBRARIES}
+    ${czmq_LIBRARIES}
     ${OPTIONAL_LIBRARIES}
 )
 set_target_properties(
@@ -233,8 +233,8 @@ add_executable(
 target_link_libraries(
     perf_remote
     zyre
-    ${LIBZMQ_LIBRARIES}
-    ${CZMQ_LIBRARIES}
+    ${ZEROMQ_LIBRARIES}
+    ${czmq_LIBRARIES}
     ${OPTIONAL_LIBRARIES}
 )
 set_target_properties(
@@ -248,8 +248,8 @@ add_executable(
 target_link_libraries(
     zpinger
     zyre
-    ${LIBZMQ_LIBRARIES}
-    ${CZMQ_LIBRARIES}
+    ${ZEROMQ_LIBRARIES}
+    ${czmq_LIBRARIES}
     ${OPTIONAL_LIBRARIES}
 )
 set_target_properties(
@@ -266,8 +266,8 @@ add_executable(
 target_link_libraries(
     ztester_beacon
     zyre
-    ${LIBZMQ_LIBRARIES}
-    ${CZMQ_LIBRARIES}
+    ${ZEROMQ_LIBRARIES}
+    ${czmq_LIBRARIES}
     ${OPTIONAL_LIBRARIES}
 )
 set_target_properties(
@@ -281,8 +281,8 @@ add_executable(
 target_link_libraries(
     ztester_gossip
     zyre
-    ${LIBZMQ_LIBRARIES}
-    ${CZMQ_LIBRARIES}
+    ${ZEROMQ_LIBRARIES}
+    ${czmq_LIBRARIES}
     ${OPTIONAL_LIBRARIES}
 )
 set_target_properties(
@@ -296,8 +296,8 @@ add_executable(
 target_link_libraries(
     zyre_selftest
     zyre
-    ${LIBZMQ_LIBRARIES}
-    ${CZMQ_LIBRARIES}
+    ${ZEROMQ_LIBRARIES}
+    ${czmq_LIBRARIES}
     ${OPTIONAL_LIBRARIES}
 )
 set_target_properties(
