--- Modules/_ctypes/cfield.c
+++ Modules/_ctypes/cfield.c
@@ -1490,7 +1490,11 @@
     return PyLong_FromVoidPtr(*(void **)ptr);
 }
 
-static struct fielddesc formattable[] = {
+#define FORMATTABLE_SIZE 30
+static struct fielddesc formattable[FORMATTABLE_SIZE];
+
+static void formattable_init(void) {
+struct fielddesc my_formattable[] = {
     { 's', s_set, s_get, &ffi_type_pointer},
     { 'b', b_set, b_get, &ffi_type_schar},
     { 'B', B_set, B_get, &ffi_type_uchar},
@@ -1542,6 +1546,11 @@
     { 'O', O_set, O_get, &ffi_type_pointer},
     { 0, NULL, NULL, NULL},
 };
+    size_t nb = 1;
+    for (struct fielddesc *pos = my_formattable; pos->code; ++pos, ++nb);
+    if (FORMATTABLE_SIZE < nb) abort();
+    memcpy(formattable, my_formattable, nb * sizeof(struct fielddesc));
+}
 
 /*
   Ideas: Implement VARIANT in this table, using 'V' code.
@@ -1556,5 +1565,6 @@
 
     if (!initialized) {
         initialized = 1;
+        formattable_init();
         if (sizeof(wchar_t) == sizeof(short))
             _ctypes_get_fielddesc("u")->pffi_type = &ffi_type_sshort;
@@ -1616,7 +1626,7 @@ typedef struct _ffi_type
     struct _ffi_type **elements;
 } ffi_type;
 */
-
+#if 0
 /* align and size are bogus for void, but they must not be zero */
 ffi_type ffi_type_void = { 1, 1, FFI_TYPE_VOID };
 
@@ -1643,5 +1653,5 @@
                                  FFI_TYPE_LONGDOUBLE };
 
 ffi_type ffi_type_pointer = { sizeof(void *), VOID_P_ALIGN, FFI_TYPE_POINTER };
-
+#endif
 /*---------------- EOF ----------------*/
