diff --git a/cmake/BsoncxxUtil.cmake b/cmake/BsoncxxUtil.cmake
index b38f9af..7ec9d72 100644
--- a/cmake/BsoncxxUtil.cmake
+++ b/cmake/BsoncxxUtil.cmake
@@ -36,7 +36,7 @@ function(bsoncxx_add_library TARGET OUTPUT_NAME LINK_TYPE)
 	if (CMAKE_VERSION VERSION_LESS 3.15.0)
 	  target_include_directories(${TARGET} PUBLIC ${Boost_INCLUDE_DIRS})
 	else()
-          target_link_libraries(${TARGET} PUBLIC Boost::boost)
+          target_link_libraries(${TARGET} PUBLIC CONAN_PKG::boost)
 	endif()
     endif()
 
