diff --git a/CMakeLists.txt b/CMakeLists.txt
index e16167a..c188915 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -47,6 +47,15 @@ ELSE()
   SET(CMAKE_C_STANDARD 99)
 ENDIF()
 
+FIND_PACKAGE(mariadb-connector-c REQUIRED)
+
+if(BUILD_SHARED_LIBS)
+  set(MARIADB_LIBRARY_TYPE SHARED)
+else()
+  set(MARIADB_LIBRARY_TYPE STATIC)
+endif(BUILD_SHARED_LIBS)
+
+
 SET(CMAKE_CXX_STANDARD 11)
 SET(CMAKE_CXX_STANDARD_REQUIRED ON)
 
@@ -62,7 +71,6 @@ LIST(REMOVE_ITEM MAODBC_LANGUAGES "RC")
 SET(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_SOURCE_DIR}/cmake)
 
 SET(LIBRARY_NAME "mariadbcpp")
-SET(STATIC_LIBRARY_NAME "${LIBRARY_NAME}-static")
 
 CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/src/Version.h.in
                ${CMAKE_SOURCE_DIR}/src/Version.h)
@@ -565,7 +573,7 @@ IF(MARIADB_LINK_DYNAMIC)
                                              ShLwApi
                 DOC "Path to ShLwApi.Lib. Usually found in $ENV{programfiles\(x86\)}\\Windows Kits\\10\\Lib")
     ELSE()
-      FIND_LIBRARY(CCLIB libmariadb.so)
+      SET(CCLIB mariadb-connector-c::mariadb-connector-c)
       IF (${CCLIB} STREQUAL "CCLIB-NOTFOUND")
         # It won't be found by linker either unless user does some magic before build, so we actually could stop here with error
         MESSAGE(STATUS "C/C library has not been found")
@@ -580,6 +588,7 @@ IF(MARIADB_LINK_DYNAMIC)
                       /usr/local/include/mariadb
                       /usr/local/include/mysql)
                 #PATH_SUFFIXES mariadb mysql)
+      FIND_FILE(CCHEADER NAMES "mysql.h")
       IF (${CCHEADER} STREQUAL "CCHEADER-NOTFOUND")
         # Again should probably stop
         MESSAGE(STATUS "C/C headers have not been found")
@@ -606,7 +615,7 @@ IF(UNIX)
 ENDIF()
 SET(${LIBRARY_NAME}_OBJECTS $<TARGET_OBJECTS:${LIBRARY_NAME}_obj>)
 
-IF(WIN32 AND NOT MINGW)
+IF(False)
   ADD_LIBRARY(${LIBRARY_NAME} SHARED ${CMAKE_SOURCE_DIR}/src/maconncpp.rc ${${LIBRARY_NAME}_OBJECTS} ${CMAKE_SOURCE_DIR}/src/maconncpp.def)
   ADD_LIBRARY(${STATIC_LIBRARY_NAME} STATIC ${${LIBRARY_NAME}_OBJECTS} ${CMAKE_SOURCE_DIR}/src/maconncpp.rc)
 
@@ -624,36 +633,39 @@ ELSE()
     SET(EMPTY_FILE ${CMAKE_BINARY_DIR}/empty.cpp)
   ENDIF()
   #  MESSAGE(STATUS "Version script: ${CMAKE_SOURCE_DIR}/src/maconncpp.def")
-  ADD_LIBRARY(${LIBRARY_NAME} SHARED ${${LIBRARY_NAME}_OBJECTS} ${EMPTY_FILE})
-  ADD_LIBRARY(${STATIC_LIBRARY_NAME} STATIC ${${LIBRARY_NAME}_OBJECTS} ${EMPTY_FILE})
+  ADD_LIBRARY(${LIBRARY_NAME} ${MARIADB_LIBRARY_TYPE} ${${LIBRARY_NAME}_OBJECTS} ${EMPTY_FILE})
 
   IF(APPLE)
+    if (MARIADB_LIBRARY_TYPE STREQUAL "SHARED")
     SET_TARGET_PROPERTIES(${LIBRARY_NAME} PROPERTIES LINK_FLAGS "-Wl"
                                      INSTALL_RPATH_USE_LINK_PATH 0
                                      BUILD_WITH_INSTALL_RPATH 1
                                      XCODE_ATTRIBUTE_ENABLE_HARDENED_RUNTIME YES
                                      XCODE_ATTRIBUTE_OTHER_CODE_SIGN_FLAGS "--timestamp -f"
                                      INSTALL_RPATH "${MAODBC_INSTALL_RPATH}")
+    else()
     SET_TARGET_PROPERTIES(${STATIC_LIBRARY_NAME} PROPERTIES 
                                      INSTALL_RPATH_USE_LINK_PATH 0
                                      BUILD_WITH_INSTALL_RPATH 1
                                      XCODE_ATTRIBUTE_ENABLE_HARDENED_RUNTIME YES
                                      XCODE_ATTRIBUTE_OTHER_CODE_SIGN_FLAGS "--timestamp -f"
                                      INSTALL_RPATH "${MAODBC_INSTALL_RPATH}")
+    endif()
     IF(WITH_SIGNCODE)
       SET_TARGET_PROPERTIES(${LIBRARY_NAME} PROPERTIES XCODE_ATTRIBUTE_CODE_SIGN_IDENTITY "Developer ID Application: ${DEVELOPER_ID}")
-      SET_TARGET_PROPERTIES(${STATIC_LIBRARY_NAME} PROPERTIES XCODE_ATTRIBUTE_CODE_SIGN_IDENTITY "Developer ID Application: ${DEVELOPER_ID}")
     ELSE()
       SET_TARGET_PROPERTIES(${LIBRARY_NAME} PROPERTIES XCODE_ATTRIBUTE_CODE_SIGN_IDENTITY "")
-      SET_TARGET_PROPERTIES(${STATIC_LIBRARY_NAME} PROPERTIES XCODE_ATTRIBUTE_CODE_SIGN_IDENTITY "")
     ENDIF()
   ELSE()
 #    SET_TARGET_PROPERTIES(${LIBRARY_NAME} PROPERTIES LINK_FLAGS "-Wl,--version-script=${CMAKE_SOURCE_DIR}/src/maconncpp.def")
   ENDIF()
 ENDIF()
 
-TARGET_LINK_LIBRARIES(${LIBRARY_NAME} ${MARIADB_CLIENT_TARGET_NAME} ${PLATFORM_DEPENDENCIES})
-TARGET_LINK_LIBRARIES(${STATIC_LIBRARY_NAME} mariadbclient)
+if (MARIADB_LIBRARY_TYPE STREQUAL "SHARED")
+  TARGET_LINK_LIBRARIES(${LIBRARY_NAME} ${MARIADB_CLIENT_TARGET_NAME} ${PLATFORM_DEPENDENCIES})
+else()
+  TARGET_LINK_LIBRARIES(${LIBRARY_NAME} mariadbclient)
+endif()
 
 # MINGW Specific includes and linking
 IF(MINGW)
@@ -676,7 +688,6 @@ IF(MINGW)
   
   IF(MARIADB_CONNECTORC_LIB)
     TARGET_LINK_DIRECTORIES(${LIBRARY_NAME} PUBLIC ${MARIADB_CONNECTORC_LIB})
-    TARGET_LINK_DIRECTORIES(${STATIC_LIBRARY_NAME} PUBLIC ${MARIADB_CONNECTORC_LIB})
     MESSAGE(STATUS "Found MariaDB Connector/C libraries - ${MARIADB_CONNECTORC_LIB}")
   ELSE()
     MESSAGE(WARNING "Did not find MariaDB Connector/C libraries")
@@ -685,14 +696,16 @@ IF(MINGW)
 ENDIF()
 
 INCLUDE(symlink)
-CREATE_SYMLINK(lib${LIBRARY_NAME}${CMAKE_STATIC_LIBRARY_SUFFIX} ${STATIC_LIBRARY_NAME} ${INSTALL_LIBDIR})
+if (MARIADB_LIBRARY_TYPE STREQUAL "STATIC")
+  CREATE_SYMLINK(lib${LIBRARY_NAME}${CMAKE_STATIC_LIBRARY_SUFFIX} ${LIBRARY_NAME} ${INSTALL_LIBDIR})
+endif()
 
 ADD_DEPENDENCIES(${LIBRARY_NAME} DEPENDENCIES_FOR_PACKAGE)
 
 ########## Packaging ##########
 
 # MSI
-IF(WIN32)
+IF(False)
   IF(WITH_MSI)
     ADD_CUSTOM_COMMAND(TARGET ${LIBRARY_NAME} POST_BUILD
       COMMAND ${CMAKE_COMMAND} ARGS -DDRIVER_LIB_DIR=$<TARGET_FILE_DIR:${LIBRARY_NAME}>
@@ -709,13 +722,16 @@ ELSE()
     #MESSAGE(STATUS "Configuring to generate PKG package")
     #ADD_SUBDIRECTORY(osxinstall)
   ENDIF()
-  INSTALL(TARGETS ${LIBRARY_NAME}
-          LIBRARY DESTINATION ${INSTALL_LIBDIR}
-          COMPONENT SharedLibraries)
-  INSTALL(TARGETS
-          ${STATIC_LIBRARY_NAME}
-          ARCHIVE DESTINATION ${INSTALL_LIBDIR}
-          COMPONENT Development)
+  if (MARIADB_LIBRARY_TYPE STREQUAL "SHARED")
+    INSTALL(TARGETS ${LIBRARY_NAME}
+            LIBRARY DESTINATION ${INSTALL_LIBDIR}
+            COMPONENT SharedLibraries)
+  else()
+    INSTALL(TARGETS
+            ${LIBRARY_NAME}
+            ARCHIVE DESTINATION ${INSTALL_LIBDIR}
+            COMPONENT Development)
+  endif()
 
   MESSAGE(STATUS "Documentation installed to ${INSTALL_DOCDIR}")
   MESSAGE(STATUS "License file installed to ${INSTALL_LICENSEDIR}")
