--- a/cmake/options_defaults.cmake	    2024-07-26 18:41:28.417453206 +0500
+++ b/cmake/options.cmake	            2024-07-26 18:41:52.224451154 +0500
@@ -12,11 +12,11 @@
 OPTION(WITH_MSAN "Enable memory sanitizer" OFF)
 
 IF(WIN32 AND NOT MINGW)
-  OPTION(WITH_MSI "Build MSI installation package" ON)
+  OPTION(WITH_MSI "Build MSI installation package" OFF)
   OPTION(CONC_WITH_MSI "Build C/C MSI installation package" OFF)
   OPTION(WITH_SIGNCODE "Digitally sign files" OFF)
 
-  OPTION(MARIADB_LINK_DYNAMIC "Link Connector/C library dynamically" OFF)
+  OPTION(MARIADB_LINK_DYNAMIC "Link Connector/C library dynamically" ON)
   OPTION(ALL_PLUGINS_STATIC "Compile all plugins in, i.e. make them static" OFF)
   SET(CLIENT_PLUGIN_PVIO_NPIPE "STATIC")
   # We don't provide its support in ODBC yet, thus there is no need to bloat the library size
@@ -26,7 +26,7 @@
 ELSE()
   OPTION(WITH_MSI "Build MSI installation package" OFF)
   IF(APPLE)
-    OPTION(MARIADB_LINK_DYNAMIC "Link Connector/C library dynamically" OFF)
+    OPTION(MARIADB_LINK_DYNAMIC "Link Connector/C library dynamically" ON)
     OPTION(WITH_SIGNCODE "Digitally sign files" OFF)
     IF(NOT "${DEVELOPER_ID}" STREQUAL "")
       SET(WITH_SIGNCODE ON)
@@ -99,7 +99,7 @@
     SET(CLIENT_PLUGIN_SHA256_PASSWORD "STATIC")
     SET(CLIENT_PLUGIN_MYSQL_CLEAR_PASSWORD "STATIC")
     SET(CLIENT_PLUGIN_MYSQL_OLD_PASSWORD "STATIC")
-    SET(MARIADB_LINK_DYNAMIC OFF)
+    SET(MARIADB_LINK_DYNAMIC ON)
   ENDIF()
 ELSE()
   SET(WITH_MSI OFF)
