diff --git a/CMake/BuildFlatBuffers.cmake b/CMake/BuildFlatBuffers.cmake
index 835f4b80..edc2fde7 100644
--- a/CMake/BuildFlatBuffers.cmake
+++ b/CMake/BuildFlatBuffers.cmake
@@ -60,7 +60,7 @@ function(build_flatbuffers flatbuffers_schemas
     set(FLATC_TARGET "")
     set(FLATC ${FLATBUFFERS_FLATC_EXECUTABLE})
   else()
-    set(FLATC_TARGET flatc)
+    set(FLATC_TARGET "")
     set(FLATC flatc)
   endif()
   set(FLATC_SCHEMA_ARGS --gen-mutable)
