--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -514,6 +514,7 @@ function(compile_flatbuffers_schema_to_embedded_binary SRC_FBS OPT)
   register_generated_output(${GEN_BFBS_HEADER})
 endfunction()
 
+if(0)
 # Look if we have python 3 installed so that we can run the generate code python
 # script after flatc is built.
 find_package(PythonInterp 3)
@@ -544,6 +545,7 @@ if(PYTHONINTERP_FOUND AND
 else()
   message("No Python3 interpreter found! Unable to generate files automatically.")
 endif()
+endif()
 
 if(FLATBUFFERS_BUILD_TESTS)
   file(COPY "${CMAKE_CURRENT_SOURCE_DIR}/tests" DESTINATION "${CMAKE_CURRENT_BINARY_DIR}")
