--- a/lib/CMakeLists.txt
+++ b/lib/CMakeLists.txt
@@ -84,6 +84,10 @@ if (NOT BUILD_FRAMEWORK)
     set_target_properties(vorbisfile PROPERTIES SOVERSION ${VORBISFILE_VERSION_INFO})
 
     target_link_libraries(vorbis ${OGG_LIBRARIES})
+    find_library(M_LIB m)
+    if(M_LIB)
+        target_link_libraries(vorbis ${M_LIB})
+    endif()
     target_link_libraries(vorbisenc ${OGG_LIBRARIES} vorbis)
     target_link_libraries(vorbisfile ${OGG_LIBRARIES} vorbis)
 
