--- a/lib/CMakeLists.txt
+++ b/lib/CMakeLists.txt
@@ -84,6 +84,9 @@ if (NOT BUILD_FRAMEWORK)
     set_target_properties(vorbisfile PROPERTIES SOVERSION ${VORBISFILE_VERSION_INFO})
 
     target_link_libraries(vorbis ${OGG_LIBRARIES})
+    if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
+        target_link_libraries(vorbis m)
+    endif()
     target_link_libraries(vorbisenc ${OGG_LIBRARIES} vorbis)
     target_link_libraries(vorbisfile ${OGG_LIBRARIES} vorbis)
 
