--- Makefile.am
+++ Makefile.am
@@ -2,7 +2,7 @@
 # copyright : Copyright (c) 2013-2014 Code Synthesis Tools CC
 # license   : MIT; see accompanying LICENSE file
 
-SUBDIRS = xml tests examples doc
-dist_doc_DATA = LICENSE NEWS README INSTALL version
+SUBDIRS = xml
+dist_doc_DATA = LICENSE NEWS README INSTALL
 EXTRA_DIST = tester.bat libstudxml-vc9.sln libstudxml-vc10.sln libstudxml-vc11.sln libstudxml-vc12.sln bootstrap tester.in
 ACLOCAL_AMFLAGS = -I m4
