--- a/loader/CMakeLists.txt
+++ b/loader/CMakeLists.txt
@@ -160,9 +160,7 @@ if(WIN32)
         target_link_libraries(asm_offset PRIVATE loader_specific_options)
         # If not cross compiling, run asm_offset to generage gen_defines.asm
         if (NOT CMAKE_CROSSCOMPILING)
-            add_custom_command(TARGET asm_offset POST_BUILD
-                COMMAND asm_offset MASM
-                BYPRODUCTS gen_defines.asm)
+            add_custom_command(OUTPUT gen_defines.asm DEPENDS asm_offset COMMAND asm_offset MASM)
         else()
             # Forces compiler to write the intermediate asm file, needed so that we can get sizeof/offset of info out of it.
             target_compile_options(asm_offset PRIVATE "/Fa$<TARGET_FILE_DIR:asm_offset>/asm_offset.asm" /FA)
@@ -236,9 +234,7 @@ else() # i.e.: Linux
         target_link_libraries(asm_offset loader_specific_options)
         # If not cross compiling, run asm_offset to generage gen_defines.asm
         if (NOT CMAKE_CROSSCOMPILING)
-            add_custom_command(TARGET asm_offset POST_BUILD
-                COMMAND asm_offset GAS
-                BYPRODUCTS gen_defines.asm)
+            add_custom_command(OUTPUT gen_defines.asm DEPENDS asm_offset COMMAND asm_offset GAS)
         else()
             # Forces compiler to write the intermediate asm file, needed so that we can get sizeof/offset of info out of it.
             target_compile_options(asm_offset PRIVATE -save-temps=obj)
