From 62670da269486511bfc5afc44488baddce9f64f2 Mon Sep 17 00:00:00 2001
From: ItsBasi <5033630+ItsBasi@users.noreply.github.com>
Date: Sat, 1 Aug 2020 18:45:23 +0200
Subject: [PATCH 5/9] Only use find_package OpenSSL for avoiding pkgconfig

---
 CMakeLists.txt | 19 +++++--------------
 1 file changed, 5 insertions(+), 14 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 59cfdf7..4f29604 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -94,20 +94,11 @@ endif()
 
 # Check to see if openssl installation is present
 if(MZ_OPENSSL)
-    find_package(PkgConfig)
-    if(PKGCONFIG_FOUND)
-        pkg_check_modules(OPENSSL openssl)
-    endif()
-    if(NOT OPENSSL_FOUND)
-        find_package(OpenSSL)
-    endif()
-    if(OPENSSL_FOUND)
-        message(STATUS "Using OpenSSL ${OPENSSL_VERSION}")
-
-        list(APPEND MINIZIP_INC ${OPENSSL_INCLUDE_DIR})
-        if(OPENSSL_INCLUDE_DIRS)
-            list(APPEND MINIZIP_INC ${OPENSSL_INCLUDE_DIRS})
-        endif()
+    find_package(OpenSSL REQUIRED)
+    message(STATUS "Using OpenSSL ${OPENSSL_VERSION}")
+    list(APPEND MINIZIP_INC ${OPENSSL_INCLUDE_DIR})
+    if(OPENSSL_INCLUDE_DIRS)
+        list(APPEND MINIZIP_INC ${OPENSSL_INCLUDE_DIRS})
     endif()
 endif()
 
-- 
2.28.0.windows.1

