From 49ad6d4f835ec229d4214d5eabf98ef4ec976d2b Mon Sep 17 00:00:00 2001
From: ItsBasi <5033630+ItsBasi@users.noreply.github.com>
Date: Tue, 4 Aug 2020 21:59:52 +0200
Subject: [PATCH] Require finding libbsd and iconv if it is enabled

---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 772da9e..a03fc54 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -250,7 +250,7 @@ if(UNIX)
         if(MZ_LIBBSD AND NOT HAVE_ARC4RANDOM_BUF)
             find_package(PkgConfig REQUIRED)
 
-            pkg_check_modules(LIBBSD libbsd)
+            pkg_check_modules(LIBBSD REQUIRED libbsd)
             if(LIBBSD_FOUND)
                 check_library_exists("${LIBBSD_LIBRARIES}" "arc4random_buf"
                     "${LIBBSD_LIBRARY_DIRS}" HAVE_LIBBSD_ARC4RANDOM_BUF)
@@ -270,7 +270,7 @@ if(UNIX)
 
     # Iconv is only necessary when it is not already built-in
     # FindIconv requires cmake 3.11 or higher
-    find_package(Iconv QUIET)
+    find_package(Iconv REQUIRED)
     if(Iconv_FOUND)
         message(STATUS "Using Iconv")
         list(APPEND MINIZIP_DEF -DHAVE_ICONV)
-- 
2.28.0.windows.1

