From 85d479d15f4c4dc8fc4889b9f7d2af6352892734 Mon Sep 17 00:00:00 2001
From: ItsBasi <5033630+ItsBasi@users.noreply.github.com>
Date: Tue, 4 Aug 2020 22:10:18 +0200
Subject: [PATCH] Iconv can be assumed to be found since find_package is
 required

---
 CMakeLists.txt | 12 ++++--------
 1 file changed, 4 insertions(+), 8 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index a03fc54..8941946 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -271,14 +271,10 @@ if(UNIX)
     # Iconv is only necessary when it is not already built-in
     # FindIconv requires cmake 3.11 or higher
     find_package(Iconv REQUIRED)
-    if(Iconv_FOUND)
-        message(STATUS "Using Iconv")
-        list(APPEND MINIZIP_DEF -DHAVE_ICONV)
-        list(APPEND MINIZIP_INC ${Iconv_INCLUDE_DIRS})
-        set(PC_PRIVATE_LIBS "${PC_PRIVATE_LIBS} -liconv")
-    else()
-        message(WARNING "Limited encoding support due to missing iconv")
-    endif()
+	message(STATUS "Using Iconv ${Iconv_VERSION}")
+	list(APPEND MINIZIP_DEF -DHAVE_ICONV)
+	list(APPEND MINIZIP_INC ${Iconv_INCLUDE_DIRS})
+	set(PC_PRIVATE_LIBS "${PC_PRIVATE_LIBS} -liconv")
 endif()
 
 # Include compatibility layer
-- 
2.28.0.windows.1

