From e672b8638cb0350c85071ade737fcb6e5f0bbaa7 Mon Sep 17 00:00:00 2001
From: ItsBasi <5033630+ItsBasi@users.noreply.github.com>
Date: Mon, 17 Aug 2020 18:06:17 +0200
Subject: [PATCH] Guard the usage of __has_include behind the defined check

---
 mz.h | 16 ++++++++++++----
 1 file changed, 12 insertions(+), 4 deletions(-)

diff --git a/mz.h b/mz.h
index 4d3732b..002555f 100644
--- a/mz.h
+++ b/mz.h
@@ -158,10 +158,14 @@
 #include <string.h> /* memset, strncpy, strlen */
 #include <limits.h>
 
-#if defined(HAVE_STDINT_H) || \
-   (defined(__has_include) && __has_include(<stdint.h>))
+#if defined(HAVE_STDINT_H)
 #  include <stdint.h>
 #endif
+#if defined(__has_include)
+#if __has_include(<stdint.h>)
+#  include <stdint.h>
+#endif
+#endif
 
 #ifndef __INT8_TYPE__
 typedef signed char        int8_t;
@@ -188,10 +192,14 @@ typedef unsigned int       uint32_t;
 typedef unsigned long long uint64_t;
 #endif
 
-#if defined(HAVE_INTTYPES_H) || \
-   (defined(__has_include) && __has_include(<inttypes.h>))
+#if defined(HAVE_INTTYPES_H)
 #  include <inttypes.h>
 #endif
+#if defined(__has_include)
+#if __has_include(<inttypes.h>)
+#  include <inttypes.h>
+#endif
+#endif
 
 #ifndef PRId8
 #  define PRId8  "hhd"
-- 
2.28.0.windows.1

