From 3ae3faacb4fcc83c9b505ac0765fdc56eb11beb0 Mon Sep 17 00:00:00 2001
From: ItsBasi <5033630+ItsBasi@users.noreply.github.com>
Date: Tue, 25 Aug 2020 00:06:49 +0200
Subject: [PATCH] Watch option variables for silent and unexpected changes

---
 CMakeLists.txt | 44 ++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 44 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9f5ef82..5b6e23f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -32,6 +32,50 @@ set(MZ_PROJECT_SUFFIX "" CACHE STRING "Project name suffix for package managers"
 option(ZLIB_FORCE_FETCH "Skips find package for ZLIB" OFF)
 option(ZSTD_FORCE_FETCH "Skips find package for ZSTD" OFF)
 
+function(assert_true variable access value current_list_file stack)
+    if(NOT ${${variable}})
+        message(FATAL_ERROR "cmake silently changed variable ${variable}, should have been true.")
+    endif()
+endfunction()
+
+function(assert_false variable access value current_list_file stack)
+    if(${${variable}})
+        message(FATAL_ERROR "cmake silently changed variable ${variable}, should have been false.")
+    endif()
+endfunction()
+
+function(assert_unchanging)
+    set(options)
+    set(oneValueArgs)
+    set(multiValueArgs VARIABLES)
+    cmake_parse_arguments(ARG "${options}" "${oneValueArgs}"
+                          "${multiValueArgs}" ${ARGN} )
+
+    foreach(variable ${ARG_VARIABLES})
+        if(${${variable}})
+            variable_watch(${variable} "assert_true")
+        else()
+            variable_watch(${variable} "assert_false")
+        endif()
+    endforeach()
+endfunction()
+
+assert_unchanging(VARIABLES
+    MZ_COMPAT
+    MZ_ZLIB
+    MZ_BZIP2
+    MZ_LZMA
+    MZ_ZSTD
+    MZ_PKCRYPT
+    MZ_WZAES
+    MZ_LIBCOMP
+    MZ_OPENSSL
+    MZ_LIBBSD
+    MZ_BRG
+    MZ_SIGNING
+    MZ_DECOMPRESS
+    MZ_COMPRESS)
+
 mark_as_advanced(MZ_FILE32_API MZ_PROJECT_SUFFIX ZLIB_FORCE_FETCH ZSTD_FORCE_FETCH)
 
 if(POLICY CMP0074)
