From 54fc50a8f146a81a5ea36be8455296055943bc30 Mon Sep 17 00:00:00 2001
From: ItsBasi <5033630+ItsBasi@users.noreply.github.com>
Date: Tue, 11 Aug 2020 21:49:28 +0200
Subject: [PATCH] Fix find_package for zstd

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d6a06b2..bc34908 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -77,7 +77,7 @@ endif()
 
 # Check if zstd installation is present
 if(MZ_ZSTD)
-    find_package(ZSTD REQUIRED)
+    find_package(zstd REQUIRED)
     message(STATUS "Using ZSTD ${zstd_VERSION}")
     list(APPEND MINIZIP_INC ${zstd_INCLUDE_DIRS})
     set(PC_PRIVATE_LIBS "${PC_PRIVATE_LIBS} -lzstd")
-- 
2.28.0.windows.1

