From 31f86202ac1f9a9fad5c9b275289a5e608d281ed Mon Sep 17 00:00:00 2001
From: ItsBasi <5033630+ItsBasi@users.noreply.github.com>
Date: Fri, 14 Aug 2020 21:27:37 +0200
Subject: [PATCH] Use libbsd-overlay component

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0212a88..61da18e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -248,7 +248,7 @@ if(UNIX)
         endif()
 
         if(MZ_LIBBSD AND NOT HAVE_ARC4RANDOM_BUF)
-            find_package(libbsd REQUIRED)
+            find_package(libbsd REQUIRED COMPONENTS libbsd-overlay)
 
             check_library_exists("${libbsd_LIBRARIES}" "arc4random_buf"
                 "${libbsd_LIB_DIRS}" HAVE_LIBBSD_ARC4RANDOM_BUF)
-- 
2.28.0.windows.1

