--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -731,12 +731,7 @@ if(JAS_ENABLE_LIBJPEG)
 		# (e.g., stdio.h and stdint.h).  So, we cannot reliably use
 		# check_include_file here.
 		jas_get_includes_from_targets(CMAKE_REQUIRED_INCLUDES JPEG::JPEG)
-		check_c_source_compiles("
-			#include <stdio.h>
-			#include <stdint.h>
-			#include <jpeglib.h>
-			int main() {}
-		" JAS_HAVE_JPEGLIB_H)
+		set(JAS_HAVE_JPEGLIB_H 1)
 		if(JAS_HAVE_JPEGLIB_H)
 			set(JAS_HAVE_LIBJPEG 1)
 			set(JAS_LIBJPEG_TARGET JPEG::JPEG)
