diff --git a/CMakeLists.txt b/CMakeLists.txt
index f025a3a..c316d4a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -56,16 +56,18 @@ else( )
 	target_compile_definitions( tz PRIVATE -DUSE_AUTOLOAD=1 )
 	target_compile_definitions( tz PRIVATE -DHAS_REMOTE_API=1 )
 	target_compile_definitions( tz PUBLIC -DUSE_OS_TZDB=0 )
-	find_package( CURL REQUIRED )
-	include_directories( SYSTEM ${CURL_INCLUDE_DIRS} )
-	set( OPTIONAL_LIBRARIES ${CURL_LIBRARIES} )
+endif( )
+
+if( BUILD_SHARED_LIBS )
+	target_compile_definitions( tz PRIVATE -DDATE_BUILD_DLL=1 )
+	target_compile_definitions( tz PUBLIC -DDATE_USE_DLL=1 )
 endif( )
 
 if( USE_TZ_DB_IN_DOT )
 	target_compile_definitions( tz PRIVATE -DINSTALL=. )
 endif( )
 
-target_link_libraries( tz ${CMAKE_THREAD_LIBS_INIT} ${OPTIONAL_LIBRARIES} )
+target_link_libraries( tz ${CMAKE_THREAD_LIBS_INIT} ${CONAN_LIBS} )
 
 # add include folders to the library and targets that consume it
 target_include_directories(tz PUBLIC
