diff --git a/include/date/date.h b/include/date/date.h
index cb115a9..fca9661 100644
--- a/include/date/date.h
+++ b/include/date/date.h
@@ -62,7 +62,11 @@
 #include <stdexcept>
 #include <string>
 #if HAS_STRING_VIEW
-# include <string_view>
+# if __has_include (<string_view>)
+#  include <string_view>
+# else
+#  include <experimental/string_view>
+# endif
 #endif
 #include <utility>
 #include <type_traits>
