diff -Naur a/crnlib/crn_core.h b/crnlib/crn_core.h
--- a/crnlib/crn_core.h	2019-06-15 03:59:38.000000000 -0400
+++ b/crnlib/crn_core.h	2020-09-04 23:14:27.538724100 -0400
@@ -82,7 +82,7 @@

    #define CRNLIB_RESTRICT

-   #define CRNLIB_FORCE_INLINE inline __attribute__((__always_inline__,__gnu_inline__))
+   #define CRNLIB_FORCE_INLINE inline __attribute__((always_inline))

    #define CRNLIB_INT64_FORMAT_SPECIFIER "%lli"
    #define CRNLIB_UINT64_FORMAT_SPECIFIER "%llu"
