--- a/foxi/onnxifi_dummy.c
+++ b/foxi/onnxifi_dummy.c
@@ -103,7 +103,9 @@ ONNXIFI_PUBLIC ONNXIFI_CHECK_RESULT onnxStatus ONNXIFI_ABI onnxInitGraph(
     const void* onnxModel,
     uint32_t weightCount,
     const onnxTensorDescriptorV1* weightDescriptors,
-    onnxGraph* graph) {
+    onnxGraph* graph,
+    uint32_t maxSeqLength,
+    void* deferredWeightReader) {
   if (graph == NULL) {
     return ONNXIFI_STATUS_INVALID_POINTER;
   }
@@ -215,6 +217,8 @@ ONNXIFI_PUBLIC ONNXIFI_CHECK_RESULT onnxStatus ONNXIFI_ABI
 onnxWaitEventFor(onnxEvent event,
                  uint32_t timeoutMs,
                  onnxEventState* eventState,
-                 onnxStatus* eventStatus) {
+                 onnxStatus* eventStatus,
+                 char* message,
+                 size_t* messageLength) {
   return ONNXIFI_STATUS_SUCCESS;
 }
\ No newline at end of file
--- a/foxi/onnxifi_wrapper.c
+++ b/foxi/onnxifi_wrapper.c
@@ -761,7 +761,9 @@ ONNXIFI_PUBLIC onnxStatus ONNXIFI_ABI onnxInitGraph(
     const void* onnxModel,
     uint32_t weightsCount,
     const onnxTensorDescriptorV1* weightDescriptors,
-    onnxGraph* graph)
+    onnxGraph* graph,
+    uint32_t maxSeqLength,
+    void* deferredWeightReader)
 {
   if (graph == NULL) {
     return ONNXIFI_STATUS_INVALID_POINTER;
@@ -797,7 +799,9 @@ ONNXIFI_PUBLIC onnxStatus ONNXIFI_ABI onnxInitGraph(
     onnxModel,
     weightsCount,
     weightDescriptors,
-    &graph_wrapper->graph);
+    &graph_wrapper->graph,
+    maxSeqLength,
+    deferredWeightReader);
   switch (status) {
     case ONNXIFI_STATUS_SUCCESS:
     case ONNXIFI_STATUS_FALLBACK:
