diff --git a/common/CMakeLists.txt b/common/CMakeLists.txt
index 761971d6..3afd9759 100644
--- a/common/CMakeLists.txt
+++ b/common/CMakeLists.txt
@@ -66,7 +66,7 @@ add_library(${TARGET} STATIC
     train.cpp
     ngram-cache.h
     ngram-cache.cpp
-    )
+)
 
 if (BUILD_SHARED_LIBS)
     set_target_properties(${TARGET} PROPERTIES POSITION_INDEPENDENT_CODE ON)
@@ -78,10 +78,8 @@ set(LLAMA_COMMON_EXTRA_LIBS build_info)
 if (LLAMA_CURL)
     find_package(CURL REQUIRED)
     add_definitions(-DLLAMA_USE_CURL)
-    include_directories(${CURL_INCLUDE_DIRS})
-    find_library(CURL_LIBRARY curl REQUIRED)
-    set(LLAMA_COMMON_EXTRA_LIBS ${LLAMA_COMMON_EXTRA_LIBS} ${CURL_LIBRARY})
-endif ()
+    list(APPEND LLAMA_COMMON_EXTRA_LIBS CURL::libcurl)
+endif()
 
 target_include_directories(${TARGET} PUBLIC .)
 target_compile_features   (${TARGET} PUBLIC cxx_std_11)
