diff --git a/common/CMakeLists.txt b/common/CMakeLists.txt
index 24b7f87..e332bec 100644
--- a/common/CMakeLists.txt
+++ b/common/CMakeLists.txt
@@ -84,9 +84,7 @@ set(LLAMA_COMMON_EXTRA_LIBS build_info)
 if (LLAMA_CURL)
     find_package(CURL REQUIRED)
     target_compile_definitions(${TARGET} PUBLIC LLAMA_USE_CURL)
-    include_directories(${CURL_INCLUDE_DIRS})
-    find_library(CURL_LIBRARY curl REQUIRED)
-    set(LLAMA_COMMON_EXTRA_LIBS ${LLAMA_COMMON_EXTRA_LIBS} ${CURL_LIBRARY})
+    list(APPEND LLAMA_COMMON_EXTRA_LIBS CURL::libcurl)
 endif ()
 
 target_include_directories(${TARGET} PUBLIC .)
