diff --git a/cpr/CMakeLists.txt b/cpr/CMakeLists.txt
diff --git a/a/cpr/CMakeLists.txt b/b/cpr/CMakeLists.txt
index c4f9b5b..7e2279d 100644
--- a/a/cpr/CMakeLists.txt
+++ b/b/cpr/CMakeLists.txt
@@ -44,8 +44,7 @@ add_library(${CPR_LIBRARIES}
     "${CPR_INCLUDE_DIRS}/cpr/unix_socket.h")
 
 message(STATUS "Using CURL_LIBRARIES: ${CURL_LIBRARIES}.")
-target_link_libraries(${CPR_LIBRARIES}
-    ${CURL_LIBRARIES})
+target_link_libraries(cpr PUBLIC CURL::CURL)

 include(GNUInstallDirs)
 install(TARGETS cpr
