diff --git a/a/CMakeLists.txt b/b/CMakeLists.txt
index aca0ab0..c4c81d3 100644
--- a/a/CMakeLists.txt
+++ b/b/CMakeLists.txt
@@ -75,13 +75,6 @@ if(NOT USE_SYSTEM_CURL OR NOT CURL_FOUND)
     set_property(TARGET libcurl PROPERTY FOLDER "external")
 endif()
 
-
-# SSL configuration
-if(CMAKE_USE_OPENSSL)
-    find_package(OpenSSL REQUIRED)
-    message(STATUS "using system OpenSSL library.")
-endif()
-
 # GTest configuration
 if(BUILD_CPR_TESTS)
     if(USE_SYSTEM_GTEST)
diff --git a/a/cpr/CMakeLists.txt b/b/cpr/CMakeLists.txt
index 981f1c5..14c5824 100644
--- a/a/cpr/CMakeLists.txt
+++ b/b/cpr/CMakeLists.txt
@@ -21,10 +21,9 @@ add_library(cpr
     )
 
 add_library(cpr::cpr ALIAS cpr)
-
 message(STATUS "Using CURL_LIBRARIES: ${CURL_LIBRARIES}.")
-target_link_libraries(cpr PUBLIC CURL::libcurl) # todo should be private, but first dependencys in ssl_options need to be removed
-
+target_link_libraries(cpr PUBLIC ${CURL_LIBRARIES}) # todo should be private, but first dependencys in ssl_options need to be removed
+target_include_directories(cpr PUBLIC ${CURL_INCLUDE_DIRS})
 
 # Set version for shared libraries.
 set_target_properties(cpr
