diff --git a/cpr/CMakeLists.txt b/cpr/CMakeLists.txt
index 48f4ee1..8793573 100644
--- a/cpr/CMakeLists.txt
+++ b/cpr/CMakeLists.txt
@@ -43,8 +43,8 @@ add_library(${CPR_LIBRARIES}
     "${CPR_INCLUDE_DIRS}/cpr/ssl_options.h")
 
 message(STATUS "Using CURL_LIBRARIES: ${CURL_LIBRARIES}.")
-target_link_libraries(${CPR_LIBRARIES}
-    ${CURL_LIBRARIES})
+target_link_libraries(cpr
+    PUBLIC CURL::CURL)
 
 include(GNUInstallDirs)
 install(TARGETS cpr
