diff --git a/CMakeLists.txt b/CMakeLists.txt
index c065227d..c1c19155 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -55,15 +55,7 @@ include(cmake/code_coverage.cmake)
 
 # Curl configuration
 if(USE_SYSTEM_CURL)
-    find_package(CURL COMPONENTS PROTOCOLS HTTP HTTPS FEATURES SSL)
-    if (CURL_FOUND)
-        set(SSL_ENABLED ON CACHE INTERNAL "" FORCE)
-    else()
-        find_package(CURL COMPONENTS PROTOCOLS HTTP)
-        if(CURL_FOUND)
-            set(SSL_ENABLED OFF CACHE INTERNAL "" FORCE)
-        endif()
-    endif()
+    find_package(CURL)
 endif()
 if(NOT USE_SYSTEM_CURL OR NOT CURL_FOUND)
     message(STATUS "Not using system Curl, using built-in curl project instead.")
