--- a/cpr/CMakeLists.txt
+++ b/cpr/CMakeLists.txt
@@ -36,6 +36,7 @@ include(GNUInstallDirs)
 if(CPR_FORCE_USE_SYSTEM_CURL)
     install(TARGETS cpr
             EXPORT cprTargets
+            RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
             ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
             LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
 
@@ -62,6 +63,7 @@ if(CPR_FORCE_USE_SYSTEM_CURL)
      
 else()
     install(TARGETS cpr
+            RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
             ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
             LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
 endif()
