diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3f6e1d0..ba830cb 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -77,7 +77,7 @@ cpr_option(CPR_DEBUG_SANITIZER_FLAG_UB "Enables the UndefinedBehaviorSanitizer f
 cpr_option(CPR_DEBUG_SANITIZER_FLAG_ALL "Enables all sanitizers for debug builds except the ThreadSanitizer since it is incompatible with the other sanitizers." OFF)
 message(STATUS "=======================================================")
 
-if (MSVC)
+if (0)
     if (BUILD_SHARED_LIBS)
         message(STATUS "Build windows dynamic libs.")
     else()
