diff --git a/cpp/Platform.Hashing/Platform.Hashing.Crc32.h b/cpp/Platform.Hashing/Platform.Hashing.Crc32.h
index 2c90b40..c7708c7 100644
--- a/cpp/Platform.Hashing/Platform.Hashing.Crc32.h
+++ b/cpp/Platform.Hashing/Platform.Hashing.Crc32.h
@@ -13,7 +13,7 @@
 
 #if defined(__GNUC__) && !defined(__INTEL_COMPILER)
   #define target_feature(feature) __attribute__((target(feature)))
-#elif defined(__MSVC__)
+#elif defined(_MSC_VER)
   #define target_feature(feature) // empty
 #else
   #error "compiler is not supported"
