from https://github.com/fnc12/sqlite_orm/pull/821

--- a/include/sqlite_orm/sqlite_orm.h
+++ b/include/sqlite_orm/sqlite_orm.h
@@ -10037,6 +10037,9 @@ namespace sqlite_orm {
 
         template<class T>
         using is_insert_constraint = std::is_same<insert_constraint, T>;
+
+        template<class T>
+        struct is_upsert_clause;
     }
 
     inline internal::insert_constraint or_rollback() {
