diff --git a/include/packio/json_rpc/hash.h b/include/packio/json_rpc/hash.h
index 9f6e3dd..664696d 100644
--- a/include/packio/json_rpc/hash.h
+++ b/include/packio/json_rpc/hash.h
@@ -6,11 +6,15 @@
 #define PACKIO_JSON_RPC_HASH_H
 
 #include <boost/json.hpp>
+#include <boost/version.hpp>
+
+#if BOOST_VERSION < 107700
 
 namespace packio {
 namespace json_rpc {
 namespace internal {
 
+
 inline std::size_t combine(std::size_t seed, std::size_t h) noexcept
 {
     seed ^= h + 0x9e3779b9 + (seed << 6U) + (seed >> 2U);
@@ -72,4 +76,6 @@ struct hash<boost::json::value> {
 
 } // std
 
+#endif // BOOST_VERSION < 107700
+
 #endif // PACKIO_JSON_RPC_HASH_H
