diff --git a/CMake/FollyCompilerUnix.cmake b/CMake/FollyCompilerUnix.cmake
index 9c76fdc..73829df 100644
--- a/CMake/FollyCompilerUnix.cmake
+++ b/CMake/FollyCompilerUnix.cmake
@@ -30,7 +30,6 @@ function(apply_folly_compile_options_to_target THETARGET)
   )
   target_compile_options(${THETARGET}
     PRIVATE
-      -g
       -std=${CXX_STD}
       -finput-charset=UTF-8
       -fsigned-char
diff --git a/CMake/folly-deps.cmake b/CMake/folly-deps.cmake
index 0fb8a90..d45c515 100644
--- a/CMake/folly-deps.cmake
+++ b/CMake/folly-deps.cmake
@@ -23,12 +23,12 @@ find_package(DoubleConversion MODULE REQUIRED)
 list(APPEND FOLLY_LINK_LIBRARIES ${DOUBLE_CONVERSION_LIBRARY})
 list(APPEND FOLLY_INCLUDE_DIRECTORIES ${DOUBLE_CONVERSION_INCLUDE_DIR})
 
-find_package(Gflags MODULE)
-set(FOLLY_HAVE_LIBGFLAGS ${LIBGFLAGS_FOUND})
-list(APPEND FOLLY_LINK_LIBRARIES ${LIBGFLAGS_LIBRARY})
-list(APPEND FOLLY_INCLUDE_DIRECTORIES ${LIBGFLAGS_INCLUDE_DIR})
-list(APPEND CMAKE_REQUIRED_LIBRARIES ${LIBGFLAGS_LIBRARY})
-list(APPEND CMAKE_REQUIRED_INCLUDES ${LIBGFLAGS_INCLUDE_DIR})
+find_package(gflags MODULE)
+set(FOLLY_HAVE_LIBGFLAGS ${gflags_FOUND})
+list(APPEND FOLLY_LINK_LIBRARIES ${gflags_LIBRARY})
+list(APPEND FOLLY_INCLUDE_DIRECTORIES ${gflags_INCLUDE_DIR})
+list(APPEND CMAKE_REQUIRED_LIBRARIES ${gflags_LIBRARY})
+list(APPEND CMAKE_REQUIRED_INCLUDES ${gflags_INCLUDE_DIR})
 
 find_package(Glog MODULE)
 set(FOLLY_HAVE_LIBGLOG ${GLOG_FOUND})
