diff --git a/folly/portability/Builtins.h b/folly/portability/Builtins.h
index 971cb88..102fda1 100644
--- a/folly/portability/Builtins.h
+++ b/folly/portability/Builtins.h
@@ -41,7 +41,7 @@ FOLLY_ALWAYS_INLINE void __builtin___clear_cache(char* begin, char* end) {
   }
 }
 
-#if !defined(_MSC_VER) || (_MSC_VER < 1923)
+#if !defined(_MSC_VER) || (_MSC_VER < 1930)
 FOLLY_ALWAYS_INLINE int __builtin_clz(unsigned int x) {
   unsigned long index;
   return int(_BitScanReverse(&index, (unsigned long)x) ? 31 - index : 32);
