diff --git a/folly/portability/Time.h b/folly/portability/Time.h
index 994a09e5d70..e4f0d101ca9 100644
--- a/folly/portability/Time.h
+++ b/folly/portability/Time.h
@@ -49,7 +49,6 @@
 #define CLOCK_PROCESS_CPUTIME_ID 2
 #define CLOCK_THREAD_CPUTIME_ID 3
 
-typedef uint8_t clockid_t;
 extern "C" int clock_gettime(clockid_t clk_id, struct timespec* ts);
 extern "C" int clock_getres(clockid_t clk_id, struct timespec* ts);
 #endif
