diff --git a/CMakeLists.txt b/CMakeLists.txt
index d77de0c..7e3ca48 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -100,17 +100,6 @@ set(FRUIT_VERSION "3.4.0")
 add_subdirectory(configuration)
 add_subdirectory(src)
 
-if("${CMAKE_BUILD_TYPE}" STREQUAL "Debug")
-  # Do not exclude these from "make all" in debug mode, they must build.
-  add_subdirectory(examples)
-  add_subdirectory(tests)
-else()
-  add_subdirectory(examples EXCLUDE_FROM_ALL)
-  add_subdirectory(tests)
-endif()
-
-add_subdirectory(extras EXCLUDE_FROM_ALL)
-
 install(DIRECTORY include/fruit/
   DESTINATION "${INSTALL_INCLUDE_DIR}"
   FILES_MATCHING PATTERN "*.h")
