diff --git a/build/cmake/BuildVersion.cmake b/build/cmake/BuildVersion.cmake
index 4b04eb8..dfaf394 100644
--- a/build/cmake/BuildVersion.cmake
+++ b/build/cmake/BuildVersion.cmake
@@ -47,7 +47,7 @@ function(compute_build_version outvar)
 endfunction()
 
 # Define the BUILD_VERSION:
-compute_build_version(BUILD_VERSION)
+#compute_build_version(BUILD_VERSION)
 
 # Set a BUILD_VERSION_SIMPLE, which is just a three-number-triple that CMake understands
 string (REGEX REPLACE "([0-9]+\\.[0-9]+\\.[0-9]+).*$" "\\1" BUILD_VERSION_SIMPLE "${BUILD_VERSION}")
 