diff --git a/src/libbson/src/bson/bson-iter.c b/src/libbson/src/bson/bson-iter.c
index c9e839e..1611c82 100644
--- a/src/libbson/src/bson/bson-iter.c
+++ b/src/libbson/src/bson/bson-iter.c
@@ -1894,8 +1894,8 @@ bson_iter_visit_all (bson_iter_t *iter,             /* INOUT */
                      const bson_visitor_t *visitor, /* IN */
                      void *data)                    /* IN */
 {
-   uint32_t bson_type;
-   const char *key;
+   uint32_t bson_type = BSON_TYPE_EOD;
+   const char *key = NULL;
    bool unsupported;
 
    BSON_ASSERT (iter);
@@ -2141,7 +2141,7 @@ bson_iter_visit_all (bson_iter_t *iter,             /* INOUT */
 
    if (iter->err_off) {
       if (unsupported && visitor->visit_unsupported_type &&
-          bson_utf8_validate (key, strlen (key), false)) {
+          key != NULL && bson_utf8_validate (key, strlen (key), false)) {
          visitor->visit_unsupported_type (iter, key, bson_type, data);
          return false;
       }
