diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8096487..96b72f3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -243,7 +243,8 @@ endif()
 # ---- Generate Build Info Headers ----
 
 # used in export header generated below
-if(NOT CPPTRACE_BUILD_SHARED)
+if(build_type STREQUAL "STATIC")
+  message(WARNING "Test")
   target_compile_definitions(${target_name} PUBLIC CPPTRACE_STATIC_DEFINE)
 endif()
 