diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index a4fc2d3..51736bc 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -41,7 +41,7 @@ endif()

 if (CIO_LIB_SHARED)
   add_library(chunkio-shared SHARED ${src})
-  target_link_libraries(chunkio-static ${libs})
+  target_link_libraries(chunkio-shared ${libs})
   if(CIO_SANITIZE_ADDRESS)
     add_sanitizers(chunkio-shared)
   endif()

