diff --git a/CMakeLists.txt b/CMakeLists.txt
index e6d5dcb..ebf3c4c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -109,10 +109,15 @@ include_directories(

 add_subdirectory(deps/crc32)
 add_subdirectory(src)
-add_subdirectory(tools)
+#add_subdirectory(tools)

 # Tests
 if(CIO_TESTS)
   enable_testing()
   add_subdirectory(tests)
 endif()
+
+include(GNUInstallDirs)
+
+install(DIRECTORY include/chunkio DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+install(DIRECTORY deps/monkey/include/monkey DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
diff --git a/deps/crc32/CMakeLists.txt b/deps/crc32/CMakeLists.txt
index 8ae1823..0b7dba0 100644
--- a/deps/crc32/CMakeLists.txt
+++ b/deps/crc32/CMakeLists.txt
@@ -3,3 +3,12 @@ set(src
   )

 add_library(cio-crc32 STATIC ${src})
+
+if(CIO_LIB_STATIC)
+  include(GNUInstallDirs)
+  install(
+    TARGETS cio-crc32
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  )
+endif()
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index a4fc2d3..51736bc 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -50,3 +50,22 @@ endif()
 if (NOT CIO_LIB_STATIC AND NOT CIO_LIB_SHARED)
   message(FATAL_ERROR "What are you doing?, you should build something")
 endif()
+
+include(GNUInstallDirs)
+
+if(CIO_LIB_STATIC)
+    install(
+        TARGETS chunkio-static
+        LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+        ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+        )
+endif()
+
+if(CIO_LIB_SHARED)
+    install(
+        TARGETS chunkio-shared
+        RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+        LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+        ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    )
+endif()
