--- a/httplib.h
+++ b/httplib.h
@@ -2838,7 +2838,7 @@ inline bool redirect(T &cli, const Request &req, Response &res,
 inline bool contains_header(const std::string &header, const std::string &name) {
   if (header.length() >= name.length()) {
     for (int i = 0; i < name.length(); ++i) {
-      if (std::tolower(header[i]) != std::tolower(name[i])) {
+      if (::tolower(header[i]) != ::tolower(name[i])) {
         return false;
       }
     }
