diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5e828ab..8ff05d6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -66,9 +66,11 @@ set(JSON_C_PUBLIC_HEADERS
     ./json_c_version.h
     ./json_inttypes.h
     ./json_object.h
+    ./json_object_iterator.h
     ./json_pointer.h
     ./json_tokener.h
     ./json_util.h
+    ./json_visit.h
     ./linkhash.h
     ./printbuf.h
 )
@@ -101,22 +103,12 @@ set(JSON_C_SOURCES
 )
 
 add_library(json-c
-    SHARED
     ${JSON_C_SOURCES}
     ${JSON_C_HEADERS}
 )
-
-add_library(json-c-static
-    STATIC
-    ${JSON_C_SOURCES}
-    ${JSON_C_HEADERS}
-)
-
 set_property(TARGET json-c PROPERTY C_STANDARD 99)
-set_property(TARGET json-c-static PROPERTY C_STANDARD 99)
-set_target_properties(json-c-static PROPERTIES OUTPUT_NAME json-c)
 
-install(TARGETS json-c json-c-static
+install(TARGETS json-c
     RUNTIME DESTINATION bin
     LIBRARY DESTINATION lib
     ARCHIVE DESTINATION lib
@@ -132,5 +124,5 @@ if (UNIX)
 	set(VERSION ${PROJECT_VERSION})
 	configure_file(json-c.pc.in json-c.pc @ONLY)
 	set(INSTALL_PKGCONFIG_DIR "${CMAKE_INSTALL_PREFIX}/lib/pkgconfig" CACHE PATH "Installation directory for pkgconfig (.pc) files")
-	install(FILES ${CMAKE_BINARY_DIR}/json-c.pc DESTINATION "${INSTALL_PKGCONFIG_DIR}")
+	install(FILES ${CMAKE_CURRENT_BINARY_DIR}/json-c.pc DESTINATION "${INSTALL_PKGCONFIG_DIR}")
 endif ()
