--- src/location.c
+++ src/location.c
@@ -17,23 +17,23 @@
 
    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <http://www.gnu.org/licenses/>.  */
-
 #include <config.h>
 #include "system.h"
-
 #include <mbfile.h>
 #include <mbswidth.h>
 #include <quotearg.h>
 #include <stdio.h>    /* fileno */
+#if !defined(_WINDOWS)
 #include <sys/ioctl.h>
+#endif
 #include <sys/stat.h> /* fstat */
+#if !defined(_WINDOWS)
 #include <termios.h>
-
+#endif
 #ifdef WINSIZE_IN_PTEM
 # include <sys/stream.h>
 # include <sys/ptem.h>
 #endif
-
 #include "complain.h"
 #include "getargs.h"
 #include "location.h"
