diff --git a/CMakeLists.txt b/CMakeLists.txt
index abe12e0..73d443f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,3 +1,5 @@
+cmake_minimum_required (VERSION 3.1)
+
 # This block needs to stay before the project (log4cplus) line so that
 #  the output files placed into Android's libs directory.
 if (CMAKE_TOOLCHAIN_FILE)
@@ -12,10 +14,6 @@ endif ()
 set (CMAKE_LEGACY_CYGWIN_WIN32 0)
 
 project (log4cplus)
-cmake_minimum_required (VERSION 3.1)
-
-# Use "-fPIC" / "-fPIE" for all targets by default, including static libs.
-set (CMAKE_POSITION_INDEPENDENT_CODE ON)
 
 enable_language (CXX)
 if (MSVC)
