diff --git a/avahi-common/domain.c b/avahi-common/domain.c
index 3b1ab68..3a8f8de 100644
--- a/avahi-common/domain.c
+++ b/avahi-common/domain.c
@@ -477,7 +477,7 @@ int avahi_service_name_join(char *p, size_t size, const char *name, const char *
     return AVAHI_OK;
 }
 
-#ifndef HAVE_STRLCPY
+#if 1
 
 static size_t strlcpy(char *dest, const char *src, size_t n) {
     assert(dest);
diff --git a/avahi-compat-howl/text.c b/avahi-compat-howl/text.c
index 7ef4df3..a22a98c 100644
--- a/avahi-compat-howl/text.c
+++ b/avahi-compat-howl/text.c
@@ -37,7 +37,7 @@ struct _sw_text_record {
     int buffer_valid;
 };
 
-#ifndef HAVE_STRLCPY
+#if 1
 
 static size_t strlcpy(char *dest, const char *src, size_t n) {
     assert(dest);
