diff --git a/src/Wt/Dbo/backend/CMakeLists.txt b/src/Wt/Dbo/backend/CMakeLists.txt
index 29a72c3..8efdd2b 100644
--- a/src/Wt/Dbo/backend/CMakeLists.txt
+++ b/src/Wt/Dbo/backend/CMakeLists.txt
@@ -128,11 +128,12 @@ IF(ENABLE_POSTGRES AND POSTGRES_FOUND)
     SET_PROPERTY(TARGET wtdbopostgres PROPERTY VISIBILITY_INLINES_HIDDEN YES)
   endif()
 
+  find_package(PostgreSQL REQUIRED CONFIG)
   TARGET_LINK_LIBRARIES(wtdbopostgres
     PUBLIC
       wtdbo
     PRIVATE
-      ${POSTGRES_LIBRARIES}
+      PostgreSQL::PostgreSQL
   )
 
   IF(TARGET Boost::headers)
@@ -308,11 +309,12 @@ IF(ENABLE_MYSQL AND MYSQL_FOUND)
     SET_PROPERTY(TARGET wtdbomysql PROPERTY VISIBILITY_INLINES_HIDDEN YES)
   endif()
 
+  find_package(libmysqlclient CONFIG REQUIRED)
   TARGET_LINK_LIBRARIES(wtdbomysql
     PUBLIC
       wtdbo
     PRIVATE
-      ${MYSQL_LIBRARIES}
+      libmysqlclient::libmysqlclient
   )
 
   INCLUDE_DIRECTORIES(${MYSQL_INCLUDE})
