diff --git a/cmake/modules/TableGen.cmake b/cmake/modules/TableGen.cmake
index 5e9e2674405..ca7c83132c7 100644
--- a/cmake/modules/TableGen.cmake
+++ b/cmake/modules/TableGen.cmake
@@ -93,7 +93,7 @@ function(tablegen project ofn)
   list(TRANSFORM tblgen_includes PREPEND -I)
 
   add_custom_command(OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/${ofn}
-    COMMAND ${${project}_TABLEGEN_EXE} ${ARGN} -I ${CMAKE_CURRENT_SOURCE_DIR}
+	  COMMAND ${CMAKE_COMMAND} -E env DYLD_LIBRARY_PATH="$ENV{DYLD_LIBRARY_PATH}" $<TARGET_FILE:${${project}_TABLEGEN_EXE}> ${ARGN} -I ${CMAKE_CURRENT_SOURCE_DIR}
     ${tblgen_includes}
     ${LLVM_TABLEGEN_FLAGS}
     ${LLVM_TARGET_DEFINITIONS_ABSOLUTE}
