--- poppler/GlobalParams.h
+++ poppler/GlobalParams.h
@@ -60,14 +60,14 @@
 class GfxFont;
 class Stream;
 class SysFontList;
-
-//------------------------------------------------------------------------
-
-// The global parameters object.
-extern std::unique_ptr<GlobalParams> globalParams;
+#if defined(_WIN32) && defined(poppler_EXPORTS)
+#define POPPLER_API __declspec(dllexport)
+#elif  defined(_WIN32) && !defined(POPPLER_STATIC)
+#define POPPLER_API __declspec(dllimport)
+#else
+#define POPPLER_API
+#endif
+extern POPPLER_API std::unique_ptr<GlobalParams> globalParams;
-
-//------------------------------------------------------------------------
-
 enum SysFontType
 {
     sysFontPFA,
