From be12d3ca12f4127c2ca20e4be144d34d9f9825e7 Mon Sep 17 00:00:00 2001
From: Jordan Williams <jordan@jwillikers.com>
Date: Tue, 7 May 2024 10:30:18 -0500
Subject: [PATCH] Insert freertos_config CMake target for the FreeRTOSConfig.h
 header

---
 CMakeLists.txt | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6fd1ff56b..9f06a088d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -15,6 +15,27 @@ cmake_minimum_required(VERSION 3.15)
 # heap implementation. If the option is not set, the cmake will use no heap
 # implementation (e.g. when only static allocation is used).
 
+project(freertos_kernel LANGUAGES C)
+
+add_library(freertos_config INTERFACE)
+
+set_property(
+  TARGET freertos_config
+  APPEND
+  PROPERTY PUBLIC_HEADER
+           ${_FREERTOS_CONFIG_DIR}/FreeRTOSConfig.h
+)
+
+target_compile_definitions(freertos_config INTERFACE projCOVERAGE_TEST=0)
+
+target_include_directories(freertos_config INTERFACE ${_FREERTOS_CONFIG_DIR})
+
+include(GNUInstallDirs)
+
+install(TARGETS freertos_config
+        PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
+)
+
 # `freertos_config` target defines the path to FreeRTOSConfig.h and optionally other freertos based config files
 if(NOT TARGET freertos_config )
     if (NOT DEFINED FREERTOS_CONFIG_FILE_DIRECTORY )
-- 
2.45.0

