From b8c533b0fb2b92e9bd2aada5e195d7a0b3c0c6a9 Mon Sep 17 00:00:00 2001
From: Joakim Haugen <joakim.haugen@sintef.no>
Date: Wed, 10 May 2023 13:17:11 +0200
Subject: [PATCH] fix find asio and tinyxml2

---
 CMakeLists.txt | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b01b2c470..7867feff3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -232,9 +232,11 @@ if(NOT BUILD_SHARED_LIBS)
     set(FASTDDS_STATIC ON)
 endif()
 
-eprosima_find_package(fastcdr REQUIRED)
-eprosima_find_thirdparty(Asio asio VERSION 1.10.8)
-eprosima_find_thirdparty(TinyXML2 tinyxml2)
+eprosima_find_thirdparty(fastcdr REQUIRED)
+eprosima_find_thirdparty(asio REQUIRED)
+eprosima_find_thirdparty(tinyxml2 REQUIRED)
+set(TINYXML2_LIBRARY tinyxml2::tinyxml2)
+set(Asio_INCLUDE_DIR ${asio_INCLUDE_DIR})
 
 find_package(foonathan_memory REQUIRED)
 message(STATUS "Found foonathan_memory: ${foonathan_memory_DIR}")
-- 
2.30.2

