From 0c0a182229dc4816ec5b5f73ef2e638e3341082a Mon Sep 17 00:00:00 2001
From: ivovk <adikue@gmail.com>
Date: Tue, 11 May 2021 09:10:11 +0300
Subject: [PATCH 4/4] Fix msvc install

---
 CMakeLists.txt | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5388d8d..57d1c5e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -79,7 +79,9 @@ if(NOT URITEMPLATE_BUNDLED_MODE)
     install(TARGETS uri-template
         EXPORT uritemplateTargets
         LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
-        ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}")
+        ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+        RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
+    )
 
     install(DIRECTORY include/uri-template
         DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
-- 
2.31.1

