--- poppler/GlobalParams.cc
+++ poppler/GlobalParams.cc
@@ -377,5 +377,8 @@
 GlobalParams::GlobalParams(const char *customPopplerDataDir) : popplerDataDir(customPopplerDataDir)
 {
+    if (!popplerDataDir) {
+        popplerDataDir = ::getenv("POPPLER_DATADIR");
+    }
     // scan the encoding in reverse because we want the lowest-numbered
     // index for each char name ('space' is encoded twice)
     macRomanReverseMap = new NameToCharCode();
