--- CMakeLists.txt
+++ CMakeLists.txt
@@ -43,8 +43,10 @@ string(REGEX REPLACE "^0?(.+)$" "\\1" POPPLER_MINOR_VERSION "${POPPLER_MINOR_VER
 set(POPPLER_MICRO_VERSION "0")
 set(POPPLER_VERSION "${POPPLER_MAJOR_VERSION}.${POPPLER_MINOR_VERSION_STRING}.${POPPLER_MICRO_VERSION}")
 
-set (CMAKE_CXX_STANDARD 14)
-set (CMAKE_CXX_EXTENSIONS OFF)
+if (NOT CMAKE_CXX_STANDARD)
+  set (CMAKE_CXX_STANDARD 14)
+  set (CMAKE_CXX_EXTENSIONS OFF)
+endif()
 
 # command line switches
 option(ENABLE_UNSTABLE_API_ABI_HEADERS "Install API/ABI unstable xpdf headers." OFF)
@@ -154,23 +156,11 @@ else()
 endif()
 
 if (ENABLE_QT5)
-  find_package(Qt5Core 5.5)  # Update QT_DISABLE_DEPRECATED_BEFORE in qt5/CMakeLists.txt when increasing this
-  find_package(Qt5Gui)
-  find_package(Qt5Xml)
-  find_package(Qt5Widgets)
-  find_package(Qt5Test)
-  if (NOT (Qt5Core_FOUND AND Qt5Gui_FOUND AND Qt5Xml_FOUND AND Qt5Widgets_FOUND AND Qt5Test_FOUND))
-    message("-- Package Qt5Core or Qt5Gui or Qt5Xml or Qt5Widgets or Qt5Test not found")
-    set(ENABLE_QT5 OFF)
-  endif()
+  find_package(qt 5.5 COMPONENTS Core Gui Xml Widgets Test REQUIRED)  # Update QT_DISABLE_DEPRECATED_BEFORE in qt5/CMakeLists.txt when increasing this
 endif()
 
 if (ENABLE_QT6)
-  find_package(Qt6 COMPONENTS Core Gui Widgets Test QUIET)
-  if (NOT (Qt6Core_FOUND AND Qt6Gui_FOUND AND Qt6Widgets_FOUND AND Qt6Test_FOUND))
-    message("-- Package Qt6Core or Qt6Gui or Qt6Widgets or Qt6Test not found")
-    set(ENABLE_QT6 OFF)
-  endif()
+  find_package(Qt6 COMPONENTS Core Gui Widgets Test REQUIRED)
 endif()
 
 macro_optional_find_package(Cairo ${CAIRO_VERSION})
--- qt5/src/CMakeLists.txt
+++ qt5/src/CMakeLists.txt
@@ -1,10 +1,7 @@
-add_definitions(${QT5_DEFINITIONS})
-add_definitions(${Qt5Core_DEFINITIONS})
 add_definitions(-DQT_NO_SIGNALS_SLOTS_KEYWORDS)
 
 include_directories(
   ${CMAKE_CURRENT_SOURCE_DIR}
-  ${Qt5Core_INCLUDE_DIRS}
   ${CMAKE_CURRENT_BINARY_DIR}
 )
 
@@ -45,7 +42,7 @@ if(MINGW AND BUILD_SHARED_LIBS)
     get_target_property(POPPLER_QT5_SOVERSION poppler-qt5 SOVERSION)
     set_target_properties(poppler-qt5 PROPERTIES SUFFIX "-${POPPLER_QT5_SOVERSION}${CMAKE_SHARED_LIBRARY_SUFFIX}")
 endif()
-target_link_libraries(poppler-qt5 poppler ${Qt5Core_LIBRARIES} ${Qt5Gui_LIBRARIES} ${Qt5Xml_LIBRARIES} CONAN_PKG::freetype)
+target_link_libraries(poppler-qt5 poppler CONAN_PKG::qt CONAN_PKG::freetype)
 if(MSVC)
 target_link_libraries(poppler-qt5 poppler ${poppler_LIBS})
 endif()
