diff --git a/cmake/unwind.cmake b/cmake/unwind.cmake
index 460a43c15..aa4356070 100644
--- a/cmake/unwind.cmake
+++ b/cmake/unwind.cmake
@@ -1,4 +1,4 @@
 #
 # Unwind
 #
-find_package(Libunwind QUIET)
+find_package(libunwind QUIET)
diff --git a/cmake/zstd.cmake b/cmake/zstd.cmake
index bfaa5b20e..afba1f5d8 100644
--- a/cmake/zstd.cmake
+++ b/cmake/zstd.cmake
@@ -4,10 +4,10 @@
 option(WITH_ZSTD
     "Build support for compression/decompression with Zstd." TRUE)
 if (WITH_ZSTD)
-    find_package(ZSTD QUIET)
-    set_package_properties(ZSTD PROPERTIES TYPE
+    find_package(zstd QUIET)
+    set_package_properties(zstd PROPERTIES TYPE
         PURPOSE "General compression support")
-    if (ZSTD_FOUND)
+    if (zstd_FOUND)
         set(CMAKE_REQUIRED_LIBRARIES ${CMAKE_REQUIRED_LIBRARIES}
             "${ZSTD_STATIC_LIB}")
         mark_as_advanced(CLEAR ZSTD_INCLUDE_DIRS)
diff --git a/pdal/util/CMakeLists.txt b/pdal/util/CMakeLists.txt
index db42c7f53..c28f08046 100644
--- a/pdal/util/CMakeLists.txt
+++ b/pdal/util/CMakeLists.txt
@@ -9,9 +9,9 @@ endif()
 include(${PDAL_CMAKE_DIR}/execinfo.cmake)
 include(${PDAL_CMAKE_DIR}/unwind.cmake)
 
-if(LIBUNWIND_FOUND)
+if(libunwind_FOUND)
     set(BACKTRACE_SOURCE BacktraceUnwind.cpp)
-    set(BACKTRACE_LIBRARIES ${LIBUNWIND_LIBRARIES})
+    set(BACKTRACE_LIBRARIES ${libunwind_LIBRARIES})
 elseif(LIBEXECINFO_FOUND)
     set(BACKTRACE_SOURCE BacktraceExecinfo.cpp)
     set(BACKTRACE_LIBRARIES ${LIBEXECINFO_LIBRARIES})
