--- a/cmake/draco_install.cmake
+++ b/cmake/draco_install.cmake
@@ -70,14 +70,12 @@ macro(draco_setup_install_target)
                   "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_BINDIR}")
 
   if(MSVC)
-    install(TARGETS draco DESTINATION
-                    "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}")
+    install(TARGETS draco RUNTIME DESTINATION bin ARCHIVE DESTINATION lib LIBRARY DESTINATION lib)
   else()
     install(TARGETS draco_static DESTINATION
                     "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}")
     if(BUILD_SHARED_LIBS)
-      install(TARGETS draco_shared DESTINATION
-                      "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}")
+      install(TARGETS draco_shared RUNTIME DESTINATION bin ARCHIVE DESTINATION lib LIBRARY DESTINATION lib)
     endif()
   endif()
 
