--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -4,7 +4,6 @@ cmake_minimum_required(VERSION 3.0)
 # new feature to Cmake Version > 2.8.12
 # Mac ONLY. Define Relative Path on Mac OS
 if(NOT DEFINED CMAKE_MACOSX_RPATH)
-  set(CMAKE_MACOSX_RPATH 0)
 endif()
 
 # Set the version number for the library
--- a/gtsam/CMakeLists.txt
+++ b/gtsam/CMakeLists.txt
@@ -177,9 +177,6 @@ if(WIN32) # Add 'lib' prefix to static library to avoid filename collision with
 endif()
 
 if (APPLE AND BUILD_SHARED_LIBS)
-	set_target_properties(gtsam PROPERTIES
-		INSTALL_NAME_DIR
-		"${CMAKE_INSTALL_PREFIX}/lib")
 endif()
 
 install(
--- a/gtsam/3rdparty/metis/libmetis/CMakeLists.txt
+++ b/gtsam/3rdparty/metis/libmetis/CMakeLists.txt
@@ -16,9 +16,6 @@ if(WIN32)
 endif()
 
 if (APPLE)
-		set_target_properties(metis PROPERTIES
-			INSTALL_NAME_DIR
-			"${CMAKE_INSTALL_PREFIX}/lib")
  endif()
 
 install(TARGETS metis EXPORT GTSAM-exports LIBRARY DESTINATION lib ARCHIVE DESTINATION lib RUNTIME DESTINATION bin)
