--- cmake/SociBackend.cmake
+++ cmake/SociBackend.cmake
@@ -31,14 +31,13 @@
     if(NOT DEPEND_FOUND)
       list(APPEND DEPS_NOT_FOUND ${dep})
     else()
-      string(TOUPPER "${dep}" DEPU)
-      if( ${DEPU}_INCLUDE_DIR )
-        list(APPEND DEPS_INCLUDE_DIRS ${${DEPU}_INCLUDE_DIR})
+      if( ${dep}_INCLUDE_DIR )
+        list(APPEND DEPS_INCLUDE_DIRS ${${dep}_INCLUDE_DIR})
       endif()
-      if( ${DEPU}_INCLUDE_DIRS )
-        list(APPEND DEPS_INCLUDE_DIRS ${${DEPU}_INCLUDE_DIRS})
+      if( ${dep}_INCLUDE_DIRS )
+        list(APPEND DEPS_INCLUDE_DIRS ${${dep}_INCLUDE_DIRS})
       endif()
-      list(APPEND DEPS_LIBRARIES ${${DEPU}_LIBRARIES})
+      list(APPEND DEPS_LIBRARIES ${${dep}_LIBRARIES})
     endif()
   endforeach()
 
--- cmake/SociDependencies.cmake	2020-05-18 22:39:36.264753367 +0000
+++ cmake/SociDependencies.cmake	2020-05-18 22:44:06.246758889 +0000
@@ -25,7 +25,7 @@
 
 set(SOCI_BACKENDS_DB_DEPENDENCIES
   MySQL
-  ODBC
+  odbc
   Oracle
   PostgreSQL
   SQLite3
@@ -55,9 +55,7 @@
   endif()
 
   foreach(variable ${VARNAMES})
-    if(${VARNAMEU}_FOUND)
-      boost_report_value(${VARNAMEU}_${variable})
-    elseif(${VARNAME}_FOUND)
+    if(${VARNAME}_FOUND)
       boost_report_value(${VARNAME}_${variable})
     endif()
   endforeach()
--- cmake/dependencies/ODBC.cmake	2020-05-18 22:39:36.267753367 +0000
+++ cmake/dependencies/ODBC.cmake	1970-01-01 00:00:00.000000000 +0000
@@ -1,5 +0,0 @@
-set(ODBC_FIND_QUIETLY TRUE)
-
-find_package(ODBC)
-
-boost_external_report(ODBC INCLUDE_DIR LIBRARIES)
--- cmake/dependencies/odbc.cmake	1970-01-01 00:00:00.000000000 +0000
+++ cmake/dependencies/odbc.cmake	2020-05-18 22:42:54.861758574 +0000
@@ -0,0 +1,5 @@
+set(ODBC_FIND_QUIETLY TRUE)
+
+find_package(ODBC)
+
+boost_external_report(ODBC INCLUDE_DIR LIBRARIES)

--- src/backends/odbc/CMakeLists.txt
+++ src/backends/odbc/CMakeLists.txt
@@ -9,7 +9,7 @@
 #
 ###############################################################################
 soci_backend(ODBC
-  DEPENDS ODBC
+  DEPENDS odbc
   DESCRIPTION "SOCI backend for ODBC"
   AUTHORS "Maciej Sobczak, Stephen Hutton, David Courtney"
   MAINTAINERS "Vadim Zeitlin, Mateusz Loskot, Maciej Sobczak")

